// @primer/components/lib/stories/Button.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.buttonTableList = exports.buttonGroup = exports.closeButton = exports.invisibleButton = exports.primaryButton = exports.outlineButton = exports.dangerButton = exports.defaultButton = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ = require("..");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var _default = {
  title: 'Composite components/Button',
  decorators: [Story => {
    return /*#__PURE__*/_react.default.createElement(_.ThemeProvider, null, /*#__PURE__*/_react.default.createElement(_.BaseStyles, null, /*#__PURE__*/_react.default.createElement(Story, null)));
  }],
  argTypes: {
    as: {
      table: {
        disable: true
      }
    },
    theme: {
      table: {
        disable: true
      }
    },
    sx: {
      table: {
        disable: true
      }
    },
    variant: {
      control: {
        type: 'radio',
        options: ['small', 'medium', 'large']
      }
    }
  }
};
exports.default = _default;

const defaultButton = args => /*#__PURE__*/_react.default.createElement(_.Button, args, "Default Button");

exports.defaultButton = defaultButton;
defaultButton.displayName = "defaultButton";

const dangerButton = args => /*#__PURE__*/_react.default.createElement(_.ButtonDanger, args, "Danger Button");

exports.dangerButton = dangerButton;
dangerButton.displayName = "dangerButton";

const outlineButton = args => /*#__PURE__*/_react.default.createElement(_.ButtonOutline, args, "Outline Button");

exports.outlineButton = outlineButton;
outlineButton.displayName = "outlineButton";

const primaryButton = args => /*#__PURE__*/_react.default.createElement(_.ButtonPrimary, args, "Primary Button");

exports.primaryButton = primaryButton;
primaryButton.displayName = "primaryButton";

const invisibleButton = args => /*#__PURE__*/_react.default.createElement(_.ButtonInvisible, args, "Invisible Button");

exports.invisibleButton = invisibleButton;
invisibleButton.displayName = "invisibleButton";

const closeButton = args => /*#__PURE__*/_react.default.createElement(_.ButtonClose, _extends({}, args, {
  onClick: () => alert('button clicked.')
}));

exports.closeButton = closeButton;
closeButton.displayName = "closeButton";

const buttonGroup = args => /*#__PURE__*/_react.default.createElement(_.ButtonGroup, {
  display: "block",
  my: 2
}, /*#__PURE__*/_react.default.createElement(_.Button, args, "Button 1"), /*#__PURE__*/_react.default.createElement(_.Button, args, "Button 2"), /*#__PURE__*/_react.default.createElement(_.Button, args, "Button 3"));

exports.buttonGroup = buttonGroup;
buttonGroup.displayName = "buttonGroup";

const buttonTableList = args => /*#__PURE__*/_react.default.createElement(_.ButtonTableList, args, "Button Table List");

exports.buttonTableList = buttonTableList;
buttonTableList.displayName = "buttonTableList";
defaultButton.args = {
  variant: 'medium'
};
dangerButton.args = {
  variant: 'medium'
};
outlineButton.args = {
  variant: 'medium'
};
primaryButton.args = {
  variant: 'medium'
};
invisibleButton.args = {
  variant: 'medium'
};
closeButton.args = {
  variant: 'medium'
};
buttonGroup.args = {
  variant: 'medium'
};
buttonTableList.args = {
  variant: 'medium'
};
