// @primer/components/lib/stories/Portal.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MultiplePortalRoots = exports.CustomPortalRootByRegistration = exports.customPortalRootById = exports.defaultPortal = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ = require("..");

var _Portal = _interopRequireWildcard(require("../Portal"));

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function () { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Generic behaviors/Portal',
  component: _Portal.default,
  decorators: [Story => {
    return /*#__PURE__*/_react.default.createElement(_.ThemeProvider, null, /*#__PURE__*/_react.default.createElement(_.BaseStyles, null, /*#__PURE__*/_react.default.createElement(Story, null)));
  }]
};
exports.default = _default;

const defaultPortal = () => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, "Root position", /*#__PURE__*/_react.default.createElement(_.Box, {
  bg: "red.2",
  p: 3
}, "Outer container", /*#__PURE__*/_react.default.createElement(_.Box, {
  bg: "green.2",
  p: 3
}, "Inner container", /*#__PURE__*/_react.default.createElement(_Portal.default, null, "Portaled content rendered at ", /*#__PURE__*/_react.default.createElement("code", null, "<BaseStyles>"), " root."))));

exports.defaultPortal = defaultPortal;

const customPortalRootById = () => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, "Root position", /*#__PURE__*/_react.default.createElement(_.Box, {
  bg: "red.2",
  p: 3,
  id: "__primerPortalRoot__"
}, "Outer container", /*#__PURE__*/_react.default.createElement(_.Box, {
  bg: "green.2",
  p: 3
}, "Inner container", /*#__PURE__*/_react.default.createElement(_Portal.default, null, "Portaled content rendered at the outer container."))));

exports.customPortalRootById = customPortalRootById;

const CustomPortalRootByRegistration = () => {
  const outerContainerRef = _react.default.useRef(null);

  const [mounted, setMounted] = _react.default.useState(false);

  _react.default.useEffect(() => {
    if (outerContainerRef.current instanceof HTMLElement) {
      (0, _Portal.registerPortalRoot)(outerContainerRef.current);
      setMounted(true);
    }
  }, []);

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, "Root position", /*#__PURE__*/_react.default.createElement(_.Box, {
    bg: "red.2",
    p: 3,
    ref: outerContainerRef
  }, mounted ? /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, "Outer container", /*#__PURE__*/_react.default.createElement(_.Box, {
    bg: "green.2",
    p: 3
  }, "Inner container", /*#__PURE__*/_react.default.createElement(_Portal.default, null, "Portaled content rendered at the outer container."))) : null));
};

exports.CustomPortalRootByRegistration = CustomPortalRootByRegistration;

const MultiplePortalRoots = () => {
  const outerContainerRef = _react.default.useRef(null);

  const innerContainerRef = _react.default.useRef(null);

  const [mounted, setMounted] = _react.default.useState(false);

  _react.default.useEffect(() => {
    if (outerContainerRef.current instanceof HTMLElement && innerContainerRef.current instanceof HTMLElement) {
      (0, _Portal.registerPortalRoot)(outerContainerRef.current, 'outer');
      (0, _Portal.registerPortalRoot)(innerContainerRef.current, 'inner');
      setMounted(true);
    }
  }, [outerContainerRef]);

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, "Root position", /*#__PURE__*/_react.default.createElement(_.Box, {
    bg: "red.2",
    p: 3,
    ref: outerContainerRef
  }, "Outer container", /*#__PURE__*/_react.default.createElement(_.Box, {
    bg: "green.2",
    p: 3,
    ref: innerContainerRef
  }, mounted ? /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_Portal.default, {
    containerName: "outer"
  }, "Portaled content rendered at the outer container."), /*#__PURE__*/_react.default.createElement(_Portal.default, {
    containerName: "inner"
  }, "Portaled content rendered at the end of the inner container."), /*#__PURE__*/_react.default.createElement(_Portal.default, null, "Portaled content rendered at ", /*#__PURE__*/_react.default.createElement("code", null, "<BaseStyles>"), " root.")) : null, "Inner container")));
};

exports.MultiplePortalRoots = MultiplePortalRoots;
