// @primer/components/lib-esm/stories/Button.stories.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import { BaseStyles, Button, ButtonClose, ButtonDanger, ButtonGroup, ButtonInvisible, ButtonOutline, ButtonPrimary, ButtonTableList, ThemeProvider } from '..';
export default {
  title: 'Composite components/Button',
  decorators: [Story => {
    return /*#__PURE__*/React.createElement(ThemeProvider, null, /*#__PURE__*/React.createElement(BaseStyles, null, /*#__PURE__*/React.createElement(Story, null)));
  }],
  argTypes: {
    as: {
      table: {
        disable: true
      }
    },
    theme: {
      table: {
        disable: true
      }
    },
    sx: {
      table: {
        disable: true
      }
    },
    variant: {
      control: {
        type: 'radio',
        options: ['small', 'medium', 'large']
      }
    }
  }
};
export const defaultButton = args => /*#__PURE__*/React.createElement(Button, args, "Default Button");
defaultButton.displayName = "defaultButton";
export const dangerButton = args => /*#__PURE__*/React.createElement(ButtonDanger, args, "Danger Button");
dangerButton.displayName = "dangerButton";
export const outlineButton = args => /*#__PURE__*/React.createElement(ButtonOutline, args, "Outline Button");
outlineButton.displayName = "outlineButton";
export const primaryButton = args => /*#__PURE__*/React.createElement(ButtonPrimary, args, "Primary Button");
primaryButton.displayName = "primaryButton";
export const invisibleButton = args => /*#__PURE__*/React.createElement(ButtonInvisible, args, "Invisible Button");
invisibleButton.displayName = "invisibleButton";
export const closeButton = args => /*#__PURE__*/React.createElement(ButtonClose, _extends({}, args, {
  onClick: () => alert('button clicked.')
}));
closeButton.displayName = "closeButton";
export const buttonGroup = args => /*#__PURE__*/React.createElement(ButtonGroup, {
  display: "block",
  my: 2
}, /*#__PURE__*/React.createElement(Button, args, "Button 1"), /*#__PURE__*/React.createElement(Button, args, "Button 2"), /*#__PURE__*/React.createElement(Button, args, "Button 3"));
buttonGroup.displayName = "buttonGroup";
export const buttonTableList = args => /*#__PURE__*/React.createElement(ButtonTableList, args, "Button Table List");
buttonTableList.displayName = "buttonTableList";
defaultButton.args = {
  variant: 'medium'
};
dangerButton.args = {
  variant: 'medium'
};
outlineButton.args = {
  variant: 'medium'
};
primaryButton.args = {
  variant: 'medium'
};
invisibleButton.args = {
  variant: 'medium'
};
closeButton.args = {
  variant: 'medium'
};
buttonGroup.args = {
  variant: 'medium'
};
buttonTableList.args = {
  variant: 'medium'
};
