// @primer/components/lib/ActionList/Item.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Item = Item;

var _octiconsReact = require("@primer/octicons-react");

var _react = _interopRequireWildcard(require("react"));

var _constants = require("../constants");

var _sx = _interopRequireDefault(require("../sx"));

var _Flex = _interopRequireDefault(require("../Flex"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Header = require("./Header");

var _Divider = require("./Divider");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function () { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const getItemVariant = (variant = 'default', disabled) => {
  if (disabled) {
    return {
      color: (0, _constants.get)('colors.text.disabled'),
      iconColor: (0, _constants.get)('colors.text.disabled'),
      annotationColor: (0, _constants.get)('colors.text.disabled'),
      hoverBackground: 'inherit',
      hoverCursor: 'default'
    };
  }

  switch (variant) {
    case 'danger':
      return {
        color: (0, _constants.get)('colors.text.danger'),
        iconColor: (0, _constants.get)('colors.icon.danger'),
        annotationColor: (0, _constants.get)('colors.text.disabled'),
        hoverBackground: (0, _constants.get)('colors.bg.danger'),
        hoverCursor: 'pointer'
      };

    default:
      return {
        color: 'inherit',
        iconColor: (0, _constants.get)('colors.text.disabled'),
        annotationColor: (0, _constants.get)('colors.text.disabled'),
        hoverBackground: (0, _constants.get)('colors.selectMenu.tapHighlight'),
        hoverCursor: 'pointer'
      };
  }
};

const StyledItemContent = _styledComponents.default.div.withConfig({
  displayName: "Item__StyledItemContent",
  componentId: "jqpvy8-0"
})(["width:100%;"]);

const StyledItem = _styledComponents.default.div.withConfig({
  displayName: "Item__StyledItem",
  componentId: "jqpvy8-1"
})(["padding:6px ", ";display:flex;border-radius:", ";color:", ";@media (hover:hover) and (pointer:fine){:hover{background:", ";cursor:", ";}}:not(:first-of-type):not(", " + &):not(", " + &){margin-top:", ";", "::before{content:' ';display:block;position:relative;top:-7px;border:0 solid ", ";border-top-width:", ";}}", ""], (0, _constants.get)('space.2'), (0, _constants.get)('radii.2'), ({
  variant,
  item
}) => getItemVariant(variant, item === null || item === void 0 ? void 0 : item.disabled).color, ({
  variant,
  item
}) => getItemVariant(variant, item === null || item === void 0 ? void 0 : item.disabled).hoverBackground, ({
  variant,
  item
}) => getItemVariant(variant, item === null || item === void 0 ? void 0 : item.disabled).hoverCursor, _Divider.StyledDivider, _Header.StyledHeader, ({
  showDivider
}) => showDivider ? `1px` : '0', StyledItemContent, (0, _constants.get)('colors.selectMenu.borderSecondary'), ({
  showDivider
}) => showDivider ? `1px` : '0', _sx.default);

const StyledTextContainer = _styledComponents.default.div.withConfig({
  displayName: "Item__StyledTextContainer",
  componentId: "jqpvy8-2"
})(["display:flex;flex-direction:", ";"], ({
  descriptionVariant
}) => descriptionVariant === 'inline' ? 'row' : 'column');

const BaseVisualContainer = _styledComponents.default.div.withConfig({
  displayName: "Item__BaseVisualContainer",
  componentId: "jqpvy8-3"
})(["height:20px;width:", ";display:flex;flex-direction:column;justify-content:center;margin-right:", ";svg{fill:", ";font-size:", ";}"], (0, _constants.get)('space.3'), (0, _constants.get)('space.2'), ({
  variant,
  disabled
}) => getItemVariant(variant, disabled).iconColor, (0, _constants.get)('fontSizes.0'));

const LeadingVisualContainer = (0, _styledComponents.default)(BaseVisualContainer).withConfig({
  displayName: "Item__LeadingVisualContainer",
  componentId: "jqpvy8-4"
})([""]);
const TrailingVisualContainer = (0, _styledComponents.default)(BaseVisualContainer).withConfig({
  displayName: "Item__TrailingVisualContainer",
  componentId: "jqpvy8-5"
})(["color:", "};margin-left:auto;margin-right:0;div:nth-child(2){margin-left:", ";}display:flex;flex-direction:row;justify-content:flex-end;"], ({
  variant,
  disabled
}) => getItemVariant(variant, disabled).annotationColor, (0, _constants.get)('space.2'));

const DescriptionContainer = _styledComponents.default.span.withConfig({
  displayName: "Item__DescriptionContainer",
  componentId: "jqpvy8-6"
})(["color:", ";margin-left:", ";"], (0, _constants.get)('colors.text.secondary'), ({
  descriptionVariant
}) => descriptionVariant === 'inline' ? (0, _constants.get)('space.2') : 0);
/**
 * An actionable or selectable `Item` with an optional icon and description.
 */


function Item(itemProps) {
  const {
    text,
    description,
    descriptionVariant = 'inline',
    selected,
    selectionVariant,
    leadingVisual: LeadingVisual,
    trailingIcon: TrailingIcon,
    trailingText,
    variant = 'default',
    showDivider,
    disabled,
    onAction,
    onKeyPress,
    children,
    onClick,
    id,
    ...props
  } = itemProps;
  const keyPressHandler = (0, _react.useCallback)(event => {
    if (disabled) {
      return;
    }

    onKeyPress === null || onKeyPress === void 0 ? void 0 : onKeyPress(event);
    const isCheckbox = event.target instanceof HTMLInputElement && event.target.type === 'checkbox';

    if (isCheckbox && event.key === ' ') {
      // space key on a checkbox will also trigger a click event.  Ignore the space key so we don't get double events
      return;
    }

    if (!event.defaultPrevented && [' ', 'Enter'].includes(event.key)) {
      onAction === null || onAction === void 0 ? void 0 : onAction(itemProps, event);
    }
  }, [onAction, disabled, itemProps, onKeyPress]);
  const clickHandler = (0, _react.useCallback)(event => {
    if (disabled) {
      return;
    }

    onClick === null || onClick === void 0 ? void 0 : onClick(event);

    if (!event.defaultPrevented) {
      onAction === null || onAction === void 0 ? void 0 : onAction(itemProps, event);
    }
  }, [onAction, disabled, itemProps, onClick]);
  return /*#__PURE__*/_react.default.createElement(StyledItem, _extends({
    tabIndex: disabled ? undefined : -1,
    variant: variant,
    showDivider: showDivider,
    "aria-selected": selected
  }, props, {
    "data-id": id,
    onKeyPress: keyPressHandler,
    onClick: clickHandler
  }), !!selected === selected && /*#__PURE__*/_react.default.createElement(LeadingVisualContainer, null, selectionVariant === 'multiple' ? /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("input", {
    type: "checkbox",
    checked: selected,
    "aria-label": text,
    readOnly: true,
    "aria-readonly": "false"
  })) : selected && /*#__PURE__*/_react.default.createElement(_octiconsReact.CheckIcon, null)), LeadingVisual && /*#__PURE__*/_react.default.createElement(LeadingVisualContainer, {
    variant: variant,
    disabled: disabled
  }, /*#__PURE__*/_react.default.createElement(LeadingVisual, null)), /*#__PURE__*/_react.default.createElement(StyledItemContent, null, /*#__PURE__*/_react.default.createElement(_Flex.default, null, children, (text || description) && /*#__PURE__*/_react.default.createElement(StyledTextContainer, {
    descriptionVariant: descriptionVariant
  }, text && /*#__PURE__*/_react.default.createElement("div", null, text), description && /*#__PURE__*/_react.default.createElement(DescriptionContainer, {
    descriptionVariant: descriptionVariant
  }, description)), (TrailingIcon || trailingText) && /*#__PURE__*/_react.default.createElement(TrailingVisualContainer, {
    variant: variant,
    disabled: disabled
  }, trailingText && /*#__PURE__*/_react.default.createElement("div", null, trailingText), TrailingIcon && /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement(TrailingIcon, null))))));
}

Item.displayName = "Item";
