// @primer/components/lib-esm/SideNav.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

// eslint-disable-next-line import/no-namespace
import { COMMON, get } from './constants';
import styled, { css } from 'styled-components';
import BorderBox from './BorderBox';
import Link from './Link';
import React from 'react';
import classnames from 'classnames';
import sx from './sx';

function SideNavBase({
  variant,
  className,
  bordered,
  children,
  ...props
}) {
  const variantClassName = variant === 'lightweight' ? 'lightweight' : 'normal';
  const newClassName = classnames(className, `variant-${variantClassName}`);

  if (!bordered) {
    props = { ...props,
      borderWidth: 0
    };
  }

  return /*#__PURE__*/React.createElement(BorderBox, _extends({
    as: "nav",
    className: newClassName
  }, props), children);
}

SideNavBase.displayName = "SideNavBase";
const SideNav = styled(SideNavBase).withConfig({
  displayName: "SideNav",
  componentId: "sc-15k667c-0"
})(["background-color:", ";", " ", ";", ";"], get('colors.bg.secondary'), props => props.bordered && css(["& > &{border-left:0;border-right:0;border-bottom:0;}"]), COMMON, sx);
// used for variant normal hover, focus pseudo selectors
const CommonAccessibilityVariantNormalStyles = css(["background-color:", ";outline:none;text-decoration:none;"], get('colors.state.hover.secondaryBg')); // used for light weight hover, focus pseudo selectors

const CommonAccessibilityVariantLightWeightStyles = css(["color:", ";text-decoration:none;outline:none;"], get('colors.text.primary'));
const SideNavLink = styled(Link).attrs(props => {
  const isReactRouter = typeof props.to === 'string';

  if (isReactRouter || props.selected) {
    // according to their docs, NavLink supports aria-current:
    // https://reacttraining.com/react-router/web/api/NavLink/aria-current-string
    return {
      'aria-current': 'page'
    };
  } else {
    return {};
  }
}).withConfig({
  displayName: "SideNav__SideNavLink",
  componentId: "sc-15k667c-1"
})(["position:relative;display:block;", " width:100%;text-align:left;font-size:", ";& > ", "{border-bottom:none;}", ".variant-normal > &{color:", ";padding:", ";border:0;border-top:", " solid ", ";&:first-child{border-top:0;border-top-right-radius:", ";border-top-left-radius:", ";}&:last-child{border-bottom-right-radius:", ";border-bottom-left-radius:", ";}&::before{position:absolute;top:0;bottom:0;left:0;z-index:1;width:3px;pointer-events:none;content:'';}&:hover{", "}&:focus{", " box-shadow:", ";z-index:1;}&[aria-current='page'],&[aria-selected='true']{background-color:", ";&::before{background-color:", ";}}}", ".variant-lightweight > &{padding:", " 0;color:", ";&:hover{", "}&:focus{", " box-shadow:", ";z-index:1;}&[aria-current='page'],&[aria-selected='true']{color:", ";font-weight:", ";}}", ";"], props => props.variant === 'full' && css(["display:flex;align-items:center;justify-content:space-between;"]), get('fontSizes.1'), SideNav, SideNav, get('colors.text.primary'), get('space.3'), get('borderWidths.1'), get('colors.border.secondary'), get('radii.2'), get('radii.2'), get('radii.2'), get('radii.2'), CommonAccessibilityVariantNormalStyles, CommonAccessibilityVariantNormalStyles, get('shadows.state.focus.shadow'), get('colors.sidenav.selectedBg'), get('colors.sidenav.borderActive'), SideNav, get('space.1'), get('colors.text.link'), CommonAccessibilityVariantLightWeightStyles, CommonAccessibilityVariantLightWeightStyles, get('shadows.state.focus.shadow'), get('colors.text.primary'), get('fontWeights.semibold'), sx);
SideNav.defaultProps = {
  variant: 'normal'
};
SideNavLink.defaultProps = {
  variant: 'normal'
};
SideNavLink.displayName = 'SideNav.Link';
export default Object.assign(SideNav, {
  Link: SideNavLink
});
