// @primer/components/lib/Button/ButtonBase.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.buttonSystemProps = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styledSystem = require("styled-system");

var _constants = require("../constants");

var _ButtonStyles = _interopRequireDefault(require("./ButtonStyles"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const buttonSystemProps = (0, _styledSystem.compose)(_styledSystem.fontSize, _constants.COMMON, _constants.LAYOUT);
exports.buttonSystemProps = buttonSystemProps;
const variants = (0, _styledSystem.variant)({
  variants: {
    small: {
      p: '4px 12px',
      fontSize: 0
    },
    medium: {
      fontSize: 1
    },
    large: {
      fontSize: 2,
      p: '10px 20px'
    }
  }
});

const ButtonBase = _styledComponents.default.button.attrs(({
  disabled,
  onClick
}) => ({
  onClick: disabled ? undefined : onClick
})).withConfig({
  displayName: "ButtonBase",
  componentId: "sc-181ps9o-0"
})(["", " ", ""], _ButtonStyles.default, variants);

ButtonBase.defaultProps = {
  variant: 'medium'
};
var _default = ButtonBase;
exports.default = _default;
