// @primer/components/dist/browser.esm.js
import e,{useState as r,useRef as t,useCallback as o,useEffect as a,forwardRef as n,useMemo as i,createContext as d,useContext as l}from"react";import c,{createGlobalStyle as s,ThemeProvider as f,css as b,ThemeContext as g,keyframes as p}from"styled-components";import u,{createPortal as m}from"react-dom";var h={theme:{animation:{easeOutCubic:"cubic-bezier(0.33, 1, 0.68, 1)"},borderWidths:[0,"1px"],breakpoints:["544px","768px","1012px","1280px"],fonts:{normal:'-apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji"',mono:'SFMono-Regular, Consolas, "Liberation Mono", Menlo, Courier, monospace'},fontSizes:["12px","14px","16px","20px","24px","32px","40px","48px"],fontWeights:{light:300,normal:400,semibold:500,bold:600},lineHeights:{condensedUltra:"1",condensed:"1.25",default:"1.5"},radii:["0","3px","6px","100px"],sizes:{small:"544px",medium:"768px",large:"1012px",xlarge:"1280px"},space:["0","4px","8px","16px","24px","32px","40px","48px","64px","80px","96px","112px","128px"],colorSchemes:{light:{colors:{auto:{black:"#1b1f23",white:"#ffffff",gray:["#fafbfc","#f6f8fa","#e1e4e8","#d1d5da","#959da5","#6a737d","#586069","#444d56","#2f363d","#24292e"],blue:["#f1f8ff","#dbedff","#c8e1ff","#79b8ff","#2188ff","#0366d6","#005cc5","#044289","#032f62","#05264c"],green:["#f0fff4","#dcffe4","#bef5cb","#85e89d","#34d058","#28a745","#22863a","#176f2c","#165c26","#144620"],yellow:["#fffdef","#fffbdd","#fff5b1","#ffea7f","#ffdf5d","#ffd33d","#f9c513","#dbab09","#b08800","#735c0f"],orange:["#fff8f2","#ffebda","#ffd1ac","#ffab70","#fb8532","#f66a0a","#e36209","#d15704","#c24e00","#a04100"],red:["#ffeef0","#ffdce0","#fdaeb7","#f97583","#ea4a5a","#d73a49","#cb2431","#b31d28","#9e1c23","#86181d"],purple:["#f5f0ff","#e6dcfd","#d1bcf9","#b392f0","#8a63d2","#6f42c1","#5a32a3","#4c2889","#3a1d6e","#29134e"],pink:["#ffeef8","#fedbf0","#f9b3dd","#f692ce","#ec6cb9","#ea4aaa","#d03592","#b93a86","#99306f","#6d224f"]},text:{primary:"#24292e",secondary:"#586069",tertiary:"#6a737d",placeholder:"#6a737d",disabled:"#959da5",inverse:"#ffffff",link:"#0366d6",danger:"#cb2431",success:"#22863a",warning:"#b08800",white:"#ffffff"},icon:{primary:"#24292e",secondary:"#586069",tertiary:"#959da5",info:"#0366d6",danger:"#d73a49",success:"#22863a",warning:"#b08800"},border:{primary:"#e1e4e8",secondary:"#eaecef",tertiary:"#d1d5da",overlay:"#e1e4e8",inverse:"#ffffff",info:"#0366d6",danger:"#d73a49",success:"#34d058",warning:"#f9c513"},bg:{canvas:"#ffffff",canvasMobile:"#ffffff",canvasInverse:"#24292e",canvasInset:"#f6f8fa",primary:"#ffffff",secondary:"#fafbfc",tertiary:"#f6f8fa",overlay:"#ffffff",backdrop:"rgba(27,31,35,0.5)",info:"#f1f8ff",infoInverse:"#0366d6",danger:"#ffeef0",dangerInverse:"#d73a49",success:"#dcffe4",successInverse:"#28a745",warning:"#fff5b1",warningInverse:"#ffd33d"},state:{hover:{primaryBg:"#0366d6",primaryBorder:"#0366d6",primaryText:"#ffffff",primaryIcon:"#ffffff",secondaryBg:"#f6f8fa",secondaryBorder:"#f6f8fa"},selected:{primaryBg:"#0366d6",primaryBorder:"#0366d6",primaryText:"#ffffff",primaryIcon:"#ffffff"},focus:{border:"#0366d6"}},fade:{fg10:"rgba(27,31,35,0.1)",fg15:"rgba(27,31,35,0.15)",fg30:"rgba(27,31,35,0.3)",fg50:"rgba(27,31,35,0.5)",fg70:"rgba(27,31,35,0.7)",fg85:"rgba(27,31,35,0.85)",black10:"rgba(27,31,35,0.1)",black15:"rgba(27,31,35,0.15)",black30:"rgba(27,31,35,0.3)",black50:"rgba(27,31,35,0.5)",black70:"rgba(27,31,35,0.7)",black85:"rgba(27,31,35,0.85)",white10:"rgba(255,255,255,0.1)",white15:"rgba(255,255,255,0.15)",white30:"rgba(255,255,255,0.3)",white50:"rgba(255,255,255,0.5)",white70:"rgba(255,255,255,0.7)",white85:"rgba(255,255,255,0.85)"},alert:{info:{text:"#24292e",icon:"rgba(4,66,137,0.6)",bg:"#dbedff",border:"rgba(4,66,137,0.2)"},warn:{text:"#24292e",icon:"#b08800",bg:"#fffbdd",border:"rgba(176,136,0,0.2)"},error:{text:"#24292e",icon:"rgba(158,28,35,0.6)",bg:"#ffe3e6",border:"rgba(158,28,35,0.2)"},success:{text:"#24292e",icon:"rgba(23,111,44,0.8)",bg:"#dcffe4",border:"rgba(23,111,44,0.2)"}},autocomplete:{rowBorder:"#eaecef"},blankslate:{icon:"#a3aab1"},btn:{text:"#24292e",bg:"#fafbfc",border:"rgba(27,31,35,0.15)",hoverBg:"#f3f4f6",hoverBorder:"rgba(27,31,35,0.15)",selectedBg:"#edeff2",focusBg:"#fafbfc",focusBorder:"rgba(27,31,35,0.15)",primary:{text:"#ffffff",bg:"#2ea44f",border:"rgba(27,31,35,0.15)",hoverBg:"#2c974b",hoverBorder:"rgba(27,31,35,0.15)",selectedBg:"#2a8f47",disabledText:"rgba(255,255,255,0.8)",disabledBg:"#94d3a2",disabledBorder:"rgba(27,31,35,0.1)",focusBg:"#2ea44f",focusBorder:"rgba(27,31,35,0.15)",icon:"rgba(255,255,255,0.8)",counterBg:"rgba(255,255,255,0.2)"},outline:{text:"#0366d6",hoverText:"#ffffff",hoverBg:"#0366d6",hoverBorder:"rgba(27,31,35,0.15)",hoverCounterBg:"rgba(255,255,255,0.2)",selectedText:"#ffffff",selectedBg:"#035fc7",selectedBorder:"rgba(27,31,35,0.15)",disabledText:"rgba(3,102,214,0.5)",disabledBg:"#fafbfc",disabledCounterBg:"rgba(3,102,214,0.05)",focusBorder:"rgba(27,31,35,0.15)",counterBg:"rgba(3,102,214,0.1)"},danger:{text:"#d73a49",hoverText:"#ffffff",hoverBg:"#cb2431",hoverBorder:"rgba(27,31,35,0.15)",hoverCounterBg:"rgba(255,255,255,0.2)",selectedText:"#ffffff",selectedBg:"#d42d3d",selectedBorder:"rgba(27,31,35,0.15)",disabledText:"rgba(215,58,73,0.5)",disabledBg:"#fafbfc",disabledCounterBg:"rgba(215,58,73,0.05)",focusBorder:"rgba(27,31,35,0.15)",counterBg:"rgba(215,58,73,0.1)",icon:"#d73a49",hoverIcon:"#ffffff"},counterBg:"rgba(27,31,35,0.08)"},counter:{text:"#24292e",bg:"rgba(209,213,218,0.5)",primary:{text:"#ffffff",bg:"#6a737d"},secondary:{text:"#6a737d"}},label:{border:"#e1e4e8",primary:{text:"#24292e",border:"#6a737d"},secondary:{text:"#586069",border:"#e1e4e8"},info:{text:"#0366d6",border:"#0366d6"},success:{text:"#22863a",border:"#28a745"},warning:{text:"#735c0f",border:"#b08800"},danger:{text:"#d73a49",border:"#cb2431"},orange:{text:"#c24e00",border:"#f66a0a"}},input:{bg:"#ffffff",contrastBg:"#fafbfc",border:"#e1e4e8",disabledBg:"#f6f8fa",disabledBorder:"#e1e4e8",warningBorder:"#f9c513",errorBorder:"#cb2431",tooltip:{success:{text:"#144620",bg:"#dcffe4",border:"#34d058"},warning:{text:"#735c0f",bg:"#fff5b1",border:"#f9c513"},error:{text:"#86181d",bg:"#ffeef0",border:"#f97583"}}},avatar:{bg:"#ffffff",border:"rgba(0,0,0,0)",stackFade:"#d1d5da",stackFadeMore:"#e1e4e8"},toast:{text:"#24292e",bg:"#ffffff",border:"#e1e4e8",icon:"#ffffff",iconBg:"#0366d6",iconBorder:"rgba(0,0,0,0)",success:{text:"#24292e",border:"#e1e4e8",icon:"#ffffff",iconBg:"#28a745",iconBorder:"rgba(0,0,0,0)"},warning:{text:"#24292e",border:"#e1e4e8",icon:"#24292e",iconBg:"#ffd33d",iconBorder:"rgba(0,0,0,0)"},danger:{text:"#24292e",border:"#e1e4e8",icon:"#ffffff",iconBg:"#d73a49",iconBorder:"rgba(0,0,0,0)"},loading:{text:"#24292e",border:"#e1e4e8",icon:"#ffffff",iconBg:"#586069",iconBorder:"rgba(0,0,0,0)"}},timeline:{text:"#444d56",badgeBg:"#e1e4e8",targetBadgeBorder:"#2188ff",targetBadgeShadow:"#c8e1ff"},selectMenu:{borderSecondary:"#eaecef",backdropBg:"rgba(27,31,35,0.5)",backdropBorder:"rgba(0,0,0,0)",tapHighlight:"rgba(209,213,218,0.5)",tapFocusBg:"#dbedff"},box:{blueBorder:"#c8e1ff",rowYellowBg:"#fffbdd",rowBlueBg:"#f1f8ff",headerBlueBg:"#f1f8ff",headerBlueBorder:"#c8e1ff",borderInfo:"rgba(3,102,214,0.2)",bgInfo:"#f1f8ff",borderWarning:"rgba(255,211,61,0.4)",bgWarning:"#fffdef"},branchName:{text:"#586069",icon:"#a8bbd0",bg:"#eaf5ff",link:{text:"#0366d6",icon:"#a8bbd0",bg:"#eaf5ff"}},markdown:{codeBg:"rgba(27,31,35,0.05)",frameBorder:"#dfe2e5",blockquoteBorder:"#dfe2e5",tableBorder:"#dfe2e5",tableTrBorder:"#c6cbd1"},menu:{headingText:"#24292e",borderActive:"#f9826c",bgActive:"rgba(0,0,0,0)"},sidenav:{selectedBg:"#ffffff",borderActive:"#f9826c"},header:{text:"rgba(255,255,255,0.7)",bg:"#24292e",logo:"#ffffff"},filterItem:{barBg:"#eff3f6"},hiddenTextExpander:{bg:"#dfe2e5",bgHover:"#c6cbd1"},dragAndDrop:{border:"#c3c8cf"},uploadEnabled:{border:"#dfe2e5",borderFocused:"#4a9eff"},previewableCommentForm:{border:"#c3c8cf"},underlinenav:{border:"rgba(209,213,218,0)",borderHover:"#d1d5da",borderActive:"#f9826c",text:"#24292e",textHover:"#24292e",textActive:"#24292e",icon:"#959da5",iconHover:"#959da5",iconActive:"#24292e",counterText:"#24292e"},verifiedBadge:{text:"#22863a",bg:"#ffffff",border:"#e1e4e8"},socialCount:{bg:"#ffffff"},tooltip:{text:"#ffffff",bg:"#24292e"},headerSearch:{bg:"#24292e",border:"#444d56"},searchKeyword:{hl:"#fffbdd"},diffstat:{neutralBg:"#d1d5da",neutralBorder:"#d1d5da",deletionBg:"#d73a49",deletionBorder:"#d73a49",additionBg:"#28a745",additionBorder:"#28a745"},mktg:{success:"#2ebc4f",info:"#1074e7",bgShadeGradient:{top:"rgba(27,31,35,0.065)",bottom:"rgba(27,31,35,0)"},btn:{bg:"#3076df",border:"#3076df",text:"#ffffff",icon:"#ffffff",hover:{bg:"#0366d6",border:"#0366d6"},disabled:{bg:"rgba(48,118,223,0.5)",border:"rgba(0,0,0,0)",text:"#ffffff",icon:"#ffffff"},primary:{bg:"#5eb85c",border:"#5eb85c",text:"#ffffff",icon:"#ffffff",hover:{bg:"#28a745",border:"#28a745"},disabled:{bg:"rgba(94,184,92,0.5)",border:"rgba(0,0,0,0)",text:"#ffffff",icon:"#ffffff"}},outline:{bg:"rgba(0,0,0,0)",border:"rgba(42,103,195,0.5)",text:"#3076df",icon:"#3076df",hover:{bg:"rgba(0,0,0,0)",border:"#0366d6",text:"#0366d6",icon:"#0366d6"},disabled:{bg:"rgba(0,0,0,0)",border:"rgba(42,103,195,0.2)",text:"rgba(48,118,223,0.5)",icon:"rgba(48,118,223,0.5)"}},dark:{bg:"rgba(0,0,0,0)",border:"rgba(255,255,255,0.5)",text:"#ffffff",icon:"#ffffff",hover:{bg:"#ffffff",border:"#ffffff",text:"#444d56",icon:"#444d56"},disabled:{bg:"rgba(0,0,0,0)",border:"rgba(255,255,255,0.2)",text:"rgba(255,255,255,0.5)",icon:"rgba(255,255,255,0.5)"}}}},filesExplorerIcon:"#79b8ff",hlAuthorBg:"#f1f8ff",hlAuthorBorder:"#c8e1ff",logoSubdued:"#d1d5da",discussionBorder:"#a2cbac",discussionBgSuccess:"#28a745",actionsWorkflowTableStickyBg:"rgba(255,255,255,0.95)",repoLanguageColorBorder:"rgba(27,31,35,0.1)",codeSelectionBg:"#c8e1ff",highlight:{text:"#442c12",bg:"#fff0bb"},blob:{lineHighlightBg:"#fffbdd",lineHighlightBorder:"rgba(0,0,0,0)"},diff:{addition:{text:"#22863a",bg:"#e6ffed",border:"#34d058"},deletion:{text:"#cb2431",bg:"#ffeef0",border:"#d73a49"},change:{text:"#b08800",bg:"#fff5b1",border:"#f9c513"}},diffBlob:{numText:"rgba(27,31,35,0.3)",numHoverText:"rgba(27,31,35,0.6)",addition:{numText:"rgba(27,31,35,0.3)",numHoverText:"rgba(27,31,35,0.6)",numBg:"#cdffd8",lineBg:"#e6ffed",wordBg:"#acf2bd"},deletion:{numText:"rgba(27,31,35,0.3)",numHoverText:"rgba(27,31,35,0.6)",numBg:"#ffdce0",lineBg:"#ffeef0",wordBg:"#fdb8c0"},hunk:{text:"rgba(27,31,35,0.7)",numBg:"#dbedff",lineBg:"#f1f8ff"},emptyBlockBg:"#fafbfc",selectedLineHighlightBg:"rgba(255,223,93,0.2)",selectedLineHighlightBorder:"#ffd33d",expander:{icon:"#586069",hoverIcon:"#ffffff",hoverBg:"#0366d6"},commentButton:{icon:"#ffffff",bg:"#0366d6",gradientBg:"#0372ef"}},globalNav:{logo:"#ffffff",bg:"#24292e",text:"#ffffff",icon:"#ffffff",inputBg:"#fafbfc",inputBorder:"#fafbfc",inputIcon:"#d1d5da",inputPlaceholder:"#959da5"},calendarGraph:{dayBg:"#ebedf0",dayBorder:"rgba(27,31,35,0.06)",dayL1Bg:"#9be9a8",dayL2Bg:"#40c463",dayL3Bg:"#30a14e",dayL4Bg:"#216e39",dayL4Border:"rgba(27,31,35,0.06)",dayL3Border:"rgba(27,31,35,0.06)",dayL2Border:"rgba(27,31,35,0.06)",dayL1Border:"rgba(27,31,35,0.06)"},footerInvertocat:{octicon:"#d1d5da",octiconHover:"#6a737d"},prState:{draft:{text:"#ffffff",bg:"#6a737d",border:"rgba(0,0,0,0)"},open:{text:"#ffffff",bg:"#28a745",border:"rgba(0,0,0,0)"},merged:{text:"#ffffff",bg:"#6f42c1",border:"rgba(0,0,0,0)"},closed:{text:"#ffffff",bg:"#d73a49",border:"rgba(0,0,0,0)"}},topicTag:{text:"#0366d6",bg:"#f1f8ff",hoverBg:"#ddeeff",activeBg:"#e7f3ff"},mergeBox:{successIconBg:"#28a745",successIconText:"#ffffff",successIconBorder:"rgba(0,0,0,0)",successIndicatorBg:"#28a745",successIndicatorBorder:"rgba(0,0,0,0)",mergedIconBg:"#6f42c1",mergedIconText:"#ffffff",mergedIconBorder:"rgba(0,0,0,0)",mergedBoxBorder:"#6f42c1",neutralIconBg:"#6a737d",neutralIconText:"#ffffff",neutralIconBorder:"rgba(0,0,0,0)",neutralIndicatorBg:"#6a737d",neutralIndicatorBorder:"rgba(0,0,0,0)",warningIconBg:"#dbab09",warningIconText:"#ffffff",warningIconBorder:"rgba(0,0,0,0)",warningBoxBorder:"#ffd33d",warningMergeHighlight:"rgba(0,0,0,0)",errorIconBg:"#d73a49",errorIconText:"#ffffff",errorIconBorder:"rgba(0,0,0,0)",errorIndicatorBg:"#d73a49",errorIndicatorBorder:"rgba(0,0,0,0)"},project:{cardBg:"#ffffff",headerBg:"#24292e",sidebarBg:"#ffffff",gradientIn:"#ffffff",gradientOut:"rgba(255,255,255,0)"},checks:{bg:"#24292e",textPrimary:"#fafbfc",textSecondary:"#959da5",textLink:"#79b8ff",btnIcon:"#d1d5da",btnHoverIcon:"#fafbfc",btnHoverBg:"rgba(255,255,255,0.125)",inputText:"#f6f8fa",inputPlaceholderText:"#959da5",inputFocusText:"#959da5",inputBg:"#2f363d",dropdownText:"#d1d5da",dropdownBg:"#2f363d",dropdownBorder:"#444d56",dropdownHoverText:"#fafbfc",dropdownHoverBg:"#444d56",dropdownBtnHoverText:"#fafbfc",dropdownBtnHoverBg:"#2f363d",scrollbarThumbBg:"#586069",headerLabelText:"#e1e4e8",headerLabelOpenText:"#fafbfc",headerBorder:"#2f363d",headerIcon:"#959da5",lineText:"#e1e4e8",lineNumText:"rgba(149,157,165,0.75)",lineTimestampText:"#959da5",lineHoverBg:"#2f363d",lineSelectedBg:"rgba(33,136,255,0.15)",lineSelectedNumText:"#79b8ff",lineDtFmText:"#24292e",lineDtFmBg:"#ffd33d",gateBg:"rgba(249,197,19,0.15)",gateText:"#e1e4e8",gateWaitingText:"#d1d5da",stepHeaderOpenBg:"#2f363d",stepErrorText:"#f97583",stepWarningText:"#ffea7f",loglineText:"#959da5",loglineNumText:"rgba(149,157,165,0.75)",loglineDebugText:"#b392f0",loglineErrorText:"#e1e4e8",loglineErrorNumText:"#f97583",loglineErrorBg:"rgba(203,36,49,0.15)",loglineWarningText:"#e1e4e8",loglineWarningNumText:"#ffea7f",loglineWarningBg:"rgba(249,197,19,0.15)",loglineCommandText:"#79b8ff",loglineSectionText:"#85e89d"},introShelf:{gradientLeft:"#f1f8ff",gradientRight:"#dcffe4",gradientIn:"#ffffff",gradientOut:"rgba(255,255,255,0)"},marketingIcon:{primary:"#2188ff",secondary:"#79b8ff"},prettylights:{syntax:{comment:"#6a737d",constant:"#005cc5",entity:"#6f42c1",storageModifierImport:"#24292e",entityTag:"#22863a",keyword:"#d73a49",string:"#032f62",variable:"#e36209",brackethighlighterUnmatched:"#b31d28",invalidIllegalText:"#fafbfc",invalidIllegalBg:"#b31d28",carriageReturnText:"#fafbfc",carriageReturnBg:"#d73a49",stringRegexp:"#22863a",markupList:"#735c0f",markupHeading:"#005cc5",markupItalic:"#24292e",markupBold:"#24292e",markupDeletedText:"#b31d28",markupDeletedBg:"#ffeef0",markupInsertedText:"#22863a",markupInsertedBg:"#f0fff4",markupChangedText:"#e36209",markupChangedBg:"#ffebda",markupIgnoredText:"#f6f8fa",markupIgnoredBg:"#005cc5",metaDiffRange:"#6f42c1",brackethighlighterAngle:"#586069",sublimelinterGutterMark:"#959da5",constantOtherReferenceLink:"#032f62"}},codemirror:{text:"#24292e",bg:"#ffffff",guttersBg:"#ffffff",guttermarkerText:"#ffffff",guttermarkerSubtleText:"#d1d5da",linenumberText:"#959da5",cursor:"#24292e",selectionBg:"#c8e1ff",activelineBg:"#fafbfc",matchingbracketText:"#24292e",linesBg:"#ffffff",syntax:{comment:"#6a737d",constant:"#005cc5",entity:"#6f42c1",keyword:"#d73a49",storage:"#d73a49",string:"#032f62",support:"#005cc5",variable:"#e36209"}},ansi:{black:"#24292e",blackBright:"#2f363d",white:"#e1e4e8",whiteBright:"#e1e4e8",gray:"#959da5",red:"#f97583",redBright:"#fdaeb7",green:"#85e89d",greenBright:"#bef5cb",yellow:"#ffea7f",yellowBright:"#fff5b1",blue:"#79b8ff",blueBright:"#c8e1ff",magenta:"#b392f0",magentaBright:"#d1bcf9",cyan:"#76e3ea",cyanBright:"#b3f0ff"}},shadows:{shadow:{small:"0 1px 0 rgba(27,31,35,0.04)",medium:"0 3px 6px rgba(149,157,165,0.15)",large:"0 8px 24px rgba(149,157,165,0.2)",extraLarge:"0 12px 48px rgba(149,157,165,0.3)",highlight:"inset 0 1px 0 rgba(255,255,255,0.25)",inset:"inset 0 1px 0 rgba(225,228,232,0.2)"},state:{focus:{shadow:"0 0 0 3px rgba(3,102,214,0.3)"}},autocomplete:{shadow:"0 3px 6px rgba(149,157,165,0.15)"},btn:{shadow:"0 1px 0 rgba(27,31,35,0.04)",insetShadow:"inset 0 1px 0 rgba(255,255,255,0.25)",focusShadow:"0 0 0 3px rgba(3,102,214,0.3)",shadowActive:"inset 0 0.15em 0.3em rgba(27,31,35,0.15)",shadowInputFocus:"0 0 0 0.2em rgba(3,102,214,0.3)",primary:{shadow:"0 1px 0 rgba(27,31,35,0.1)",insetShadow:"inset 0 1px 0 rgba(255,255,255,0.03)",selectedShadow:"inset 0 1px 0 rgba(20,70,32,0.2)",focusShadow:"0 0 0 3px rgba(46,164,79,0.4)"},outline:{hoverShadow:"0 1px 0 rgba(27,31,35,0.1)",hoverInsetShadow:"inset 0 1px 0 rgba(255,255,255,0.03)",selectedShadow:"inset 0 1px 0 rgba(5,38,76,0.2)",focusShadow:"0 0 0 3px rgba(0,92,197,0.4)"},danger:{hoverShadow:"0 1px 0 rgba(27,31,35,0.1)",hoverInsetShadow:"inset 0 1px 0 rgba(255,255,255,0.03)",selectedShadow:"inset 0 1px 0 rgba(134,24,29,0.2)",focusShadow:"0 0 0 3px rgba(203,36,49,0.4)"}},dropdown:{shadow:"0 8px 24px rgba(149,157,165,0.2)"},input:{shadow:"inset 0 1px 2px rgba(27,31,35,0.075)"},avatar:{childShadow:"-2px -2px 0 rgba(255,255,255,0.8)"},toast:{shadow:"0 8px 24px rgba(149,157,165,0.2)"},selectMenu:{shadow:"0 0 18px rgba(27,31,35,0.4)"},mktg:{btn:{focusShadow:"0 0 0 3px rgba(48,118,223,0.3)",primary:{focusShadow:"0 0 0 3px rgba(94,184,92,0.3)"},outline:{focusShadow:"0 0 0 3px rgba(42,103,195,0.3)"},dark:{focusShadow:"0 0 0 3px rgba(255,255,255,0.3)"}}}}},dark:{colors:{auto:{black:"#f0f6fc",white:"#010409",gray:["#0d1117","#161b22","#21262d","#30363d","#484f58","#6e7681","#8b949e","#b1bac4","#c9d1d9","#f0f6fc"],blue:["#051d4d","#0c2d6b","#0d419d","#1158c7","#1f6feb","#388bfd","#58a6ff","#79c0ff","#a5d6ff","#cae8ff"],green:["#04260f","#033a16","#0f5323","#196c2e","#238636","#2ea043","#3fb950","#56d364","#7ee787","#aff5b4"],yellow:["#341a00","#4b2900","#693e00","#845306","#9e6a03","#bb8009","#d29922","#e3b341","#f2cc60","#f8e3a1"],orange:["#3d1300","#5a1e02","#762d0a","#9b4215","#bd561d","#db6d28","#f0883e","#ffa657","#ffc680","#ffdfb6"],red:["#490202","#67060c","#8e1519","#b62324","#da3633","#f85149","#ff7b72","#ffa198","#ffc1ba","#ffdcd7"],purple:["#271052","#3c1e70","#553098","#6e40c9","#8957e5","#a371f7","#bc8cff","#d2a8ff","#e2c5ff","#eddeff"],pink:["#42062a","#5e103e","#7d2457","#9e3670","#bf4b8a","#db61a2","#f778ba","#ff9bce","#ffbedd","#ffdaec"]},text:{primary:"#c9d1d9",secondary:"#8b949e",tertiary:"#8b949e",placeholder:"#484f58",disabled:"#484f58",inverse:"#0d1117",link:"#58a6ff",danger:"#f85149",success:"#56d364",warning:"#e3b341",white:"#f0f6fc"},icon:{primary:"#c9d1d9",secondary:"#6e7681",tertiary:"#484f58",info:"#79c0ff",danger:"#f85149",success:"#56d364",warning:"#e3b341"},border:{primary:"#30363d",secondary:"#21262d",tertiary:"#6e7681",overlay:"#30363d",inverse:"#f0f6fc",info:"rgba(56,139,253,0.4)",danger:"rgba(248,81,73,0.4)",success:"rgba(63,185,80,0.4)",warning:"rgba(187,128,9,0.4)"},bg:{canvas:"#0d1117",canvasMobile:"#010409",canvasInverse:"#f0f6fc",canvasInset:"#090c10",primary:"#0d1117",secondary:"#0d1117",tertiary:"#161b22",overlay:"#21262d",backdrop:"rgba(1,4,9,0.8)",info:"rgba(56,139,253,0.1)",infoInverse:"#388bfd",danger:"rgba(248,81,73,0.1)",dangerInverse:"#da3633",success:"rgba(46,160,67,0.1)",successInverse:"#2ea043",warning:"rgba(187,128,9,0.1)",warningInverse:"#bb8009"},state:{hover:{primaryBg:"#1f6feb",primaryBorder:"#388bfd",primaryText:"#f0f6fc",primaryIcon:"#f0f6fc",secondaryBg:"#161b22",secondaryBorder:"#161b22"},selected:{primaryBg:"#1f6feb",primaryBorder:"#388bfd",primaryText:"#f0f6fc",primaryIcon:"#f0f6fc"},focus:{border:"#388bfd"}},fade:{fg10:"rgba(240,246,252,0.1)",fg15:"rgba(240,246,252,0.15)",fg30:"rgba(240,246,252,0.3)",fg50:"rgba(240,246,252,0.5)",fg70:"rgba(240,246,252,0.7)",fg85:"rgba(240,246,252,0.85)",black10:"rgba(1,4,9,0.1)",black15:"rgba(1,4,9,0.15)",black30:"rgba(1,4,9,0.3)",black50:"rgba(1,4,9,0.5)",black70:"rgba(1,4,9,0.7)",black85:"rgba(1,4,9,0.85)",white10:"rgba(240,246,252,0.1)",white15:"rgba(240,246,252,0.15)",white30:"rgba(240,246,252,0.3)",white50:"rgba(240,246,252,0.5)",white70:"rgba(240,246,252,0.7)",white85:"rgba(240,246,252,0.85)"},alert:{info:{text:"#79c0ff",icon:"#79c0ff",bg:"rgba(56,139,253,0.1)",border:"rgba(56,139,253,0.4)"},warn:{text:"#e3b341",icon:"#e3b341",bg:"rgba(187,128,9,0.1)",border:"rgba(187,128,9,0.4)"},error:{text:"#ff7b72",icon:"#ff7b72",bg:"rgba(248,81,73,0.1)",border:"rgba(248,81,73,0.4)"},success:{text:"#56d364",icon:"#56d364",bg:"rgba(46,160,67,0.1)",border:"rgba(46,160,67,0.4)"}},autocomplete:{rowBorder:"#30363d"},blankslate:{icon:"#535c66"},btn:{text:"#c9d1d9",bg:"#21262d",border:"#30363d",hoverBg:"#30363d",hoverBorder:"#8b949e",selectedBg:"#161b22",focusBg:"#21262d",focusBorder:"#8b949e",primary:{text:"#ffffff",bg:"#238636",border:"#2ea043",hoverBg:"#2ea043",hoverBorder:"#3fb950",selectedBg:"#238636",disabledText:"rgba(240,246,252,0.5)",disabledBg:"rgba(35,134,54,0.6)",disabledBorder:"rgba(0,0,0,0)",focusBg:"#238636",focusBorder:"#3fb950",icon:"#f0f6fc",counterBg:"rgba(240,246,252,0.2)"},outline:{text:"#58a6ff",hoverText:"#58a6ff",hoverBg:"#30363d",hoverBorder:"#58a6ff",hoverCounterBg:"rgba(240,246,252,0.2)",selectedText:"#f0f6fc",selectedBg:"#0d419d",selectedBorder:"rgba(240,246,252,0.1)",disabledText:"rgba(88,166,255,0.5)",disabledBg:"#0d1117",disabledCounterBg:"rgba(31,111,235,0.05)",focusBorder:"#58a6ff",counterBg:"rgba(31,111,235,0.1)"},danger:{text:"#f85149",hoverText:"#ffffff",hoverBg:"#da3633",hoverBorder:"#f85149",hoverCounterBg:"rgba(255,255,255,0.2)",selectedText:"#ffffff",selectedBg:"#b62324",selectedBorder:"rgba(240,246,252,0.1)",disabledText:"rgba(248,81,73,0.5)",disabledBg:"#0d1117",disabledCounterBg:"rgba(218,54,51,0.05)",focusBorder:"#f85149",counterBg:"rgba(218,54,51,0.1)",icon:"#f85149",hoverIcon:"#f0f6fc"},counterBg:"#30363d"},counter:{text:"#c9d1d9",bg:"#30363d",primary:{text:"#c9d1d9",bg:"#6e7681"},secondary:{text:"#8b949e"}},label:{border:"#30363d",primary:{text:"#b1bac4",border:"#6e7681"},secondary:{text:"#8b949e",border:"#30363d"},info:{text:"#388bfd",border:"rgba(56,139,253,0.4)"},success:{text:"#3fb950",border:"rgba(46,160,67,0.4)"},warning:{text:"#e3b341",border:"rgba(242,211,91,0.4)"},danger:{text:"#f85149",border:"rgba(248,81,73,0.4)"},orange:{text:"#db6d28",border:"rgba(219,109,40,0.4)"}},input:{bg:"#0d1117",contrastBg:"rgba(1,4,9,0.5)",border:"#21262d",disabledBg:"#161b22",disabledBorder:"#30363d",warningBorder:"#d29922",errorBorder:"#f85149",tooltip:{success:{text:"#56d364",bg:"#101f1b",border:"#1c532b"},warning:{text:"#e3b341",bg:"#1e1c16",border:"#5d4411"},error:{text:"#ff7b72",bg:"#25171c",border:"#792e2e"}}},avatar:{bg:"rgba(240,246,252,0.1)",border:"rgba(240,246,252,0.1)",stackFade:"#30363d",stackFadeMore:"#21262d"},toast:{text:"#79c0ff",bg:"#010409",border:"#214981",icon:"#79c0ff",iconBg:"#121d2e",iconBorder:"#214981",success:{text:"#56d364",border:"#1c532c",icon:"#56d364",iconBg:"#111f1c",iconBorder:"#1c532c"},warning:{text:"#e3b341",border:"#5e4411",icon:"#e3b341",iconBg:"#1f1c16",iconBorder:"#5e4411"},danger:{text:"#ff7b72",border:"#792e2e",icon:"#ff7b72",iconBg:"#25171c",iconBorder:"#792e2e"},loading:{text:"#c9d1d9",border:"#30363d",icon:"#f0f6fc",iconBg:"#30363d",iconBorder:"#30363d"}},timeline:{text:"#b1bac4",badgeBg:"#0d1117",targetBadgeBorder:"#1f6feb",targetBadgeShadow:"#0d419d"},selectMenu:{borderSecondary:"#30363d",backdropBg:"rgba(1,4,9,0.5)",backdropBorder:"#484f58",tapHighlight:"rgba(48,54,61,0.5)",tapFocusBg:"#0c2d6b"},box:{blueBorder:"#0d419d",rowYellowBg:"rgba(235,196,64,0.1)",rowBlueBg:"rgba(121,192,255,0.1)",headerBlueBg:"#0d1117",headerBlueBorder:"#30363d",borderInfo:"rgba(56,139,253,0.4)",bgInfo:"rgba(56,139,253,0.1)",borderWarning:"rgba(187,128,9,0.4)",bgWarning:"rgba(187,128,9,0.1)"},branchName:{text:"#c9d1d9",icon:"#b1bac4",bg:"rgba(88,166,255,0.1)",link:{text:"#58a6ff",icon:"#58a6ff",bg:"rgba(88,166,255,0.1)"}},markdown:{codeBg:"rgba(240,246,252,0.15)",frameBorder:"#3b434b",blockquoteBorder:"#3b434b",tableBorder:"#3b434b",tableTrBorder:"#272c32"},menu:{headingText:"#8b949e",borderActive:"#f78166",bgActive:"#161b22"},sidenav:{selectedBg:"#21262d",borderActive:"#f78166"},header:{text:"rgba(240,246,252,0.7)",bg:"#161b22",logo:"#f0f6fc"},filterItem:{barBg:"#12161c"},hiddenTextExpander:{bg:"#21262d",bgHover:"#30363d"},dragAndDrop:{border:"#25292f"},uploadEnabled:{border:"#3b434b",borderFocused:"#4487ee"},previewableCommentForm:{border:"#25292f"},underlinenav:{border:"rgba(48,54,61,0)",borderHover:"#30363d",borderActive:"#f78166",text:"#8b949e",textHover:"#c9d1d9",textActive:"#c9d1d9",icon:"#6e7681",iconHover:"#c9d1d9",iconActive:"#c9d1d9",counterText:"#8b949e"},verifiedBadge:{text:"#3fb950",bg:"rgba(63,185,80,0.1)",border:"rgba(63,185,80,0.4)"},socialCount:{bg:"#21262d"},tooltip:{text:"#f0f6fc",bg:"#21262d"},headerSearch:{bg:"#0d1117",border:"#21262d"},searchKeyword:{hl:"rgba(187,128,9,0.4)"},diffstat:{neutralBg:"#30363d",neutralBorder:"rgba(240,246,252,0.1)",deletionBg:"#da3633",deletionBorder:"#f85149",additionBg:"#238636",additionBorder:"#2ea043"},mktg:{success:"#29933d",info:"#2a7bf3",bgShadeGradient:{top:"rgba(1,4,9,0.065)",bottom:"rgba(1,4,9,0)"},btn:{bg:"#1f6feb",border:"#1f6feb",text:"#f0f6fc",icon:"#f0f6fc",hover:{bg:"#388bfd",border:"#388bfd"},disabled:{bg:"rgba(56,139,253,0.5)",border:"rgba(0,0,0,0)",text:"rgba(240,246,252,0.5)",icon:"rgba(240,246,252,0.5)"},primary:{bg:"#238636",border:"#238636",text:"#f0f6fc",icon:"#f0f6fc",hover:{bg:"#2ea043",border:"#2ea043"},disabled:{bg:"rgba(46,160,67,0.5)",border:"rgba(0,0,0,0)",text:"rgba(240,246,252,0.5)",icon:"rgba(240,246,252,0.5)"}},outline:{bg:"rgba(0,0,0,0)",border:"rgba(56,139,253,0.5)",text:"#388bfd",icon:"#388bfd",hover:{bg:"rgba(0,0,0,0)",border:"#388bfd",text:"#58a6ff",icon:"#58a6ff"},disabled:{bg:"rgba(0,0,0,0)",border:"rgba(56,139,253,0.2)",text:"rgba(56,139,253,0.5)",icon:"rgba(56,139,253,0.5)"}},dark:{bg:"rgba(0,0,0,0)",border:"rgba(201,209,217,0.5)",text:"#c9d1d9",icon:"#c9d1d9",hover:{bg:"rgba(201,209,217,0.5)",border:"rgba(201,209,217,0.5)",text:"#0d1117",icon:"#0d1117"},disabled:{bg:"rgba(0,0,0,0)",border:"rgba(201,209,217,0.2)",text:"rgba(201,209,217,0.5)",icon:"rgba(201,209,217,0.5)"}}}},filesExplorerIcon:"#6e7681",hlAuthorBg:"#051d4d",hlAuthorBorder:"#0d419d",logoSubdued:"#30363d",discussionBorder:"#34513b",discussionBgSuccess:"rgba(46,160,67,0.1)",actionsWorkflowTableStickyBg:"rgba(13,17,23,0.95)",repoLanguageColorBorder:"rgba(240,246,252,0.2)",codeSelectionBg:"rgba(121,192,255,0.3)",highlight:{text:"#ffd467",bg:"rgba(204,143,44,0.38)"},blob:{lineHighlightBg:"rgba(210,153,34,0.15)",lineHighlightBorder:"#e3b341"},diff:{addition:{text:"#56d364",bg:"rgba(46,160,67,0.2)",border:"#196c2e"},deletion:{text:"#f85149",bg:"rgba(218,54,51,0.2)",border:"#b62324"},change:{text:"#e3b341",bg:"#341a00",border:"#9e6a03"}},diffBlob:{numText:"rgba(240,246,252,0.3)",numHoverText:"rgba(240,246,252,0.6)",addition:{numText:"#3fb950",numHoverText:"#7ee787",numBg:"rgba(46,160,67,0.1)",lineBg:"rgba(46,160,67,0.2)",wordBg:"rgba(46,160,67,0.55)"},deletion:{numText:"#f85149",numHoverText:"#ffa198",numBg:"rgba(218,54,51,0.1)",lineBg:"rgba(218,54,51,0.2)",wordBg:"rgba(218,54,51,0.5)"},hunk:{text:"#8b949e",numBg:"rgba(88,166,255,0.15)",lineBg:"rgba(88,166,255,0.1)"},emptyBlockBg:"#161b22",selectedLineHighlightBg:"rgba(187,128,9,0.1)",selectedLineHighlightBorder:"#bb8009",expander:{icon:"#8b949e",hoverIcon:"#f0f6fc",hoverBg:"#1f6feb"},commentButton:{icon:"#f0f6fc",bg:"#1f6feb",gradientBg:"#367eed"}},globalNav:{logo:"#f0f6fc",bg:"#161b22",text:"#c9d1d9",icon:"#c9d1d9",inputBg:"#0d1117",inputBorder:"#21262d",inputIcon:"#21262d",inputPlaceholder:"#484f58"},calendarGraph:{dayBg:"#161b22",dayBorder:"rgba(27,31,35,0.06)",dayL1Bg:"#003820",dayL2Bg:"#00602d",dayL3Bg:"#10983d",dayL4Bg:"#27d545",dayL4Border:"rgba(255,255,255,0.05)",dayL3Border:"rgba(255,255,255,0.05)",dayL2Border:"rgba(255,255,255,0.05)",dayL1Border:"rgba(255,255,255,0.05)"},footerInvertocat:{octicon:"#30363d",octiconHover:"#6e7681"},prState:{draft:{text:"#8b949e",bg:"rgba(139,148,158,0.1)",border:"rgba(139,148,158,0.4)"},open:{text:"#3fb950",bg:"rgba(63,185,80,0.1)",border:"rgba(63,185,80,0.4)"},merged:{text:"#a371f7",bg:"rgba(188,140,255,0.1)",border:"rgba(188,140,255,0.4)"},closed:{text:"#f85149",bg:"rgba(218,54,51,0.1)",border:"rgba(218,54,51,0.4)"}},topicTag:{text:"#58a6ff",bg:"rgba(56,139,253,0.1)",hoverBg:"rgba(56,139,253,0.2)",activeBg:"rgba(56,139,253,0.15)"},mergeBox:{successIconBg:"rgba(46,160,67,0.1)",successIconText:"#3fb950",successIconBorder:"rgba(46,160,67,0.4)",successIndicatorBg:"#238636",successIndicatorBorder:"#2ea043",mergedIconBg:"rgba(188,140,255,0.1)",mergedIconText:"#a371f7",mergedIconBorder:"rgba(188,140,255,0.4)",mergedBoxBorder:"rgba(188,140,255,0.4)",neutralIconBg:"rgba(201,209,217,0.1)",neutralIconText:"#8b949e",neutralIconBorder:"rgba(201,209,217,0.4)",neutralIndicatorBg:"#484f58",neutralIndicatorBorder:"#6e7681",warningIconBg:"rgba(187,128,9,0.1)",warningIconText:"#e3b341",warningIconBorder:"rgba(187,128,9,0.4)",warningBoxBorder:"rgba(187,128,9,0.4)",warningMergeHighlight:"rgba(187,128,9,0.1)",errorIconBg:"rgba(248,81,73,0.1)",errorIconText:"#f85149",errorIconBorder:"rgba(248,81,73,0.4)",errorIndicatorBg:"#da3633",errorIndicatorBorder:"#f85149"},project:{cardBg:"#161b22",headerBg:"#0d1117",sidebarBg:"#161b22",gradientIn:"#161b22",gradientOut:"rgba(22,27,34,0)"},checks:{bg:"#090c10",textPrimary:"#c9d1d9",textSecondary:"#8b949e",textLink:"#58a6ff",btnIcon:"#6e7681",btnHoverIcon:"#c9d1d9",btnHoverBg:"#30363d",inputText:"#8b949e",inputPlaceholderText:"#484f58",inputFocusText:"#c9d1d9",inputBg:"#0d1117",dropdownText:"#c9d1d9",dropdownBg:"#21262d",dropdownBorder:"#30363d",dropdownHoverText:"#f0f6fc",dropdownHoverBg:"#1f6feb",dropdownBtnHoverText:"#f0f6fc",dropdownBtnHoverBg:"#161b22",scrollbarThumbBg:"#30363d",headerLabelText:"#8b949e",headerLabelOpenText:"#c9d1d9",headerBorder:"#21262d",headerIcon:"#6e7681",lineText:"#8b949e",lineNumText:"#8b949e",lineTimestampText:"#8b949e",lineHoverBg:"#161b22",lineSelectedBg:"rgba(56,139,253,0.1)",lineSelectedNumText:"#58a6ff",lineDtFmText:"#0d1117",lineDtFmBg:"#d29922",gateBg:"rgba(132,83,6,0.15)",gateText:"#8b949e",gateWaitingText:"#e3b341",stepHeaderOpenBg:"#161b22",stepErrorText:"#f85149",stepWarningText:"#e3b341",loglineText:"#6e7681",loglineNumText:"#8b949e",loglineDebugText:"#bc8cff",loglineErrorText:"#8b949e",loglineErrorNumText:"#8b949e",loglineErrorBg:"rgba(248,81,73,0.1)",loglineWarningText:"#8b949e",loglineWarningNumText:"#e3b341",loglineWarningBg:"rgba(187,128,9,0.1)",loglineCommandText:"#58a6ff",loglineSectionText:"#56d364"},introShelf:{gradientLeft:"rgba(56,139,253,0.1)",gradientRight:"rgba(46,160,67,0.1)",gradientIn:"#0d1117",gradientOut:"rgba(13,17,23,0)"},marketingIcon:{primary:"#79c0ff",secondary:"#1f6feb"},prettylights:{syntax:{comment:"#8b949e",constant:"#79c0ff",entity:"#d2a8ff",storageModifierImport:"#c9d1d9",entityTag:"#7ee787",keyword:"#ff7b72",string:"#a5d6ff",variable:"#ffa657",brackethighlighterUnmatched:"#f85149",invalidIllegalText:"#f0f6fc",invalidIllegalBg:"#8e1519",carriageReturnText:"#f0f6fc",carriageReturnBg:"#b62324",stringRegexp:"#7ee787",markupList:"#f2cc60",markupHeading:"#1f6feb",markupItalic:"#c9d1d9",markupBold:"#c9d1d9",markupDeletedText:"#ffdcd7",markupDeletedBg:"#67060c",markupInsertedText:"#aff5b4",markupInsertedBg:"#033a16",markupChangedText:"#ffdfb6",markupChangedBg:"#5a1e02",markupIgnoredText:"#c9d1d9",markupIgnoredBg:"#1158c7",metaDiffRange:"#d2a8ff",brackethighlighterAngle:"#8b949e",sublimelinterGutterMark:"#484f58",constantOtherReferenceLink:"#a5d6ff"}},codemirror:{text:"#c9d1d9",bg:"#0d1117",guttersBg:"#0d1117",guttermarkerText:"#0d1117",guttermarkerSubtleText:"#6e7681",linenumberText:"#8b949e",cursor:"#f0f6fc",selectionBg:"rgba(121,192,255,0.3)",activelineBg:"#161b22",matchingbracketText:"#c9d1d9",linesBg:"#0d1117",syntax:{comment:"#8b949e",constant:"#79c0ff",entity:"#d2a8ff",keyword:"#ff7b72",storage:"#ff7b72",string:"#a5d6ff",support:"#79c0ff",variable:"#ffa657"}},ansi:{black:"#0d1117",blackBright:"#161b22",white:"#b1bac4",whiteBright:"#b1bac4",gray:"#6e7681",red:"#ff7b72",redBright:"#ffa198",green:"#3fb950",greenBright:"#56d364",yellow:"#d29922",yellowBright:"#e3b341",blue:"#58a6ff",blueBright:"#79c0ff",magenta:"#bc8cff",magentaBright:"#d2a8ff",cyan:"#76e3ea",cyanBright:"#b3f0ff"}},shadows:{shadow:{small:"0 0 rgba(0,0,0,0)",highlight:"0 0 rgba(0,0,0,0)",inset:"0 0 rgba(0,0,0,0)"},autocomplete:{shadow:"0 16px 32px rgba(1,4,9,0.85)"},btn:{shadow:"0 0 rgba(0,0,0,0)",insetShadow:"0 0 rgba(0,0,0,0)",focusShadow:"0 0 0 3px rgba(139,148,158,0.3)",shadowActive:"inset 0 0.15em 0.3em rgba(1,4,9,0.15)",shadowInputFocus:"0 0 0 0.2em rgba(31,111,235,0.3)",primary:{shadow:"0 0 rgba(0,0,0,0)",insetShadow:"0 0 rgba(0,0,0,0)",selectedShadow:"0 0 rgba(0,0,0,0)",focusShadow:"0 0 0 3px rgba(46,164,79,0.4)"},outline:{hoverShadow:"0 1px 0 rgba(1,4,9,0.1)",hoverInsetShadow:"inset 0 1px 0 rgba(240,246,252,0.03)",selectedShadow:"0 0 rgba(0,0,0,0)",focusShadow:"0 0 0 3px rgba(17,88,199,0.4)"},danger:{hoverShadow:"0 0 rgba(0,0,0,0)",hoverInsetShadow:"0 0 rgba(0,0,0,0)",selectedShadow:"0 0 rgba(0,0,0,0)",focusShadow:"0 0 0 3px rgba(182,35,36,0.4)"}},dropdown:{shadow:"0 16px 32px rgba(1,4,9,0.85)"},input:{shadow:"0 0 rgba(0,0,0,0)"},selectMenu:{shadow:"0 0 18px rgba(1,4,9,0.4)"},mktg:{btn:{focusShadow:"0 0 0 3px rgba(31,111,235,0.3)",primary:{focusShadow:"0 0 0 3px rgba(35,134,54,0.3)"},outline:{focusShadow:"0 0 0 3px rgba(56,139,253,0.3)"},dark:{focusShadow:"0 0 0 3px rgba(201,209,217,0.3)"}}}}},dark_dimmed:{colors:{auto:{black:"#cdd9e5",white:"#1c2128",gray:["#22272e","#2d333b","#373e47","#444c56","#545d68","#636e7b","#768390","#909dab","#adbac7","#cdd9e5"],blue:["#0f2d5c","#143d79","#1b4b91","#255ab2","#316dca","#4184e4","#539bf5","#6cb6ff","#96d0ff","#c6e6ff"],green:["#113417","#1b4721","#245829","#2b6a30","#347d39","#46954a","#57ab5a","#6bc46d","#8ddb8c","#b4f1b4"],yellow:["#452700","#593600","#6c4400","#805400","#966600","#ae7c14","#c69026","#daaa3f","#eac55f","#fbe090"],orange:["#4d210c","#682d0f","#7f3913","#94471b","#ae5622","#cc6b2c","#e0823d","#f69d50","#ffbc6f","#ffddb0"],red:["#78191b","#78191b","#922323","#ad2e2c","#c93c37","#e5534b","#f47067","#ff938a","#ffb8b0","#ffd8d3"],purple:["#352160","#472c82","#5936a2","#6b44bc","#8256d0","#986ee2","#b083f0","#dcbdfb","#dcbdfb","#eedcff"],pink:["#551639","#69264a","#7e325a","#983b6e","#ae4c82","#c96198","#e275ad","#fc8dc7","#ffb3d8","#ffd7eb"]},text:{primary:"#adbac7",secondary:"#768390",tertiary:"#768390",placeholder:"#545d68",disabled:"#545d68",inverse:"#22272e",link:"#539bf5",danger:"#e5534b",success:"#6bc46d",warning:"#daaa3f",white:"#cdd9e5"},icon:{primary:"#adbac7",secondary:"#636e7b",tertiary:"#545d68",info:"#6cb6ff",danger:"#e5534b",success:"#6bc46d",warning:"#daaa3f"},border:{primary:"#444c56",secondary:"#373e47",tertiary:"#636e7b",overlay:"#444c56",inverse:"#cdd9e5",info:"rgba(65,132,228,0.4)",danger:"rgba(229,83,75,0.4)",success:"rgba(87,171,90,0.4)",warning:"rgba(174,124,20,0.4)"},bg:{canvas:"#22272e",canvasMobile:"#1c2128",canvasInverse:"#cdd9e5",canvasInset:"#1e2228",primary:"#22272e",secondary:"#22272e",tertiary:"#2d333b",overlay:"#373e47",backdrop:"rgba(28,33,40,0.8)",info:"rgba(65,132,228,0.1)",infoInverse:"#4184e4",danger:"rgba(229,83,75,0.1)",dangerInverse:"#c93c37",success:"rgba(70,149,74,0.1)",successInverse:"#46954a",warning:"rgba(174,124,20,0.1)",warningInverse:"#ae7c14"},state:{hover:{primaryBg:"#316dca",primaryBorder:"#4184e4",primaryText:"#cdd9e5",primaryIcon:"#cdd9e5",secondaryBg:"#2d333b",secondaryBorder:"#2d333b"},selected:{primaryBg:"#316dca",primaryBorder:"#4184e4",primaryText:"#cdd9e5",primaryIcon:"#cdd9e5"},focus:{border:"#4184e4"}},fade:{fg10:"rgba(205,217,229,0.1)",fg15:"rgba(205,217,229,0.15)",fg30:"rgba(205,217,229,0.3)",fg50:"rgba(205,217,229,0.5)",fg70:"rgba(205,217,229,0.7)",fg85:"rgba(205,217,229,0.85)",black10:"rgba(28,33,40,0.1)",black15:"rgba(28,33,40,0.15)",black30:"rgba(28,33,40,0.3)",black50:"rgba(28,33,40,0.5)",black70:"rgba(28,33,40,0.7)",black85:"rgba(28,33,40,0.85)",white10:"rgba(205,217,229,0.1)",white15:"rgba(205,217,229,0.15)",white30:"rgba(205,217,229,0.3)",white50:"rgba(205,217,229,0.5)",white70:"rgba(205,217,229,0.7)",white85:"rgba(205,217,229,0.85)"},alert:{info:{text:"#6cb6ff",icon:"#6cb6ff",bg:"rgba(65,132,228,0.1)",border:"rgba(65,132,228,0.4)"},warn:{text:"#daaa3f",icon:"#daaa3f",bg:"rgba(174,124,20,0.1)",border:"rgba(174,124,20,0.4)"},error:{text:"#f47067",icon:"#f47067",bg:"rgba(229,83,75,0.1)",border:"rgba(229,83,75,0.4)"},success:{text:"#6bc46d",icon:"#6bc46d",bg:"rgba(70,149,74,0.1)",border:"rgba(70,149,74,0.4)"}},autocomplete:{rowBorder:"#444c56"},blankslate:{icon:"#5f6a76"},btn:{text:"#adbac7",bg:"#373e47",border:"#444c56",hoverBg:"#444c56",hoverBorder:"#768390",selectedBg:"#2d333b",focusBg:"#373e47",focusBorder:"#768390",primary:{text:"#ffffff",bg:"#347d39",border:"#46954a",hoverBg:"#46954a",hoverBorder:"#57ab5a",selectedBg:"#347d39",disabledText:"rgba(205,217,229,0.5)",disabledBg:"rgba(52,125,57,0.6)",disabledBorder:"rgba(0,0,0,0)",focusBg:"#347d39",focusBorder:"#57ab5a",icon:"#cdd9e5",counterBg:"rgba(205,217,229,0.2)"},outline:{text:"#539bf5",hoverText:"#539bf5",hoverBg:"#444c56",hoverBorder:"#539bf5",hoverCounterBg:"rgba(205,217,229,0.2)",selectedText:"#cdd9e5",selectedBg:"#1b4b91",selectedBorder:"rgba(205,217,229,0.1)",disabledText:"rgba(83,155,245,0.5)",disabledBg:"#22272e",disabledCounterBg:"rgba(49,109,202,0.05)",focusBorder:"#539bf5",counterBg:"rgba(49,109,202,0.1)"},danger:{text:"#e5534b",hoverText:"#ffffff",hoverBg:"#c93c37",hoverBorder:"#e5534b",hoverCounterBg:"rgba(255,255,255,0.2)",selectedText:"#ffffff",selectedBg:"#ad2e2c",selectedBorder:"rgba(205,217,229,0.1)",disabledText:"rgba(229,83,75,0.5)",disabledBg:"#22272e",disabledCounterBg:"rgba(201,60,55,0.05)",focusBorder:"#e5534b",counterBg:"rgba(201,60,55,0.1)",icon:"#e5534b",hoverIcon:"#cdd9e5"},counterBg:"#444c56"},counter:{text:"#adbac7",bg:"#444c56",primary:{text:"#adbac7",bg:"#636e7b"},secondary:{text:"#768390"}},label:{border:"#444c56",primary:{text:"#909dab",border:"#636e7b"},secondary:{text:"#768390",border:"#444c56"},info:{text:"#4184e4",border:"rgba(65,132,228,0.4)"},success:{text:"#57ab5a",border:"rgba(70,149,74,0.4)"},warning:{text:"#daaa3f",border:"rgba(242,211,91,0.4)"},danger:{text:"#e5534b",border:"rgba(229,83,75,0.4)"},orange:{text:"#cc6b2c",border:"rgba(204,107,44,0.4)"}},input:{bg:"#22272e",contrastBg:"rgba(28,33,40,0.5)",border:"#373e47",disabledBg:"#2d333b",disabledBorder:"#444c56",warningBorder:"#c69026",errorBorder:"#e5534b",tooltip:{success:{text:"#6bc46d",bg:"#263231",border:"#335a3b"},warning:{text:"#daaa3f",bg:"#30302b",border:"#624e22"},error:{text:"#f47067",bg:"#362b31",border:"#7c3b3b"}}},avatar:{bg:"rgba(205,217,229,0.1)",border:"rgba(205,217,229,0.1)",stackFade:"#444c56",stackFadeMore:"#373e47"},toast:{text:"#6cb6ff",bg:"#1c2128",border:"#214981",icon:"#6cb6ff",iconBg:"#121d2e",iconBorder:"#214981",success:{text:"#6bc46d",border:"#1c532c",icon:"#6bc46d",iconBg:"#111f1c",iconBorder:"#1c532c"},warning:{text:"#daaa3f",border:"#5e4411",icon:"#daaa3f",iconBg:"#1f1c16",iconBorder:"#5e4411"},danger:{text:"#f47067",border:"#792e2e",icon:"#f47067",iconBg:"#25171c",iconBorder:"#792e2e"},loading:{text:"#adbac7",border:"#444c56",icon:"#cdd9e5",iconBg:"#444c56",iconBorder:"#444c56"}},timeline:{text:"#909dab",badgeBg:"#22272e",targetBadgeBorder:"#316dca",targetBadgeShadow:"#1b4b91"},selectMenu:{borderSecondary:"#444c56",backdropBg:"rgba(28,33,40,0.5)",backdropBorder:"#545d68",tapHighlight:"rgba(68,76,86,0.5)",tapFocusBg:"#143d79"},box:{blueBorder:"#1b4b91",rowYellowBg:"rgba(235,196,64,0.1)",rowBlueBg:"rgba(108,182,255,0.1)",headerBlueBg:"#22272e",headerBlueBorder:"#444c56",borderInfo:"rgba(65,132,228,0.4)",bgInfo:"rgba(65,132,228,0.1)",borderWarning:"rgba(174,124,20,0.4)",bgWarning:"rgba(174,124,20,0.1)"},branchName:{text:"#adbac7",icon:"#909dab",bg:"rgba(83,155,245,0.1)",link:{text:"#539bf5",icon:"#539bf5",bg:"rgba(83,155,245,0.1)"}},markdown:{codeBg:"rgba(205,217,229,0.15)",frameBorder:"#4f5964",blockquoteBorder:"#4f5964",tableBorder:"#4f5964",tableTrBorder:"#3b424b"},menu:{headingText:"#768390",borderActive:"#f78166",bgActive:"#2d333b"},sidenav:{selectedBg:"#373e47",borderActive:"#f78166"},header:{text:"rgba(205,217,229,0.7)",bg:"#2d333b",logo:"#cdd9e5"},filterItem:{barBg:"#292e35"},hiddenTextExpander:{bg:"#373e47",bgHover:"#444c56"},dragAndDrop:{border:"#393f48"},uploadEnabled:{border:"#4f5964",borderFocused:"#4f84d4"},previewableCommentForm:{border:"#393f48"},underlinenav:{border:"rgba(68,76,86,0)",borderHover:"#444c56",borderActive:"#f78166",text:"#768390",textHover:"#adbac7",textActive:"#adbac7",icon:"#636e7b",iconHover:"#adbac7",iconActive:"#adbac7",counterText:"#768390"},verifiedBadge:{text:"#57ab5a",bg:"rgba(87,171,90,0.1)",border:"rgba(87,171,90,0.4)"},socialCount:{bg:"#373e47"},tooltip:{text:"#cdd9e5",bg:"#373e47"},headerSearch:{bg:"#22272e",border:"#373e47"},searchKeyword:{hl:"rgba(174,124,20,0.4)"},diffstat:{neutralBg:"#444c56",neutralBorder:"rgba(205,217,229,0.1)",deletionBg:"#c93c37",deletionBorder:"#e5534b",additionBg:"#347d39",additionBorder:"#46954a"},mktg:{success:"#3d8942",info:"#3877d5",bgShadeGradient:{top:"rgba(28,33,40,0.065)",bottom:"rgba(28,33,40,0)"},btn:{bg:"#316dca",border:"#316dca",text:"#cdd9e5",icon:"#cdd9e5",hover:{bg:"#4184e4",border:"#4184e4"},disabled:{bg:"rgba(65,132,228,0.5)",border:"rgba(0,0,0,0)",text:"rgba(205,217,229,0.5)",icon:"rgba(205,217,229,0.5)"},primary:{bg:"#347d39",border:"#347d39",text:"#cdd9e5",icon:"#cdd9e5",hover:{bg:"#46954a",border:"#46954a"},disabled:{bg:"rgba(70,149,74,0.5)",border:"rgba(0,0,0,0)",text:"rgba(205,217,229,0.5)",icon:"rgba(205,217,229,0.5)"}},outline:{bg:"rgba(0,0,0,0)",border:"rgba(65,132,228,0.5)",text:"#4184e4",icon:"#4184e4",hover:{bg:"rgba(0,0,0,0)",border:"#4184e4",text:"#539bf5",icon:"#539bf5"},disabled:{bg:"rgba(0,0,0,0)",border:"rgba(65,132,228,0.2)",text:"rgba(65,132,228,0.5)",icon:"rgba(65,132,228,0.5)"}},dark:{bg:"rgba(0,0,0,0)",border:"rgba(173,186,199,0.5)",text:"#adbac7",icon:"#adbac7",hover:{bg:"rgba(173,186,199,0.5)",border:"rgba(173,186,199,0.5)",text:"#22272e",icon:"#22272e"},disabled:{bg:"rgba(0,0,0,0)",border:"rgba(173,186,199,0.2)",text:"rgba(173,186,199,0.5)",icon:"rgba(173,186,199,0.5)"}}}},filesExplorerIcon:"#636e7b",hlAuthorBg:"#0f2d5c",hlAuthorBorder:"#1b4b91",logoSubdued:"#444c56",discussionBorder:"#494c49",discussionBgSuccess:"rgba(70,149,74,0.1)",actionsWorkflowTableStickyBg:"rgba(34,39,46,0.95)",repoLanguageColorBorder:"rgba(205,217,229,0.2)",codeSelectionBg:"rgba(108,182,255,0.3)",highlight:{text:"#ffd467",bg:"rgba(204,143,44,0.38)"},blob:{lineHighlightBg:"rgba(198,144,38,0.15)",lineHighlightBorder:"#daaa3f"},diff:{addition:{text:"#6bc46d",bg:"rgba(70,149,74,0.2)",border:"#2b6a30"},deletion:{text:"#e5534b",bg:"rgba(201,60,55,0.2)",border:"#ad2e2c"},change:{text:"#daaa3f",bg:"#452700",border:"#966600"}},diffBlob:{numText:"rgba(205,217,229,0.3)",numHoverText:"rgba(205,217,229,0.6)",addition:{numText:"#57ab5a",numHoverText:"#8ddb8c",numBg:"rgba(70,149,74,0.1)",lineBg:"rgba(70,149,74,0.2)",wordBg:"rgba(70,149,74,0.55)"},deletion:{numText:"#e5534b",numHoverText:"#ff938a",numBg:"rgba(201,60,55,0.1)",lineBg:"rgba(201,60,55,0.2)",wordBg:"rgba(201,60,55,0.5)"},hunk:{text:"#768390",numBg:"rgba(83,155,245,0.15)",lineBg:"rgba(83,155,245,0.1)"},emptyBlockBg:"#2d333b",selectedLineHighlightBg:"rgba(174,124,20,0.1)",selectedLineHighlightBorder:"#ae7c14",expander:{icon:"#768390",hoverIcon:"#cdd9e5",hoverBg:"#316dca"},commentButton:{icon:"#cdd9e5",bg:"#316dca",gradientBg:"#437bd1"}},globalNav:{logo:"#cdd9e5",bg:"#2d333b",text:"#adbac7",icon:"#adbac7",inputBg:"#22272e",inputBorder:"#373e47",inputIcon:"#373e47",inputPlaceholder:"#545d68"},calendarGraph:{dayBg:"#2d333b",dayBorder:"rgba(27,31,35,0.06)",dayL1Bg:"#003820",dayL2Bg:"#00602d",dayL3Bg:"#10983d",dayL4Bg:"#27d545",dayL4Border:"rgba(255,255,255,0.05)",dayL3Border:"rgba(255,255,255,0.05)",dayL2Border:"rgba(255,255,255,0.05)",dayL1Border:"rgba(255,255,255,0.05)"},footerInvertocat:{octicon:"#444c56",octiconHover:"#636e7b"},prState:{draft:{text:"#768390",bg:"rgba(118,131,144,0.1)",border:"rgba(118,131,144,0.4)"},open:{text:"#57ab5a",bg:"rgba(87,171,90,0.1)",border:"rgba(87,171,90,0.4)"},merged:{text:"#986ee2",bg:"rgba(176,131,240,0.1)",border:"rgba(176,131,240,0.4)"},closed:{text:"#e5534b",bg:"rgba(201,60,55,0.1)",border:"rgba(201,60,55,0.4)"}},topicTag:{text:"#539bf5",bg:"rgba(65,132,228,0.1)",hoverBg:"rgba(65,132,228,0.2)",activeBg:"rgba(65,132,228,0.15)"},mergeBox:{successIconBg:"rgba(70,149,74,0.1)",successIconText:"#57ab5a",successIconBorder:"rgba(70,149,74,0.4)",successIndicatorBg:"#347d39",successIndicatorBorder:"#46954a",mergedIconBg:"rgba(176,131,240,0.1)",mergedIconText:"#986ee2",mergedIconBorder:"rgba(176,131,240,0.4)",mergedBoxBorder:"rgba(176,131,240,0.4)",neutralIconBg:"rgba(173,186,199,0.1)",neutralIconText:"#768390",neutralIconBorder:"rgba(173,186,199,0.4)",neutralIndicatorBg:"#545d68",neutralIndicatorBorder:"#636e7b",warningIconBg:"rgba(174,124,20,0.1)",warningIconText:"#daaa3f",warningIconBorder:"rgba(174,124,20,0.4)",warningBoxBorder:"rgba(174,124,20,0.4)",warningMergeHighlight:"rgba(174,124,20,0.1)",errorIconBg:"rgba(229,83,75,0.1)",errorIconText:"#e5534b",errorIconBorder:"rgba(229,83,75,0.4)",errorIndicatorBg:"#c93c37",errorIndicatorBorder:"#e5534b"},project:{cardBg:"#2d333b",headerBg:"#22272e",sidebarBg:"#2d333b",gradientIn:"#2d333b",gradientOut:"rgba(45,51,59,0)"},checks:{bg:"#1e2228",textPrimary:"#adbac7",textSecondary:"#768390",textLink:"#539bf5",btnIcon:"#636e7b",btnHoverIcon:"#adbac7",btnHoverBg:"#444c56",inputText:"#768390",inputPlaceholderText:"#545d68",inputFocusText:"#adbac7",inputBg:"#22272e",dropdownText:"#adbac7",dropdownBg:"#373e47",dropdownBorder:"#444c56",dropdownHoverText:"#cdd9e5",dropdownHoverBg:"#316dca",dropdownBtnHoverText:"#cdd9e5",dropdownBtnHoverBg:"#2d333b",scrollbarThumbBg:"#444c56",headerLabelText:"#768390",headerLabelOpenText:"#adbac7",headerBorder:"#373e47",headerIcon:"#636e7b",lineText:"#768390",lineNumText:"#768390",lineTimestampText:"#768390",lineHoverBg:"#2d333b",lineSelectedBg:"rgba(65,132,228,0.1)",lineSelectedNumText:"#539bf5",lineDtFmText:"#22272e",lineDtFmBg:"#c69026",gateBg:"rgba(128,84,0,0.15)",gateText:"#768390",gateWaitingText:"#daaa3f",stepHeaderOpenBg:"#2d333b",stepErrorText:"#e5534b",stepWarningText:"#daaa3f",loglineText:"#636e7b",loglineNumText:"#768390",loglineDebugText:"#b083f0",loglineErrorText:"#768390",loglineErrorNumText:"#768390",loglineErrorBg:"rgba(229,83,75,0.1)",loglineWarningText:"#768390",loglineWarningNumText:"#daaa3f",loglineWarningBg:"rgba(174,124,20,0.1)",loglineCommandText:"#539bf5",loglineSectionText:"#6bc46d"},introShelf:{gradientLeft:"rgba(65,132,228,0.1)",gradientRight:"rgba(70,149,74,0.1)",gradientIn:"#22272e",gradientOut:"rgba(34,39,46,0)"},marketingIcon:{primary:"#6cb6ff",secondary:"#316dca"},prettylights:{syntax:{comment:"#768390",constant:"#6cb6ff",entity:"#dcbdfb",storageModifierImport:"#adbac7",entityTag:"#8ddb8c",keyword:"#f47067",string:"#96d0ff",variable:"#f69d50",brackethighlighterUnmatched:"#e5534b",invalidIllegalText:"#cdd9e5",invalidIllegalBg:"#922323",carriageReturnText:"#cdd9e5",carriageReturnBg:"#ad2e2c",stringRegexp:"#8ddb8c",markupList:"#eac55f",markupHeading:"#316dca",markupItalic:"#adbac7",markupBold:"#adbac7",markupDeletedText:"#ffd8d3",markupDeletedBg:"#78191b",markupInsertedText:"#b4f1b4",markupInsertedBg:"#1b4721",markupChangedText:"#ffddb0",markupChangedBg:"#682d0f",markupIgnoredText:"#adbac7",markupIgnoredBg:"#255ab2",metaDiffRange:"#dcbdfb",brackethighlighterAngle:"#768390",sublimelinterGutterMark:"#545d68",constantOtherReferenceLink:"#96d0ff"}},codemirror:{text:"#adbac7",bg:"#22272e",guttersBg:"#22272e",guttermarkerText:"#22272e",guttermarkerSubtleText:"#636e7b",linenumberText:"#768390",cursor:"#cdd9e5",selectionBg:"rgba(108,182,255,0.3)",activelineBg:"#2d333b",matchingbracketText:"#adbac7",linesBg:"#22272e",syntax:{comment:"#768390",constant:"#6cb6ff",entity:"#dcbdfb",keyword:"#f47067",storage:"#f47067",string:"#96d0ff",support:"#6cb6ff",variable:"#f69d50"}},ansi:{black:"#22272e",blackBright:"#2d333b",white:"#909dab",whiteBright:"#909dab",gray:"#636e7b",red:"#f47067",redBright:"#ff938a",green:"#57ab5a",greenBright:"#6bc46d",yellow:"#c69026",yellowBright:"#daaa3f",blue:"#539bf5",blueBright:"#6cb6ff",magenta:"#b083f0",magentaBright:"#dcbdfb",cyan:"#76e3ea",cyanBright:"#b3f0ff"}},shadows:{shadow:{small:"0 0 rgba(0,0,0,0)",highlight:"0 0 rgba(0,0,0,0)",inset:"0 0 rgba(0,0,0,0)"},autocomplete:{shadow:"0 16px 32px rgba(28,33,40,0.85)"},btn:{shadow:"0 0 rgba(0,0,0,0)",insetShadow:"0 0 rgba(0,0,0,0)",focusShadow:"0 0 0 3px rgba(118,131,144,0.3)",shadowActive:"inset 0 0.15em 0.3em rgba(28,33,40,0.15)",shadowInputFocus:"0 0 0 0.2em rgba(49,109,202,0.3)",primary:{shadow:"0 0 rgba(0,0,0,0)",insetShadow:"0 0 rgba(0,0,0,0)",selectedShadow:"0 0 rgba(0,0,0,0)",focusShadow:"0 0 0 3px rgba(46,164,79,0.4)"},outline:{hoverShadow:"0 1px 0 rgba(28,33,40,0.1)",hoverInsetShadow:"inset 0 1px 0 rgba(205,217,229,0.03)",selectedShadow:"0 0 rgba(0,0,0,0)",focusShadow:"0 0 0 3px rgba(37,90,178,0.4)"},danger:{hoverShadow:"0 0 rgba(0,0,0,0)",hoverInsetShadow:"0 0 rgba(0,0,0,0)",selectedShadow:"0 0 rgba(0,0,0,0)",focusShadow:"0 0 0 3px rgba(173,46,44,0.4)"}},dropdown:{shadow:"0 16px 32px rgba(28,33,40,0.85)"},input:{shadow:"0 0 rgba(0,0,0,0)"},selectMenu:{shadow:"0 0 18px rgba(28,33,40,0.4)"},mktg:{btn:{focusShadow:"0 0 0 3px rgba(49,109,202,0.3)",primary:{focusShadow:"0 0 0 3px rgba(52,125,57,0.3)"},outline:{focusShadow:"0 0 0 3px rgba(65,132,228,0.3)"},dark:{focusShadow:"0 0 0 3px rgba(173,186,199,0.3)"}}}}}}}},x=Object.getOwnPropertySymbols,v=Object.prototype.hasOwnProperty,y=Object.prototype.propertyIsEnumerable;
/*
object-assign
(c) Sindre Sorhus
@license MIT
*/function w(e){if(null==e)throw new TypeError("Object.assign cannot be called with null or undefined");return Object(e)}var B=function(){try{if(!Object.assign)return!1;var e=new String("abc");if(e[5]="de","5"===Object.getOwnPropertyNames(e)[0])return!1;for(var r={},t=0;t<10;t++)r["_"+String.fromCharCode(t)]=t;if("0123456789"!==Object.getOwnPropertyNames(r).map((function(e){return r[e]})).join(""))return!1;var o={};return"abcdefghijklmnopqrst".split("").forEach((function(e){o[e]=e})),"abcdefghijklmnopqrst"===Object.keys(Object.assign({},o)).join("")}catch(e){return!1}}()?Object.assign:function(e,r){for(var t,o,a=w(e),n=1;n<arguments.length;n++){for(var i in t=Object(arguments[n]))v.call(t,i)&&(a[i]=t[i]);if(x){o=x(t);for(var d=0;d<o.length;d++)y.call(t,o[d])&&(a[o[d]]=t[o[d]])}}return a},$=function(e,r){var t=B({},e,r);for(var o in e){var a;e[o]&&"object"==typeof r[o]&&B(t,((a={})[o]=B(e[o],r[o]),a))}return t},k={breakpoints:[40,52,64].map((function(e){return e+"em"}))},S=function(e){return"@media screen and (min-width: "+e+")"},T=function(e,r){return E(r,e,e)},E=function(e,r,t,o,a){for(r=r&&r.split?r.split("."):[r],o=0;o<r.length;o++)e=e?e[r[o]]:a;return e===a?t:e},C=function e(r){var t={},o=function(e){var o,a,n={},i=!1,d=e.theme&&e.theme.disableStyledSystemCache;for(var l in e)if(r[l]){var c=r[l],s=e[l],f=E(e.theme,c.scale,c.defaults);if("object"!=typeof s)B(n,c(s,f,e));else{if(t.breakpoints=!d&&t.breakpoints||E(e.theme,"breakpoints",k.breakpoints),Array.isArray(s)){t.media=!d&&t.media||[null].concat(t.breakpoints.map(S)),n=$(n,I(t.media,c,f,s,e));continue}null!==s&&(n=$(n,N(t.breakpoints,c,f,s,e)),i=!0)}}return i&&(o=n,a={},Object.keys(o).sort((function(e,r){return e.localeCompare(r,void 0,{numeric:!0,sensitivity:"base"})})).forEach((function(e){a[e]=o[e]})),n=a),n};o.config=r,o.propNames=Object.keys(r),o.cache=t;var a=Object.keys(r).filter((function(e){return"config"!==e}));return a.length>1&&a.forEach((function(t){var a;o[t]=e(((a={})[t]=r[t],a))})),o},I=function(e,r,t,o,a){var n={};return o.slice(0,e.length).forEach((function(o,i){var d,l=e[i],c=r(o,t,a);l?B(n,((d={})[l]=B({},n[l],c),d)):B(n,c)})),n},N=function(e,r,t,o,a){var n={};for(var i in o){var d=e[i],l=r(o[i],t,a);if(d){var c,s=S(d);B(n,((c={})[s]=B({},n[s],l),c))}else B(n,l)}return n},A=function(e){var r=e.properties,t=e.property,o=e.scale,a=e.transform,n=void 0===a?T:a,i=e.defaultScale;r=r||[t];var d=function(e,t,o){var a={},i=n(e,t,o);if(null!==i)return r.forEach((function(e){a[e]=i})),a};return d.scale=o,d.defaults=i,d},z=function(e){void 0===e&&(e={});var r={};return Object.keys(e).forEach((function(t){var o=e[t];r[t]=!0!==o?"function"!=typeof o?A(o):o:A({property:t,scale:t})})),C(r)},L=function(){for(var e={},r=arguments.length,t=new Array(r),o=0;o<r;o++)t[o]=arguments[o];t.forEach((function(r){r&&r.config&&B(e,r.config)}));var a=C(e);return a},R=z({width:{property:"width",scale:"sizes",transform:function(e,r){return E(r,e,!function(e){return"number"==typeof e&&!isNaN(e)}(e)||e>1?e:100*e+"%")}},height:{property:"height",scale:"sizes"},minWidth:{property:"minWidth",scale:"sizes"},minHeight:{property:"minHeight",scale:"sizes"},maxWidth:{property:"maxWidth",scale:"sizes"},maxHeight:{property:"maxHeight",scale:"sizes"},size:{properties:["width","height"],scale:"sizes"},overflow:!0,overflowX:!0,overflowY:!0,display:!0,verticalAlign:!0}),M={color:{property:"color",scale:"colors"},backgroundColor:{property:"backgroundColor",scale:"colors"},opacity:!0};M.bg=M.backgroundColor;var H=z(M),O=z({fontFamily:{property:"fontFamily",scale:"fonts"},fontSize:{property:"fontSize",scale:"fontSizes",defaultScale:[12,14,16,20,24,32,48,64,72]},fontWeight:{property:"fontWeight",scale:"fontWeights"},lineHeight:{property:"lineHeight",scale:"lineHeights"},letterSpacing:{property:"letterSpacing",scale:"letterSpacings"},textAlign:!0,fontStyle:!0}),P=z({alignItems:!0,alignContent:!0,justifyItems:!0,justifyContent:!0,flexWrap:!0,flexDirection:!0,flex:!0,flexGrow:!0,flexShrink:!0,flexBasis:!0,justifySelf:!0,alignSelf:!0,order:!0}),D={space:[0,4,8,16,32,64,128,256,512]},W=z({gridGap:{property:"gridGap",scale:"space",defaultScale:D.space},gridColumnGap:{property:"gridColumnGap",scale:"space",defaultScale:D.space},gridRowGap:{property:"gridRowGap",scale:"space",defaultScale:D.space},gridColumn:!0,gridRow:!0,gridAutoFlow:!0,gridAutoColumns:!0,gridAutoRows:!0,gridTemplateColumns:!0,gridTemplateRows:!0,gridTemplateAreas:!0,gridArea:!0}),j={border:{property:"border",scale:"borders"},borderWidth:{property:"borderWidth",scale:"borderWidths"},borderStyle:{property:"borderStyle",scale:"borderStyles"},borderColor:{property:"borderColor",scale:"colors"},borderRadius:{property:"borderRadius",scale:"radii"},borderTop:{property:"borderTop",scale:"borders"},borderTopLeftRadius:{property:"borderTopLeftRadius",scale:"radii"},borderTopRightRadius:{property:"borderTopRightRadius",scale:"radii"},borderRight:{property:"borderRight",scale:"borders"},borderBottom:{property:"borderBottom",scale:"borders"},borderBottomLeftRadius:{property:"borderBottomLeftRadius",scale:"radii"},borderBottomRightRadius:{property:"borderBottomRightRadius",scale:"radii"},borderLeft:{property:"borderLeft",scale:"borders"},borderX:{properties:["borderLeft","borderRight"],scale:"borders"},borderY:{properties:["borderTop","borderBottom"],scale:"borders"},borderTopWidth:{property:"borderTopWidth",scale:"borderWidths"},borderTopColor:{property:"borderTopColor",scale:"colors"},borderTopStyle:{property:"borderTopStyle",scale:"borderStyles"}};j.borderTopLeftRadius={property:"borderTopLeftRadius",scale:"radii"},j.borderTopRightRadius={property:"borderTopRightRadius",scale:"radii"},j.borderBottomWidth={property:"borderBottomWidth",scale:"borderWidths"},j.borderBottomColor={property:"borderBottomColor",scale:"colors"},j.borderBottomStyle={property:"borderBottomStyle",scale:"borderStyles"},j.borderBottomLeftRadius={property:"borderBottomLeftRadius",scale:"radii"},j.borderBottomRightRadius={property:"borderBottomRightRadius",scale:"radii"},j.borderLeftWidth={property:"borderLeftWidth",scale:"borderWidths"},j.borderLeftColor={property:"borderLeftColor",scale:"colors"},j.borderLeftStyle={property:"borderLeftStyle",scale:"borderStyles"},j.borderRightWidth={property:"borderRightWidth",scale:"borderWidths"},j.borderRightColor={property:"borderRightColor",scale:"colors"},j.borderRightStyle={property:"borderRightStyle",scale:"borderStyles"};var F=z(j),_={background:!0,backgroundImage:!0,backgroundSize:!0,backgroundPosition:!0,backgroundRepeat:!0};_.bgImage=_.backgroundImage,_.bgSize=_.backgroundSize,_.bgPosition=_.backgroundPosition,_.bgRepeat=_.backgroundRepeat;var U=z(_),G={space:[0,4,8,16,32,64,128,256,512]},K=z({position:!0,zIndex:{property:"zIndex",scale:"zIndices"},top:{property:"top",scale:"space",defaultScale:G.space},right:{property:"right",scale:"space",defaultScale:G.space},bottom:{property:"bottom",scale:"space",defaultScale:G.space},left:{property:"left",scale:"space",defaultScale:G.space}}),V={space:[0,4,8,16,32,64,128,256,512]},X=function(e){return"number"==typeof e&&!isNaN(e)},Y=function(e,r){if(!X(e))return E(r,e,e);var t=e<0,o=Math.abs(e),a=E(r,o,o);return X(a)?a*(t?-1:1):t?"-"+a:a},q={};q.margin={margin:{property:"margin",scale:"space",transform:Y,defaultScale:V.space},marginTop:{property:"marginTop",scale:"space",transform:Y,defaultScale:V.space},marginRight:{property:"marginRight",scale:"space",transform:Y,defaultScale:V.space},marginBottom:{property:"marginBottom",scale:"space",transform:Y,defaultScale:V.space},marginLeft:{property:"marginLeft",scale:"space",transform:Y,defaultScale:V.space},marginX:{properties:["marginLeft","marginRight"],scale:"space",transform:Y,defaultScale:V.space},marginY:{properties:["marginTop","marginBottom"],scale:"space",transform:Y,defaultScale:V.space}},q.margin.m=q.margin.margin,q.margin.mt=q.margin.marginTop,q.margin.mr=q.margin.marginRight,q.margin.mb=q.margin.marginBottom,q.margin.ml=q.margin.marginLeft,q.margin.mx=q.margin.marginX,q.margin.my=q.margin.marginY,q.padding={padding:{property:"padding",scale:"space",defaultScale:V.space},paddingTop:{property:"paddingTop",scale:"space",defaultScale:V.space},paddingRight:{property:"paddingRight",scale:"space",defaultScale:V.space},paddingBottom:{property:"paddingBottom",scale:"space",defaultScale:V.space},paddingLeft:{property:"paddingLeft",scale:"space",defaultScale:V.space},paddingX:{properties:["paddingLeft","paddingRight"],scale:"space",defaultScale:V.space},paddingY:{properties:["paddingTop","paddingBottom"],scale:"space",defaultScale:V.space}},q.padding.p=q.padding.padding,q.padding.pt=q.padding.paddingTop,q.padding.pr=q.padding.paddingRight,q.padding.pb=q.padding.paddingBottom,q.padding.pl=q.padding.paddingLeft,q.padding.px=q.padding.paddingX,q.padding.py=q.padding.paddingY;var J=z(q.margin),Z=z(q.padding),Q=L(J,Z),ee=z({boxShadow:{property:"boxShadow",scale:"shadows"},textShadow:{property:"textShadow",scale:"shadows"}});function re(){return(re=Object.assign||function(e){for(var r=1;r<arguments.length;r++){var t=arguments[r];for(var o in t)Object.prototype.hasOwnProperty.call(t,o)&&(e[o]=t[o])}return e}).apply(this,arguments)}var te=function(e,r,t,o,a){for(r=r&&r.split?r.split("."):[r],o=0;o<r.length;o++)e=e?e[r[o]]:a;return e===a?t:e},oe=[40,52,64].map((function(e){return e+"em"})),ae={space:[0,4,8,16,32,64,128,256,512],fontSizes:[12,14,16,20,24,32,48,64,72]},ne={bg:"backgroundColor",m:"margin",mt:"marginTop",mr:"marginRight",mb:"marginBottom",ml:"marginLeft",mx:"marginX",my:"marginY",p:"padding",pt:"paddingTop",pr:"paddingRight",pb:"paddingBottom",pl:"paddingLeft",px:"paddingX",py:"paddingY"},ie={marginX:["marginLeft","marginRight"],marginY:["marginTop","marginBottom"],paddingX:["paddingLeft","paddingRight"],paddingY:["paddingTop","paddingBottom"],size:["width","height"]},de={color:"colors",backgroundColor:"colors",borderColor:"colors",margin:"space",marginTop:"space",marginRight:"space",marginBottom:"space",marginLeft:"space",marginX:"space",marginY:"space",padding:"space",paddingTop:"space",paddingRight:"space",paddingBottom:"space",paddingLeft:"space",paddingX:"space",paddingY:"space",top:"space",right:"space",bottom:"space",left:"space",gridGap:"space",gridColumnGap:"space",gridRowGap:"space",gap:"space",columnGap:"space",rowGap:"space",fontFamily:"fonts",fontSize:"fontSizes",fontWeight:"fontWeights",lineHeight:"lineHeights",letterSpacing:"letterSpacings",border:"borders",borderTop:"borders",borderRight:"borders",borderBottom:"borders",borderLeft:"borders",borderWidth:"borderWidths",borderStyle:"borderStyles",borderRadius:"radii",borderTopRightRadius:"radii",borderTopLeftRadius:"radii",borderBottomRightRadius:"radii",borderBottomLeftRadius:"radii",borderTopWidth:"borderWidths",borderTopColor:"colors",borderTopStyle:"borderStyles",borderBottomWidth:"borderWidths",borderBottomColor:"colors",borderBottomStyle:"borderStyles",borderLeftWidth:"borderWidths",borderLeftColor:"colors",borderLeftStyle:"borderStyles",borderRightWidth:"borderWidths",borderRightColor:"colors",borderRightStyle:"borderStyles",outlineColor:"colors",boxShadow:"shadows",textShadow:"shadows",zIndex:"zIndices",width:"sizes",minWidth:"sizes",maxWidth:"sizes",height:"sizes",minHeight:"sizes",maxHeight:"sizes",flexBasis:"sizes",size:"sizes",fill:"colors",stroke:"colors"},le=function(e,r){if("number"!=typeof r||r>=0)return te(e,r,r);var t=Math.abs(r),o=te(e,t,t);return"string"==typeof o?"-"+o:-1*o},ce=["margin","marginTop","marginRight","marginBottom","marginLeft","marginX","marginY","top","bottom","left","right"].reduce((function(e,r){var t;return re({},e,((t={})[r]=le,t))}),{}),se=function e(r){return function(t){void 0===t&&(t={});var o=re({},ae,{},t.theme||t),a={},n=function(e){return function(r){var t={},o=te(r,"breakpoints",oe),a=[null].concat(o.map((function(e){return"@media screen and (min-width: "+e+")"})));for(var n in e){var i="function"==typeof e[n]?e[n](r):e[n];if(null!=i)if(Array.isArray(i))for(var d=0;d<i.slice(0,a.length).length;d++){var l=a[d];l?(t[l]=t[l]||{},null!=i[d]&&(t[l][n]=i[d])):t[n]=i[d]}else t[n]=i}return t}}("function"==typeof r?r(o):r)(o);for(var i in n){var d=n[i],l="function"==typeof d?d(o):d;if("variant"!==i)if(l&&"object"==typeof l)a[i]=e(l)(o);else{var c=te(ne,i,i),s=te(de,c),f=te(o,s,te(o,c,{})),b=te(ce,c,te)(f,l,l);if(ie[c])for(var g=ie[c],p=0;p<g.length;p++)a[g[p]]=b;else a[c]=b}else a=re({},a,{},e(te(o,l))(o))}return a}},fe=function(e){var r,t,o=e.scale,a=e.prop,n=void 0===a?"variant":a,i=e.variants,d=void 0===i?{}:i,l=e.key;(t=Object.keys(d).length?function(e,r,t){return se(E(r,e,null))(t.theme)}:function(e,r){return E(r,e,null)}).scale=o||l,t.defaults=d;var c=((r={})[n]=t,r);return C(c)},be=fe({key:"buttons"}),ge=fe({key:"textStyles",prop:"textStyle"}),pe=fe({key:"colorStyles",prop:"colors"}),ue=R.width,me=R.height,he=R.minWidth,xe=R.minHeight,ve=R.maxWidth,ye=R.maxHeight,we=R.size,Be=R.verticalAlign,$e=R.display,ke=R.overflow,Se=R.overflowX,Te=R.overflowY,Ee=H.opacity,Ce=O.fontSize,Ie=O.fontFamily,Ne=O.fontWeight,Ae=O.lineHeight,ze=O.textAlign,Le=O.fontStyle,Re=O.letterSpacing,Me=P.alignItems,He=P.alignContent,Oe=P.justifyItems,Pe=P.justifyContent,De=P.flexWrap,We=P.flexDirection,je=P.flex,Fe=P.flexGrow,_e=P.flexShrink,Ue=P.flexBasis,Ge=P.justifySelf,Ke=P.alignSelf,Ve=P.order,Xe=W.gridGap,Ye=W.gridColumnGap,qe=W.gridRowGap,Je=W.gridColumn,Ze=W.gridRow,Qe=W.gridAutoFlow,er=W.gridAutoColumns,rr=W.gridAutoRows,tr=W.gridTemplateColumns,or=W.gridTemplateRows,ar=W.gridTemplateAreas,nr=W.gridArea,ir=F.borderWidth,dr=F.borderStyle,lr=F.borderColor,cr=F.borderTop,sr=F.borderRight,fr=F.borderBottom,br=F.borderLeft,gr=F.borderRadius,pr=U.backgroundImage,ur=U.backgroundSize,mr=U.backgroundPosition,hr=U.backgroundRepeat,xr=K.zIndex,vr=K.top,yr=K.right,wr=K.bottom,Br=K.left,$r=function(e){var r=e.prop,t=e.cssProperty,o=e.alias,a=e.key,n=e.transformValue,i=e.scale,d=e.properties,l={};return l[r]=A({properties:d,property:t||r,scale:a,defaultScale:i,transform:n}),o&&(l[o]=l[r]),C(l)},kr=Object.freeze({__proto__:null,width:ue,height:me,minWidth:he,minHeight:xe,maxWidth:ve,maxHeight:ye,size:we,verticalAlign:Be,display:$e,overflow:ke,overflowX:Se,overflowY:Te,opacity:Ee,fontSize:Ce,fontFamily:Ie,fontWeight:Ne,lineHeight:Ae,textAlign:ze,fontStyle:Le,letterSpacing:Re,alignItems:Me,alignContent:He,justifyItems:Oe,justifyContent:Pe,flexWrap:De,flexDirection:We,flex:je,flexGrow:Fe,flexShrink:_e,flexBasis:Ue,justifySelf:Ge,alignSelf:Ke,order:Ve,gridGap:Xe,gridColumnGap:Ye,gridRowGap:qe,gridColumn:Je,gridRow:Ze,gridAutoFlow:Qe,gridAutoColumns:er,gridAutoRows:rr,gridTemplateColumns:tr,gridTemplateRows:or,gridTemplateAreas:ar,gridArea:nr,borderWidth:ir,borderStyle:dr,borderColor:lr,borderTop:cr,borderRight:sr,borderBottom:fr,borderLeft:br,borderRadius:gr,backgroundImage:pr,backgroundSize:ur,backgroundPosition:mr,backgroundRepeat:hr,zIndex:xr,top:vr,right:yr,bottom:wr,left:Br,style:$r,get:E,createParser:C,createStyleFunction:A,compose:L,system:z,margin:J,padding:Z,space:Q,color:H,layout:R,typography:O,flexbox:P,border:F,background:U,position:K,grid:W,shadow:ee,boxShadow:ee,textShadow:ee,variant:fe,buttonStyle:be,textStyle:ge,colorStyle:pe,borders:F});const{get:Sr,compose:Tr,system:Er}=kr,Cr=e=>{return r=e,void 0===(t=Sr(h.theme,e))&&(t=null),function(e){return E(e.theme,r,t)};var r,t},Ir=Tr(Q,H,$e),Nr=Tr(O,Er({whiteSpace:{property:"whiteSpace"}})),Ar=Tr(F,ee),zr=R,Lr=K,Rr=P,Mr=W;function Hr(){return(Hr=Object.assign||function(e){for(var r=1;r<arguments.length;r++){var t=arguments[r];for(var o in t)Object.prototype.hasOwnProperty.call(t,o)&&(e[o]=t[o])}return e}).apply(this,arguments)}const Or=s`
  * { box-sizing: border-box; }
  body { margin: 0; }
  table { border-collapse: collapse; }
  
  [role="button"]:focus:not(:focus-visible):not(.focus-visible),
  [role="tabpanel"][tabindex="0"]:focus:not(:focus-visible):not(.focus-visible),
  button:focus:not(:focus-visible):not(.focus-visible),
  summary:focus:not(:focus-visible):not(.focus-visible),
  a:focus:not(:focus-visible):not(.focus-visible) {
    outline: none;
    box-shadow: none;
  }

  [tabindex="0"]:focus:not(:focus-visible):not(.focus-visible),
  details-dialog:focus:not(:focus-visible):not(.focus-visible) {
    outline: none;
  }
`,Pr=c.div`
  ${Nr};
  ${Ir};
`;function Dr(r){const{children:t,...o}=r;return require("focus-visible"),e.createElement(Pr,Hr({},o,{"data-portal-root":!0}),e.createElement(Or,null),t)}Dr.displayName="BaseStyles",Dr.defaultProps={color:"text.primary",fontFamily:"normal",lineHeight:"default"};var Wr=function(e){return function(e){return!!e&&"object"==typeof e}(e)&&!function(e){var r=Object.prototype.toString.call(e);return"[object RegExp]"===r||"[object Date]"===r||function(e){return e.$$typeof===jr}(e)}(e)};var jr="function"==typeof Symbol&&Symbol.for?Symbol.for("react.element"):60103;function Fr(e,r){return!1!==r.clone&&r.isMergeableObject(e)?Vr((t=e,Array.isArray(t)?[]:{}),e,r):e;var t}function _r(e,r,t){return e.concat(r).map((function(e){return Fr(e,t)}))}function Ur(e){return Object.keys(e).concat(function(e){return Object.getOwnPropertySymbols?Object.getOwnPropertySymbols(e).filter((function(r){return e.propertyIsEnumerable(r)})):[]}(e))}function Gr(e,r){try{return r in e}catch(e){return!1}}function Kr(e,r,t){var o={};return t.isMergeableObject(e)&&Ur(e).forEach((function(r){o[r]=Fr(e[r],t)})),Ur(r).forEach((function(a){(function(e,r){return Gr(e,r)&&!(Object.hasOwnProperty.call(e,r)&&Object.propertyIsEnumerable.call(e,r))})(e,a)||(Gr(e,a)&&t.isMergeableObject(r[a])?o[a]=function(e,r){if(!r.customMerge)return Vr;var t=r.customMerge(e);return"function"==typeof t?t:Vr}(a,t)(e[a],r[a],t):o[a]=Fr(r[a],t))})),o}function Vr(e,r,t){(t=t||{}).arrayMerge=t.arrayMerge||_r,t.isMergeableObject=t.isMergeableObject||Wr,t.cloneUnlessOtherwiseSpecified=Fr;var o=Array.isArray(r);return o===Array.isArray(e)?o?t.arrayMerge(e,r,t):Kr(e,r,t):Fr(r,t)}Vr.all=function(e,r){if(!Array.isArray(e))throw new Error("first argument should be an array");return e.reduce((function(e,t){return Vr(e,t,r)}),{})};var Xr=Vr;const Yr=e.createContext({setColorMode:()=>null,setDayScheme:()=>null,setNightScheme:()=>null}),qr=({children:r,...t})=>{var o,a,n,i,d,l,c,s;const{theme:b,colorMode:g,dayScheme:p,nightScheme:u}=Jr(),m=null!==(o=null!==(a=t.theme)&&void 0!==a?a:b)&&void 0!==o?o:h.theme,[x,v]=e.useState(null!==(n=null!==(i=t.colorMode)&&void 0!==i?i:g)&&void 0!==n?n:"day"),[y,w]=e.useState(null!==(d=null!==(l=t.dayScheme)&&void 0!==l?l:p)&&void 0!==d?d:"light"),[B,$]=e.useState(null!==(c=null!==(s=t.nightScheme)&&void 0!==s?s:u)&&void 0!==c?c:"dark"),k=function(e,r){switch(e){case"auto":return r;default:return e}}(x,function(){const[r,t]=e.useState(Qr);return e.useEffect(()=>{var e,r;const o=null===(e=window)||void 0===e||null===(r=e.matchMedia)||void 0===r?void 0:r.call(e,"(prefers-color-scheme: dark)");function a(e){const r=e.matches;t(r?"night":"day")}return null==o||o.addEventListener("change",a),function(){null==o||o.removeEventListener("change",a)}},[]),r}()),S=function(e,r,t){switch(e){case"day":return r;case"night":return t}}(k,y,B),T=e.useMemo(()=>function(e,r){if(!e.colorSchemes)return e;if(!e.colorSchemes[r])return console.error(`\`${r}\` scheme not defined in \`theme.colorSchemes\``),e;return Xr(e,e.colorSchemes[r])}(m,S),[m,S]);return e.useEffect(()=>{var e,r;v(null!==(e=null!==(r=t.colorMode)&&void 0!==r?r:g)&&void 0!==e?e:"day")},[t.colorMode,g]),e.useEffect(()=>{var e,r;w(null!==(e=null!==(r=t.dayScheme)&&void 0!==r?r:p)&&void 0!==e?e:"light")},[t.dayScheme,p]),e.useEffect(()=>{var e,r;$(null!==(e=null!==(r=t.nightScheme)&&void 0!==r?r:u)&&void 0!==e?e:"dark")},[t.nightScheme,u]),e.createElement(Yr.Provider,{value:{theme:T,colorScheme:S,colorMode:x,resolvedColorMode:k,dayScheme:y,nightScheme:B,setColorMode:v,setDayScheme:w,setNightScheme:$}},e.createElement(f,{theme:T},r))};function Jr(){return e.useContext(Yr)}function Zr(e,r){var t;const{colorScheme:o=""}=Jr();return null!==(t=e[o])&&void 0!==t?t:r}function Qr(){var e,r,t;return"undefined"!=typeof window&&null!==(e=(r=window).matchMedia)&&void 0!==e&&null!==(t=e.call(r,"(prefers-color-scheme: dark)"))&&void 0!==t&&t.matches?"night":"day"}qr.displayName="ThemeProvider";const et=e=>se(e.sx),rt=c.div`
  ${Ir}
  ${Rr}
  ${zr}
  ${et};
`,tt=c(rt)`
  ${Ar};
  ${et};
`;tt.defaultProps={borderWidth:"1px",borderStyle:"solid",borderColor:"border.primary",borderRadius:2};const ot=c(rt)``;ot.defaultProps={display:"flex"};const at=c(rt)`
  ${Mr};
`;at.defaultProps={display:"grid"};const nt=c(rt)`
  ${Lr};
  ${et};
`,it=e.forwardRef((r,t)=>e.createElement(nt,Hr({},r,{position:"absolute",ref:t})));it.displayName="Absolute";const dt=e.forwardRef((r,t)=>e.createElement(nt,Hr({},r,{position:"fixed",ref:t})));dt.displayName="Fixed";const lt=e.forwardRef((r,t)=>e.createElement(nt,Hr({},r,{position:"relative",ref:t})));lt.displayName="Relative";const ct=e.forwardRef((r,t)=>e.createElement(nt,Hr({},r,{position:"sticky",ref:t})));function st({ref:e,closeOnOutsideClick:n,defaultOpen:i,onClickOutside:d}){const[l,c]=r(i),s=t(null),f=null!=e?e:s,b=o(e=>{const{current:r}=f;e.target.closest("details")!==r&&(d&&d(e),e.defaultPrevented||c(!1))},[f,c,d]);a(()=>{if(l&&n)return document.addEventListener("click",b),()=>{document.removeEventListener("click",b)}},[l,n,b]);const g=e=>{if(!e.defaultPrevented){const r=e.target;c(r.open)}};return{open:l,setOpen:c,getDetailsProps:()=>({onToggle:g,open:l,ref:f})}}function ft(){const e=t(new Set),r=o((r,t,...o)=>{const a=window.setTimeout(r,t,...o);return e.current.add(a),a},[]),n=o(r=>{clearTimeout(r),e.current.delete(r)},[]);return a(()=>()=>{for(const r of e.current)clearTimeout(r)},[]),{safeSetTimeout:r,safeClearTimeout:n}}ct.defaultProps={top:0,zIndex:1},ct.displayName="Sticky";const bt=({containerRef:e,ignoreClickRefs:r,onClickOutside:t})=>{const n=o(o=>{(({ignoreClickRefs:e,containerRef:r,e:t})=>{var o;let a=!0;if(t instanceof MouseEvent&&t.button>0&&(a=!1),null!==(o=r.current)&&void 0!==o&&o.contains(t.target))a=!1;else if(e)for(const r of e){var n;if(r&&null!==(n=r.current)&&void 0!==n&&n.contains(t.target)){a=!1;break}}return a})({ignoreClickRefs:r,containerRef:e,e:o})&&t(o)},[t,e,r]);a(()=>(document.addEventListener("click",n),()=>{document.removeEventListener("click",n)}),[n])};function*gt(e,r={}){var t,o;const a=null!==(t=r.strict)&&void 0!==t&&t,n=null!==(o=r.onlyTabbable)&&void 0!==o&&o?ut:pt,i=document.createTreeWalker(e,NodeFilter.SHOW_ELEMENT,{acceptNode:e=>e instanceof HTMLElement&&n(e,a)?NodeFilter.FILTER_ACCEPT:NodeFilter.FILTER_SKIP});let d=null;if(!r.reverse&&n(e,a)&&(yield e),r.reverse){let e=i.lastChild();for(;e;)d=e,e=i.lastChild()}else d=i.firstChild();for(;d instanceof HTMLElement;)yield d,d=r.reverse?i.previousNode():i.nextNode();r.reverse&&n(e,a)&&(yield e)}function pt(e,r=!1){const t=["BUTTON","INPUT","SELECT","TEXTAREA","OPTGROUP","OPTION","FIELDSET"].includes(e.tagName)&&e.disabled,o=e.hidden,a=e instanceof HTMLInputElement&&"hidden"===e.type;if(t||o||a)return!1;if(r){const r=0===e.offsetWidth||0===e.offsetHeight,t=["hidden","collapse"].includes(getComputedStyle(e).visibility),o=0===e.getClientRects().length;if(r||t||o)return!1}return null!=e.getAttribute("tabindex")||!(e instanceof HTMLAnchorElement&&null==e.getAttribute("href"))&&-1!==e.tabIndex}function ut(e,r=!1){return pt(e,r)&&"-1"!==e.getAttribute("tabindex")}function mt({initialFocusRef:e,returnFocusRef:r,containerRef:t}){a(()=>{const o=r.current;if(e&&e.current)e.current.focus();else if(t&&t.current){const e=gt(t.current).next().value;null==e||e.focus()}return function(){null==o||o.focus()}},[e,r,t])}const ht=[];function xt(e){if("Escape"===e.key&&!e.defaultPrevented)for(let r=ht.length-1;r>=0&&(ht[r](e),!e.defaultPrevented);--r);}const vt=(e,r=[e])=>{const t=o(e,r);a(()=>(0===ht.length&&document.addEventListener("keydown",xt),ht.push(t),()=>{ht.splice(ht.findIndex(e=>e===t),1),0===ht.length&&document.removeEventListener("keydown",xt)}),[t])};function yt(r){const t=e.useRef(null);return null!=r?r:t}const wt=({overlayRef:e,returnFocusRef:r,initialFocusRef:t,onEscape:o,ignoreClickRefs:a,onClickOutside:n})=>{const i=yt(e);return mt({containerRef:i,returnFocusRef:r,initialFocusRef:t}),bt({containerRef:i,ignoreClickRefs:a,onClickOutside:n}),vt(o),{ref:i}};let Bt=!1;function $t(){}try{const e=Object.create({},{signal:{get(){Bt=!0}}});window.addEventListener("test",$t,e),window.removeEventListener("test",$t,e)}catch(e){}function kt(){Bt||(!function(){if("undefined"==typeof window)return;const e=EventTarget.prototype.addEventListener;EventTarget.prototype.addEventListener=function(r,t,o){return"object"==typeof o&&"signal"in o&&o.signal instanceof AbortSignal&&e.call(o.signal,"abort",()=>{this.removeEventListener(r,t,o)}),e.call(this,r,t,o)}}(),Bt=!0)}let St=void 0;function Tt(){return void 0===St&&(St=/^mac/i.test(window.navigator.platform)),St}let Et,Ct=1e4;function It(){return"__primer_id_"+Ct++}kt(),function(e){e[e.ArrowHorizontal=1]="ArrowHorizontal",e[e.ArrowVertical=2]="ArrowVertical",e[e.JK=4]="JK",e[e.HL=8]="HL",e[e.HomeAndEnd=16]="HomeAndEnd",e[e.PageUpDown=256]="PageUpDown",e[e.WS=32]="WS",e[e.AD=64]="AD",e[e.Tab=128]="Tab",e[e.ArrowAll=e.ArrowHorizontal|e.ArrowVertical]="ArrowAll",e[e.HJKL=e.HL|e.JK]="HJKL",e[e.WASD=e.WS|e.AD]="WASD",e[e.All=e.ArrowAll|e.HJKL|e.HomeAndEnd|e.PageUpDown|e.WASD|e.Tab]="All"}(Et||(Et={}));const Nt={ArrowLeft:Et.ArrowHorizontal,ArrowDown:Et.ArrowVertical,ArrowUp:Et.ArrowVertical,ArrowRight:Et.ArrowHorizontal,h:Et.HL,j:Et.JK,k:Et.JK,l:Et.HL,a:Et.AD,s:Et.WS,w:Et.WS,d:Et.AD,Tab:Et.Tab,Home:Et.HomeAndEnd,End:Et.HomeAndEnd,PageUp:Et.PageUpDown,PageDown:Et.PageUpDown},At={ArrowLeft:"previous",ArrowDown:"next",ArrowUp:"previous",ArrowRight:"next",h:"previous",j:"next",k:"previous",l:"next",a:"previous",s:"next",w:"previous",d:"next",Tab:"next",Home:"start",End:"end",PageUp:"start",PageDown:"end"};function zt(e){const r=At[e.key];if("Tab"===e.key&&e.shiftKey)return"previous";const t=Tt();if(t&&e.metaKey||!t&&e.ctrlKey){if("ArrowLeft"===e.key||"ArrowUp"===e.key)return"start";if("ArrowRight"===e.key||"ArrowDown"===e.key)return"end"}return r}const Lt=[];function Rt(e){for(const r of Lt)r(e)}function Mt(e,r){var t,o,a,n;const i=[],d=new WeakMap,l=null!==(t=null==r?void 0:r.bindKeys)&&void 0!==t?t:(null!=r&&r.getNextFocusable?Et.ArrowAll:Et.ArrowVertical)|Et.HomeAndEnd,c=null!==(o=null==r?void 0:r.focusOutBehavior)&&void 0!==o?o:"stop",s=null!==(a=null==r?void 0:r.focusInStrategy)&&void 0!==a?a:"previous",f=null==r?void 0:r.activeDescendantControl,b=null==r?void 0:r.onActiveDescendantChanged;let g,p=Boolean(f);function u(e){const r=g;g=e,f||(r&&r!==e&&d.has(r)&&r.setAttribute("tabindex","-1"),null==e||e.setAttribute("tabindex","0"))}function m(){null==f||f.removeAttribute("aria-activedescendant"),p=!0,null==b||b(void 0,g),"first"===s&&(g=void 0)}function h(...e){const t=e.filter(e=>{var t,o;return null===(t=null==r||null===(o=r.focusableElementFilter)||void 0===o?void 0:o.call(r,e))||void 0===t||t});if(0===t.length)return;const o=i.findIndex(e=>(e.compareDocumentPosition(t[0])&Node.DOCUMENT_POSITION_PRECEDING)>0);i.splice(-1===o?i.length:o,0,...t);for(const e of t)d.has(e)||d.set(e,e.getAttribute("tabindex")),e.setAttribute("tabindex","-1");g||u(i[0])}function x(...e){for(const r of e){const e=i.indexOf(r);e>=0&&(i.splice(e,1),r===g&&u(i[0]));const t=d.get(r);void 0!==t&&(null===t?r.removeAttribute("tabindex"):r.setAttribute("tabindex",t),d.delete(r))}}h(...gt(e));const v=(y=e=>{if("previous"===s){const r=i.indexOf(e);r>=0&&u(i[r])}},Lt.push(y),()=>{const e=Lt.indexOf(y);Lt.splice(e,1)});var y;u(i[0]);new MutationObserver(e=>{for(const r of e)for(const e of r.removedNodes)e instanceof HTMLElement&&x(...gt(e));for(const r of e)for(const e of r.addedNodes)e instanceof HTMLElement&&h(...gt(e))}).observe(e,{subtree:!0,childList:!0});const w=new AbortController,B=null!==(n=null==r?void 0:r.abortSignal)&&void 0!==n?n:w.signal;B.addEventListener("abort",()=>{x(...i),v()});let $=void 0;e.addEventListener("mousedown",e=>{e.target instanceof HTMLElement&&e.target!==document.activeElement&&($=i.indexOf(e.target))},{signal:B}),f||e.addEventListener("focusin",r=>{if(r.target instanceof HTMLElement){if(void 0!==$)$>=0&&i[$]!==g&&u(i[$]),$=void 0;else if("previous"===s)u(r.target);else if("closest"===s||"first"===s){if(r.relatedTarget instanceof Element&&!e.contains(r.relatedTarget)){const e="previous"===S?i.length-1:0,r=i[e];return void(null==r||r.focus())}u(r.target)}else if("function"==typeof s)if(r.relatedTarget instanceof Element&&!e.contains(r.relatedTarget)){const e=s(r.relatedTarget);if((e?i.indexOf(e):-1)>=0&&e instanceof HTMLElement)return void e.focus();console.warn("Element requested is not a known focusable element.")}else u(r.target);Rt(r.target)}S=void 0},{signal:B});const k=null!=f?f:e;let S=void 0;return"closest"===s&&document.addEventListener("keydown",e=>{"Tab"===e.key&&(S=zt(e))},{signal:B,capture:!0}),k.addEventListener("keydown",e=>{if(e.key in At){const n=Nt[e.key];if(!e.defaultPrevented&&(n&l)>0&&!function(e,r){const t=e.key,o=[...t].length,a=r instanceof HTMLInputElement&&"text"===r.type||r instanceof HTMLTextAreaElement;if(a&&(1===o||"Home"===t||"End"===t))return!0;if(r instanceof HTMLSelectElement){if(1===o)return!0;if("ArrowDown"===t&&Tt()&&!e.metaKey)return!0;if("ArrowDown"===t&&!Tt()&&e.altKey)return!0}if(r instanceof HTMLTextAreaElement&&("PageUp"===t||"PageDown"===t))return!0;if(a){const e=r,o=0===e.selectionStart&&0===e.selectionEnd,a=e.selectionStart===e.value.length&&e.selectionEnd===e.value.length;if("ArrowLeft"===t&&!o)return!0;if("ArrowRight"===t&&!a)return!0;if(e instanceof HTMLTextAreaElement){if("ArrowUp"===t&&!o)return!0;if("ArrowDown"===t&&!a)return!0}}return!1}(e,document.activeElement)){const n=zt(e);let d=void 0;if(p)p=!1,d=g||i[0];else{var t;if(null!=r&&r.getNextFocusable)d=r.getNextFocusable(n,null!==(t=document.activeElement)&&void 0!==t?t:void 0,e);if(!d){const r=function(){if(!g)return 0;const e=i.indexOf(g);return-1===e?0:e}();let t=r;"previous"===n?t-=1:"start"===n?t=0:"next"===n?t+=1:"end"===n&&(t=i.length-1),t<0&&("wrap"===c&&"Tab"!==e.key?t=i.length-1:(f&&m(),t=0)),t>=i.length&&(t="wrap"===c&&"Tab"!==e.key?0:i.length-1),r!==t&&(d=i[t])}}d&&(f?(o=g,(a=d).id||a.setAttribute("id",It()),g=a,null==f||f.setAttribute("aria-activedescendant",a.id),Rt(a),null==b||b(a,o===a?void 0:o)):(S=n,d.focus())),("Tab"!==e.key||d)&&e.preventDefault()}}var o,a;"Escape"===e.key&&!p&&f&&m()},{signal:B}),f&&f.addEventListener("focusout",()=>{m()}),w}var Ht=b(["position:relative;display:inline-block;padding:6px 16px;font-family:inherit;font-weight:",";line-height:20px;white-space:nowrap;vertical-align:middle;cursor:pointer;user-select:none;border-radius:",";appearance:none;text-decoration:none;text-align:center;&:hover{text-decoration:none;}&:focus{outline:none;}&:disabled{cursor:default;}&:disabled svg{opacity:0.6;}"],Cr("fontWeights.bold"),Cr("radii.2"));const Ot=L(Ce,Ir,zr),Pt=fe({variants:{small:{p:"4px 12px",fontSize:0},medium:{fontSize:1},large:{fontSize:2,p:"10px 20px"}}}),Dt=c.button.attrs(({disabled:e,onClick:r})=>({onClick:e?void 0:r}))`
  ${Ht}
  ${Pt}
`;Dt.defaultProps={variant:"medium"};const Wt=c(Dt)`
  color: ${Cr("colors.btn.text")};
  background-color: ${Cr("colors.btn.bg")};
  border: 1px solid ${Cr("colors.btn.border")};
  box-shadow: ${Cr("shadows.btn.shadow")}, ${Cr("shadows.btn.insetShadow")}};

  &:hover {
    background-color: ${Cr("colors.btn.hoverBg")};
    border-color: ${Cr("colors.btn.hoverBorder")};
  }

  // focus must come before :active so that the active box shadow overrides
  &:focus {
    border-color: ${Cr("colors.btn.focusBorder")};
    box-shadow: ${Cr("shadows.btn.focusShadow")};
  }

  &:active {
    background-color: ${Cr("colors.btn.selectedBg")};
    box-shadow: ${Cr("shadows.btn.shadowActive")};
  }

  &:disabled {
    color: ${Cr("colors.text.disabled")};
    background-color: ${Cr("colors.btn.bg")};
    border-color: ${Cr("colors.btn.border")};
  }

  ${Ot};
  ${et};
`,jt=c(Dt)`
  color: ${Cr("colors.btn.danger.text")};
  border: 1px solid ${Cr("colors.btn.border")};
  background-color: ${Cr("colors.btn.bg")};
  box-shadow: ${Cr("shadows.btn.shadow")};

  &:hover {
    color: ${Cr("colors.btn.danger.hoverText")};
    background-color: ${Cr("colors.btn.danger.hoverBg")};
    border-color: ${Cr("colors.btn.danger.hoverBorder")};
    box-shadow: ${Cr("shadows.btn.danger.hoverShadow")};
  }
  // focus must come before :active so that the active box shadow overrides
  &:focus {
    border-color: ${Cr("colors.btn.danger.focusBorder")};
    box-shadow: ${Cr("shadows.btn.danger.focusShadow")};
  }

  &:active {
    color: ${Cr("colors.btn.danger.selectedText")};
    background-color: ${Cr("colors.btn.danger.selectedBg")};
    box-shadow: ${Cr("shadows.btn.danger.selectedShadow")};
    border-color: ${Cr("colors.btn.danger.selectedBorder")};
  }

  &:disabled {
    color: ${Cr("colors.btn.danger.disabledText")};
    background-color: ${Cr("colors.btn.danger.disabledBg")};
    border-color: ${Cr("colors.btn.danger.disabledBorder")};
  }

  ${Ot};
  ${et};
`,Ft=c(rt)`
  vertical-align: middle;

  && > * {
    position: relative;
    border-right-width: 0;
    border-radius: 0;

    :first-child {
      border-top-left-radius: ${Cr("radii.2")};
      border-bottom-left-radius: ${Cr("radii.2")};
      margin-right: 0;
    }

    :not(:first-child) {
      margin-left: 0;
      margin-right: 0;
    }

    :last-child {
      border-right-width: 1px;
      border-top-right-radius: ${Cr("radii.2")};
      border-bottom-right-radius: ${Cr("radii.2")};
    }

    :focus,
    :active,
    :hover {
      border-right-width: 1px;
      + * {
        border-left-width: 0;
      }
    }

    :focus,
    :active {
      z-index: 1;
    }
  }

  ${et};
`;Ft.defaultProps={display:"inline-block"};const _t=c(Dt)`
  color: ${Cr("colors.btn.outline.text")};
  border: 1px solid ${Cr("colors.btn.border")};
  background-color: ${Cr("colors.btn.bg")};
  box-shadow: ${Cr("shadows.btn.shadow")};

  &:hover {
    color: ${Cr("colors.btn.outline.hoverText")};
    background-color: ${Cr("colors.btn.outline.hoverBg")};
    border-color: ${Cr("colors.btn.outline.hoverBorder")};
    box-shadow: ${Cr("shadows.btn.outline.hoverShadow")};
  }
  // focus must come before :active so that the active box shadow overrides
  &:focus {
    border-color: ${Cr("colors.btn.outline.focusBorder")};
    box-shadow: ${Cr("shadows.btn.outline.focusShadow")};
  }

  &:active {
    color: ${Cr("colors.btn.outline.selectedText")};
    background-color: ${Cr("colors.btn.outline.selectedBg")};
    box-shadow: ${Cr("shadows.btn.outline.selectedShadow")};
    border-color: ${Cr("colors.btn.outline.selectedBorder")};
  }

  &:disabled {
    color: ${Cr("colors.btn.outline.disabledText")};
    background-color: ${Cr("colors.btn.outline.disabledBg")};
    border-color: ${Cr("colors.btn.border")};
  }

  ${Ot};
  ${et};
`,Ut=c(Dt)`
  color: ${Cr("colors.btn.primary.text")};
  border: 1px solid ${Cr("colors.btn.primary.border")};
  background-color: ${Cr("colors.btn.primary.bg")};
  box-shadow: ${Cr("shadows.btn.primary.shadow")};

  &:hover {
    color: ${Cr("colors.btn.primary.hoverText")};
    background-color: ${Cr("colors.btn.primary.hoverBg")};
    border-color: ${Cr("colors.btn.primary.hoverBorder")};
    box-shadow: ${Cr("shadows.btn.primary.hoverShadow")};
  }
  // focus must come before :active so that the active box shadow overrides
  &:focus {
    border-color: ${Cr("colors.btn.primary.focusBorder")};
    box-shadow: ${Cr("shadows.btn.primary.focusShadow")};
  }

  &:active {
    background-color: ${Cr("colors.btn.primary.selectedBg")};
    box-shadow: ${Cr("shadows.btn.primary.selectedShadow")};
  }

  &:disabled {
    color: ${Cr("colors.btn.primary.disabledText")};
    background-color: ${Cr("colors.btn.primary.disabledBg")};
    border-color: ${Cr("colors.btn.primary.disabledBorder")};
  }

  ${Ot};
  ${et};
`,Gt=c(Dt)`
  color: ${Cr("colors.text.link")};
  background-color: transparent;
  border: 0;
  border-radius: ${Cr("radii.2")};
  box-shadow: none;

  &:disabled {
    color: ${Cr("colors.text.disabled")};
  }

  &:focus {
    box-shadow: ${Cr("shadows.btn.focusShadow")};
  }

  ${Ot};
  ${et}
`,Kt=c.summary`
  display: inline-block;
  padding: 0;
  font-size: ${Cr("fontSizes.1")};
  color: ${Cr("colors.text.secondary")};
  text-decoration: none;
  white-space: nowrap;
  cursor: pointer;
  user-select: none;
  background-color: transparent;
  border: 0;
  appearance: none; // Corrects inability to style clickable input types in iOS.

  &:hover {
    text-decoration: underline;
  }

  &:disabled {
    &,
    &:hover {
      color: rgba(${Cr("colors.text.secondary")}, 0.5);
      cursor: default;
    }
  }

  &:after {
    display: inline-block;
    margin-left: ${Cr("space.1")};
    width: 0;
    height: 0;
    vertical-align: -2px;
    content: '';
    border: 4px solid transparent;
    border-top-color: currentcolor;
  }
  ${Ir}
  ${Nr}
  ${zr}
  ${et};
`;var Vt={small:16,medium:32,large:64};function Xt(e){var r=e["aria-label"],t=e.className,o=e.size,a=e.verticalAlign,n=e.svgDataByHeight,i=Vt[o]||o,d=function(e,r){return e.map((function(e){return parseInt(e,10)})).reduce((function(e,t){return t<=r?t:e}),e[0])}(Object.keys(n),i),l=n[d].width;return{"aria-hidden":r?"false":"true","aria-label":r,role:"img",className:t,viewBox:"0 0 "+l+" "+d,width:i*(l/d),height:i,fill:"currentColor",style:{display:"inline-block",userSelect:"none",verticalAlign:a},dangerouslySetInnerHTML:{__html:n[d].path}}}var Yt=Object.assign||function(e){for(var r=1;r<arguments.length;r++){var t=arguments[r];for(var o in t)Object.prototype.hasOwnProperty.call(t,o)&&(e[o]=t[o])}return e};function qt(r){return e.createElement("svg",Xt(Yt({},r,{svgDataByHeight:{16:{width:16,path:'<path fill-rule="evenodd" d="M13.78 4.22a.75.75 0 010 1.06l-7.25 7.25a.75.75 0 01-1.06 0L2.22 9.28a.75.75 0 011.06-1.06L6 10.94l6.72-6.72a.75.75 0 011.06 0z"></path>'},24:{width:24,path:'<path fill-rule="evenodd" d="M21.03 5.72a.75.75 0 010 1.06l-11.5 11.5a.75.75 0 01-1.072-.012l-5.5-5.75a.75.75 0 111.084-1.036l4.97 5.195L19.97 5.72a.75.75 0 011.06 0z"></path>'}}})))}function Jt(r){return e.createElement("svg",Xt(Yt({},r,{svgDataByHeight:{16:{width:16,path:'<path fill-rule="evenodd" d="M5 3.254V3.25v.005a.75.75 0 110-.005v.004zm.45 1.9a2.25 2.25 0 10-1.95.218v5.256a2.25 2.25 0 101.5 0V7.123A5.735 5.735 0 009.25 9h1.378a2.251 2.251 0 100-1.5H9.25a4.25 4.25 0 01-3.8-2.346zM12.75 9a.75.75 0 100-1.5.75.75 0 000 1.5zm-8.5 4.5a.75.75 0 100-1.5.75.75 0 000 1.5z"></path>'},24:{width:24,path:'<path fill-rule="evenodd" d="M5.75 21a1.75 1.75 0 110-3.5 1.75 1.75 0 010 3.5zM2.5 19.25a3.25 3.25 0 106.5 0 3.25 3.25 0 00-6.5 0zM5.75 6.5a1.75 1.75 0 110-3.5 1.75 1.75 0 010 3.5zM2.5 4.75a3.25 3.25 0 106.5 0 3.25 3.25 0 00-6.5 0zM18.25 15a1.75 1.75 0 110-3.5 1.75 1.75 0 010 3.5zM15 13.25a3.25 3.25 0 106.5 0 3.25 3.25 0 00-6.5 0z"></path><path fill-rule="evenodd" d="M6.5 7.25c0 2.9 2.35 5.25 5.25 5.25h4.5V14h-4.5A6.75 6.75 0 015 7.25h1.5z"></path><path fill-rule="evenodd" d="M5.75 16.75A.75.75 0 006.5 16V8A.75.75 0 005 8v8c0 .414.336.75.75.75z"></path>'}}})))}function Zt(r){return e.createElement("svg",Xt(Yt({},r,{svgDataByHeight:{16:{width:16,path:'<path fill-rule="evenodd" d="M7.177 3.073L9.573.677A.25.25 0 0110 .854v4.792a.25.25 0 01-.427.177L7.177 3.427a.25.25 0 010-.354zM3.75 2.5a.75.75 0 100 1.5.75.75 0 000-1.5zm-2.25.75a2.25 2.25 0 113 2.122v5.256a2.251 2.251 0 11-1.5 0V5.372A2.25 2.25 0 011.5 3.25zM11 2.5h-1V4h1a1 1 0 011 1v5.628a2.251 2.251 0 101.5 0V5A2.5 2.5 0 0011 2.5zm1 10.25a.75.75 0 111.5 0 .75.75 0 01-1.5 0zM3.75 12a.75.75 0 100 1.5.75.75 0 000-1.5z"></path>'},24:{width:24,path:'<path fill-rule="evenodd" d="M4.75 3a1.75 1.75 0 100 3.5 1.75 1.75 0 000-3.5zM1.5 4.75a3.25 3.25 0 116.5 0 3.25 3.25 0 01-6.5 0zM4.75 17.5a1.75 1.75 0 100 3.5 1.75 1.75 0 000-3.5zM1.5 19.25a3.25 3.25 0 116.5 0 3.25 3.25 0 01-6.5 0zm17.75-1.75a1.75 1.75 0 100 3.5 1.75 1.75 0 000-3.5zM16 19.25a3.25 3.25 0 116.5 0 3.25 3.25 0 01-6.5 0z"></path><path fill-rule="evenodd" d="M4.75 7.25A.75.75 0 015.5 8v8A.75.75 0 014 16V8a.75.75 0 01.75-.75zm8.655-5.53a.75.75 0 010 1.06L12.185 4h4.065A3.75 3.75 0 0120 7.75v8.75a.75.75 0 01-1.5 0V7.75a2.25 2.25 0 00-2.25-2.25h-4.064l1.22 1.22a.75.75 0 01-1.061 1.06l-2.5-2.5a.75.75 0 010-1.06l2.5-2.5a.75.75 0 011.06 0z"></path>'}}})))}function Qt(r){return e.createElement("svg",Xt(Yt({},r,{svgDataByHeight:{16:{width:16,path:'<path fill-rule="evenodd" d="M1.5 8a6.5 6.5 0 0110.65-5.003.75.75 0 00.959-1.153 8 8 0 102.592 8.33.75.75 0 10-1.444-.407A6.5 6.5 0 011.5 8zM8 12a1 1 0 100-2 1 1 0 000 2zm0-8a.75.75 0 01.75.75v3.5a.75.75 0 11-1.5 0v-3.5A.75.75 0 018 4zm4.78 4.28l3-3a.75.75 0 00-1.06-1.06l-2.47 2.47-.97-.97a.749.749 0 10-1.06 1.06l1.5 1.5a.75.75 0 001.06 0z"></path>'},24:{width:24,path:'<path d="M2.5 12c0-5.24 4.288-9.5 9.593-9.5a9.608 9.608 0 017.197 3.219.75.75 0 001.12-.998A11.108 11.108 0 0012.093 1C5.973 1 1 5.919 1 12s4.973 11 11.093 11c5.403 0 9.91-3.832 10.893-8.915a.75.75 0 10-1.472-.285c-.848 4.381-4.74 7.7-9.421 7.7C6.788 21.5 2.5 17.24 2.5 12z"></path><path d="M12 17a1 1 0 100-2 1 1 0 000 2zm0-10a.75.75 0 01.75.75v4.5a.75.75 0 01-1.5 0v-4.5A.75.75 0 0112 7zm11.28.78a.75.75 0 00-1.06-1.06l-3.47 3.47-1.47-1.47a.75.75 0 10-1.06 1.06l2 2a.75.75 0 001.06 0l4-4z"></path>'}}})))}function eo(r){return e.createElement("svg",Xt(Yt({},r,{svgDataByHeight:{16:{width:16,path:'<path fill-rule="evenodd" d="M8 1.5a6.5 6.5 0 100 13 6.5 6.5 0 000-13zM0 8a8 8 0 1116 0A8 8 0 010 8zm9 3a1 1 0 11-2 0 1 1 0 012 0zm-.25-6.25a.75.75 0 00-1.5 0v3.5a.75.75 0 001.5 0v-3.5z"></path>'},24:{width:24,path:'<path d="M12 7a.75.75 0 01.75.75v4.5a.75.75 0 01-1.5 0v-4.5A.75.75 0 0112 7zm1 9a1 1 0 11-2 0 1 1 0 012 0z"></path><path fill-rule="evenodd" d="M12 1C5.925 1 1 5.925 1 12s4.925 11 11 11 11-4.925 11-11S18.075 1 12 1zM2.5 12a9.5 9.5 0 1119 0 9.5 9.5 0 01-19 0z"></path>'}}})))}function ro(r){return e.createElement("svg",Xt(Yt({},r,{svgDataByHeight:{16:{width:16,path:'<path fill-rule="evenodd" d="M1.326 1.973a1.2 1.2 0 011.49-.832c.387.112.977.307 1.575.602.586.291 1.243.71 1.7 1.296.022.027.042.056.061.084A13.22 13.22 0 018 3c.67 0 1.289.037 1.861.108l.051-.07c.457-.586 1.114-1.004 1.7-1.295a9.654 9.654 0 011.576-.602 1.2 1.2 0 011.49.832c.14.493.356 1.347.479 2.29.079.604.123 1.28.07 1.936.541.977.773 2.11.773 3.301C16 13 14.5 15 8 15s-8-2-8-5.5c0-1.034.238-2.128.795-3.117-.08-.712-.034-1.46.052-2.12.122-.943.34-1.797.479-2.29zM8 13.065c6 0 6.5-2 6-4.27C13.363 5.905 11.25 5 8 5s-5.363.904-6 3.796c-.5 2.27 0 4.27 6 4.27z"></path><path d="M4 8a1 1 0 012 0v1a1 1 0 01-2 0V8zm2.078 2.492c-.083-.264.146-.492.422-.492h3c.276 0 .505.228.422.492C9.67 11.304 8.834 12 8 12c-.834 0-1.669-.696-1.922-1.508zM10 8a1 1 0 112 0v1a1 1 0 11-2 0V8z"></path>'},24:{width:24,path:'<path d="M7.75 11c-.69 0-1.25.56-1.25 1.25v1.5a1.25 1.25 0 102.5 0v-1.5C9 11.56 8.44 11 7.75 11zm1.27 4.5a.469.469 0 01.48-.5h5a.47.47 0 01.48.5c-.116 1.316-.759 2.5-2.98 2.5s-2.864-1.184-2.98-2.5zm7.23-4.5c-.69 0-1.25.56-1.25 1.25v1.5a1.25 1.25 0 102.5 0v-1.5c0-.69-.56-1.25-1.25-1.25z"></path><path fill-rule="evenodd" d="M21.255 3.82a1.725 1.725 0 00-2.141-1.195c-.557.16-1.406.44-2.264.866-.78.386-1.647.93-2.293 1.677A18.442 18.442 0 0012 5c-.93 0-1.784.059-2.569.17-.645-.74-1.505-1.28-2.28-1.664a13.876 13.876 0 00-2.265-.866 1.725 1.725 0 00-2.141 1.196 23.645 23.645 0 00-.69 3.292c-.125.97-.191 2.07-.066 3.112C1.254 11.882 1 13.734 1 15.527 1 19.915 3.13 23 12 23c8.87 0 11-3.053 11-7.473 0-1.794-.255-3.647-.99-5.29.127-1.046.06-2.15-.066-3.125a23.652 23.652 0 00-.689-3.292zM20.5 14c.5 3.5-1.5 6.5-8.5 6.5s-9-3-8.5-6.5c.583-4 3-6 8.5-6s7.928 2 8.5 6z"></path>'}}})))}function to(r){return e.createElement("svg",Xt(Yt({},r,{svgDataByHeight:{16:{width:16,path:'<path fill-rule="evenodd" d="M8 1.5a6.5 6.5 0 100 13 6.5 6.5 0 000-13zM0 8a8 8 0 1116 0A8 8 0 010 8zm9 3a1 1 0 11-2 0 1 1 0 012 0zM6.92 6.085c.081-.16.19-.299.34-.398.145-.097.371-.187.74-.187.28 0 .553.087.738.225A.613.613 0 019 6.25c0 .177-.04.264-.077.318a.956.956 0 01-.277.245c-.076.051-.158.1-.258.161l-.007.004a7.728 7.728 0 00-.313.195 2.416 2.416 0 00-.692.661.75.75 0 001.248.832.956.956 0 01.276-.245 6.3 6.3 0 01.26-.16l.006-.004c.093-.057.204-.123.313-.195.222-.149.487-.355.692-.662.214-.32.329-.702.329-1.15 0-.76-.36-1.348-.863-1.725A2.76 2.76 0 008 4c-.631 0-1.155.16-1.572.438-.413.276-.68.638-.849.977a.75.75 0 101.342.67z"></path>'},24:{width:24,path:'<path d="M10.97 8.265a1.45 1.45 0 00-.487.57.75.75 0 01-1.341-.67c.2-.402.513-.826.997-1.148C10.627 6.69 11.244 6.5 12 6.5c.658 0 1.369.195 1.934.619a2.45 2.45 0 011.004 2.006c0 1.033-.513 1.72-1.027 2.215-.19.183-.399.358-.579.508l-.147.123a4.329 4.329 0 00-.435.409v1.37a.75.75 0 11-1.5 0v-1.473c0-.237.067-.504.247-.736.22-.28.486-.517.718-.714l.183-.153.001-.001c.172-.143.324-.27.47-.412.368-.355.569-.676.569-1.136a.953.953 0 00-.404-.806C12.766 8.118 12.384 8 12 8c-.494 0-.814.121-1.03.265zM13 17a1 1 0 11-2 0 1 1 0 012 0z"></path><path fill-rule="evenodd" d="M12 1C5.925 1 1 5.925 1 12s4.925 11 11 11 11-4.925 11-11S18.075 1 12 1zM2.5 12a9.5 9.5 0 1119 0 9.5 9.5 0 01-19 0z"></path>'}}})))}function oo(r){return e.createElement("svg",Xt(Yt({},r,{svgDataByHeight:{16:{width:16,path:'<path d="M4.427 7.427l3.396 3.396a.25.25 0 00.354 0l3.396-3.396A.25.25 0 0011.396 7H4.604a.25.25 0 00-.177.427z"></path>'},24:{width:24,path:'<path d="M11.646 15.146L5.854 9.354a.5.5 0 01.353-.854h11.586a.5.5 0 01.353.854l-5.793 5.792a.5.5 0 01-.707 0z"></path>'}}})))}function ao(r){return e.createElement("svg",Xt(Yt({},r,{svgDataByHeight:{16:{width:16,path:'<path fill-rule="evenodd" d="M3.72 3.72a.75.75 0 011.06 0L8 6.94l3.22-3.22a.75.75 0 111.06 1.06L9.06 8l3.22 3.22a.75.75 0 11-1.06 1.06L8 9.06l-3.22 3.22a.75.75 0 01-1.06-1.06L6.94 8 3.72 4.78a.75.75 0 010-1.06z"></path>'},24:{width:24,path:'<path fill-rule="evenodd" d="M5.72 5.72a.75.75 0 011.06 0L12 10.94l5.22-5.22a.75.75 0 111.06 1.06L13.06 12l5.22 5.22a.75.75 0 11-1.06 1.06L12 13.06l-5.22 5.22a.75.75 0 01-1.06-1.06L10.94 12 5.72 6.78a.75.75 0 010-1.06z"></path>'}}})))}qt.defaultProps={className:"octicon",size:16,verticalAlign:"text-bottom"},Jt.defaultProps={className:"octicon",size:16,verticalAlign:"text-bottom"},Zt.defaultProps={className:"octicon",size:16,verticalAlign:"text-bottom"},Qt.defaultProps={className:"octicon",size:16,verticalAlign:"text-bottom"},eo.defaultProps={className:"octicon",size:16,verticalAlign:"text-bottom"},ro.defaultProps={className:"octicon",size:16,verticalAlign:"text-bottom"},to.defaultProps={className:"octicon",size:16,verticalAlign:"text-bottom"},oo.defaultProps={className:"octicon",size:16,verticalAlign:"text-bottom"},ao.defaultProps={className:"octicon",size:16,verticalAlign:"text-bottom"};const no=c.button`
  border: none;
  padding: 0;
  background: transparent;
  outline: none;
  cursor: pointer;
  border-radius: ${Cr("radii.2")};
  color: ${Cr("colors.text.secondary")};

  &:focus {
    box-shadow: ${Cr("shadows.btn.focusShadow")};
  }

  &:hover {
    color: ${Cr("colors.text.link")};
  }
  ${Ir};
  ${zr};
  ${et};
`,io=n((r,t)=>e.createElement(no,Hr({ref:t,"aria-label":"Close"},r),e.createElement(ao,null))),lo={"outside-top":["outside-bottom","outside-right","outside-left","outside-bottom"],"outside-bottom":["outside-top","outside-right","outside-left","outside-bottom"],"outside-left":["outside-right","outside-bottom","outside-top","outside-bottom"],"outside-right":["outside-left","outside-bottom","outside-top","outside-bottom"]};function co(e,r,t={}){const o=function(e){let r=e.parentNode;for(;null!==r;){if(r instanceof HTMLElement&&"static"!==getComputedStyle(r).position)return r;r=r.parentNode}return document.body}(e),a=function(e){let r=e;for(;null!==r&&r!==document.body;){if("visible"!==getComputedStyle(r).overflow)break;r=r.parentNode}const t=r!==document.body&&r instanceof HTMLElement?r:document.body,o=t.getBoundingClientRect(),a=getComputedStyle(t),[n,i,d,l]=[a.borderTopWidth,a.borderLeftWidth,a.borderRightWidth,a.borderBottomWidth].map(e=>parseInt(e,10)||0);return{top:o.top+n,left:o.left+i,width:o.width-d-i,height:Math.max(o.height-n-l,t===document.body?window.innerHeight:-1/0)}}(o),n=getComputedStyle(o),i=o.getBoundingClientRect(),[d,l]=[n.borderTopWidth,n.borderLeftWidth].map(e=>parseInt(e,10)||0);return function(e,r,t,o,{side:a,align:n,allowOutOfBounds:i,anchorOffset:d,alignmentOffset:l}){const c={top:e.top-r.top,left:e.left-r.left,width:e.width,height:e.height};let s=uo(t,o,a,n,d,l);if(s.top-=r.top,s.left-=r.left,!i){const i=lo[a];let f=0;if(i){let e=a;for(;f<i.length&&mo(e,s,c,t);){const a=i[f++];e=a,s=uo(t,o,a,n,d,l),s.top-=r.top,s.left-=r.left}}s.top<c.top&&(s.top=c.top),s.left<c.left&&(s.left=c.left),s.left+t.width>e.width+c.left&&(s.left=e.width+c.left-t.width),i&&f<i.length&&s.top+t.height>e.height+c.top&&(s.top=e.height+c.top-t.height)}return s}(a,{top:i.top+d,left:i.left+l},e.getBoundingClientRect(),r instanceof Element?r.getBoundingClientRect():r,function(e={}){var r,t,o,a,n;const i=null!==(r=e.side)&&void 0!==r?r:so,d=null!==(t=e.align)&&void 0!==t?t:fo;return{side:i,align:d,anchorOffset:null!==(o=e.anchorOffset)&&void 0!==o?o:"inside-center"===i?0:bo,alignmentOffset:null!==(a=e.alignmentOffset)&&void 0!==a?a:"center"!==d&&i.startsWith("inside")?go:0,allowOutOfBounds:null!==(n=e.allowOutOfBounds)&&void 0!==n?n:po}}(t))}const so="outside-bottom",fo="start",bo=4,go=4,po=!1;function uo(e,r,t,o,a,n){const i=r.left+r.width,d=r.top+r.height;let l=-1,c=-1;return"outside-top"===t?l=r.top-a-e.height:"outside-bottom"===t?l=d+a:"outside-left"===t?c=r.left-a-e.width:"outside-right"===t&&(c=i+a),"outside-top"!==t&&"outside-bottom"!==t||("start"===o?c=r.left+n:"center"===o?c=r.left-(e.width-r.width)/2+n:"end"===o&&(c=i-e.width-n)),"outside-left"!==t&&"outside-right"!==t||("start"===o?l=r.top+n:"center"===o?l=r.top-(e.height-r.height)/2+n:"end"===o&&(l=d-e.height-n)),"inside-top"===t?l=r.top+a:"inside-bottom"===t?l=d-a-e.height:"inside-left"===t?c=r.left+a:"inside-right"===t?c=i-a-e.width:"inside-center"===t&&(c=(i+r.left)/2-e.width/2+a),"inside-top"===t||"inside-bottom"===t?"start"===o?c=r.left+n:"center"===o?c=r.left-(e.width-r.width)/2+n:"end"===o&&(c=i-e.width-n):"inside-left"!==t&&"inside-right"!==t&&"inside-center"!==t||("start"===o?l=r.top+n:"center"===o?l=r.top-(e.height-r.height)/2+n:"end"===o&&(l=d-e.height-n)),{top:l,left:c}}function mo(e,r,t,o){return"outside-top"===e||"outside-bottom"===e?r.top<t.top||r.top+o.height>t.height+t.top:r.left<t.left||r.left+o.width>t.width+t.left}kt();const ho=[];let xo=void 0;function vo(e,r=!1){return gt(e,{reverse:r,strict:!0,onlyTabbable:!0}).next().value}function yo(e,r,t){const o=new AbortController,a=null!=t?t:o.signal;e.setAttribute("data-focus-trap","active");let n=void 0;function i(t){if(t instanceof HTMLElement){if(e.contains(t))return void(n=t);if(n&&ut(n)&&e.contains(n))return void n.focus();{const o=r&&e.contains(r)?r:vo(e);if(o)return void o.focus();t.blur()}}}const d=function(e){const r=new AbortController;return e.addEventListener("abort",()=>{r.abort()}),r}(a);if(e.addEventListener("keydown",r=>{if("Tab"!==r.key||r.defaultPrevented)return;const{target:t}=r,o=vo(e),a=vo(e,!0);t===o&&r.shiftKey?(r.preventDefault(),null==a||a.focus()):t!==a||r.shiftKey||(r.preventDefault(),null==o||o.focus())},{signal:d.signal}),xo){const e=xo;xo.container.setAttribute("data-focus-trap","suspended"),xo.controller.abort(),ho.push(e)}d.signal.addEventListener("abort",()=>{xo=void 0}),a.addEventListener("abort",()=>{e.removeAttribute("data-focus-trap");const r=ho.findIndex(r=>r.container===e);r>=0&&ho.splice(r,1),function(){const e=ho.pop();e&&yo(e.container,e.initialFocus,e.originalSignal)}()}),document.addEventListener("focusin",e=>{i(e.target)},{signal:d.signal}),i(document.activeElement),xo={container:e,controller:d,initialFocus:r,originalSignal:a};const l=ho.findIndex(r=>r.container===e);if(l>=0&&ho.splice(l,1),!t)return o}function wo(r,t=[]){const o=yt(null==r?void 0:r.containerRef),a=yt(null==r?void 0:r.initialFocusRef),n=null==r?void 0:r.disabled,i=e.useRef(),d=e.useRef(null);function l(){var e;null===(e=i.current)||void 0===e||e.abort(),null!=r&&r.restoreFocusOnCleanUp&&d.current instanceof HTMLElement&&(d.current.focus(),d.current=null)}return d.current||null!=r&&r.disabled||(d.current=document.activeElement),e.useEffect(()=>{if(o.current instanceof HTMLElement){var e;if(!n)return i.current=yo(o.current,null!==(e=a.current)&&void 0!==e?e:void 0),()=>{l()};l()}},[o,a,n,...t]),{containerRef:o,initialFocusRef:a}}function Bo({icon:r,...t}){return e.createElement(r,t)}Bo.displayName="Octicon";const $o=c(Bo)`
  ${Ir}
  ${et}
`;function ko(r={},t=[]){const o=yt(r.containerRef),n=!!r.activeDescendantFocus,i=yt("boolean"!=typeof r.activeDescendantFocus&&r.activeDescendantFocus?r.activeDescendantFocus:void 0),d=null==r?void 0:r.disabled,l=e.useRef();return a(()=>{if(o.current instanceof HTMLElement&&(!n||i.current instanceof HTMLElement)){if(!d){var e;const t={...r,activeDescendantControl:null!==(e=i.current)&&void 0!==e?e:void 0};return l.current=Mt(o.current,t),()=>{var e;null===(e=l.current)||void 0===e||e.abort()}}var t;null===(t=l.current)||void 0===t||t.abort()}},[d,...t]),{containerRef:o,activeDescendantControlRef:i}}$o.defaultProps={size:16};const So={};function To(){if(!("__default__"in So)||!document.body.contains(So.__default__)){let e=document.getElementById("__primerPortalRoot__");if(!(e instanceof Element)){e=document.createElement("div"),e.setAttribute("id","__primerPortalRoot__");const r=document.querySelector("[data-portal-root]");r?r.appendChild(e):document.body.appendChild(e)}!function(e,r="__default__"){So[r]=e}(e)}}const Eo=({children:r,onMount:t,containerName:o})=>{const a=document.createElement("div");a.style.position="relative",a.style.zIndex="1";const n=e.useRef(a);return e.useLayoutEffect(()=>{let e=o;void 0===e&&(e="__default__",To());const r=So[e];if(!r)throw new Error(`Portal container '${o}' is not yet registered. Container must be registered with registerPortal before use.`);const a=n.current;return r.appendChild(a),null==t||t(),()=>{r.removeChild(a)}},[n]),m(r,n.current)};function Co(...r){const o=t(null);return e.useEffect(()=>{function e(e=null){for(const t of r){if(!t)return;"function"==typeof t?t(e):t.current=e}}return e(o.current),()=>{e(o.current)}},[...r,o.current]),o}const Io=c("div")`
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: rgba(0, 0, 0, 0.4);
  animation: dialog-backdrop-appear ${"200ms"} ${Cr("animation.easeOutCubic")};

  @keyframes dialog-backdrop-appear {
    0% {
      opacity: 0;
    }
    100% {
      opacity: 1;
    }
  }
`,No={sm:"480px",lg:"640px",auto:"auto"},Ao={sm:"296px",md:"320px",lg:"480px",xl:"640px"},zo=c.div`
  display: flex;
  flex-direction: column;
  background-color: ${Cr("colors.bg.overlay")};
  box-shadow: 0px 1px 3px rgba(0, 0, 0, 0.12), 0px 8px 24px rgba(149, 157, 165, 0.2);
  min-width: 296px;
  max-width: calc(100vw - 64px);
  max-height: calc(100vh - 64px);
  width: ${e=>{var r;return Ao[null!==(r=e.width)&&void 0!==r?r:"xl"]}};
  height: ${e=>{var r;return No[null!==(r=e.height)&&void 0!==r?r:"auto"]}};
  border-radius: 12px;
  opacity: 1;
  animation: overlay--dialog-appear ${"200ms"} ${Cr("animation.easeOutCubic")};

  @keyframes overlay--dialog-appear {
    0% {
      opacity: 0;
      transform: scale(0.5);
    }
    100% {
      opacity: 1;
      transform: scale(1);
    }
  }

  ${Ir};
  ${Lr};
  ${et};
`,Lo=({dialogLabelId:r,title:t,subtitle:a,dialogDescriptionId:n,onClose:i})=>{const d=o(()=>{i("close-button")},[i]);return e.createElement(Go.Header,null,e.createElement(ot,null,e.createElement(ot,{px:2,py:"6px",flexDirection:"column",flexGrow:1},e.createElement(Go.Title,{id:r},null!=t?t:"Dialog"),a&&e.createElement(Go.Subtitle,{id:n},a)),e.createElement(Go.CloseButton,{onClose:d})))};Lo.displayName="DefaultHeader";const Ro=({children:r})=>e.createElement(Go.Body,null,r);Ro.displayName="DefaultBody";const Mo=({footerButtons:r})=>{const{containerRef:t}=ko({bindKeys:Et.ArrowHorizontal|Et.Tab,focusInStrategy:"closest"});return r?e.createElement(Go.Footer,{ref:t},e.createElement(Go.Buttons,{buttons:r})):null},Ho=e.forwardRef((r,o)=>{const{title:a="Dialog",subtitle:n="",renderHeader:i,renderBody:d,renderFooter:l,onClose:c,role:s="dialog",width:f="xl",height:b="auto"}=r,g=It(),p=It(),u={...r,title:a,subtitle:n,role:s,dialogLabelId:g,dialogDescriptionId:p},m=t(null),h=Co(m,o),x=t(null);wo({containerRef:m,restoreFocusOnCleanUp:!0}),vt(e=>{c("escape"),e.preventDefault()},[c]);const v=(null!=i?i:Lo)(u),y=(null!=d?d:Ro)(u),w=(null!=l?l:Mo)(u);return e.createElement(e.Fragment,null,e.createElement(Eo,null,e.createElement(Io,{ref:x},e.createElement(zo,{width:f,height:b,ref:h,role:s,"aria-labelledby":g,"aria-describedby":p},v,y,w))))});Ho.displayName="Dialog";const Oo=c(rt).attrs({as:"header"})`
  box-shadow: 0 1px 0 ${Cr("colors.border.overlay")};
  padding: ${Cr("space.2")};
  z-index: 1;
  flex-shrink: 0;
`,Po=c(rt)`
  font-size: ${Cr("fontSizes.1")};
  font-weight: ${Cr("fontWeights.bold")};
`,Do=c(rt)`
  font-size: ${Cr("fontSizes.0")};
  margin-top: ${Cr("space.1")};
  color: ${Cr("colors.text.tertiary")};
`,Wo=c(rt)`
  flex-grow: 1;
  overflow: auto;
  padding: ${Cr("space.3")};
`,jo=c(rt).attrs({as:"footer"})`
  box-shadow: 0 -1px 0 ${Cr("colors.border.overlay")};
  padding: ${Cr("space.3")};
  display: flex;
  flex-flow: wrap;
  justify-content: flex-end;
  z-index: 1;
  flex-shrink: 0;

  button {
    margin-left: ${Cr("space.1")};
    &:first-child {
      margin-left: 0;
    }
  }
`,Fo={normal:Wt,primary:Ut,danger:jt},_o=c(Wt)`
  border-radius: 4px;
  background: transparent;
  border: 0;
  vertical-align: middle;
  color: ${Cr("colors.text.secondary")};
  padding: ${Cr("space.2")};
  align-self: flex-start;
  line-height: normal;
  box-shadow: none;
`,Uo=({onClose:r})=>e.createElement(_o,{"aria-label":"Close",onClick:r},e.createElement($o,{icon:ao}));Uo.displayName="CloseButton";const Go=Object.assign(Ho,{Header:Oo,Title:Po,Subtitle:Do,Body:Wo,Footer:jo,Buttons:({buttons:r})=>{const o=t(null);let n=0;return a(()=>{o.current&&o.current.focus()},[]),e.createElement(e.Fragment,null,r.map((r,t)=>{const{content:a,buttonType:i="normal",autoFocus:d=!1,...l}=r,c=Fo[i];return e.createElement(c,Hr({key:t},l,{ref:d&&0===n?(n++,o):null}),a)}))},CloseButton:Uo}),Ko=c.header`
  padding: ${Cr("space.2")};
  display: flex;
  flex-direction: row;
`,Vo=c(rt)`
  font-size: ${Cr("fontSizes.3")};
  font-weight: 700;
  padding: 6px ${Cr("space.2")};
  flex-grow: 1;
`,Xo=({title:r,onClose:t,dialogLabelId:a})=>{const n=o(()=>{t("close-button")},[t]);return e.createElement(Ko,null,e.createElement(Vo,{id:a},r),e.createElement(Go.CloseButton,{onClose:n}))};Xo.displayName="ConfirmationHeader";const Yo=c(rt)`
  font-size: ${Cr("fontSizes.1")};
  padding: 0 ${Cr("space.3")} ${Cr("space.3")} ${Cr("space.3")};
  color: ${Cr("colors.text.tertiary")};
  flex-grow: 1;
`,qo=({children:r})=>e.createElement(Yo,null,r);qo.displayName="ConfirmationBody";const Jo=c(rt)`
  display: flex;
  flex-shrink: 0;
  button {
    font-size: ${Cr("fontSizes.1")};
    flex: 1 1 0;
    border-bottom: 0;
    margin: 0;
    border-right: 0;
    &:first-child {
      border-left: 0;
    }
    &:focus {
      // this is a bit of a hack to get the focus outlines to show up
      z-index: 1;
    }
  }
  button:first-of-type:not(:last-of-type) {
    // All except border-bottom-left-radius, which should follow the curvature of the container
    border-top-left-radius: 0;
    border-top-right-radius: 0;
    border-bottom-right-radius: 0;
  }
  button:last-of-type:not(:first-of-type) {
    // All except border-bottom-right-radius, which should follow the curvature of the container
    border-top-left-radius: 0;
    border-top-right-radius: 0;
    border-bottom-left-radius: 0;
  }
  button:first-of-type:last-of-type {
    // When the footer contains a single button, it should follow the curvature of the container
    border-top-left-radius: 0;
    border-top-right-radius: 0;
  }
  button:not(:first-of-type):not(:last-of-type) {
    border-radius: 0;
  }
`,Zo=({footerButtons:r})=>{const{containerRef:t}=ko({bindKeys:Et.ArrowHorizontal|Et.Tab,focusInStrategy:"closest"});return e.createElement(Jo,{ref:t},e.createElement(Go.Buttons,{buttons:null!=r?r:[]}))};Zo.displayName="ConfirmationFooter";const Qo=r=>{const{onClose:t,title:a,cancelButtonContent:n="Cancel",confirmButtonContent:i="OK",confirmButtonType:d="normal",children:l}=r,c=[{content:n,onClick:o(()=>{t("cancel")},[t]),autoFocus:!0,variant:"large"},{content:i,buttonType:d,variant:"large",onClick:o(()=>{t("confirm")},[t])}];return e.createElement(Go,{onClose:t,title:a,footerButtons:c,role:"alertdialog",width:"md",renderHeader:Xo,renderBody:qo,renderFooter:Zo},l)};function ea(){const{theme:r,colorMode:t,dayScheme:a,nightScheme:n}=Jr();return o(o=>async function(r,t){const{content:o,...a}=t;return new Promise(t=>{const n=document.createElement("div");u.render(e.createElement(qr,r,e.createElement(Qo,Hr({},a,{onClose:e=>{u.unmountComponentAtNode(n),t("confirm"===e)}}),o)),n)})}({theme:r,colorMode:t,dayScheme:a,nightScheme:n},o),[r,t,a,n])}Qo.displayName="ConfirmationDialog";const ra=c.div`
   {
    /* 6px vertical padding + 20px line height = 32px total height
     *
     * TODO: When rem-based spacing on a 4px scale lands, replace
     * hardcoded '6px' with 'calc((${Cr("space.s32")} - ${Cr("space.20")}) / 2)'.
     */
  }
  padding: 6px ${Cr("space.3")};
  font-size: ${Cr("fontSizes.0")};
  font-weight: ${Cr("fontWeights.bold")};
  color: ${Cr("colors.text.secondary")};

  ${({variant:e})=>"filled"===e&&b(["background:",";margin:"," 0;border-top:1px solid ",";border-bottom:1px solid ",";&:first-child{margin-top:0;}"],Cr("colors.bg.tertiary"),Cr("space.2"),Cr("colors.border.tertiary"),Cr("colors.border.tertiary"))}

  ${et}
`;function ta({variant:r="subtle",title:t,auxiliaryText:o,children:a,...n}){return e.createElement(ra,Hr({role:"heading",variant:r},n),t,o&&e.createElement("span",null,"auxiliaryText"))}ta.displayName="Header";const oa=c.div`
  ${et}
`;function aa({header:r,items:t,...o}){return e.createElement(oa,o,r&&e.createElement(ta,r),t)}aa.displayName="Group";const na=c.div`
  height: 1px;
  background: ${Cr("colors.selectMenu.borderSecondary")};
  margin-top: calc(${Cr("space.2")} - 1px);
  margin-bottom: ${Cr("space.2")};
`;function ia(){return e.createElement(na,null)}ia.displayName="Divider",ia.renderItem=ia;const da=(e="default",r)=>{if(r)return{color:Cr("colors.text.disabled"),iconColor:Cr("colors.text.disabled"),annotationColor:Cr("colors.text.disabled"),hoverBackground:"inherit",hoverCursor:"default"};switch(e){case"danger":return{color:Cr("colors.text.danger"),iconColor:Cr("colors.icon.danger"),annotationColor:Cr("colors.text.disabled"),hoverBackground:Cr("colors.bg.danger"),hoverCursor:"pointer"};default:return{color:"inherit",iconColor:Cr("colors.text.disabled"),annotationColor:Cr("colors.text.disabled"),hoverBackground:Cr("colors.selectMenu.tapHighlight"),hoverCursor:"pointer"}}},la=c.div`
  width: 100%;
`,ca=c.div`
  /* 6px vertical padding + 20px line height = 32px total height
   *
   * TODO: When rem-based spacing on a 4px scale lands, replace
   * hardcoded '6px' with 'calc((${Cr("space.s32")} - ${Cr("space.20")}) / 2)'.
   */
  padding: 6px ${Cr("space.2")};
  display: flex;
  border-radius: ${Cr("radii.2")};
  color: ${({variant:e,item:r})=>da(e,null==r?void 0:r.disabled).color};

  @media (hover: hover) and (pointer: fine) {
    :hover {
      background: ${({variant:e,item:r})=>da(e,null==r?void 0:r.disabled).hoverBackground};
      cursor: ${({variant:e,item:r})=>da(e,null==r?void 0:r.disabled).hoverCursor};
    }
  }

  // Item dividers
  :not(:first-of-type):not(${na} + &):not(${ra} + &) {
    margin-top: ${({showDivider:e})=>e?"1px":"0"};

    ${la}::before {
      content: ' ';
      display: block;
      position: relative;
      top: -7px;
      // NB: This 'get' wont execute if its moved into the arrow function below.
      border: 0 solid ${Cr("colors.selectMenu.borderSecondary")};
      border-top-width: ${({showDivider:e})=>e?"1px":"0"};
    }
  }

  ${et}
`,sa=c.div`
  display: flex;
  flex-direction: ${({descriptionVariant:e})=>"inline"===e?"row":"column"};
`,fa=c.div`
  // Match visual height to adjacent text line height.
  // TODO: When rem-based spacing on a 4px scale lands, replace
  // hardcoded '20px' with '${Cr("space.s20")}'.
  height: 20px;
  width: ${Cr("space.3")};
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin-right: ${Cr("space.2")};

  svg {
    fill: ${({variant:e,disabled:r})=>da(e,r).iconColor};
    font-size: ${Cr("fontSizes.0")};
  }
`,ba=c(fa)``,ga=c(fa)`
  color: ${({variant:e,disabled:r})=>da(e,r).annotationColor}};
  margin-left: auto;
  margin-right: 0;
  div:nth-child(2) {
    margin-left: ${Cr("space.2")};
  }
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
`,pa=c.span`
  color: ${Cr("colors.text.secondary")};
  margin-left: ${({descriptionVariant:e})=>"inline"===e?Cr("space.2"):0};
`;function ua(r){const{text:t,description:a,descriptionVariant:n="inline",selected:i,selectionVariant:d,leadingVisual:l,trailingIcon:c,trailingText:s,variant:f="default",showDivider:b,disabled:g,onAction:p,onKeyPress:u,children:m,onClick:h,id:x,...v}=r,y=o(e=>{if(g)return;null==u||u(e);e.target instanceof HTMLInputElement&&"checkbox"===e.target.type&&" "===e.key||!e.defaultPrevented&&[" ","Enter"].includes(e.key)&&(null==p||p(r,e))},[p,g,r,u]),w=o(e=>{g||(null==h||h(e),e.defaultPrevented||null==p||p(r,e))},[p,g,r,h]);return e.createElement(ca,Hr({tabIndex:g?void 0:-1,variant:f,showDivider:b,"aria-selected":i},v,{"data-id":x,onKeyPress:y,onClick:w}),!!i===i&&e.createElement(ba,null,"multiple"===d?e.createElement(e.Fragment,null,e.createElement("input",{type:"checkbox",checked:i,"aria-label":t,readOnly:!0,"aria-readonly":"false"})):i&&e.createElement(qt,null)),l&&e.createElement(ba,{variant:f,disabled:g},e.createElement(l,null)),e.createElement(la,null,e.createElement(ot,null,m,(t||a)&&e.createElement(sa,{descriptionVariant:n},t&&e.createElement("div",null,t),a&&e.createElement(pa,{descriptionVariant:n},a)),(c||s)&&e.createElement(ga,{variant:f,disabled:g},s&&e.createElement("div",null,s),c&&e.createElement("div",null,e.createElement(c,null))))))}ua.displayName="Item";const ma=c.div`
  font-size: ${Cr("fontSizes.1")};
  /* 14px font-size * 1.428571429 = 20px line height
   *
   * TODO: When rem-based spacing on a 4px scale lands, replace
   * hardcoded '20px'
   */
  line-height: 20px;
`;function ha(r){var t;const{firstGroupStyle:o,lastGroupStyle:a,headerStyle:n,itemStyle:d}=function(e="inset"){switch(e){case"full":return{headerStyle:{paddingX:Cr("space.2")},itemStyle:{borderRadius:0}};default:return{firstGroupStyle:{marginTop:Cr("space.2")},lastGroupStyle:{marginBottom:Cr("space.2")},itemStyle:{marginX:Cr("space.2")}}}}(r.variant),l=(t,o)=>{var a,n,i;const l="renderItem"in t&&t.renderItem||r.renderItem||ua,c=null!==(a=null!==(n=t.key)&&void 0!==n?n:null===(i=t.id)||void 0===i?void 0:i.toString())&&void 0!==a?a:It();return e.createElement(l,Hr({showDivider:r.showItemDividers,selectionVariant:r.selectionVariant},t,{key:c,sx:{...d,...t.sx},item:o}))};let c=[];const s=i(It,[]);if(function(e){return"groupMetadata"in e}(r)){const e=r.groupMetadata.reduce((e,r)=>e.set(r.groupId,r),new Map);for(const t of r.items){var f;const r=e.get(t.groupId);e.set(t.groupId,{...r,items:[...null!==(f=null==r?void 0:r.items)&&void 0!==f?f:[],l({showDivider:null==r?void 0:r.showItemDividers,...r&&"renderItem"in r&&{renderItem:r.renderItem},...t},t)]})}c=[...e.values()]}else{var b;c=[{items:null===(b=r.items)||void 0===b?void 0:b.map(e=>l(e,e)),groupId:s}]}return e.createElement(ma,r,null===(t=c)||void 0===t?void 0:t.map(({header:t,...i},d)=>{const l="filled"===(null==t?void 0:t.variant),s=d>0&&!l;return e.createElement(e.Fragment,{key:i.groupId},s?e.createElement(ia,{key:i.groupId+"-divider"}):null,(t=>{var o;const a=(null!==(o="renderGroup"in t&&t.renderGroup)&&void 0!==o?o:r.renderGroup)||aa;return e.createElement(a,Hr({},t,{key:t.groupId}))})({sx:{...0===d&&o,...d===c.length-1&&a,...d>0&&!s&&{mt:2}},...t&&{header:{...t,sx:{...n,...null==t?void 0:t.sx}}},...i}))}))}ha.displayName="List";const xa=Object.assign(ha,{Group:aa,Item:ua,Divider:ia}),va={sm:"480px",md:"640px",auto:"auto"},ya={sm:"256px",md:"320px",lg:"480px",xl:"640px",auto:"auto"},wa=c.div`
  background-color: ${Cr("colors.bg.overlay")};
  box-shadow: 0px 1px 3px rgba(0, 0, 0, 0.12), 0px 8px 24px rgba(149, 157, 165, 0.2);
  position: absolute;
  min-width: 192px;
  max-width: 640px;
  height: ${e=>va[e.height||"auto"]};
  width: ${e=>ya[e.width||"auto"]};
  border-radius: 12px;
  animation: overlay-appear 200ms ${Cr("animation.easeOutCubic")};

  @keyframes overlay-appear {
    0% {
      opacity: 0;
      transform: translateY(${Cr("space.2")});
    }
    100% {
      opacity: 1;
    }
  }
  visibility: ${e=>e.visibility||"visible"};
  ${Ir};
  ${Lr};
  ${et};
`,Ba=e.forwardRef(({onClickOutside:r,role:o="dialog",initialFocusRef:a,returnFocusRef:n,ignoreClickRefs:i,onEscape:d,visibility:l,...c},s)=>{const f=t(null),b=Co(f,s),g=wt({overlayRef:f,returnFocusRef:n,onEscape:d,ignoreClickRefs:i,onClickOutside:r,initialFocusRef:a});return e.createElement(Eo,null,e.createElement(wa,Hr({},g,{"aria-modal":"true",role:o},c,{ref:b,visibility:l})))});Ba.defaultProps={height:"auto",width:"auto"};const $a=({renderAnchor:a,children:n,open:d,onOpen:l,onClose:c,height:s,width:f,overlayProps:b,focusZoneSettings:g})=>{const p=t(null),[u,m]=function(){const[e,a]=r(null),n=t(null);n.current=e;const i=o(e=>{n.current=e,a(e)},[n]);return[n,i]}(),h=i(It,[]),x=o(()=>null==c?void 0:c("click-outside"),[c]),v=o(()=>null==c?void 0:c("escape"),[c]),y=o(e=>{e.defaultPrevented||!d&&["ArrowDown","ArrowUp"," ","Enter"].includes(e.key)&&(null==l||l("anchor-key-press"),e.preventDefault())},[d,l]),w=o(e=>{e.defaultPrevented||0!==e.button||d||null==l||l("anchor-click")},[d,l]),{position:B}=function(r,t=[]){const o=yt(null==r?void 0:r.floatingElementRef),a=yt(null==r?void 0:r.anchorElementRef),[n,i]=e.useState(void 0);return e.useEffect(()=>{o.current instanceof Element&&a.current instanceof Element?i(co(o.current,a.current,r)):i(void 0)},[o,a,...t]),{floatingElementRef:o,anchorElementRef:a,position:n}}({anchorElementRef:p,floatingElementRef:u},[u.current]),$=i(()=>B&&{top:B.top+"px",left:B.left+"px"},[B]);return ko({containerRef:u,disabled:!d||!B,...g}),wo({containerRef:u,disabled:!d||!B}),e.createElement(e.Fragment,null,a({ref:p,id:h,"aria-labelledby":h,"aria-haspopup":"listbox",tabIndex:0,onClick:w,onKeyDown:y}),d?e.createElement(Ba,Hr({returnFocusRef:p,onClickOutside:x,onEscape:v,ref:m,role:"listbox",visibility:B?"visible":"hidden",height:s,width:f},$,b),n):null)};$a.displayName="AnchoredOverlay";const ka=r=>e.createElement(ua,Hr({role:"menuitem"},r));ka.displayName="ActionMenuItem",ka.displayName="ActionMenu.Item";const Sa=({anchorContent:t,renderAnchor:a=(r=>e.createElement(Wt,r)),onAction:n,open:d,setOpen:l,overlayProps:c,items:s,...f})=>{const[b,g]=function(e,t,a){const[n,i]=r(a);return[null!=e?e:n,o(e=>{i(e),t&&t(e)},[t])]}(d,l,!1),p=o(()=>g(!0),[g]),u=o(()=>g(!1),[g]),m=o(e=>a({"aria-label":"menu",children:t,...e}),[t,a]),h=i(()=>s.map(e=>({...e,role:"menuitem",onAction:(r,t)=>{var o;const a=null!==(o=e.onAction)&&void 0!==o?o:n;null==a||a(r,t),t.defaultPrevented||u()}})),[s,n,u]);return e.createElement($a,{renderAnchor:m,open:b,onOpen:p,onClose:u,overlayProps:c},e.createElement(ha,Hr({},f,{role:"menu",items:h})))};Sa.displayName="ActionMenuBase",Sa.displayName="ActionMenu";const Ta=Object.assign(Sa,{Divider:ia,Item:ka});const Ea=c.img.attrs(e=>({height:e.size,width:e.size}))`
  display: inline-block;
  overflow: hidden; // Ensure page layout in Firefox should images fail to load
  line-height: ${Cr("lineHeights.condensedUltra")};
  vertical-align: middle;
  border-radius: ${e=>function({size:e,square:r}){return r?e&&e<=24?"4px":"6px":"50%"}(e)};
  ${Ir};
  ${et}
`;Ea.defaultProps={size:20,alt:"",square:!1};const Ca=c(Ea)`
  position: absolute;
  right: -15%;
  bottom: -9%;
  box-shadow: ${Cr("shadows.avatar.childShadow")};
`,Ia=({children:r,...t})=>{const o=e.Children.map(r,(r,t)=>e.isValidElement(r)?0===t?e.cloneElement(r,{size:40}):e.createElement(Ca,Hr({bg:"bg.canvas"},r.props,{size:20})):r);return e.createElement(lt,Hr({display:"inline-flex"},t),o)};Ia.displayName="AvatarPair",Ia.displayName="AvatarPair";var Na=function(e,r,t){return e(t={path:r,exports:{},require:function(e,r){return function(){throw new Error("Dynamic requires are not currently supported by @rollup/plugin-commonjs")}(null==r&&t.path)}},t.exports),t.exports}((function(e){
/*!
  Copyright (c) 2018 Jed Watson.
  Licensed under the MIT License (MIT), see
  http://jedwatson.github.io/classnames
*/
!function(){var r={}.hasOwnProperty;function t(){for(var e=[],o=0;o<arguments.length;o++){var a=arguments[o];if(a){var n=typeof a;if("string"===n||"number"===n)e.push(a);else if(Array.isArray(a)){if(a.length){var i=t.apply(null,a);i&&e.push(i)}}else if("object"===n)if(a.toString===Object.prototype.toString)for(var d in a)r.call(a,d)&&a[d]&&e.push(d);else e.push(a.toString())}}return e.join(" ")}e.exports?(t.default=t,e.exports=t):window.classNames=t}()}));const Aa=c.span`
  display: flex;
  position: relative;
  height: 20px;
  min-width: ${e=>1===e.count?"20px":2===e.count?"30px":"38px"};

  .pc-AvatarItem {
    flex-shrink: 0;
    height: 20px;
    width: 20px;
    box-shadow: 0 0 0 1px ${Cr("colors.bg.canvas")};
    margin-left: -11px;
    position: relative;
    overflow: hidden;
    transition: margin 0.2s ease-in-out, opacity 0.2s ease-in-out, visibility 0.2s ease-in-out,
      box-shadow 0.1s ease-in-out;

    &:first-child {
      margin-left: 0;
    }
    &:nth-child(n + 4) {
      display: none;
    }
  }

  &.pc-AvatarStack--two {
    min-width: 30px;
    .pc-AvatarItem {
      &:nth-child(n + 3) {
        display: none;
      }
    }
  }

  &.pc-AvatarStack--three-plus {
    min-width: 38px;
    .pc-AvatarItem {
      &:nth-child(3) {
        opacity: ${55}%;
        margin-left: -17px;
      }
      &:nth-child(4) {
        opacity: ${40}%;
        margin-left: -17px;
      }
      &:nth-child(5) {
        opacity: ${25}%;
        margin-left: -17px;
      }
      &:nth-child(n + 4) {
        display: block;
      }
      &:nth-child(n + 6) {
        opacity: 0;
        visibility: hidden;
      }
    }
  }

  &.pc-AvatarStack--right {
    justify-content: flex-end;
    .pc-AvatarItem {
      margin-left: 0 !important;
      margin-right: -11px;

      &:first-child {
        margin-right: 0;
      }
    }

    .pc-AvatarStackBody {
      flex-direction: row-reverse;

      &:hover {
        .pc-AvatarItem {
          margin-right: ${Cr("space.1")}!important;
          margin-left: 0 !important;

          &:first-child {
            margin-right: 0 !important;
          }
        }
      }
    }
  }

  &.pc-AvatarStack--three-plus.pc-AvatarStack--right {
    .pc-AvatarItem {
      &:nth-child(3) {
        margin-right: -17px;
      }
      &:nth-child(4) {
        margin-right: -17px;
      }
      &:nth-child(5) {
        margin-right: -17px;
      }
    }
  }

  .pc-AvatarStackBody:hover {
    width: auto;

    .pc-AvatarItem {
      margin-left: ${Cr("space.1")};
      opacity: 100%;
      visibility: visible;
      box-shadow: 0 0 0 4px ${Cr("colors.bg.canvas")};
      &:first-child {
        margin-left: 0;
      }
    }
  }

  ${Ir}
  ${et};
`,za=({children:r,alignRight:t,...o})=>{const a=e.Children.count(r),n=Na({"pc-AvatarStack--two":2===a,"pc-AvatarStack--three-plus":a>2,"pc-AvatarStack--right":t});return e.createElement(Aa,Hr({count:a,className:n},o),e.createElement(it,{display:"flex",width:"38px",className:"pc-AvatarStackBody"},(r=>e.Children.map(r,(r,t)=>e.isValidElement(r)?e.cloneElement(r,{className:Na(r.props.className,"pc-AvatarItem"),sx:{zIndex:10-t,...r.props.sx}}):r))(r)))};za.displayName="AvatarStack";const La=c.a`
  display: inline-block;
  padding: 2px 6px;
  font-size: ${Cr("fontSizes.0")};
  font-family: ${Cr("fonts.mono")};
  color: ${Cr("colors.branchName.text")};
  background-color: ${Cr("colors.branchName.bg")};
  border-radius: ${Cr("radii.2")};
  ${Ir};
  ${et};
`,Ra=c.li`
  display: inline-block;
  white-space: nowrap;
  list-style: none;
  &::after {
    padding-right: 0.5em;
    padding-left: 0.5em;
    color: ${Cr("colors.text.disabled")};
    font-size: ${Cr("fontSizes.1")};
    content: '/';
  }
  &:first-child {
    margin-left: 0;
  }
  &:last-child {
    &::after {
      content: none;
    }
  }
`,Ma=c.nav`
  display: flex;
  justify-content: space-between;
  ${Ir};
  ${Rr};
  ${et};
`;function Ha({className:r,children:t,theme:o,...a}){const n=Na(r,"Breadcrumb"),i=e.Children.map(t,r=>e.createElement(Ra,{theme:o},r));return e.createElement(Ma,Hr({className:n,"aria-label":"breadcrumb",theme:o},a),e.createElement(rt,{as:"ol",my:0,pl:0},i))}Ha.displayName="Breadcrumb";const Oa=c.a.attrs(e=>({activeClassName:"string"==typeof e.to?"selected":"",className:Na(e.selected&&"selected",e.className),"aria-current":e.selected?"page":null}))`
  color: ${Cr("colors.text.link")};
  display: inline-block;
  font-size: ${Cr("fontSizes.1")};
  text-decoration: none;
  &:hover {
    text-decoration: underline;
  }
  &.selected {
    color: ${Cr("colors.text.primary")};
    pointer-events: none;
  }
  ${Ir}
  ${et};
`;Ha.displayName="Breadcrumb",Oa.displayName="Breadcrumb.Item";var Pa=Object.assign(Ha,{Item:Oa});const Da={top:"Bottom",right:"Left",bottom:"Top",left:"Right"},Wa={top:"Left",right:"Top",bottom:"Left",left:"Top"};function ja(e,r,t){const o=Da[e].toLowerCase(),a=Wa[e].toLowerCase();return{[o]:"100%",[r||a]:r?t:"50%"}}const Fa=$r({prop:"bg",key:"colors"}),_a=$r({prop:"borderColor",key:"colors"}),Ua=$r({prop:"borderWidth",key:"borderWidths",scale:[0,1]});function Ga(r){var t;const o=e.useContext(g),a={...r,theme:null!==(t=r.theme)&&void 0!==t?t:o},{bg:n}=Fa(a),{borderColor:i}=_a(a),{borderWidth:d}=Ua(a),{size:l=8,location:c="bottom"}=r,[s,f]=function(e){const[r,t]=e.split("-");return[r,t]}(c),b=Wa[s],p=[-l,0],u=[0,l],m=[l,0],h=`M${p}L${u}L${m}L${p}Z`,x=`M${p}L${u}L${m}`,v={top:`translate(${[l,2*l]}) rotate(180)`,right:`translate(${[0,l]}) rotate(-90)`,bottom:`translate(${[l,0]})`,left:`translate(${[2*l,l]}) rotate(90)`}[s];return e.createElement("svg",{width:2*l,height:2*l,style:{pointerEvents:"none",position:"absolute",...ja(s,f,l),["margin"+b]:f?null:-l}},e.createElement("g",{transform:v},e.createElement("path",{d:h,fill:n}),e.createElement("path",{d:x,fill:"none",stroke:i,strokeWidth:d})))}Ga.displayName="Caret",Ga.locations=["top","top-left","top-right","right","right-top","right-bottom","bottom","bottom-left","bottom-right","left","left-top","left-bottom"],Ga.defaultProps={bg:"bg.canvas",borderColor:"border.primary",borderWidth:1};const Ka={small:56,medium:96,large:128},Va=c.div`
  display: ${e=>e.inline?"inline-flex":"flex"};
  align-items: center;
  justify-content: center;
  background-color: ${Cr("colors.bg.canvas")};
  border-radius: 50%;
  box-shadow: ${Cr("shadows.shadow.medium")};
  ${Ir};
  ${({size:e,variant:r="medium"})=>{const t=(o=e,!isNaN(parseFloat(o))&&isFinite(o)?e:Ka[r]);var o;return{width:t,height:t}}};
  ${et};
`,Xa=c($o)`
  height: auto;
  max-width: 60%;
  max-height: 55%;
`;Va.defaultProps={inline:!1},Xa.displayName="CircleBadge.Icon";var Ya=Object.assign(Va,{Icon:Xa});function qa(r){const{size:t,as:o}=r,{icon:a,bg:n,...i}=r;return e.createElement(tt,{as:o,bg:n,overflow:"hidden",borderWidth:0,size:t,borderRadius:"50%"},e.createElement(ot,Hr({},i,{alignItems:"center",justifyContent:"center"}),e.createElement(a,{size:t})))}qa.displayName="CircleOcticon",qa.defaultProps={...ot.defaultProps,size:32};const Ja=c.span`
  display: inline-block;
  padding: 2px 5px;
  font-size: ${Cr("fontSizes.0")};
  font-weight: ${Cr("fontWeights.bold")};
  line-height: ${Cr("lineHeights.condensedUltra")};
  border-radius: 20px;
  ${({scheme:e,...r})=>({color:"secondary"===e?Cr("colors.counter.text")(r):"primary"===e?Cr("colors.counter.primary.text")(r):Cr("colors.counter.text")(r)})};
  ${({scheme:e,...r})=>({backgroundColor:"secondary"===e?Cr("colors.counter.bg")(r):"primary"===e?Cr("colors.counter.primary.bg")(r):Cr("colors.counter.bg")(r)})};
  ${Ir};

  &:empty {
    display: none;
  }

  ${et};
`,Za=c.details`
  & > summary {
    list-style: none;
  }
  & > summary::-webkit-details-marker {
    display: none;
  }

  ${Ir}
  ${et};
`;Za.displayName="Details";const Qa=()=>null;function en(e){const r=e;return r.tabIndex>=0&&!r.disabled&&function(e){return!e.hidden&&(!e.type||"hidden"!==e.type)&&(e.offsetWidth>0||e.offsetHeight>0)}(r)}const rn=c.span`
  ${Nr};
  ${Ir};
  ${et};
`,tn=()=>null,on=c.div`
  box-shadow: ${Cr("shadows.shadow.large")};
  border-radius: ${Cr("radii.2")};
  position: fixed;
  top: 0;
  left: 50%;
  transform: translateX(-50%);
  max-height: 80vh;
  z-index: 999;
  margin: 10vh auto;
  background-color: ${Cr("colors.bg.primary")};
  width: ${e=>e.narrow?"320px":e.wide?"640px":"440px"};
  outline: none;

  @media screen and (max-width: 750px) {
    width: 100vw;
    margin: 0;
    border-radius: 0;
    height: 100vh;
  }

  ${zr};
  ${Ir};
  ${et};
`,an=c(ot)`
  border-radius: ${Cr("radii.2")} ${Cr("radii.2")} 0px 0px;
  border-bottom: 1px solid ${Cr("colors.border.primary")};

  @media screen and (max-width: 750px) {
    border-radius: 0px;
  }

  ${et};
`;function nn({theme:r,children:t,backgroundColor:o="gray.1",...a}){return e.Children.toArray(t).every(e=>"string"==typeof e)&&(t=e.createElement(rn,{theme:r,color:"text.primary",fontSize:1,fontWeight:"bold",fontFamily:"sans-serif"},t)),e.createElement(an,Hr({theme:r,p:3,backgroundColor:o},a),t)}nn.displayName="DialogHeader";const dn=c.span`
  &:before {
    position: fixed;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    z-index: 80;
    display: block;
    cursor: default;
    content: ' ';
    background: transparent;
    z-index: 99;
    background: ${Cr("colors.fade.black50")};
  }
`,ln=n(({children:r,onDismiss:n=tn,isOpen:i,initialFocusRef:d,returnFocusRef:l,...c},s)=>{var f;const b=t(null),g=t(null),p=null!==(f=s)&&void 0!==f?f:b,u=t(null),m=()=>{n(),l&&l.current&&l.current.focus()},{getDialogProps:h}=function({modalRef:e,overlayRef:r,isOpen:t,onDismiss:n=Qa,initialFocusRef:i,closeButtonRef:d}){const l=o(t=>{e.current&&r.current&&!e.current.contains(t.target)&&r.current.contains(t.target)&&n()},[n,e,r]);a(()=>{if(t)return document.addEventListener("click",l),()=>{document.removeEventListener("click",l)}},[t,l]),a(()=>{t&&(i&&i.current?i.current.focus():d&&d.current&&d.current.focus())},[t,i,d]);const c=o((r,t)=>{if(e&&e.current){const o=Array.from(e.current.querySelectorAll("*")).filter(en);if(0===o.length)return;r.preventDefault();const a=document.activeElement;if(!a)return;const n=o.indexOf(a)+t,i=1===t?0:o.length-1;return o[n]||o[i]}},[e]),s=o(e=>{const r=e.shiftKey?-1:1,t=c(e,r);t&&t.focus()},[c]),f=o(e=>{switch(e.key){case"Tab":s(e);break;case"Escape":n(),e.stopPropagation()}},[s,n]);return{getDialogProps:()=>({onKeyDown:f})}}({modalRef:p,onDismiss:m,isOpen:i,initialFocusRef:d,closeButtonRef:u,returnFocusRef:l,overlayRef:g});return i?e.createElement(e.Fragment,null,e.createElement(dn,{ref:g}),e.createElement(on,Hr({tabIndex:-1,ref:p,role:"dialog","aria-modal":"true"},c,h()),e.createElement(io,{ref:u,onClick:m,sx:{position:"absolute",top:"16px",right:"16px"}}),r)):null});nn.defaultProps={backgroundColor:"bg.tertiary"},nn.propTypes={...ot.propTypes},nn.displayName="Dialog.Header",ln.displayName="Dialog";var cn=Object.assign(ln,{Header:nn});const sn=c(Za)`
  position: relative;
  display: inline-block;
`,fn=({children:r,className:t,...o})=>{const{getDetailsProps:a}=st({closeOnOutsideClick:!0});return e.createElement(sn,Hr({className:t},a(),o),r)};fn.displayName="Dropdown";const bn=({children:r,...t})=>e.createElement(Wt,Hr({as:"summary","aria-haspopup":"true"},t),r,e.createElement(gn,null));bn.displayName="DropdownButton";const gn=c.div`
  border: 4px solid transparent;
  margin-left: 12px;
  border-top-color: currentcolor;
  border-bottom-width: 0;
  content: '';
  display: inline-block;
  height: 0;
  vertical-align: middle;
  width: 0;
  ${Ir};
  ${et};
`,pn=c.ul`
  background-clip: padding-box;
  background-color: ${Cr("colors.bg.overlay")};
  border: 1px solid ${Cr("colors.border.overlay")};
  border-radius: ${Cr("radii.2")};
  box-shadow: ${Cr("shadows.dropdown.shadow")};
  left: 0;
  list-style: none;
  margin-top: 2px;
  padding: 5px 0 5px 0 !important; //TODO: fix this override on our markdown styles
  position: absolute;
  top: 100%;
  width: 160px;
  z-index: 100;

  &::before {
    position: absolute;
    display: inline-block;
    content: '';
  }

  &::after {
    position: absolute;
    display: inline-block;
    content: '';
  }

  &::before {
    border: 8px solid transparent;
    border-bottom-color: ${Cr("colors.bg.overlay")};
  }

  &::after {
    border: 7px solid transparent;
    border-bottom-color: ${Cr("colors.bg.overlay")};
  }

  // stylelint-disable-next-line selector-max-type
  > ul {
    list-style: none;
  }
  ${e=>e.direction?((e,r)=>({w:`\n      top: 0;\n      right: 100%;\n      left: auto;\n      width: auto;\n      margin-top: 0;\n      margin-right: 10px;\n\n      &::before {\n        top: 10px;\n        right: -16px;\n        left: auto;\n        border-color: transparent;\n        border-left-color: ${Cr("colors.border.overlay")(e)};\n      }\n\n      &::after {\n        top: 11px;\n        right: -14px;\n        left: auto;\n        border-color: transparent;\n        border-left-color: ${Cr("colors.border.overlay")(e)};\n      }\n    `,e:`\n      top: 0;\n      left: 100%;\n      width: auto;\n      margin-top: 0;\n      margin-left: 10px;\n\n      &::before {\n        top: 10px;\n        left: -16px;\n        border-color: transparent;\n        border-right-color: ${Cr("colors.border.overlay")(e)};\n      }\n\n      &::after {\n        top: 11px;\n        left: -14px;\n        border-color: transparent;\n        border-right-color: ${Cr("colors.border.overlay")(e)};\n      }\n    `,ne:`\n      top: auto;\n      bottom: 100%;\n      left: 0;\n      margin-bottom: 3px;\n\n      &::before,\n      &::after {\n        top: auto;\n        right: auto;\n      }\n\n      &::before {\n        bottom: -8px;\n        left: 9px;\n        border-top: 8px solid ${Cr("colors.border.overlay")(e)};\n        border-bottom: 0;\n        border-left: 8px solid transparent;\n      }\n\n      &::after {\n        bottom: -7px;\n        left: 10px;\n        border-top: 7px solid ${Cr("colors.border.overlay")(e)};\n        border-right: 7px solid transparent;\n        border-bottom: 0;\n        border-left: 7px solid transparent;\n      }\n    `,s:"\n      right: 50%;\n      left: auto;\n      transform: translateX(50%);\n\n      &::before {\n        top: -16px;\n        right: 50%;\n        transform: translateX(50%);\n      }\n\n      &::after {\n        top: -14px;\n        right: 50%;\n        transform: translateX(50%);\n      }\n    ",sw:"\n      right: 0;\n      left: auto;\n\n      &::before {\n        top: -16px;\n        right: 9px;\n        left: auto;\n      }\n\n      &::after {\n        top: -14px;\n        right: 10px;\n        left: auto;\n      }\n    ",se:"\n      &::before {\n        top: -16px;\n        left: 9px;\n      }\n\n      &::after {\n        top: -14px;\n        left: 10px;\n      }\n    "}[r]))(e.theme,e.direction):""};
  ${Ir};
  ${et};
`,un=c.li`
  display: block;
  padding: ${Cr("space.1")} 10px ${Cr("space.1")} 15px;
  overflow: hidden;
  color: ${Cr("colors.text.primary")};
  text-overflow: ellipsis;
  white-space: nowrap;
  a {
    color: ${Cr("colors.text.primary")};
    text-decoration: none;
    display: block;
    overflow: hidden;
    color: ${Cr("colors.text.primary")};
    text-overflow: ellipsis;
    white-space: nowrap;
  }

  &:focus,
  a:focus {
    color: ${Cr("colors.state.hover.primaryText")};
    text-decoration: none;
    background-color: ${Cr("colors.state.hover.primaryBg")};
  }

  &:hover,
  &:hover a {
    color: ${Cr("colors.state.hover.primaryText")};
    text-decoration: none;
    background-color: ${Cr("colors.state.hover.primaryBg")};
    outline: none;
  }
  ${Ir};
  ${et};
`;pn.defaultProps={direction:"sw"},pn.displayName="Dropdown.Menu",un.displayName="Dropdown.Item",bn.defaultProps=Wt.defaultProps,bn.displayName="Dropdown.Button",gn.displayName="Dropdown.Caret",fn.defaultProps=Za.defaultProps;var mn=Object.assign(fn,{Caret:gn,Menu:pn,Item:un,Button:bn});const hn=e.forwardRef(({children:r,...t},o)=>e.createElement(Wt,Hr({ref:o},t),r,e.createElement($o,{icon:oo,ml:1})));function xn({renderAnchor:t=(r=>e.createElement(hn,r)),placeholder:a,selectedItem:n,onChange:d,overlayProps:l,items:c,...s}){const[f,b]=r(!1),g=o(()=>b(!0),[]),p=o(()=>b(!1),[]),u=o(e=>{var r;return t({...e,children:null!==(r=null==n?void 0:n.text)&&void 0!==r?r:a})},[a,t,null==n?void 0:n.text]),m=i(()=>c.map(e=>({...e,role:"option",selected:e===n,onAction:(r,t)=>{var o;null===(o=e.onAction)||void 0===o||o.call(e,r,t),t.defaultPrevented||(p(),null==d||d(e===n?void 0:e))}})),[c,d,p,n]);return e.createElement($a,{renderAnchor:u,open:f,onOpen:g,onClose:p,overlayProps:l},e.createElement(ha,Hr({},s,{role:"listbox",items:m})))}xn.displayName="DropdownMenu",xn.displayName="DropdownMenu";const vn=c.div`
  ${Ir};
  display: flex;
  align-items: stretch;

  summary {
    border-radius: 0;
    border-top-left-radius: ${Cr("radii.2")};
    border-bottom-left-radius: ${Cr("radii.2")};
    border-right: 0;
  }
  .TextInput-wrapper {
    border-radius: 0;
    border-top-right-radius: ${Cr("radii.2")};
    border-bottom-right-radius: ${Cr("radii.2")};
    z-index: 1; // Allows the focus outline to show on top of the dropdown.
  }

  ${et}
`,yn=c.ul`
  list-style-type: none;
  ${Ir};
  ${et};
`,wn=({children:r,...t})=>{const o=e.Children.map(r,r=>e.createElement("li",null,r));return e.createElement(yn,t,o)};wn.displayName="FilterList";const Bn=c.a`
  position: relative;
  display: block;
  padding: ${e=>e.small?Cr("space.1")(e)+" 10px":Cr("space.2")(e)+" 11px"};
  margin: ${e=>e.small?"0 0 2px":"0 0 5px 0"};
  overflow: hidden;
  font-size: ${Cr("fontSizes.1")};
  color: ${e=>e.selected?Cr("colors.state.selected.primaryText"):Cr("colors.text.secondary")};
  background-color: ${e=>e.selected?Cr("colors.state.selected.primaryBg"):""}!important;
  text-decoration: none;
  text-overflow: ellipsis;
  white-space: nowrap;
  cursor: pointer;
  border-radius: ${Cr("radii.1")};
  &:hover {
    text-decoration: none;
    background-color: ${Cr("colors.bg.tertiary")};
  }
  &:active {
    color: ${Cr("colors.state.selected.primaryText")};
    background-color: ${Cr("colors.state.selected.primaryBg")};
  }
  .count {
    float: right;
    font-weight: ${Cr("fontWeights.bold")};
  }
  ${Ir};
  ${et};
`;function $n({children:r,count:t,...o}){return e.createElement(Bn,o,t&&e.createElement("span",{title:"results",className:"count"},t),r)}$n.displayName="FilterListItem",wn.defaultProps={m:0,p:0},$n.displayName="FilterList.Item";var kn=Object.assign(wn,{Item:$n});const Sn=fe({variants:{default:{color:"alert.info.text",backgroundColor:"alert.info.bg",borderColor:"alert.info.border",svg:{color:"alert.info.icon"}},success:{color:"alert.success.text",backgroundColor:"alert.success.bg",borderColor:"alert.success.border",svg:{color:"alert.success.icon"}},danger:{color:"alert.error.text",backgroundColor:"alert.error.bg",borderColor:"alert.error.border",svg:{color:"alert.error.icon"}},warning:{color:"alert.warn.text",backgroundColor:"alert.warn.bg",borderColor:"alert.warn.border",svg:{color:"alert.warn.icon"}}}}),Tn=c.div`
  position: relative;
  color: ${Cr("colors.text.primary")};
  padding: ${Cr("space.3")};
  border-style: solid;
  border-width: ${e=>e.full?"1px 0px":"1px"};
  border-radius: ${e=>e.full?"0":Cr("radii.2")};
  margin-top: ${e=>e.full?"-1px":"0"};

  p:last-child {
    margin-bottom: 0;
  }

  svg {
    margin-right: ${Cr("space.2")};
  }

  ${Ir};
  ${Sn};
  ${et};
`;Tn.defaultProps={variant:"default"};const En=c.div`
  margin: ${Cr("space.3")} 0;
  font-weight: ${Cr("fontWeights.normal")};
  ${Ir};
  ${et};
`,Cn=c.label`
  display: block;
  margin: 0 0 ${Cr("space.2")};
  font-size: ${Cr("fontSizes.1")};
  font-weight: ${Cr("fontWeights.bold")};
  ${Nr};
  ${Ir};
  ${et};
`;Cn.displayName="FormGroup.Label";var In=Object.assign(En,{Label:Cn});const Nn=c.div`
  z-index: 32;
  display: flex;
  padding: ${Cr("space.3")};
  font-size: ${Cr("fontSizes.1")};
  line-height: ${Cr("lineHeights.default")};
  color: ${Cr("colors.header.text")};
  background-color: ${Cr("colors.header.bg")};
  align-items: center;
  flex-wrap: nowrap;

  ${Ir}
  ${Ar}
  ${et};
`,An=c.div`
  display: flex;
  margin-right: ${Cr("space.3")};
  align-self: stretch;
  align-items: center;
  flex-wrap: nowrap;

  ${({full:e})=>e&&b(["flex:auto;"])};

  ${Ir};
  ${Ar};
  ${et};
`;An.displayName="Header.Item";const zn=c.a.attrs(({to:e})=>"string"==typeof e?{"aria-current":"page"}:{})`
  font-weight: ${Cr("fontWeights.bold")};
  color: ${Cr("colors.header.logo")};
  white-space: nowrap;
  cursor: pointer;
  text-decoration: none;
  display: flex;
  align-items: center;

  &:hover,
  &:focus {
    color: ${Cr("colors.header.text")};
  }

  ${Ir};
  ${Ar};
  ${Nr};
  ${et};
`;zn.displayName="Header.Link";var Ln=Object.assign(Nn,{Link:zn,Item:An});const Rn=c.h2`
  font-weight: ${Cr("fontWeights.bold")};
  font-size: ${Cr("fontSizes.5")};
  margin: 0;
  ${Nr};
  ${Ir};
  ${et};
`;Rn.defaultProps={theme:h.theme};const Mn=c.span`
  ${Ir}
  & * {
    margin-right: ${Cr("space.1")};
  }
  & *:last-child {
    margin-right: 0;
  }
  ${et};
`,Hn=b(["margin-top:-1px;margin-bottom:-1px;color:",";border:"," solid ",";box-shadow:none;",";",";background-color:transparent;"],Cr("colors.label.secondary.text"),Cr("borderWidths.1"),Cr("colors.label.border"),lr,Ir),On=fe({variants:{small:{fontSize:0,lineHeight:"16px",padding:"0px 8px"},medium:{fontSize:0,lineHeight:"20px",padding:"0 8px"},large:{fontSize:0,lineHeight:"24px",padding:"0 12px"},xl:{fontSize:1,lineHeight:"16px",padding:"8px 12px"}}}),Pn=c.span`
  display: inline-block;
  font-weight: ${Cr("fontWeights.semibold")};
  color: ${Cr("colors.text.inverse")};
  border-radius: ${Cr("radii.3")};

  &:hover {
    text-decoration: none;
  }

  ${On}
  ${Ir}
  ${e=>e.dropshadow?"box-shadow: inset 0 -1px 0 rgba(27, 31, 35, 0.12)":""}
  ${e=>e.outline?Hn:""} // must be last to override other values
  ${et}
`;Pn.defaultProps={bg:"label.primary.border",variant:"medium"};const Dn=z({hoverColor:{property:"color",scale:"colors"}}),Wn=c.a`
  color: ${e=>e.muted?Cr("colors.text.secondary")(e):Cr("colors.text.link")(e)};
  text-decoration: ${e=>e.underline?"underline":"none"};
  &:hover {
    text-decoration: ${e=>e.muted?"none":"underline"};
    ${e=>e.hoverColor?Dn:e.muted?"color: "+Cr("colors.text.link")(e):""};
  }
  &:is(button) {
    display: inline-block;
    padding: 0;
    font-size: inherit;
    white-space: nowrap;
    cursor: pointer;
    user-select: none;
    background-color: transparent;
    border: 0;
    appearance: none;
  }
  ${Nr};
  ${Ir};
  ${et};
`,jn=c.div`
  position: relative;
  padding-top: ${Cr("space.4")};
  padding-bottom: ${Cr("space.4")};
  margin-bottom: ${Cr("space.4")};
  border-bottom: 1px solid ${Cr("colors.border.primary")};
  ${Ir};
  ${et};
`;const Fn=c.a`
  display: inline-block;
  min-width: 32px;
  padding: 5px 10px;
  font-style: normal;
  line-height: 20px;
  color: ${Cr("colors.text.primary")};
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  cursor: pointer;
  user-select: none;
  text-decoration: none;

  margin-right: ${Cr("space.1")};

  &:last-child {
    margin-right: 0;
  }

  border: ${Cr("borderWidths.1")} solid transparent;
  border-radius: ${Cr("radii.2")};
  transition: border-color 0.2s cubic-bezier(0.3, 0, 0.5, 1);

  &:hover,
  &:focus {
    text-decoration: none;
    border-color: ${Cr("colors.border.primary")};
    outline: 0;
    transition-duration: 0.1s;
  }

  &:active {
    border-color: ${Cr("colors.border.secondary")};
  }

  &[rel='prev'],
  &[rel='next'] {
    color: ${Cr("colors.text.link")};
  }

  &[aria-current],
  &[aria-current]:hover {
    color: ${Cr("colors.state.selected.primaryText")};
    background-color: ${Cr("colors.state.selected.primaryBg")};
    border-color: transparent;
  }

  &[aria-disabled],
  &[aria-disabled]:hover {
    color: ${Cr("colors.text.disabled")}; // check
    cursor: default;
    border-color: transparent;
  }

  @supports (clip-path: polygon(50% 0, 100% 50%, 50% 100%)) {
    &[rel='prev']::before,
    &[rel='next']::after {
      display: inline-block;
      width: 16px;
      height: 16px;
      vertical-align: text-bottom;
      content: '';
      background-color: currentColor;
    }

    // chevron-left
    &[rel='prev']::before {
      margin-right: ${Cr("space.1")};
      clip-path: polygon(
        9.8px 12.8px,
        8.7px 12.8px,
        4.5px 8.5px,
        4.5px 7.5px,
        8.7px 3.2px,
        9.8px 4.3px,
        6.1px 8px,
        9.8px 11.7px,
        9.8px 12.8px
      );
    }

    // chevron-right
    &[rel='next']::after {
      margin-left: ${Cr("space.1")};
      clip-path: polygon(
        6.2px 3.2px,
        7.3px 3.2px,
        11.5px 7.5px,
        11.5px 8.5px,
        7.3px 12.8px,
        6.2px 11.7px,
        9.9px 8px,
        6.2px 4.3px,
        6.2px 3.2px
      );
    }
  }

  ${Ir};
`;function _n({theme:r,pageCount:t,currentPage:o,onPageChange:a,hrefBuilder:n,marginPageCount:i,showPages:d,surroundingPageCount:l}){const c=e.useCallback(e=>r=>a(r,e),[a]),s=e.useMemo(()=>function(e,r,t,o,a){const n=[];if(t){const t=[],i=r=>{r>=1&&r<=e&&t.push(r)};let d=r-a,l=r+a;if(d<1&&l>e)d=1,l=e;else if(d<1)for(;d<1;)d++,l++;else if(l>e)for(;l>e;)d--,l--;for(let r=1;r<=o;r++){const t=r,o=e-(r-1);t>=d?l++:i(t),o<=l?d--:i(o)}for(let e=d;e<=l;e++)i(e);const c=t.slice().sort((e,r)=>e-r).filter((e,r,t)=>!r||e!==t[r-1]);for(let e=0;e<c.length;e++){const t=c[e],o=t===r;if(0===e)1!==t&&n.push({type:"BREAK",num:1}),n.push({type:"NUM",num:t,selected:o});else{1===t-c[e-1]||n.push({type:"BREAK",num:t-1}),n.push({type:"NUM",num:t,selected:o})}}const s=n[n.length-1];"NUM"===s.type&&s.num!==e&&n.push({type:"BREAK",num:e})}const i={type:"NEXT",num:r+1,disabled:r===e};return[{type:"PREV",num:r-1,disabled:1===r},...n,i]}(t,o,!!d,i,l),[t,o,d,i,l]);return e.useMemo(()=>s.map(t=>{const{props:o,key:a,content:i}=function(e,r,t){const o={};let a="",n="";switch(e.type){case"PREV":n="page-prev",a="Previous",e.disabled?Object.assign(o,{as:"span","aria-disabled":"true"}):Object.assign(o,{rel:"prev",href:r(e.num),"aria-label":"Previous Page",onClick:t});break;case"NEXT":n="page-next",a="Next",e.disabled?Object.assign(o,{as:"span","aria-disabled":"true"}):Object.assign(o,{rel:"next",href:r(e.num),"aria-label":"Next Page",onClick:t});break;case"NUM":n="page-"+e.num,a=String(e.num),e.selected?Object.assign(o,{as:"em","aria-current":"page"}):Object.assign(o,{href:r(e.num),"aria-label":"Page "+e.num,onClick:t});break;case"BREAK":n=`page-${e.num}-break`,a="",Object.assign(o,{as:"span","aria-disabled":!0})}return{props:o,key:n,content:a}}(t,n,c(t.num));return e.createElement(Fn,Hr({},o,{key:a,theme:r}),i)}),[s,n,c,r])}const Un=c.nav`
  margin-top: 20px;
  margin-bottom: 15px;
  text-align: center;
  ${et};
`;function Gn({theme:r,pageCount:t,currentPage:o,onPageChange:a=Vn,hrefBuilder:n=Kn,marginPageCount:i=1,showPages:d=!0,surroundingPageCount:l=2,...c}){const s=_n({theme:r,pageCount:t,currentPage:o,onPageChange:a,hrefBuilder:n,marginPageCount:i,showPages:d,surroundingPageCount:l});return e.createElement(Un,Hr({"aria-label":"Pagination"},c,{theme:r}),e.createElement(rt,{display:"inline-block",theme:r},s))}function Kn(e){return"#"+e}function Vn(){}function Xn(r){const{bg:t,border:o,borderColor:a,theme:n}=r,{caret:i,children:d,...l}=r,c={bg:t,borderColor:a,borderWidth:o,location:i,theme:n};return e.createElement(tt,Hr({sx:{position:"relative"}},l),d,e.createElement(Ga,c))}Gn.displayName="Pagination",Gn.defaultProps={hrefBuilder:Kn,marginPageCount:1,onPageChange:Vn,showPages:!0,surroundingPageCount:2},Xn.displayName="PointerBox";const Yn=c.div.attrs(({className:e,caret:r})=>({className:Na(e,"caret-pos--"+r)}))`
  position: ${e=>e.relative?"relative":"absolute"};
  z-index: 100;
  display: ${e=>e.open?"block":"none"};

  ${Ir};
  ${zr};
  ${Lr};
  ${et};
`,qn=c(tt)`
  position: relative;
  width: 232px;
  margin-right: auto;
  margin-left: auto;
  padding: ${Cr("space.4")};
  background-color: ${Cr("colors.bg.overlay")};

  ${Ir};
  ${zr};

  // Carets
  &::before,
  &::after {
    position: absolute;
    left: 50%;
    display: inline-block;
    content: '';
  }

  &::before {
    top: -${Cr("space.3")};
    margin-left: -9px;
    border: ${Cr("space.2")} solid transparent; // TODO: solid?
    border-bottom-color: ${Cr("colors.border.overlay")};
  }

  &::after {
    top: -14px;
    margin-left: -${Cr("space.2")};
    border: 7px solid transparent; // todo: solid
    border-bottom-color: ${Cr("colors.bg.overlay")};
  }

  // Bottom-oriented carets
  ${Yn}.caret-pos--bottom & ,
  ${Yn}.caret-pos--bottom-right & ,
  ${Yn}.caret-pos--bottom-left & {
    &::before,
    &::after {
      top: auto;
      border-bottom-color: transparent;
    }

    &::before {
      bottom: -${Cr("space.3")};
      border-top-color: ${Cr("colors.border.overlay")};
    }

    &::after {
      bottom: -14px;
      // stylelint-disable-next-line primer/borders
      border-top-color: ${Cr("colors.bg.overlay")};
    }
  }

  // Top & Bottom: Right-oriented carets
  ${Yn}.caret-pos--top-right & ,
  ${Yn}.caret-pos--bottom-right & {
    right: -9px;
    margin-right: 0;

    &::before,
    &::after {
      left: auto;
      margin-left: 0;
    }

    &::before {
      right: 20px;
    }

    &::after {
      right: 21px;
    }
  }

  // Top & Bottom: Left-oriented carets
  ${Yn}.caret-pos--top-left & ,
  ${Yn}.caret-pos--bottom-left & {
    left: -9px;
    margin-left: 0;

    &::before,
    &::after {
      left: ${Cr("space.4")};
      margin-left: 0;
    }

    &::after {
      left: calc(${Cr("space.4")} + 1px);
    }
  }

  // Right- & Left-oriented carets
  ${Yn}.caret-pos--right & ,
  ${Yn}.caret-pos--right-top & ,
  ${Yn}.caret-pos--right-bottom & ,
  ${Yn}.caret-pos--left & ,
  ${Yn}.caret-pos--left-top & ,
  ${Yn}.caret-pos--left-bottom & {
    &::before,
    &::after {
      top: 50%;
      left: auto;
      margin-left: 0;
      border-bottom-color: transparent;
    }

    &::before {
      // stylelint-disable-next-line primer/spacing
      margin-top: calc((${Cr("space.2")} + 1px) * -1);
    }

    &::after {
      margin-top: -${Cr("space.2")};
    }
  }

  // Right-oriented carets
  ${Yn}.caret-pos--right & ,
  ${Yn}.caret-pos--right-top & ,
  ${Yn}.caret-pos--right-bottom & {
    &::before {
      right: -${Cr("space.3")};
      border-left-color: ${Cr("colors.border.overlay")};
    }

    &::after {
      right: -14px;
      // stylelint-disable-next-line primer/borders
      border-left-color: ${Cr("colors.bg.overlay")};
    }
  }

  // Left-oriented carets
  ${Yn}.caret-pos--left & ,
  ${Yn}.caret-pos--left-top & ,
  ${Yn}.caret-pos--left-bottom & {
    &::before {
      left: -${Cr("space.3")};
      border-right-color: ${Cr("colors.border.overlay")};
    }

    &::after {
      left: -14px;
      // stylelint-disable-next-line primer/borders
      border-right-color: ${Cr("colors.bg.overlay")};
    }
  }

  // Right & Left: Top-oriented carets
  ${Yn}.caret-pos--right-top & ,
  ${Yn}.caret-pos--left-top & {
    &::before,
    &::after {
      top: ${Cr("space.4")};
    }
  }

  // Right & Left: Bottom-oriented carets
  ${Yn}.caret-pos--right-bottom & ,
  ${Yn}.caret-pos--left-bottom & {
    &::before,
    &::after {
      top: auto;
    }

    &::before {
      bottom: ${Cr("space.3")};
    }

    &::after {
      bottom: calc(${Cr("space.3")} + 1px);
    }
  }

  ${et};
`;Yn.defaultProps={caret:"top"},qn.displayName="Popover.Content";var Jn=Object.assign(Yn,{Content:qn});const Zn=c.span`
  width: ${e=>e.progress?e.progress+"%":0};
  ${Ir}
`,Qn={small:"5px",large:"10px",default:"8px"},ei=c.span`
  display: ${e=>e.inline?"inline-flex":"flex"};
  overflow: hidden;
  background-color: ${Cr("colors.border.primary")};
  border-radius: ${Cr("radii.1")};
  height: ${e=>Qn[e.barSize||"default"]};
  ${Ir}
  ${ue}
  ${et};
`;function ri({progress:r,bg:t,theme:o,...a}){return e.createElement(ei,Hr({theme:o},a),e.createElement(Zn,{progress:r,bg:t,theme:o}))}function ti(e,r,t){const n=o(o=>{const a=()=>{t(!0)},n=r=>{const t=Array.from(e.current.querySelectorAll('[role^="menuitem"]:not([hidden]):not([disabled]):not([aria-disabled="true"])')),o=document.activeElement,a=t.indexOf(o),n=r?t[a+1]:t[a-1],i=r?t[0]:t[t.length-1];return n||i};if(!(o instanceof KeyboardEvent))return;const i=o.target instanceof Element&&"SUMMARY"===o.target.tagName;switch(o.key){case"Escape":r&&((()=>{t(!1);const r=e.current.querySelector("summary");r&&r.focus()})(),o.preventDefault(),o.stopPropagation());break;case"ArrowDown":{i&&!r&&a();const e=n(!0);e&&e.focus(),o.preventDefault()}break;case"ArrowUp":{i&&!r&&a();const e=n(!1);e&&e.focus(),o.preventDefault()}break;case" ":case"Enter":{const r=document.activeElement;r&&(e=>{const r=e.getAttribute("role");return"menuitem"===r||"menuitemcheckbox"===r||"menuitemradio"===r})(r)&&r.closest("details")===e&&(o.preventDefault(),o.stopPropagation(),r.click())}}},[e,r,t]);a(()=>{const r=e.current;if(r)return r.addEventListener("keydown",n),()=>{r.removeEventListener("keydown",n)}},[e,n])}ri.displayName="ProgressBar",ri.defaultProps={bg:"bg.successInverse",barSize:"default"};const oi=d({}),ai=b(["padding:"," ",";margin:0;font-size:",";font-weight:",";color:",";background-color:",";border-bottom:"," solid ",";"],Cr("space.1"),Cr("space.3"),Cr("fontSizes.0"),Cr("fontWeights.bold"),Cr("colors.text.tertiary"),Cr("colors.bg.tertiary"),Cr("borderWidths.1"),Cr("colors.selectMenu.borderSecondary")),ni=c.div`
  ${ai}
  ${Ir}
  ${et};
`;ni.displayName="SelectMenu.Divider";var ii=L(Q,O,H,R,P,F,U,K,W,ee,be,ge,pe),di=new RegExp("^("+ii.propNames.join("|")+")$");const li=fe({variants:{small:{minHeight:"28px",px:2,py:"3px",fontSize:0,lineHeight:"20px"},large:{px:2,py:"10px",fontSize:3}}}),ci=c.input`
  border: 0;
  font-size: inherit;
  font-family: inherit;
  background-color: transparent;
  -webkit-appearance: none;
  color: inherit;
  width: 100%;
  &:focus {
    outline: 0;
  }
`,si=c.span`
  display: inline-flex;
  align-items: stretch;
  min-height: 34px;
  font-size: ${Cr("fontSizes.1")};
  line-height: 20px;
  color: ${Cr("colors.text.primary")};
  vertical-align: middle;
  background-repeat: no-repeat; // Repeat and position set for form states (success, error, etc)
  background-position: right 8px center; // For form validation. This keeps images 8px from right and centered vertically.
  border: 1px solid ${Cr("colors.input.border")};
  border-radius: ${Cr("radii.2")};
  outline: none;
  box-shadow: ${Cr("shadows.shadow.inset")};

  ${e=>e.hasIcon?b(["padding:0;"]):b(["padding:6px 12px;"])}

  .TextInput-icon {
    align-self: center;
    color: ${Cr("colors.icon.tertiary")};
    margin: 0 ${Cr("space.2")};
    flex-shrink: 0;
  }

  &:focus-within {
    border-color: ${Cr("colors.state.focus.border")};
    box-shadow: ${Cr("shadows.state.focus.shadow")};
  }

  ${e=>e.contrast&&b(["background-color:",";"],Cr("colors.input.contrastBg"))}

  ${e=>e.disabled&&b(["color:",";background-color:",";border-color:",";"],Cr("colors.text.secondary"),Cr("colors.input.disabledBg"),Cr("colors.input.disabledBorder"))}

  ${e=>e.block&&b(["display:block;width:100%;"])}

  // Ensures inputs don't zoom on mobile but are body-font size on desktop
  @media (min-width: ${Cr("breakpoints.1")}) {
    font-size: ${Cr("fontSizes.1")};
  }
  ${Ir}
  ${ue}
  ${he}
  ${ve}
  ${li}
  ${et};
`,fi=e.forwardRef(({icon:r,contrast:t,className:o,block:a,disabled:n,theme:i,sx:d,...l},c)=>{const s=Na(o,"TextInput-wrapper"),f=function(e){var r={};for(var t in e)di.test(t)&&(r[t]=e[t]);return r}(l),b=function(e){var r={};for(var t in e)di.test(t)||(r[t]=e[t]);return r}(l);return e.createElement(si,Hr({className:s,hasIcon:!!r,block:a,theme:i,disabled:n,contrast:t,sx:d},f),r&&e.createElement(r,{className:"TextInput-icon"}),e.createElement(ci,Hr({ref:c,disabled:n},b)))});fi.defaultProps={type:"text"},fi.displayName="TextInput";const bi=c.form`
  padding: ${Cr("space.3")};
  margin: 0;
  border-bottom: ${Cr("borderWidths.1")} solid ${Cr("colors.selectMenu.borderSecondary")};
  background-color: ${Cr("colors.bg.overlay")};
  ${Ir};

  @media (min-width: ${Cr("breakpoints.0")}) {
    padding: ${Cr("space.2")};
  }

  ${et};
`,gi=n(({theme:r,value:o,sx:n,...i},d)=>{const c=t(null),s=null!=d?d:c,{open:f}=l(oi);return a(()=>{var e;f&&(null===(e=c.current)||void 0===e||e.focus())},[f]),e.createElement(bi,{theme:r,sx:n},e.createElement(fi,Hr({theme:r,ref:s,width:"100%",block:!0,value:o,contrast:!0},i)))});gi.displayName="SelectMenu.Filter";const pi=b(["margin-top:-1px;padding:"," ",";font-size:",";color:",";text-align:center;border-top:"," solid ",";@media (min-width:","){padding:"," ",";}"],Cr("space.2"),Cr("space.3"),Cr("fontSizes.0"),Cr("colors.text.tertiary"),Cr("borderWidths.1"),Cr("colors.selectMenu.borderSecondary"),Cr("breakpoints.0"),Cr("space.1"),Cr("space.2")),ui=c.footer`
  ${pi}
  ${Ir}
  ${et};
`;ui.displayName="SelectMenu.Footer";const mi=c.h3`
  color: ${Cr("colors.text.primary")};
  flex: auto;
  font-size: ${Cr("fontSizes.1")};
  font-weight: ${Cr("fontWeights.bold")};
  margin: 0;

  @media (min-width: ${Cr("breakpoints.0")}) {
    font-size: inherit;
  }
`,hi=c.header`
  display: flex;
  flex: none; // fixes header from getting squeezed in Safari iOS
  padding: ${Cr("space.3")};
  border-bottom: ${Cr("borderWidths")} solid ${Cr("colors.selectMenu.borderSecondary")};
  ${Ir}
  ${Nr}

  @media (min-width: ${Cr("breakpoints.0")}) {
    padding-top: ${Cr("space.2")};
    padding-bottom: ${Cr("space.2")};
  }

  ${et};
`,xi=({children:r,theme:t,...o})=>e.createElement(hi,Hr({theme:t},o),e.createElement(mi,{theme:t},r));xi.displayName="SelectMenuHeader",xi.displayName="SelectMenu.Header";const vi=b(["display:flex;align-items:center;padding:",";overflow:hidden;text-align:left;cursor:pointer;background-color:",";border:0;border-bottom:"," solid ",";color:",";text-decoration:none;font-size:",";font-family:inherit;width:100%;&:hover{text-decoration:none;}&:focus{outline:none;}&[hidden]{display:none !important;}@media (min-width:","){padding-top:",";padding-bottom:",";}.SelectMenu-icon{width:",";margin-right:",";flex-shrink:0;}.SelectMenu-selected-icon{visibility:hidden;transition:transform 0.12s cubic-bezier(0.5,0.1,1,0.5),visibility 0s 0.12s linear;transform:scale(0);}&[aria-checked='true']{font-weight:500;color:",";.SelectMenu-selected-icon{visibility:visible;transition:transform 0.12s cubic-bezier(0,0,0.2,1),visibility 0s linear;transform:scale(1);}}@media (hover:hover){&:hover,&:active,&:focus{background-color:",";}}@media (hover:none){&:focus,&:active{background-color:",";}-webkit-tap-highlight-color:",";}"],Cr("space.3"),Cr("colors.bg.overlay"),Cr("borderWidths.1"),Cr("colors.selectMenu.borderSecondary"),Cr("colors.text.secondary"),Cr("fontSizes.0"),Cr("breakpoints.0"),Cr("space.2"),Cr("space.2"),Cr("space.3"),Cr("space.2"),Cr("colors.text.primary"),Cr("colors.state.hover.secondaryBg"),Cr("colors.bg.secondary"),Cr("colors.selectMenu.tapHighlight")),yi=c.a.attrs(()=>({role:"menuitemcheckbox"}))`
  ${vi}
  ${Ir}
  ${et};
`,wi=n(({children:r,selected:o,theme:a,onClick:n,...i},d)=>{const c=l(oi),s=t(null),f=null!=d?d:s;return e.createElement(yi,Hr({ref:f},i,{theme:a,onClick:e=>{var r;(n&&n(e),e.defaultPrevented)||(null===(r=c.setOpen)||void 0===r||r.call(c,!1))},"aria-checked":o}),e.createElement($o,{theme:a,className:"SelectMenu-icon SelectMenu-selected-icon",icon:qt}),r)});wi.defaultProps={selected:!1},wi.displayName="SelectMenu.Item";const Bi=b(["position:relative;padding:0;margin:0;flex:auto;overflow-x:hidden;overflow-y:auto;background-color:",";-webkit-overflow-scrolling:touch;@media (hover:hover){.SelectMenuTab:focus{background-color:",";}.SelectMenuTab:not([aria-checked='true']):hover{color:",";background-color:",";}.SelectMenuTab:not([aria-checked='true']):active{color:",";background-color:",";}}"],Cr("colors.bg.overlay"),Cr("colors.selectMenu.tapFocusBg"),Cr("colors.text.primary"),Cr("colors.bg.tertiary"),Cr("colors.text.primary"),Cr("colors.bg.tertiary")),$i=c.div`
  ${Bi}
  ${Ir}
  ${et};
`;$i.displayName="SelectMenu.List";const ki=p(["0%{opacity:0.3;}10%{opacity:1;}100%{opacity:0.3;}"]),Si=c.div`
  padding: ${Cr("space.6")} ${Cr("space.4")};
  text-align: center;
  background-color: ${Cr("colors.bg.overlay")};
  animation-name: ${ki};
  animation-duration: 2s;
  animation-timing-function: linear;
  animation-iteration-count: infinite;
  ${Ir}
`,Ti=r=>e.createElement(Si,r,e.createElement($o,{size:32,icon:ro}));Ti.displayName="SelectMenuLoadingAnimation";const Ei=p(["0%{opacity:0;transform:scale(0.9);}"]),Ci=b(["position:relative;z-index:99;display:flex;",";max-height:",";margin:auto 0;",";overflow:hidden;pointer-events:auto;flex-direction:column;background-color:",";border-radius:",";box-shadow:",";animation:"," 0.12s cubic-bezier(0,0.1,0.1,1) backwards;@media (min-width:","){height:auto;max-height:350px;margin:"," 0 "," 0;font-size:",";border:"," solid ",";border-radius:",";box-shadow:",";}"],e=>e.filter?"height: 80%":"",e=>e.filter?"none":"66%",e=>e.filter?"margin-top: 0":"",Cr("colors.bg.overlay"),Cr("radii.2"),Cr("shadows.shadow.small"),Ei,Cr("breakpoints.0"),Cr("space.1"),Cr("space.3"),Cr("fontSizes.0"),Cr("borderWidths.1"),Cr("colors.border.overlay"),Cr("radii.2"),Cr("shadows.shadow.small")),Ii=b(["position:fixed;top:0;right:0;bottom:0;left:0;z-index:99;display:flex;padding:",";pointer-events:none;flex-direction:column;&::before{position:absolute;top:0;right:0;bottom:0;left:0;pointer-events:none;content:'';background-color:",";@media (min-width:","){display:none;}}@media (min-width:","){position:absolute;top:auto;right:",";bottom:auto;left:auto;padding:0;}"],Cr("space.3"),Cr("colors.selectMenu.backdropBg"),Cr("breakpoints.0"),Cr("breakpoints.0"),e=>"right"===e.align?"0":"auto"),Ni=c.div`
  ${Ci}
  ${ue}
`,Ai=c.div`
  ${Ii}
  ${Ir}
  ${et};
`,zi=e.forwardRef(({children:r,theme:t,width:o,...a},n)=>e.createElement(Ai,Hr({theme:t},a,{role:"menu",ref:n}),e.createElement(Ni,{theme:t,width:o},r)));zi.defaultProps={align:"left",width:"300px"},zi.displayName="SelectMenu.Modal";const Li=b(["flex:1;padding:"," ",";font-size:",";font-weight:500;color:",";text-align:center;background-color:transparent;border:0;box-shadow:inset 0 -1px 0 ",";@media (min-width:","){flex:none;padding:"," ",";border:"," solid transparent;border-bottom-width:0;border-top-left-radius:",";border-top-right-radius:",";}&[aria-selected='true']{z-index:1;color:",";background-color:",";box-shadow:0 0 0 1px ",";@media (min-width:","){border-color:",";box-shadow:none;}}&:focus{background-color:",";}"],Cr("space.2"),Cr("space.3"),Cr("fontSizes.0"),Cr("colors.text.tertiary"),Cr("colors.selectMenu.borderSecondary"),Cr("breakpoints.0"),Cr("space.1"),Cr("space.3"),Cr("borderWidths.1"),Cr("radii.2"),Cr("radii.2"),Cr("colors.text-primary"),Cr("colors.bg.overlay"),Cr("colors.selectMenu.borderSecondary"),Cr("breakpoints.0"),Cr("colors.selectMenu.borderSecondary"),Cr("colors.state.hover.secondaryBg")),Ri=c.button`
  ${Li}
  ${Ir}
  ${et};
`,Mi=({tabName:r="",index:t,className:o,onClick:n,...i})=>{const d=l(oi);a(()=>{var e;d.selectedTab||0!==t||(null===(e=d.setSelectedTab)||void 0===e||e.call(d,r))},[t,d,r]);const c=d.selectedTab===r;return e.createElement(Ri,Hr({role:"tab",className:Na("SelectMenuTab",o),"aria-selected":c,onClick:e=>{var t;(n&&n(e),e.defaultPrevented)||(null===(t=d.setSelectedTab)||void 0===t||t.call(d,r))}},i),r)};Mi.displayName="SelectMenuTab",Mi.displayName="SelectMenu.Tab";const Hi=c.div`
  border-top: ${Cr("borderWidths.1")} solid ${Cr("colors.selectMenu.borderSecondary")};
  ${Ir}
  ${et};
`,Oi=({tabName:r,className:t,children:o,...a})=>{const n=l(oi);return e.createElement(Hi,Hr({role:"tabpanel",className:t,hidden:n.selectedTab!==r},a),e.createElement($i,null,o))};Oi.displayName="TabPanel",Oi.displayName="SelectMenu.TabPanel";const Pi=b(["display:flex;flex-shrink:0;margin-bottom:-1px;-webkit-overflow-scrolling:touch;overflow-x:auto;overflow-y:hidden;&::-webkit-scrollbar{display:none;}@media (min-width:","){padding:0 ",";margin-top:",";}"],Cr("breakpoints.0"),Cr("space.2"),Cr("space.3")),Di=c.div`
  ${Pi}
  ${Ir}
  ${et};
`,Wi=({children:r,...t})=>e.createElement(Di,Hr({role:"tablist"},t),r);Wi.displayName="SelectMenuTabs",Wi.displayName="SelectMenu.Tabs";const ji=c.details`
  ${"\n  // Remove marker added by the display: list-item browser default\n  > summary {\n    list-style: none;\n  }\n  // Remove marker added by details polyfill\n  > summary::before {\n    display: none;\n  }\n  // Remove marker added by Chrome\n  > summary::-webkit-details-marker {\n    display: none;\n  }\n"}
  ${Ir}
  ${et};
`,Fi=e.forwardRef(({children:n,initialTab:i="",as:d,...l},c)=>{const s=t(null),f=null!=c?c:s,[b,g]=r(i),[p,u]=r(!1),m={selectedTab:b,setSelectedTab:g,setOpen:u,open:p,initialTab:i},h=o(e=>{"current"in f&&f.current&&!f.current.contains(e.target)&&(e.defaultPrevented||u(!1))},[f,u]);return a(()=>{if(p)return document.addEventListener("click",h),()=>{document.removeEventListener("click",h)}},[p,h]),ti(f,p,u),e.createElement(oi.Provider,{value:m},e.createElement(ji,Hr({ref:f},l,{open:p,onToggle:function(e){u(e.target.open)}}),n))});Fi.displayName="SelectMenu";var _i=Object.assign(Fi,{MenuContext:oi,List:$i,Divider:ni,Filter:gi,Footer:ui,Item:wi,Modal:zi,Tabs:Wi,Tab:Mi,TabPanel:Oi,Header:xi,LoadingAnimation:Ti});function Ui({variant:r,className:t,bordered:o,children:a,...n}){const i=Na(t,"variant-"+("lightweight"===r?"lightweight":"normal"));return o||(n={...n,borderWidth:0}),e.createElement(tt,Hr({as:"nav",className:i},n),a)}Ui.displayName="SideNavBase";const Gi=c(Ui)`
  background-color: ${Cr("colors.bg.secondary")};

  ${e=>e.bordered&&b(["& > &{border-left:0;border-right:0;border-bottom:0;}"])}

  ${Ir};
  ${et};
`,Ki=b(["background-color:",";outline:none;text-decoration:none;"],Cr("colors.state.hover.secondaryBg")),Vi=b(["color:",";text-decoration:none;outline:none;"],Cr("colors.text.primary")),Xi=c(Wn).attrs(e=>"string"==typeof e.to||e.selected?{"aria-current":"page"}:{})`
  position: relative;
  display: block;
  ${e=>"full"===e.variant&&b(["display:flex;align-items:center;justify-content:space-between;"])}
  width: 100%;
  text-align: left;
  font-size: ${Cr("fontSizes.1")};

  & > ${Gi} {
    border-bottom: none;
  }

  ${Gi}.variant-normal > & {
    color: ${Cr("colors.text.primary")};
    padding: ${Cr("space.3")};
    border: 0;
    border-top: ${Cr("borderWidths.1")} solid ${Cr("colors.border.secondary")};

    &:first-child {
      border-top: 0;
      border-top-right-radius: ${Cr("radii.2")};
      border-top-left-radius: ${Cr("radii.2")};
    }

    &:last-child {
      border-bottom-right-radius: ${Cr("radii.2")};
      border-bottom-left-radius: ${Cr("radii.2")};
    }

    // Bar on the left
    &::before {
      position: absolute;
      top: 0;
      bottom: 0;
      left: 0;
      z-index: 1;
      width: 3px;
      pointer-events: none;
      content: '';
    }

    &:hover {
      ${Ki}
    }

    &:focus {
      ${Ki}
      box-shadow: ${Cr("shadows.state.focus.shadow")};
      z-index: 1;
    }

    &[aria-current='page'],
    &[aria-selected='true'] {
      background-color: ${Cr("colors.sidenav.selectedBg")};

      // Bar on the left
      &::before {
        background-color: ${Cr("colors.sidenav.borderActive")};
      }
    }
  }

  ${Gi}.variant-lightweight > & {
    padding: ${Cr("space.1")} 0;
    color: ${Cr("colors.text.link")};

    &:hover {
      ${Vi}
    }

    &:focus {
      ${Vi}
      box-shadow: ${Cr("shadows.state.focus.shadow")};
      z-index: 1;
    }

    &[aria-current='page'],
    &[aria-selected='true'] {
      color: ${Cr("colors.text.primary")};
      font-weight: ${Cr("fontWeights.semibold")};
    }
  }

  ${et};
`;Gi.defaultProps={variant:"normal"},Xi.defaultProps={variant:"normal"},Xi.displayName="SideNav.Link";var Yi=Object.assign(Gi,{Link:Xi});const qi={issueOpened:eo,pullOpened:Zt,issueClosed:Qt,pullClosed:Zt,pullMerged:Jt,draft:Zt},Ji=fe({prop:"status",variants:{issueClosed:{backgroundColor:"prState.closed.bg",color:"prState.closed.text",borderColor:"prState.closed.border"},pullClosed:{backgroundColor:"prState.closed.bg",color:"prState.closed.text",borderColor:"prState.closed.border"},pullMerged:{backgroundColor:"prState.merged.bg",color:"prState.merged.text",borderColor:"prState.merged.border"},issueOpened:{backgroundColor:"prState.open.bg",color:"prState.open.text",borderColor:"prState.open.border"},pullOpened:{backgroundColor:"prState.open.bg",color:"prState.open.text",borderColor:"prState.open.border"},draft:{backgroundColor:"prState.draft.bg",color:"prState.draft.text",borderColor:"prState.draft.border"}}}),Zi=fe({prop:"variant",variants:{small:{paddingX:2,paddingY:1,fontSize:0},normal:{paddingX:"12px",paddingY:2,fontSize:1}}}),Qi=c.span`
  display: inline-flex;
  align-items: center;
  font-weight: ${Cr("fontWeights.bold")};
  line-height: 16px;
  color: ${Cr("colors.bg.primary")};
  text-align: center;
  border-radius: ${Cr("radii.3")};
  border-width: 1px;
  border-style: solid;
  ${Ji};
  ${Zi};
  ${Ir};
  ${et};
`;function ed({children:r,status:t,variant:o,...a}){const n="small"===o?{width:"1em"}:{};return e.createElement(Qi,Hr({},a,{variant:o,status:t}),t&&e.createElement($o,Hr({mr:1},n,{icon:qi[t]||to})),r)}ed.displayName="StateLabel",ed.defaultProps={variant:"normal"};const rd=c.nav`
  display: flex;
  justify-content: space-between;

  .SubNav-body {
    display: flex;
    margin-bottom: -1px;

    > * {
      margin-left: ${Cr("space.2")};
    }

    > *:first-child {
      margin-left: 0;
    }
  }

  .SubNav-actions {
    align-self: center;
  }

  ${Ir};
  ${Rr};
  ${et};
`;function td({actions:r,className:t,children:o,label:a,...n}){const i=Na(t,"SubNav");return e.createElement(rd,Hr({className:i,"aria-label":a},n),e.createElement("div",{className:"SubNav-body"},o),r&&e.createElement("div",{className:"SubNav-actions"},r))}function od(r){return e.createElement(ot,r)}td.displayName="SubNav",od.displayName="SubNavLinks";const ad=c.a.attrs(e=>({activeClassName:"string"==typeof e.to?"selected":"",className:Na("SubNav-item",e.selected&&"selected",e.className)}))`
  padding-left: ${Cr("space.3")};
  padding-right: ${Cr("space.3")};
  font-weight: ${Cr("fontWeights.semibold")};
  font-size: ${Cr("fontSizes.1")};
  line-height: 20px; //custom value for SubNav
  min-height: 34px; //custom value for SubNav
  color: ${Cr("colors.text.primary")};
  text-align: center;
  text-decoration: none;
  border-top: 1px solid ${Cr("colors.border.primary")};
  border-bottom: 1px solid ${Cr("colors.border.primary")};
  border-right: 1px solid ${Cr("colors.border.primary")};
  display: flex;
  align-items: center;

  &:first-of-type {
    border-top-left-radius: ${Cr("radii.2")};
    border-bottom-left-radius: ${Cr("radii.2")};
    border-left: 1px solid ${Cr("colors.border.primary")};
  }

  &:last-of-type {
    border-top-right-radius: ${Cr("radii.2")};
    border-bottom-right-radius: ${Cr("radii.2")};
  }

  &:hover,
  &:focus {
    text-decoration: none;
    background-color: ${Cr("colors.bg.tertiary")};
    transition: 0.2s ease;

    .SubNav-octicon {
      color: ${Cr("colors.icon.secondary")};
    }
  }

  &.selected {
    color: ${Cr("colors.state.selected.primaryText")};
    background-color: ${Cr("colors.state.selected.primaryBg")};
    border-color: ${Cr("colors.state.selected.primaryBorder")};
    .SubNav-octicon {
      color: ${Cr("colors.state.selected.primaryText")};
    }
  }

  ${Ir};
  ${et};
`;ad.displayName="SubNav.Link",od.displayName="SubNav.Links";var nd=Object.assign(td,{Link:ad,Links:od});const id=c.div`
  margin-top: 0;
  border-bottom: 1px solid ${Cr("colors.border.primary")};
  ${Ir}
  ${et}
`,dd=c.nav`
  display: flex;
  margin-bottom: -1px;
  overflow: auto;
`;function ld({children:r,"aria-label":t,...o}){return e.createElement(id,o,e.createElement(dd,{"aria-label":t},r))}ld.displayName="TabNav";const cd=c.a.attrs(e=>({activeClassName:"string"==typeof e.to?"selected":"",className:Na("TabNav-item",e.selected&&"selected",e.className)}))`
  padding: 8px 12px;
  font-size: ${Cr("fontSizes.1")};
  line-height: 20px;
  color: ${Cr("colors.text.primary")};
  text-decoration: none;
  background-color: transparent;
  border: 1px solid transparent;
  border-bottom: 0;

  &:hover,
  &:focus {
    color: ${Cr("colors.text.primary")};
    text-decoration: none;
  }

  &.selected {
    color: ${Cr("colors.text.primary")};
    border-color: ${Cr("colors.border.primary")};
    border-top-right-radius: ${Cr("radii.2")};
    border-top-left-radius: ${Cr("radii.2")};
    background-color: ${Cr("colors.bg.canvas")};
  }

  ${Ir};
  ${et};
`;cd.displayName="TabNav.Link";var sd=Object.assign(ld,{Link:cd});const fd=c(ot).withConfig({displayName:"Timeline",componentId:"c4dq2e-0"})(["flex-direction:column;"," ",";"],e=>e.clipSidebar&&b([".Timeline-Item:first-child{padding-top:0;}.Timeline-Item:last-child{padding-bottom:0;}"]),et),bd=c(ot).attrs(e=>({className:Na("Timeline-Item",e.className)})).withConfig({displayName:"Timeline__TimelineItem",componentId:"c4dq2e-1"})(["position:relative;padding:"," 0;margin-left:",";&::before{position:absolute;top:0;bottom:0;left:0;display:block;width:2px;content:'';background-color:",";}"," ",";",";"],Cr("space.3"),Cr("space.3"),Cr("colors.border.secondary"),e=>e.condensed&&b(["padding-top:",";padding-bottom:0;&:last-child{padding-bottom:",";}.TimelineItem-Badge{height:16px;margin-top:",";margin-bottom:",";color:",";background-color:",";border:0;}"],Cr("space.1"),Cr("space.3"),Cr("space.2"),Cr("space.2"),Cr("colors.icon.tertiary"),Cr("colors.bg.canvas")),Ir,et);var gd=c(ot).withConfig({displayName:"Timeline___StyledFlex",componentId:"c4dq2e-2"})(["border-radius:50%;border:2px solid ",";"],e=>e._css);const pd=r=>e.createElement(lt,{zIndex:1},e.createElement(gd,Hr({className:Na(r.className,"TimelineItem-Badge"),flexShrink:0,overflow:"hidden",color:"icon.secondary",bg:"timeline.badgeBg",width:"32px",height:"32px",mr:2,ml:"-15px",alignItems:"center",justifyContent:"center"},r,{_css:Cr("colors.bg.canvas")}),r.children));pd.displayName="TimelineBadge";const ud=c(rt).withConfig({displayName:"Timeline__TimelineBody",componentId:"c4dq2e-3"})(["min-width:0;max-width:100%;margin-top:",";color:",";flex:auto;font-size:",";",";"],Cr("space.1"),Cr("colors.timeline.text"),Cr("fontSizes.1"),et),md=c(lt).withConfig({displayName:"Timeline__TimelineBreak",componentId:"c4dq2e-4"})(["z-index:1;height:24px;margin:0;margin-bottom:-",";margin-left:0;background-color:",";border:0;border-top:"," solid ",";",";"],Cr("space.3"),Cr("colors.bg.canvas"),Cr("space.1"),Cr("colors.border.primary"),et);bd.displayName="Timeline.Item",pd.displayName="Timeline.Badge",ud.displayName="Timeline.Body",md.displayName="Timeline.Break";var hd=Object.assign(fd,{Item:bd,Badge:pd,Body:ud,Break:md});const xd=c.span`
  position: relative;

  &::before {
    position: absolute;
    z-index: 1000001;
    display: none;
    width: 0px;
    height: 0px;
    color: ${Cr("colors.tooltip.bg")};
    pointer-events: none;
    content: '';
    border: 6px solid transparent;
    opacity: 0;
  }

  &::after {
    position: absolute;
    z-index: 1000000;
    display: none;
    padding: 0.5em 0.75em;
    font: normal normal 11px/1.5 ${Cr("fonts.normal")};
    -webkit-font-smoothing: subpixel-antialiased;
    color: ${Cr("colors.tooltip.text")};
    text-align: center;
    text-decoration: none;
    text-shadow: none;
    text-transform: none;
    letter-spacing: normal;
    word-wrap: break-word;
    white-space: pre;
    pointer-events: none;
    content: attr(aria-label);
    background: ${Cr("colors.tooltip.bg")};
    border-radius: ${Cr("radii.1")};
    opacity: 0;
  }

  // delay animation for tooltip
  @keyframes tooltip-appear {
    from {
      opacity: 0;
    }

    to {
      opacity: 1;
    }
  }

  &:hover,
  &:active,
  &:focus {
    &::before,
    &::after {
      display: inline-block;
      text-decoration: none;
      animation-name: tooltip-appear;
      animation-duration: 0.1s;
      animation-fill-mode: forwards;
      animation-timing-function: ease-in;
      animation-delay: 0.4s;
    }
  }

  &.tooltipped-no-delay:hover,
  &.tooltipped-no-delay:active,
  &.tooltipped-no-delay:focus {
    &::before,
    &::after {
      animation-delay: 0s;
    }
  }

  &.tooltipped-multiline:hover,
  &.tooltipped-multiline:active,
  &.tooltipped-multiline:focus {
    &::after {
      display: table-cell;
    }
  }

  // Tooltipped south
  &.tooltipped-s,
  &.tooltipped-se,
  &.tooltipped-sw {
    &::after {
      top: 100%;
      right: 50%;
      margin-top: 6px;
    }

    &::before {
      top: auto;
      right: 50%;
      bottom: -7px;
      margin-right: -6px;
      border-bottom-color: ${Cr("colors.tooltip.bg")};
    }
  }

  &.tooltipped-se {
    &::after {
      right: auto;
      left: 50%;
      margin-left: -${Cr("space.3")};
    }
  }

  &.tooltipped-sw::after {
    margin-right: -${Cr("space.3")};
  }

  // Tooltips above the object
  &.tooltipped-n,
  &.tooltipped-ne,
  &.tooltipped-nw {
    &::after {
      right: 50%;
      bottom: 100%;
      margin-bottom: 6px;
    }

    &::before {
      top: -7px;
      right: 50%;
      bottom: auto;
      margin-right: -6px;
      border-top-color: ${Cr("colors.tooltip.bg")};
    }
  }

  &.tooltipped-ne {
    &::after {
      right: auto;
      left: 50%;
      margin-left: -${Cr("space.3")};
    }
  }

  &.tooltipped-nw::after {
    margin-right: -${Cr("space.3")};
  }

  // Move the tooltip body to the center of the object.
  &.tooltipped-s::after,
  &.tooltipped-n::after {
    transform: translateX(50%);
  }

  // Tooltipped to the left
  &.tooltipped-w {
    &::after {
      right: 100%;
      bottom: 50%;
      margin-right: 6px;
      transform: translateY(50%);
    }

    &::before {
      top: 50%;
      bottom: 50%;
      left: -7px;
      margin-top: -6px;
      border-left-color: ${Cr("colors.tooltip.bg")};
    }
  }

  // tooltipped to the right
  &.tooltipped-e {
    &::after {
      bottom: 50%;
      left: 100%;
      margin-left: 6px;
      transform: translateY(50%);
    }

    &::before {
      top: 50%;
      right: -7px;
      bottom: 50%;
      margin-top: -6px;
      border-right-color: ${Cr("colors.tooltip.bg")};
    }
  }

  &.tooltipped-multiline {
    &::after {
      width: max-content;
      max-width: 250px;
      word-wrap: break-word;
      white-space: pre-line;
      border-collapse: separate;
    }

    &.tooltipped-s::after,
    &.tooltipped-n::after {
      right: auto;
      left: 50%;
      transform: translateX(-50%);
    }

    &.tooltipped-w::after,
    &.tooltipped-e::after {
      right: 100%;
    }
  }

  &.tooltipped-align-right-2::after {
    right: 0;
    margin-right: 0;
  }

  &.tooltipped-align-right-2::before {
    right: 15px;
  }

  &.tooltipped-align-left-2::after {
    left: 0;
    margin-left: 0;
  }

  &.tooltipped-align-left-2::before {
    left: 10px;
  }
  ${Ir};
  ${et};
`;function vd({direction:r="n",children:t,className:o,text:a,noDelay:n,align:i,wrap:d,...l}){const c=Na(o,"tooltipped-"+r,i&&`tooltipped-align-${i}-2`,n&&"tooltipped-no-delay",d&&"tooltipped-multiline");return e.createElement(xd,Hr({"aria-label":a},l,{className:c}),t)}vd.displayName="Tooltip",vd.alignments=["left","right"],vd.directions=["n","ne","e","se","s","sw","w","nw"];const yd=c.div`
  ${Nr}
  ${Ir}
  display: ${e=>e.inline?"inline-block":"inherit"};
  overflow: hidden;
  text-overflow: ellipsis;
  vertical-align: ${e=>e.inline?"top":"initial"};
  white-space: nowrap;
  ${ve}
  ${e=>e.expandable?"&:hover { max-width: 10000px; }":""}
  ${et};
`;yd.defaultProps={expandable:!1,inline:!1,maxWidth:125};const wd=c.nav`
  display: flex;
  justify-content: space-between;
  border-bottom: 1px solid ${Cr("colors.border.secondary")};
  &.UnderlineNav--right {
    justify-content: flex-end;

    .UnderlineNav-item {
      margin-right: 0;
      margin-left: ${Cr("space.3")};
    }

    .UnderlineNav-actions {
      flex: 1 1 auto;
    }
  }
  &.UnderlineNav--full {
    display: block;
  }

  .UnderlineNav-body {
    display: flex;
    margin-bottom: -1px;
  }

  .UnderlineNav-actions {
    align-self: center;
  }

  ${Ir};
  ${et};
`;function Bd({actions:r,className:t,align:o,children:a,full:n,label:i,theme:d,...l}){const c=Na(t,"UnderlineNav",o&&"UnderlineNav--"+o,n&&"UnderlineNav--full");return e.createElement(wd,Hr({className:c,"aria-label":i,theme:d},l),e.createElement("div",{className:"UnderlineNav-body"},a),r&&e.createElement("div",{className:"UnderlineNav-actions"},r))}Bd.displayName="UnderlineNav";const $d=c.a.attrs(e=>({activeClassName:"string"==typeof e.to?"selected":"",className:Na("UnderlineNav-item",e.selected&&"selected",e.className)}))`
  padding: ${Cr("space.3")} ${Cr("space.2")};
  margin-right: ${Cr("space.3")};
  font-size: ${Cr("fontSizes.1")};
  line-height: ${Cr("lineHeights.default")};
  color: ${Cr("colors.underlinenav.text")};
  text-align: center;
  border-bottom: 2px solid transparent;
  text-decoration: none;

  &:hover,
  &:focus {
    color: ${Cr("colors.underlinenav.textHover")};
    text-decoration: none;
    border-bottom-color: ${Cr("colors.border.tertiary")};
    transition: 0.2s ease;

    .UnderlineNav-octicon {
      color: ${Cr("colors.text.tertiary")};
    }
  }

  &.selected {
    color: ${Cr("colors.underlinenav.textActive")};
    border-bottom-color: ${Cr("colors.underlinenav.borderActive")};

    .UnderlineNav-octicon {
      color: ${Cr("colors.underlinenav.iconActive")};
    }
  }

  ${Ir};
  ${et};
`;$d.displayName="UnderlineNav.Link";var kd=Object.assign(Bd,{Link:$d}),Sd=h.theme;export{it as Absolute,xa as ActionList,Ta as ActionMenu,Ea as Avatar,Ia as AvatarPair,za as AvatarStack,Dr as BaseStyles,tt as BorderBox,rt as Box,La as BranchName,Pa as Breadcrumb,Wt as Button,io as ButtonClose,jt as ButtonDanger,Ft as ButtonGroup,Gt as ButtonInvisible,_t as ButtonOutline,Ut as ButtonPrimary,Kt as ButtonTableList,Ga as Caret,Ya as CircleBadge,qa as CircleOcticon,Qo as ConfirmationDialog,Ja as CounterLabel,Za as Details,cn as Dialog,mn as Dropdown,hn as DropdownButton,xn as DropdownMenu,kn as FilterList,vn as FilteredSearch,dt as Fixed,Tn as Flash,ot as Flex,In as FormGroup,at as Grid,Ln as Header,Rn as Heading,Pn as Label,Mn as LabelGroup,Wn as Link,Ba as Overlay,jn as Pagehead,Gn as Pagination,Xn as PointerBox,Jn as Popover,nt as Position,ri as ProgressBar,lt as Relative,_i as SelectMenu,Yi as SideNav,ed as StateLabel,ct as Sticky,$o as StyledOcticon,nd as SubNav,sd as TabNav,rn as Text,fi as TextInput,qr as ThemeProvider,hd as Timeline,vd as Tooltip,yd as Truncate,kd as UnderlineNav,Sd as theme,Cr as themeGet,Zr as useColorSchemeVar,ea as useConfirm,st as useDetails,vt as useOnEscapePress,bt as useOnOutsideClick,mt as useOpenAndCloseFocus,wt as useOverlay,ft as useSafeTimeout,Jr as useTheme};
