// @primer/components/dist/browser.umd.js
!function(e,t){"object"==typeof exports&&"undefined"!=typeof module?t(exports,require("react"),require("styled-components"),require("react-dom")):"function"==typeof define&&define.amd?define(["exports","react","styled-components","react-dom"],t):t((e="undefined"!=typeof globalThis?globalThis:e||self).primer={},e.React,e.styled,e.ReactDOM)}(this,(function(e,t,r,o){"use strict";function a(e){return e&&"object"==typeof e&&"default"in e?e:{default:e}}var n=a(t),i=a(r),d=a(o),l={animation:{easeOutCubic:"cubic-bezier(0.33, 1, 0.68, 1)"},borderWidths:[0,"1px"],breakpoints:["544px","768px","1012px","1280px"],fonts:{normal:'-apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji"',mono:'SFMono-Regular, Consolas, "Liberation Mono", Menlo, Courier, monospace'},fontSizes:["12px","14px","16px","20px","24px","32px","40px","48px"],fontWeights:{light:300,normal:400,semibold:500,bold:600},lineHeights:{condensedUltra:1,condensed:1.25,default:1.5},radii:["0","3px","6px","100px"],sizes:{small:"544px",medium:"768px",large:"1012px",xlarge:"1280px"},space:["0","4px","8px","16px","24px","32px","40px","48px","64px","80px","96px","112px","128px"],colorSchemes:{light:{colors:{auto:{black:"#1b1f23",white:"#ffffff",gray:["#fafbfc","#f6f8fa","#e1e4e8","#d1d5da","#959da5","#6a737d","#586069","#444d56","#2f363d","#24292e"],blue:["#f1f8ff","#dbedff","#c8e1ff","#79b8ff","#2188ff","#0366d6","#005cc5","#044289","#032f62","#05264c"],green:["#f0fff4","#dcffe4","#bef5cb","#85e89d","#34d058","#28a745","#22863a","#176f2c","#165c26","#144620"],yellow:["#fffdef","#fffbdd","#fff5b1","#ffea7f","#ffdf5d","#ffd33d","#f9c513","#dbab09","#b08800","#735c0f"],orange:["#fff8f2","#ffebda","#ffd1ac","#ffab70","#fb8532","#f66a0a","#e36209","#d15704","#c24e00","#a04100"],red:["#ffeef0","#ffdce0","#fdaeb7","#f97583","#ea4a5a","#d73a49","#cb2431","#b31d28","#9e1c23","#86181d"],purple:["#f5f0ff","#e6dcfd","#d1bcf9","#b392f0","#8a63d2","#6f42c1","#5a32a3","#4c2889","#3a1d6e","#29134e"],pink:["#ffeef8","#fedbf0","#f9b3dd","#f692ce","#ec6cb9","#ea4aaa","#d03592","#b93a86","#99306f","#6d224f"]},text:{primary:"#24292e",secondary:"#586069",tertiary:"#6a737d",placeholder:"#6a737d",disabled:"#959da5",inverse:"#ffffff",link:"#0366d6",danger:"#cb2431",success:"#22863a",warning:"#b08800",white:"#ffffff"},icon:{primary:"#24292e",secondary:"#586069",tertiary:"#959da5",info:"#0366d6",danger:"#d73a49",success:"#22863a",warning:"#b08800"},border:{primary:"#e1e4e8",secondary:"hsla(214,13%,93%,1)",tertiary:"#d1d5da",overlay:"#e1e4e8",inverse:"#ffffff",info:"#0366d6",danger:"#d73a49",success:"#34d058",warning:"#f9c513"},bg:{canvas:"#ffffff",canvasMobile:"#ffffff",canvasInverse:"#24292e",canvasInset:"#f6f8fa",primary:"#ffffff",secondary:"#fafbfc",tertiary:"#f6f8fa",overlay:"#ffffff",backdrop:"rgba(27,31,35,0.5)",info:"#f1f8ff",infoInverse:"#0366d6",danger:"#ffeef0",dangerInverse:"#d73a49",success:"#dcffe4",successInverse:"#28a745",warning:"#fff5b1",warningInverse:"#ffd33d"},state:{hover:{primaryBg:"#0366d6",primaryBorder:"#0366d6",primaryText:"#ffffff",primaryIcon:"#ffffff",secondaryBg:"#f6f8fa",secondaryBorder:"#f6f8fa"},selected:{primaryBg:"#0366d6",primaryBorder:"#0366d6",primaryText:"#ffffff",primaryIcon:"#ffffff"},focus:{border:"#0366d6"}},fade:{fg10:"rgba(27,31,35,0.1)",fg15:"rgba(27,31,35,0.15)",fg30:"rgba(27,31,35,0.3)",fg50:"rgba(27,31,35,0.5)",fg70:"rgba(27,31,35,0.7)",fg85:"rgba(27,31,35,0.85)",black10:"rgba(27,31,35,0.1)",black15:"rgba(27,31,35,0.15)",black30:"rgba(27,31,35,0.3)",black50:"rgba(27,31,35,0.5)",black70:"rgba(27,31,35,0.7)",black85:"rgba(27,31,35,0.85)",white10:"rgba(255,255,255,0.1)",white15:"rgba(255,255,255,0.15)",white30:"rgba(255,255,255,0.3)",white50:"rgba(255,255,255,0.5)",white70:"rgba(255,255,255,0.7)",white85:"rgba(255,255,255,0.85)"},alert:{info:{text:"#24292e",icon:"rgba(4,66,137,0.6)",bg:"#dbedff",border:"rgba(4,66,137,0.2)"},warn:{text:"#24292e",icon:"#b08800",bg:"#fffbdd",border:"rgba(176,136,0,0.2)"},error:{text:"#24292e",icon:"rgba(158,28,35,0.6)",bg:"#ffe3e6",border:"rgba(158,28,35,0.2)"},success:{text:"#24292e",icon:"rgba(23,111,44,0.8)",bg:"#dcffe4",border:"rgba(23,111,44,0.2)"}},autocomplete:{rowBorder:"hsla(214,13%,93%,1)"},blankslate:{icon:"hsla(212,8%,50%,1)"},btn:{text:"#24292e",bg:"#fafbfc",border:"rgba(27,31,35,0.15)",hoverBg:"#f3f4f6",hoverBorder:"rgba(27,31,35,0.15)",activeBg:"hsla(220,14%,93%,1)",activeBorder:"rgba(27,31,35,0.1)",selectedBg:"hsla(220,14%,94%,1)",focusBg:"#fafbfc",focusBorder:"rgba(27,31,35,0.15)",counterBg:"rgba(27,31,35,0.08)",primary:{text:"#ffffff",bg:"#2ea44f",border:"rgba(27,31,35,0.15)",hoverBg:"#2c974b",hoverBorder:"rgba(27,31,35,0.15)",selectedBg:"hsla(137,55%,36%,1)",disabledText:"rgba(255,255,255,0.8)",disabledBg:"#94d3a2",disabledBorder:"rgba(27,31,35,0.1)",focusBg:"#2ea44f",focusBorder:"rgba(27,31,35,0.15)",icon:"rgba(255,255,255,0.8)",counterBg:"rgba(255,255,255,0.2)"},outline:{text:"#0366d6",hoverText:"#ffffff",hoverBg:"#0366d6",hoverBorder:"rgba(27,31,35,0.15)",hoverCounterBg:"rgba(255,255,255,0.2)",selectedText:"#ffffff",selectedBg:"hsla(212,97%,40%,1)",selectedBorder:"rgba(27,31,35,0.15)",disabledText:"rgba(3,102,214,0.5)",disabledBg:"#fafbfc",disabledCounterBg:"rgba(3,102,214,0.05)",focusBorder:"rgba(27,31,35,0.15)",counterBg:"rgba(3,102,214,0.1)"},danger:{text:"#d73a49",hoverText:"#ffffff",hoverBg:"#cb2431",hoverBorder:"rgba(27,31,35,0.15)",hoverCounterBg:"rgba(255,255,255,0.2)",selectedText:"#ffffff",selectedBg:"hsla(354,66%,51%,1)",selectedBorder:"rgba(27,31,35,0.15)",disabledText:"rgba(215,58,73,0.5)",disabledBg:"#fafbfc",disabledCounterBg:"rgba(215,58,73,0.05)",focusBorder:"rgba(27,31,35,0.15)",counterBg:"rgba(215,58,73,0.1)",icon:"#d73a49",hoverIcon:"#ffffff"}},counter:{text:"#24292e",bg:"rgba(209,213,218,0.5)",primary:{text:"#ffffff",bg:"#6a737d"},secondary:{text:"#6a737d",bg:"rgba(209,213,218,0.5)"}},label:{border:"#e1e4e8",primary:{text:"#24292e",border:"#6a737d"},secondary:{text:"#586069",border:"#e1e4e8"},info:{text:"#0366d6",border:"#0366d6"},success:{text:"#22863a",border:"#28a745"},warning:{text:"#735c0f",border:"#b08800"},danger:{text:"#d73a49",border:"#cb2431"},orange:{text:"#c24e00",border:"#f66a0a"}},input:{bg:"#ffffff",contrastBg:"#fafbfc",border:"#e1e4e8",disabledBg:"#f6f8fa",disabledBorder:"#e1e4e8",warningBorder:"#f9c513",errorBorder:"#cb2431",tooltip:{success:{text:"#144620",bg:"#dcffe4",border:"#34d058"},warning:{text:"#735c0f",bg:"#fff5b1",border:"#f9c513"},error:{text:"#86181d",bg:"#ffeef0",border:"#f97583"}}},avatar:{bg:"#ffffff",border:"rgba(0,0,0,0)",stackFade:"#d1d5da",stackFadeMore:"#e1e4e8"},toast:{text:"#24292e",bg:"#ffffff",border:"#e1e4e8",icon:"#ffffff",iconBg:"#0366d6",iconBorder:"rgba(0,0,0,0)",success:{text:"#24292e",border:"#e1e4e8",icon:"#ffffff",iconBg:"#28a745",iconBorder:"rgba(0,0,0,0)"},warning:{text:"#24292e",border:"#e1e4e8",icon:"#24292e",iconBg:"#ffd33d",iconBorder:"rgba(0,0,0,0)"},danger:{text:"#24292e",border:"#e1e4e8",icon:"#ffffff",iconBg:"#d73a49",iconBorder:"rgba(0,0,0,0)"},loading:{text:"#24292e",border:"#e1e4e8",icon:"#ffffff",iconBg:"#586069",iconBorder:"rgba(0,0,0,0)"}},timeline:{text:"#444d56",badgeBg:"#e1e4e8",badgeSuccessBorder:"rgba(0,0,0,0)",targetBadgeBorder:"#2188ff",targetBadgeShadow:"#c8e1ff"},selectMenu:{borderSecondary:"hsla(214,13%,93%,1)",backdropBg:"rgba(27,31,35,0.5)",backdropBorder:"rgba(0,0,0,0)",tapHighlight:"rgba(209,213,218,0.5)",tapFocusBg:"#dbedff"},box:{blueBorder:"#c8e1ff",rowYellowBg:"#fffbdd",rowBlueBg:"#f1f8ff",headerBlueBg:"#f1f8ff",headerBlueBorder:"#c8e1ff",borderInfo:"rgba(3,102,214,0.2)",bgInfo:"#f1f8ff",borderWarning:"rgba(255,211,61,0.4)",bgWarning:"#fffdef"},branchName:{text:"#586069",icon:"hsla(212,30%,74%,1)",bg:"hsla(210,100%,96%,1)",link:{text:"#0366d6",icon:"hsla(212,30%,74%,1)",bg:"hsla(210,100%,96%,1)"}},markdown:{codeBg:"rgba(27,31,35,0.05)",frameBorder:"hsla(213,11%,89%,1)",blockquoteBorder:"hsla(213,11%,89%,1)",tableBorder:"hsla(213,11%,89%,1)",tableTrBorder:"hsla(213,11%,80%,1)"},menu:{headingText:"#24292e",borderActive:"#f9826c",bgActive:"rgba(0,0,0,0)"},sidenav:{selectedBg:"#ffffff",borderActive:"#f9826c"},header:{text:"rgba(255,255,255,0.7)",bg:"#24292e",logo:"#ffffff"},filterItem:{barBg:"hsla(210,29%,95%,1)"},hiddenTextExpander:{bg:"hsla(213,11%,89%,1)",bgHover:"hsla(213,11%,80%,1)"},dragAndDrop:{border:"hsla(213,11%,79%,1)"},uploadEnabled:{border:"hsla(213,11%,89%,1)",borderFocused:"hsla(212,100%,64%,1)"},previewableCommentForm:{border:"hsla(213,11%,79%,1)"},underlinenav:{border:"rgba(209,213,218,0)",borderHover:"#d1d5da",borderActive:"#f9826c",text:"#24292e",textHover:"#24292e",textActive:"#24292e",icon:"#959da5",iconHover:"#959da5",iconActive:"#24292e",counterText:"#24292e",counterBg:"rgba(209,213,218,0.5)"},verifiedBadge:{text:"#22863a",bg:"#ffffff",border:"#e1e4e8"},socialCount:{bg:"#ffffff"},tooltip:{text:"#ffffff",bg:"#24292e"},headerSearch:{bg:"#24292e",border:"#444d56"},searchKeyword:{hl:"#fffbdd"},diffstat:{neutralBg:"#d1d5da",neutralBorder:"#d1d5da",deletionBg:"#d73a49",deletionBorder:"#d73a49",additionBg:"#28a745",additionBorder:"#28a745"},mktg:{success:"rgba(46,188,79,1)",info:"rgba(16,116,231,1)",bgShadeGradient:{top:"rgba(27,31,35,0.065)",bottom:"rgba(27,31,35,0)"},btn:{bg:{top:"hsla(228,82%,66%,1)",bottom:"#4969ed"},bgOverlay:{top:"hsla(228,74%,59%,1)",bottom:"#3355e0"},text:"#ffffff",primary:{bg:{top:"hsla(137,56%,46%,1)",bottom:"#2ea44f"},bgOverlay:{top:"hsla(134,60%,38%,1)",bottom:"#22863a"},text:"#ffffff"},enterprise:{bg:{top:"hsla(249,100%,72%,1)",bottom:"#6f57ff"},bgOverlay:{top:"hsla(248,65%,63%,1)",bottom:"#614eda"},text:"#ffffff"},outline:{text:"#4969ed",border:"rgba(73,105,237,0.3)",hover:{text:"#3355e0",border:"rgba(51,85,224,0.5)"},focus:{border:"#4969ed",borderInset:"rgba(73,105,237,0.5)"}},dark:{text:"#ffffff",border:"rgba(255,255,255,0.3)",hover:{text:"#ffffff",border:"rgba(255,255,255,0.5)"},focus:{border:"#ffffff",borderInset:"rgba(255,255,255,0.5)"}}}},filesExplorerIcon:"#79b8ff",hlAuthorBg:"#f1f8ff",hlAuthorBorder:"#c8e1ff",logoSubdued:"#d1d5da",discussionBorder:"hsla(135,28%,72%,1)",discussionBgSuccess:"#28a745",actionsWorkflowTableStickyBg:"rgba(255,255,255,0.95)",repoLanguageColorBorder:"rgba(27,31,35,0.1)",codeSelectionBg:"#c8e1ff",highlight:{text:"#442c12",bg:"#fff0bb"},blob:{lineHighlightBg:"#fffbdd",lineHighlightBorder:"rgba(0,0,0,0)"},diff:{addition:{text:"#22863a",bg:"hsla(136,100%,95%,1)",border:"#34d058"},deletion:{text:"#cb2431",bg:"#ffeef0",border:"#d73a49"},change:{text:"#b08800",bg:"#fff5b1",border:"#f9c513"}},diffBlob:{numText:"rgba(27,31,35,0.3)",numHoverText:"rgba(27,31,35,0.6)",addition:{numText:"rgba(27,31,35,0.3)",numHoverText:"rgba(27,31,35,0.6)",numBg:"hsla(134,100%,90%,1)",lineBg:"hsla(136,100%,95%,1)",wordBg:"hsla(134,73%,81%,1)"},deletion:{numText:"rgba(27,31,35,0.3)",numHoverText:"rgba(27,31,35,0.6)",numBg:"#ffdce0",lineBg:"#ffeef0",wordBg:"hsla(353,95%,86%,1)"},hunk:{text:"rgba(27,31,35,0.7)",numBg:"#dbedff",lineBg:"#f1f8ff"},emptyBlockBg:"#fafbfc",selectedLineHighlightBg:"rgba(255,223,93,0.2)",selectedLineHighlightBorder:"#ffd33d",expander:{icon:"#586069",hoverIcon:"#ffffff",hoverBg:"#0366d6"},commentButton:{icon:"#ffffff",bg:"#0366d6",gradientBg:"hsla(212,97%,48%,1)"}},globalNav:{logo:"#ffffff",bg:"#24292e",text:"#ffffff",icon:"#ffffff",inputBg:"#fafbfc",inputBorder:"#fafbfc",inputIcon:"#d1d5da",inputPlaceholder:"#959da5"},footerInvertocat:{octicon:"#d1d5da",octiconHover:"#6a737d"},prState:{draft:{text:"#ffffff",bg:"#6a737d",border:"rgba(0,0,0,0)"},open:{text:"#ffffff",bg:"#28a745",border:"rgba(0,0,0,0)"},merged:{text:"#ffffff",bg:"#6f42c1",border:"rgba(0,0,0,0)"},closed:{text:"#ffffff",bg:"#d73a49",border:"rgba(0,0,0,0)"}},topicTag:{text:"#0366d6",bg:"#f1f8ff",hoverBg:"hsla(210,100%,93%,1)",activeBg:"hsla(210,100%,95%,1)"},mergeBox:{successIconBg:"#28a745",successIconText:"#ffffff",successIconBorder:"rgba(0,0,0,0)",successIndicatorBg:"#28a745",successIndicatorBorder:"rgba(0,0,0,0)",mergedIconBg:"#6f42c1",mergedIconText:"#ffffff",mergedIconBorder:"rgba(0,0,0,0)",mergedBoxBorder:"#6f42c1",neutralIconBg:"#6a737d",neutralIconText:"#ffffff",neutralIconBorder:"rgba(0,0,0,0)",neutralIndicatorBg:"#6a737d",neutralIndicatorBorder:"rgba(0,0,0,0)",warningIconBg:"#dbab09",warningIconText:"#ffffff",warningIconBorder:"rgba(0,0,0,0)",warningBoxBorder:"#ffd33d",warningMergeHighlight:"rgba(0,0,0,0)",errorIconBg:"#d73a49",errorIconText:"#ffffff",errorIconBorder:"rgba(0,0,0,0)",errorIndicatorBg:"#d73a49",errorIndicatorBorder:"rgba(0,0,0,0)"},project:{cardBg:"#ffffff",headerBg:"#24292e",sidebarBg:"#ffffff",gradientIn:"#ffffff",gradientOut:"rgba(255,255,255,0)"},checks:{bg:"#24292e",textPrimary:"#fafbfc",textSecondary:"#959da5",textLink:"#79b8ff",btnIcon:"#d1d5da",btnHoverIcon:"#fafbfc",btnHoverBg:"rgba(255,255,255,0.125)",inputText:"#f6f8fa",inputPlaceholderText:"#959da5",inputFocusText:"#959da5",inputBg:"#2f363d",donutError:"#cb2431",donutPending:"#dbab09",donutSuccess:"#28a745",donutNeutral:"#959da5",dropdownText:"#d1d5da",dropdownBg:"#2f363d",dropdownBorder:"#444d56",dropdownHoverText:"#fafbfc",dropdownHoverBg:"#444d56",dropdownBtnHoverText:"#fafbfc",dropdownBtnHoverBg:"#2f363d",scrollbarThumbBg:"#586069",headerLabelText:"#e1e4e8",headerLabelOpenText:"#fafbfc",headerBorder:"#2f363d",headerIcon:"#959da5",lineText:"#e1e4e8",lineNumText:"rgba(149,157,165,0.75)",lineTimestampText:"#959da5",lineHoverBg:"#2f363d",lineSelectedBg:"rgba(33,136,255,0.15)",lineSelectedNumText:"#79b8ff",lineDtFmText:"#24292e",lineDtFmBg:"#ffd33d",gateBg:"rgba(249,197,19,0.15)",gateText:"#e1e4e8",gateWaitingText:"#d1d5da",stepHeaderOpenBg:"#2f363d",stepErrorText:"#f97583",stepWarningText:"#ffea7f",loglineText:"#959da5",loglineNumText:"rgba(149,157,165,0.75)",loglineDebugText:"#b392f0",loglineErrorText:"#e1e4e8",loglineErrorNumText:"#f97583",loglineErrorBg:"rgba(203,36,49,0.15)",loglineWarningText:"#e1e4e8",loglineWarningNumText:"#ffea7f",loglineWarningBg:"rgba(249,197,19,0.15)",loglineCommandText:"#79b8ff",loglineSectionText:"#85e89d",ansi:{black:"#24292e",blackBright:"#2f363d",white:"#e1e4e8",whiteBright:"#e1e4e8",gray:"#959da5",red:"#f97583",redBright:"#fdaeb7",green:"#85e89d",greenBright:"#bef5cb",yellow:"#ffea7f",yellowBright:"#fff5b1",blue:"#79b8ff",blueBright:"#c8e1ff",magenta:"#b392f0",magentaBright:"#d1bcf9",cyan:"#76e3ea",cyanBright:"#b3f0ff"}},introShelf:{gradientLeft:"#f1f8ff",gradientRight:"#dcffe4",gradientIn:"#ffffff",gradientOut:"rgba(255,255,255,0)"},marketingIcon:{primary:"#2188ff",secondary:"#79b8ff"},prettylights:{syntax:{comment:"#6a737d",constant:"#005cc5",entity:"#6f42c1",storageModifierImport:"#24292e",entityTag:"#22863a",keyword:"#d73a49",string:"#032f62",variable:"#e36209",brackethighlighterUnmatched:"#b31d28",invalidIllegalText:"#fafbfc",invalidIllegalBg:"#b31d28",carriageReturnText:"#fafbfc",carriageReturnBg:"#d73a49",stringRegexp:"#22863a",markupList:"#735c0f",markupHeading:"#005cc5",markupItalic:"#24292e",markupBold:"#24292e",markupDeletedText:"#b31d28",markupDeletedBg:"#ffeef0",markupInsertedText:"#22863a",markupInsertedBg:"#f0fff4",markupChangedText:"#e36209",markupChangedBg:"#ffebda",markupIgnoredText:"#f6f8fa",markupIgnoredBg:"#005cc5",metaDiffRange:"#6f42c1",brackethighlighterAngle:"#586069",sublimelinterGutterMark:"#959da5",constantOtherReferenceLink:"#032f62"}},codemirror:{text:"#24292e",bg:"#ffffff",guttersBg:"#ffffff",guttermarkerText:"#ffffff",guttermarkerSubtleText:"#d1d5da",linenumberText:"#959da5",cursor:"#24292e",selectionBg:"#c8e1ff",activelineBg:"#fafbfc",matchingbracketText:"#24292e",linesBg:"#ffffff",syntax:{comment:"#6a737d",constant:"#005cc5",entity:"#6f42c1",keyword:"#d73a49",storage:"#d73a49",string:"#032f62",support:"#005cc5",variable:"#e36209"}},ansi:{black:"#24292e",blackBright:"#586069",white:"#6a737d",whiteBright:"#959da5",gray:"#6a737d",red:"#d73a49",redBright:"#cb2431",green:"#22863a",greenBright:"#28a745",yellow:"#b08800",yellowBright:"#dbab09",blue:"#0366d6",blueBright:"#2188ff",magenta:"#6f42c1",magentaBright:"#8a63d2",cyan:"#1b7c83",cyanBright:"#3192aa"}},shadows:{shadow:{small:"0 1px 0 rgba(27,31,35,0.04)",medium:"0 3px 6px rgba(149,157,165,0.15)",large:"0 8px 24px rgba(149,157,165,0.2)",extraLarge:"0 12px 48px rgba(149,157,165,0.3)",highlight:"inset 0 1px 0 rgba(255,255,255,0.25)",inset:"inset 0 1px 0 rgba(225,228,232,0.2)"},state:{focus:{shadow:"0 0 0 3px rgba(3,102,214,0.3)"}},autocomplete:{shadow:"0 3px 6px rgba(149,157,165,0.15)"},btn:{shadow:"0 1px 0 rgba(27,31,35,0.04)",insetShadow:"inset 0 1px 0 rgba(255,255,255,0.25)",focusShadow:"0 0 0 3px rgba(3,102,214,0.3)",shadowActive:"inset 0 0.15em 0.3em rgba(27,31,35,0.15)",shadowInputFocus:"0 0 0 0.2em rgba(3,102,214,0.3)",primary:{shadow:"0 1px 0 rgba(27,31,35,0.1)",insetShadow:"inset 0 1px 0 rgba(255,255,255,0.03)",selectedShadow:"inset 0 1px 0 rgba(20,70,32,0.2)",focusShadow:"0 0 0 3px rgba(46,164,79,0.4)"},outline:{hoverShadow:"0 1px 0 rgba(27,31,35,0.1)",hoverInsetShadow:"inset 0 1px 0 rgba(255,255,255,0.03)",selectedShadow:"inset 0 1px 0 rgba(5,38,76,0.2)",focusShadow:"0 0 0 3px rgba(0,92,197,0.4)"},danger:{hoverShadow:"0 1px 0 rgba(27,31,35,0.1)",hoverInsetShadow:"inset 0 1px 0 rgba(255,255,255,0.03)",selectedShadow:"inset 0 1px 0 rgba(134,24,29,0.2)",focusShadow:"0 0 0 3px rgba(203,36,49,0.4)"}},dropdown:{shadow:"0 8px 24px rgba(149,157,165,0.2)"},overlay:{shadow:"0 1px 3px rgba(27,31,35,0.12), 0 8px 24px rgba(68,77,86,0.12)"},input:{shadow:"inset 0 1px 2px rgba(27,31,35,0.075)"},avatar:{childShadow:"-2px -2px 0 rgba(255,255,255,0.8)"},toast:{shadow:"0 8px 24px rgba(149,157,165,0.2)"},selectMenu:{shadow:"0 0 18px rgba(27,31,35,0.4)"}}},dark:{colors:{auto:{black:"#f0f6fc",white:"#010409",gray:["#0d1117","#161b22","#21262d","#30363d","#484f58","#6e7681","#8b949e","#b1bac4","#c9d1d9","#f0f6fc"],blue:["#051d4d","#0c2d6b","#0d419d","#1158c7","#1f6feb","#388bfd","#58a6ff","#79c0ff","#a5d6ff","#cae8ff"],green:["#04260f","#033a16","#0f5323","#196c2e","#238636","#2ea043","#3fb950","#56d364","#7ee787","#aff5b4"],yellow:["#341a00","#4b2900","#693e00","#845306","#9e6a03","#bb8009","#d29922","#e3b341","#f2cc60","#f8e3a1"],orange:["#3d1300","#5a1e02","#762d0a","#9b4215","#bd561d","#db6d28","#f0883e","#ffa657","#ffc680","#ffdfb6"],red:["#490202","#67060c","#8e1519","#b62324","#da3633","#f85149","#ff7b72","#ffa198","#ffc1ba","#ffdcd7"],purple:["#271052","#3c1e70","#553098","#6e40c9","#8957e5","#a371f7","#bc8cff","#d2a8ff","#e2c5ff","#eddeff"],pink:["#42062a","#5e103e","#7d2457","#9e3670","#bf4b8a","#db61a2","#f778ba","#ff9bce","#ffbedd","#ffdaec"]},text:{primary:"#c9d1d9",secondary:"#8b949e",tertiary:"#8b949e",placeholder:"#484f58",disabled:"#484f58",inverse:"#0d1117",link:"#58a6ff",danger:"#f85149",success:"#56d364",warning:"#e3b341",white:"#f0f6fc"},icon:{primary:"#c9d1d9",secondary:"#6e7681",tertiary:"#484f58",info:"#58a6ff",danger:"#f85149",success:"#56d364",warning:"#e3b341"},border:{primary:"#30363d",secondary:"#21262d",tertiary:"#6e7681",overlay:"#30363d",inverse:"#f0f6fc",info:"rgba(56,139,253,0.4)",danger:"rgba(248,81,73,0.4)",success:"rgba(63,185,80,0.4)",warning:"rgba(187,128,9,0.4)"},bg:{canvas:"#0d1117",canvasMobile:"#010409",canvasInverse:"#f0f6fc",canvasInset:"hsla(216,28%,5%,1)",primary:"#0d1117",secondary:"#0d1117",tertiary:"#161b22",overlay:"rgba(28,33,40,1)",backdrop:"rgba(1,4,9,0.8)",info:"rgba(56,139,253,0.1)",infoInverse:"#388bfd",danger:"rgba(248,81,73,0.1)",dangerInverse:"#da3633",success:"rgba(46,160,67,0.1)",successInverse:"#2ea043",warning:"rgba(187,128,9,0.1)",warningInverse:"#bb8009"},state:{hover:{primaryBg:"#1f6feb",primaryBorder:"#388bfd",primaryText:"#f0f6fc",primaryIcon:"#f0f6fc",secondaryBg:"#161b22",secondaryBorder:"#161b22"},selected:{primaryBg:"#1f6feb",primaryBorder:"#388bfd",primaryText:"#f0f6fc",primaryIcon:"#f0f6fc"},focus:{border:"#388bfd"}},fade:{fg10:"rgba(240,246,252,0.1)",fg15:"rgba(240,246,252,0.15)",fg30:"rgba(240,246,252,0.3)",fg50:"rgba(240,246,252,0.5)",fg70:"rgba(240,246,252,0.7)",fg85:"rgba(240,246,252,0.85)",black10:"rgba(1,4,9,0.1)",black15:"rgba(1,4,9,0.15)",black30:"rgba(1,4,9,0.3)",black50:"rgba(1,4,9,0.5)",black70:"rgba(1,4,9,0.7)",black85:"rgba(1,4,9,0.85)",white10:"rgba(240,246,252,0.1)",white15:"rgba(240,246,252,0.15)",white30:"rgba(240,246,252,0.3)",white50:"rgba(240,246,252,0.5)",white70:"rgba(240,246,252,0.7)",white85:"rgba(240,246,252,0.85)"},alert:{info:{text:"#79c0ff",icon:"#79c0ff",bg:"rgba(56,139,253,0.1)",border:"rgba(56,139,253,0.4)"},warn:{text:"#e3b341",icon:"#e3b341",bg:"rgba(187,128,9,0.1)",border:"rgba(187,128,9,0.4)"},error:{text:"#ff7b72",icon:"#ff7b72",bg:"rgba(248,81,73,0.1)",border:"rgba(248,81,73,0.4)"},success:{text:"#56d364",icon:"#56d364",bg:"rgba(46,160,67,0.1)",border:"rgba(46,160,67,0.4)"}},autocomplete:{rowBorder:"#30363d"},blankslate:{icon:"hsla(214,10%,36%,1)"},btn:{text:"#c9d1d9",bg:"#21262d",border:"#30363d",hoverBg:"#30363d",hoverBorder:"#8b949e",activeBg:"hsla(212,12%,18%,1)",activeBorder:"#6e7681",selectedBg:"#161b22",focusBg:"#21262d",focusBorder:"#8b949e",primary:{text:"#ffffff",bg:"#238636",border:"#2ea043",hoverBg:"#2ea043",hoverBorder:"#3fb950",selectedBg:"#238636",disabledText:"rgba(240,246,252,0.5)",disabledBg:"rgba(35,134,54,0.6)",disabledBorder:"rgba(0,0,0,0)",focusBg:"#238636",focusBorder:"#3fb950",icon:"#f0f6fc",counterBg:"rgba(240,246,252,0.2)"},outline:{text:"#58a6ff",hoverText:"#58a6ff",hoverBg:"#30363d",hoverBorder:"#58a6ff",hoverCounterBg:"rgba(240,246,252,0.2)",selectedText:"#f0f6fc",selectedBg:"#0d419d",selectedBorder:"rgba(240,246,252,0.1)",disabledText:"rgba(88,166,255,0.5)",disabledBg:"#0d1117",disabledCounterBg:"rgba(31,111,235,0.05)",focusBorder:"#58a6ff",counterBg:"rgba(31,111,235,0.1)"},danger:{text:"#f85149",hoverText:"#ffffff",hoverBg:"#da3633",hoverBorder:"#f85149",hoverCounterBg:"rgba(255,255,255,0.2)",selectedText:"#ffffff",selectedBg:"#b62324",selectedBorder:"rgba(240,246,252,0.1)",disabledText:"rgba(248,81,73,0.5)",disabledBg:"#0d1117",disabledCounterBg:"rgba(218,54,51,0.05)",focusBorder:"#f85149",counterBg:"rgba(218,54,51,0.1)",icon:"#f85149",hoverIcon:"#f0f6fc"},counterBg:"#30363d"},counter:{text:"#c9d1d9",bg:"#30363d",primary:{text:"#c9d1d9",bg:"#6e7681"},secondary:{text:"#8b949e",bg:"rgba(139,148,158,0.2)"}},label:{border:"#30363d",primary:{text:"#b1bac4",border:"#6e7681"},secondary:{text:"#8b949e",border:"#30363d"},info:{text:"#388bfd",border:"rgba(56,139,253,0.4)"},success:{text:"#3fb950",border:"rgba(46,160,67,0.4)"},warning:{text:"#e3b341",border:"rgba(242,211,91,0.4)"},danger:{text:"#f85149",border:"rgba(248,81,73,0.4)"},orange:{text:"#db6d28",border:"rgba(219,109,40,0.4)"}},input:{bg:"#0d1117",contrastBg:"rgba(1,4,9,0.5)",border:"#21262d",disabledBg:"#161b22",disabledBorder:"#30363d",warningBorder:"#d29922",errorBorder:"#f85149",tooltip:{success:{text:"#56d364",bg:"rgba(16,31,27,1)",border:"rgba(28,83,43,1)"},warning:{text:"#e3b341",bg:"rgba(30,28,22,1)",border:"rgba(93,68,17,1)"},error:{text:"#ff7b72",bg:"rgba(37,23,28,1)",border:"rgba(121,46,46,1)"}}},avatar:{bg:"rgba(240,246,252,0.1)",border:"rgba(240,246,252,0.1)",stackFade:"#30363d",stackFadeMore:"#21262d"},toast:{text:"#c9d1d9",bg:"#30363d",border:"#6e7681",icon:"#f0f6fc",iconBg:"#1f6feb",iconBorder:"#388bfd",success:{text:"#c9d1d9",border:"#6e7681",icon:"#f0f6fc",iconBg:"#2ea043",iconBorder:"#3fb950"},warning:{text:"#c9d1d9",border:"#6e7681",icon:"#f0f6fc",iconBg:"#bb8009",iconBorder:"#d29922"},danger:{text:"#c9d1d9",border:"#6e7681",icon:"#f0f6fc",iconBg:"#da3633",iconBorder:"#f85149"},loading:{text:"#c9d1d9",border:"#6e7681",icon:"#f0f6fc",iconBg:"#6e7681",iconBorder:"#8b949e"}},timeline:{text:"#b1bac4",badgeBg:"#0d1117",badgeSuccessBorder:"#2ea043",targetBadgeBorder:"#1f6feb",targetBadgeShadow:"#0d419d"},selectMenu:{borderSecondary:"#30363d",backdropBg:"rgba(1,4,9,0.5)",backdropBorder:"#484f58",tapHighlight:"rgba(48,54,61,0.5)",tapFocusBg:"#0c2d6b"},box:{blueBorder:"#0d419d",rowYellowBg:"rgba(235,196,64,0.1)",rowBlueBg:"rgba(121,192,255,0.1)",headerBlueBg:"#0d1117",headerBlueBorder:"#30363d",borderInfo:"rgba(56,139,253,0.4)",bgInfo:"rgba(56,139,253,0.1)",borderWarning:"rgba(187,128,9,0.4)",bgWarning:"rgba(187,128,9,0.1)"},branchName:{text:"#c9d1d9",icon:"#b1bac4",bg:"rgba(88,166,255,0.1)",link:{text:"#58a6ff",icon:"#58a6ff",bg:"rgba(88,166,255,0.1)"}},markdown:{codeBg:"rgba(240,246,252,0.15)",frameBorder:"hsla(212,12%,26%,1)",blockquoteBorder:"hsla(212,12%,26%,1)",tableBorder:"hsla(212,12%,26%,1)",tableTrBorder:"hsla(212,12%,17%,1)"},menu:{headingText:"#8b949e",borderActive:"#f78166",bgActive:"#161b22"},sidenav:{selectedBg:"#21262d",borderActive:"#f78166"},header:{text:"rgba(240,246,252,0.7)",bg:"#161b22",logo:"#f0f6fc"},filterItem:{barBg:"hsla(215,21%,9%,1)"},hiddenTextExpander:{bg:"#21262d",bgHover:"#30363d"},dragAndDrop:{border:"hsla(212,12%,16%,1)"},uploadEnabled:{border:"hsla(212,12%,26%,1)",borderFocused:"hsla(216,84%,60%,1)"},previewableCommentForm:{border:"hsla(212,12%,16%,1)"},underlinenav:{border:"rgba(48,54,61,0)",borderHover:"#30363d",borderActive:"#f78166",text:"#8b949e",textHover:"#c9d1d9",textActive:"#c9d1d9",icon:"#6e7681",iconHover:"#c9d1d9",iconActive:"#c9d1d9",counterText:"#8b949e",counterBg:"rgba(139,148,158,0.2)"},verifiedBadge:{text:"#3fb950",bg:"rgba(63,185,80,0.1)",border:"rgba(63,185,80,0.4)"},socialCount:{bg:"#21262d"},tooltip:{text:"#f0f6fc",bg:"#6e7681"},headerSearch:{bg:"#0d1117",border:"#21262d"},searchKeyword:{hl:"rgba(187,128,9,0.4)"},diffstat:{neutralBg:"#30363d",neutralBorder:"rgba(240,246,252,0.1)",deletionBg:"#da3633",deletionBorder:"#f85149",additionBg:"#238636",additionBorder:"#2ea043"},mktg:{success:"rgba(41,147,61,1)",info:"rgba(42,123,243,1)",bgShadeGradient:{top:"rgba(1,4,9,0.065)",bottom:"rgba(1,4,9,0)"},btn:{bg:{top:"hsla(228,82%,66%,1)",bottom:"#4969ed"},bgOverlay:{top:"hsla(228,74%,59%,1)",bottom:"#3355e0"},text:"#f0f6fc",primary:{bg:{top:"hsla(137,56%,46%,1)",bottom:"#2ea44f"},bgOverlay:{top:"hsla(134,60%,38%,1)",bottom:"#22863a"},text:"#f0f6fc"},enterprise:{bg:{top:"hsla(249,100%,72%,1)",bottom:"#6f57ff"},bgOverlay:{top:"hsla(248,65%,63%,1)",bottom:"#614eda"},text:"#f0f6fc"},outline:{text:"#f0f6fc",border:"rgba(240,246,252,0.3)",hover:{text:"#f0f6fc",border:"rgba(240,246,252,0.5)"},focus:{border:"#f0f6fc",borderInset:"rgba(240,246,252,0.5)"}},dark:{text:"#f0f6fc",border:"rgba(240,246,252,0.3)",hover:{text:"#f0f6fc",border:"rgba(240,246,252,0.5)"},focus:{border:"#f0f6fc",borderInset:"rgba(240,246,252,0.5)"}}}},filesExplorerIcon:"#6e7681",hlAuthorBg:"#051d4d",hlAuthorBorder:"#0d419d",logoSubdued:"#30363d",discussionBorder:"hsla(135,22%,26%,1)",discussionBgSuccess:"rgba(46,160,67,0.1)",actionsWorkflowTableStickyBg:"rgba(13,17,23,0.95)",repoLanguageColorBorder:"rgba(240,246,252,0.2)",codeSelectionBg:"rgba(121,192,255,0.3)",highlight:{text:"#ffd467",bg:"rgba(204,143,44,0.38)"},blob:{lineHighlightBg:"rgba(210,153,34,0.15)",lineHighlightBorder:"#e3b341"},diff:{addition:{text:"#56d364",bg:"rgba(46,160,67,0.2)",border:"#196c2e"},deletion:{text:"#f85149",bg:"rgba(218,54,51,0.2)",border:"#b62324"},change:{text:"#e3b341",bg:"#341a00",border:"#9e6a03"}},diffBlob:{numText:"rgba(240,246,252,0.3)",numHoverText:"rgba(240,246,252,0.6)",addition:{numText:"#3fb950",numHoverText:"#7ee787",numBg:"rgba(46,160,67,0.1)",lineBg:"rgba(46,160,67,0.2)",wordBg:"rgba(46,160,67,0.55)"},deletion:{numText:"#f85149",numHoverText:"#ffa198",numBg:"rgba(218,54,51,0.1)",lineBg:"rgba(218,54,51,0.2)",wordBg:"rgba(218,54,51,0.5)"},hunk:{text:"#8b949e",numBg:"rgba(88,166,255,0.15)",lineBg:"rgba(88,166,255,0.1)"},emptyBlockBg:"#161b22",selectedLineHighlightBg:"rgba(187,128,9,0.1)",selectedLineHighlightBorder:"#bb8009",expander:{icon:"#8b949e",hoverIcon:"#f0f6fc",hoverBg:"#1f6feb"},commentButton:{icon:"#f0f6fc",bg:"#1f6feb",gradientBg:"hsla(216,84%,57%,1)"}},globalNav:{logo:"#f0f6fc",bg:"#161b22",text:"#c9d1d9",icon:"#c9d1d9",inputBg:"#0d1117",inputBorder:"#21262d",inputIcon:"#21262d",inputPlaceholder:"#484f58"},footerInvertocat:{octicon:"#30363d",octiconHover:"#6e7681"},prState:{draft:{text:"#8b949e",bg:"rgba(139,148,158,0.1)",border:"rgba(139,148,158,0.4)"},open:{text:"#3fb950",bg:"rgba(63,185,80,0.1)",border:"rgba(63,185,80,0.4)"},merged:{text:"#a371f7",bg:"rgba(188,140,255,0.1)",border:"rgba(188,140,255,0.4)"},closed:{text:"#f85149",bg:"rgba(218,54,51,0.1)",border:"rgba(218,54,51,0.4)"}},topicTag:{text:"#58a6ff",bg:"rgba(56,139,253,0.1)",hoverBg:"rgba(56,139,253,0.2)",activeBg:"rgba(56,139,253,0.15)"},mergeBox:{successIconBg:"rgba(46,160,67,0.1)",successIconText:"#3fb950",successIconBorder:"rgba(46,160,67,0.4)",successIndicatorBg:"#238636",successIndicatorBorder:"#2ea043",mergedIconBg:"rgba(188,140,255,0.1)",mergedIconText:"#a371f7",mergedIconBorder:"rgba(188,140,255,0.4)",mergedBoxBorder:"rgba(188,140,255,0.4)",neutralIconBg:"rgba(201,209,217,0.1)",neutralIconText:"#8b949e",neutralIconBorder:"rgba(201,209,217,0.4)",neutralIndicatorBg:"#484f58",neutralIndicatorBorder:"#6e7681",warningIconBg:"rgba(187,128,9,0.1)",warningIconText:"#e3b341",warningIconBorder:"rgba(187,128,9,0.4)",warningBoxBorder:"rgba(187,128,9,0.4)",warningMergeHighlight:"rgba(187,128,9,0.1)",errorIconBg:"rgba(248,81,73,0.1)",errorIconText:"#f85149",errorIconBorder:"rgba(248,81,73,0.4)",errorIndicatorBg:"#da3633",errorIndicatorBorder:"#f85149"},project:{cardBg:"#161b22",headerBg:"#0d1117",sidebarBg:"#161b22",gradientIn:"#161b22",gradientOut:"rgba(22,27,34,0)"},checks:{bg:"hsla(216,28%,5%,1)",textPrimary:"#c9d1d9",textSecondary:"#8b949e",textLink:"#58a6ff",btnIcon:"#6e7681",btnHoverIcon:"#c9d1d9",btnHoverBg:"#30363d",inputText:"#8b949e",inputPlaceholderText:"#484f58",inputFocusText:"#c9d1d9",inputBg:"#0d1117",donutError:"#f85149",donutPending:"#d29922",donutSuccess:"#2ea043",donutNeutral:"#8b949e",dropdownText:"#c9d1d9",dropdownBg:"rgba(28,33,40,1)",dropdownBorder:"#30363d",dropdownHoverText:"#f0f6fc",dropdownHoverBg:"#1f6feb",dropdownBtnHoverText:"#f0f6fc",dropdownBtnHoverBg:"#161b22",scrollbarThumbBg:"#30363d",headerLabelText:"#8b949e",headerLabelOpenText:"#c9d1d9",headerBorder:"#21262d",headerIcon:"#6e7681",lineText:"#8b949e",lineNumText:"#8b949e",lineTimestampText:"#8b949e",lineHoverBg:"#161b22",lineSelectedBg:"rgba(56,139,253,0.1)",lineSelectedNumText:"#58a6ff",lineDtFmText:"#0d1117",lineDtFmBg:"#d29922",gateBg:"rgba(132,83,6,0.15)",gateText:"#8b949e",gateWaitingText:"#e3b341",stepHeaderOpenBg:"#161b22",stepErrorText:"#f85149",stepWarningText:"#e3b341",loglineText:"#6e7681",loglineNumText:"#8b949e",loglineDebugText:"#bc8cff",loglineErrorText:"#8b949e",loglineErrorNumText:"#8b949e",loglineErrorBg:"rgba(248,81,73,0.1)",loglineWarningText:"#8b949e",loglineWarningNumText:"#e3b341",loglineWarningBg:"rgba(187,128,9,0.1)",loglineCommandText:"#58a6ff",loglineSectionText:"#56d364",ansi:{black:"#0d1117",blackBright:"#161b22",white:"#b1bac4",whiteBright:"#b1bac4",gray:"#6e7681",red:"#ff7b72",redBright:"#ffa198",green:"#3fb950",greenBright:"#56d364",yellow:"#d29922",yellowBright:"#e3b341",blue:"#58a6ff",blueBright:"#79c0ff",magenta:"#bc8cff",magentaBright:"#d2a8ff",cyan:"#76e3ea",cyanBright:"#b3f0ff"}},introShelf:{gradientLeft:"rgba(56,139,253,0.1)",gradientRight:"rgba(46,160,67,0.1)",gradientIn:"#0d1117",gradientOut:"rgba(13,17,23,0)"},marketingIcon:{primary:"#79c0ff",secondary:"#1f6feb"},prettylights:{syntax:{comment:"#8b949e",constant:"#79c0ff",entity:"#d2a8ff",storageModifierImport:"#c9d1d9",entityTag:"#7ee787",keyword:"#ff7b72",string:"#a5d6ff",variable:"#ffa657",brackethighlighterUnmatched:"#f85149",invalidIllegalText:"#f0f6fc",invalidIllegalBg:"#8e1519",carriageReturnText:"#f0f6fc",carriageReturnBg:"#b62324",stringRegexp:"#7ee787",markupList:"#f2cc60",markupHeading:"#1f6feb",markupItalic:"#c9d1d9",markupBold:"#c9d1d9",markupDeletedText:"#ffdcd7",markupDeletedBg:"#67060c",markupInsertedText:"#aff5b4",markupInsertedBg:"#033a16",markupChangedText:"#ffdfb6",markupChangedBg:"#5a1e02",markupIgnoredText:"#c9d1d9",markupIgnoredBg:"#1158c7",metaDiffRange:"#d2a8ff",brackethighlighterAngle:"#8b949e",sublimelinterGutterMark:"#484f58",constantOtherReferenceLink:"#a5d6ff"}},codemirror:{text:"#c9d1d9",bg:"#0d1117",guttersBg:"#0d1117",guttermarkerText:"#0d1117",guttermarkerSubtleText:"#6e7681",linenumberText:"#8b949e",cursor:"#f0f6fc",selectionBg:"rgba(121,192,255,0.3)",activelineBg:"#161b22",matchingbracketText:"#c9d1d9",linesBg:"#0d1117",syntax:{comment:"#8b949e",constant:"#79c0ff",entity:"#d2a8ff",keyword:"#ff7b72",storage:"#ff7b72",string:"#a5d6ff",support:"#79c0ff",variable:"#ffa657"}},ansi:{black:"#484f58",blackBright:"#6e7681",white:"#b1bac4",whiteBright:"#f0f6fc",gray:"#6e7681",red:"#ff7b72",redBright:"#ffa198",green:"#3fb950",greenBright:"#56d364",yellow:"#d29922",yellowBright:"#e3b341",blue:"#58a6ff",blueBright:"#79c0ff",magenta:"#bc8cff",magentaBright:"#d2a8ff",cyan:"#39c5cf",cyanBright:"#56d4dd"}},shadows:{shadow:{small:"0 0 transparent",medium:"0 3px 6px #010409",large:"0 8px 24px #010409",extraLarge:"0 12px 48px #010409",highlight:"0 0 transparent",inset:"0 0 transparent"},state:{focus:{shadow:"0 0 0 3px #0c2d6b"}},autocomplete:{shadow:"0 16px 32px rgba(1,4,9,0.85)"},btn:{shadow:"0 0 transparent",insetShadow:"0 0 transparent",focusShadow:"0 0 0 3px rgba(139,148,158,0.3)",shadowActive:"inset 0 0.15em 0.3em rgba(1,4,9,0.15)",shadowInputFocus:"0 0 0 0.2em rgba(31,111,235,0.3)",primary:{shadow:"0 0 transparent",insetShadow:"0 0 transparent",selectedShadow:"0 0 transparent",focusShadow:"0 0 0 3px rgba(46,164,79,0.4)"},outline:{hoverShadow:"0 1px 0 rgba(1,4,9,0.1)",hoverInsetShadow:"inset 0 1px 0 rgba(240,246,252,0.03)",selectedShadow:"0 0 transparent",focusShadow:"0 0 0 3px rgba(17,88,199,0.4)"},danger:{hoverShadow:"0 0 transparent",hoverInsetShadow:"0 0 transparent",selectedShadow:"0 0 transparent",focusShadow:"0 0 0 3px rgba(182,35,36,0.4)"}},dropdown:{shadow:"0 16px 32px rgba(1,4,9,0.85)"},overlay:{shadow:"0 0 0 1px #30363d, 0 16px 32px rgba(1,4,9,0.85)"},input:{shadow:"0 0 transparent"},avatar:{childShadow:"-2px -2px 0 #0d1117"},toast:{shadow:"0 8px 24px #010409"},selectMenu:{shadow:"0 0 18px rgba(1,4,9,0.4)"},checks:{inputShadow:"0 0 0 1px #21262d"}}},dark_dimmed:{colors:{auto:{black:"#cdd9e5",white:"#1c2128",gray:["#22272e","#2d333b","#373e47","#444c56","#545d68","#636e7b","#768390","#909dab","#adbac7","#cdd9e5"],blue:["#0f2d5c","#143d79","#1b4b91","#255ab2","#316dca","#4184e4","#539bf5","#6cb6ff","#96d0ff","#c6e6ff"],green:["#113417","#1b4721","#245829","#2b6a30","#347d39","#46954a","#57ab5a","#6bc46d","#8ddb8c","#b4f1b4"],yellow:["#452700","#593600","#6c4400","#805400","#966600","#ae7c14","#c69026","#daaa3f","#eac55f","#fbe090"],orange:["#4d210c","#682d0f","#7f3913","#94471b","#ae5622","#cc6b2c","#e0823d","#f69d50","#ffbc6f","#ffddb0"],red:["#78191b","#78191b","#922323","#ad2e2c","#c93c37","#e5534b","#f47067","#ff938a","#ffb8b0","#ffd8d3"],purple:["#352160","#472c82","#5936a2","#6b44bc","#8256d0","#986ee2","#b083f0","#dcbdfb","#dcbdfb","#eedcff"],pink:["#551639","#69264a","#7e325a","#983b6e","#ae4c82","#c96198","#e275ad","#fc8dc7","#ffb3d8","#ffd7eb"]},text:{primary:"#adbac7",secondary:"#768390",tertiary:"#768390",placeholder:"#545d68",disabled:"#545d68",inverse:"#22272e",link:"#539bf5",danger:"#e5534b",success:"#6bc46d",warning:"#daaa3f",white:"#cdd9e5"},icon:{primary:"#adbac7",secondary:"#636e7b",tertiary:"#545d68",info:"#539bf5",danger:"#e5534b",success:"#6bc46d",warning:"#daaa3f"},border:{primary:"#444c56",secondary:"#373e47",tertiary:"#636e7b",overlay:"#444c56",inverse:"#cdd9e5",info:"rgba(65,132,228,0.4)",danger:"rgba(229,83,75,0.4)",success:"rgba(87,171,90,0.4)",warning:"rgba(174,124,20,0.4)"},bg:{canvas:"#22272e",canvasMobile:"#1c2128",canvasInverse:"#cdd9e5",canvasInset:"hsla(215,15%,14%,1)",primary:"#22272e",secondary:"#22272e",tertiary:"#2d333b",overlay:"rgba(50,57,65,1)",backdrop:"rgba(28,33,40,0.8)",info:"rgba(65,132,228,0.1)",infoInverse:"#4184e4",danger:"rgba(229,83,75,0.1)",dangerInverse:"#c93c37",success:"rgba(70,149,74,0.1)",successInverse:"#46954a",warning:"rgba(174,124,20,0.1)",warningInverse:"#ae7c14"},state:{hover:{primaryBg:"#316dca",primaryBorder:"#4184e4",primaryText:"#cdd9e5",primaryIcon:"#cdd9e5",secondaryBg:"#2d333b",secondaryBorder:"#2d333b"},selected:{primaryBg:"#316dca",primaryBorder:"#4184e4",primaryText:"#cdd9e5",primaryIcon:"#cdd9e5"},focus:{border:"#4184e4"}},fade:{fg10:"rgba(205,217,229,0.1)",fg15:"rgba(205,217,229,0.15)",fg30:"rgba(205,217,229,0.3)",fg50:"rgba(205,217,229,0.5)",fg70:"rgba(205,217,229,0.7)",fg85:"rgba(205,217,229,0.85)",black10:"rgba(28,33,40,0.1)",black15:"rgba(28,33,40,0.15)",black30:"rgba(28,33,40,0.3)",black50:"rgba(28,33,40,0.5)",black70:"rgba(28,33,40,0.7)",black85:"rgba(28,33,40,0.85)",white10:"rgba(205,217,229,0.1)",white15:"rgba(205,217,229,0.15)",white30:"rgba(205,217,229,0.3)",white50:"rgba(205,217,229,0.5)",white70:"rgba(205,217,229,0.7)",white85:"rgba(205,217,229,0.85)"},alert:{info:{text:"#6cb6ff",icon:"#6cb6ff",bg:"rgba(65,132,228,0.1)",border:"rgba(65,132,228,0.4)"},warn:{text:"#daaa3f",icon:"#daaa3f",bg:"rgba(174,124,20,0.1)",border:"rgba(174,124,20,0.4)"},error:{text:"#f47067",icon:"#f47067",bg:"rgba(229,83,75,0.1)",border:"rgba(229,83,75,0.4)"},success:{text:"#6bc46d",icon:"#6bc46d",bg:"rgba(70,149,74,0.1)",border:"rgba(70,149,74,0.4)"}},autocomplete:{rowBorder:"#444c56"},blankslate:{icon:"hsla(213,11%,42%,1)"},btn:{text:"#adbac7",bg:"#373e47",border:"#444c56",hoverBg:"#444c56",hoverBorder:"#768390",activeBg:"hsla(213,12%,27%,1)",activeBorder:"#636e7b",selectedBg:"#2d333b",focusBg:"#373e47",focusBorder:"#768390",primary:{text:"#ffffff",bg:"#347d39",border:"#46954a",hoverBg:"#46954a",hoverBorder:"#57ab5a",selectedBg:"#347d39",disabledText:"rgba(205,217,229,0.5)",disabledBg:"rgba(52,125,57,0.6)",disabledBorder:"rgba(0,0,0,0)",focusBg:"#347d39",focusBorder:"#57ab5a",icon:"#cdd9e5",counterBg:"rgba(205,217,229,0.2)"},outline:{text:"#539bf5",hoverText:"#539bf5",hoverBg:"#444c56",hoverBorder:"#539bf5",hoverCounterBg:"rgba(205,217,229,0.2)",selectedText:"#cdd9e5",selectedBg:"#1b4b91",selectedBorder:"rgba(205,217,229,0.1)",disabledText:"rgba(83,155,245,0.5)",disabledBg:"#22272e",disabledCounterBg:"rgba(49,109,202,0.05)",focusBorder:"#539bf5",counterBg:"rgba(49,109,202,0.1)"},danger:{text:"#e5534b",hoverText:"#ffffff",hoverBg:"#c93c37",hoverBorder:"#e5534b",hoverCounterBg:"rgba(255,255,255,0.2)",selectedText:"#ffffff",selectedBg:"#ad2e2c",selectedBorder:"rgba(205,217,229,0.1)",disabledText:"rgba(229,83,75,0.5)",disabledBg:"#22272e",disabledCounterBg:"rgba(201,60,55,0.05)",focusBorder:"#e5534b",counterBg:"rgba(201,60,55,0.1)",icon:"#e5534b",hoverIcon:"#cdd9e5"},counterBg:"#444c56"},counter:{text:"#adbac7",bg:"#444c56",primary:{text:"#adbac7",bg:"#636e7b"},secondary:{text:"#768390",bg:"rgba(118,131,144,0.2)"}},label:{border:"#444c56",primary:{text:"#909dab",border:"#636e7b"},secondary:{text:"#768390",border:"#444c56"},info:{text:"#4184e4",border:"rgba(65,132,228,0.4)"},success:{text:"#57ab5a",border:"rgba(70,149,74,0.4)"},warning:{text:"#daaa3f",border:"rgba(242,211,91,0.4)"},danger:{text:"#e5534b",border:"rgba(229,83,75,0.4)"},orange:{text:"#cc6b2c",border:"rgba(204,107,44,0.4)"}},input:{bg:"#22272e",contrastBg:"rgba(28,33,40,0.5)",border:"#373e47",disabledBg:"#2d333b",disabledBorder:"#444c56",warningBorder:"#c69026",errorBorder:"#e5534b",tooltip:{success:{text:"#6bc46d",bg:"rgba(38,50,49,1)",border:"rgba(51,90,59,1)"},warning:{text:"#daaa3f",bg:"rgba(48,48,43,1)",border:"rgba(98,78,34,1)"},error:{text:"#f47067",bg:"rgba(53,43,49,1)",border:"rgba(124,59,59,1)"}}},avatar:{bg:"rgba(205,217,229,0.1)",border:"rgba(205,217,229,0.1)",stackFade:"#444c56",stackFadeMore:"#373e47"},toast:{text:"#adbac7",bg:"#444c56",border:"#636e7b",icon:"#cdd9e5",iconBg:"#316dca",iconBorder:"#4184e4",success:{text:"#adbac7",border:"#636e7b",icon:"#cdd9e5",iconBg:"#46954a",iconBorder:"#57ab5a"},warning:{text:"#adbac7",border:"#636e7b",icon:"#cdd9e5",iconBg:"#ae7c14",iconBorder:"#c69026"},danger:{text:"#adbac7",border:"#636e7b",icon:"#cdd9e5",iconBg:"#c93c37",iconBorder:"#e5534b"},loading:{text:"#adbac7",border:"#636e7b",icon:"#cdd9e5",iconBg:"#636e7b",iconBorder:"#768390"}},timeline:{text:"#909dab",badgeBg:"#22272e",badgeSuccessBorder:"#46954a",targetBadgeBorder:"#316dca",targetBadgeShadow:"#1b4b91"},selectMenu:{borderSecondary:"#444c56",backdropBg:"rgba(28,33,40,0.5)",backdropBorder:"#545d68",tapHighlight:"rgba(68,76,86,0.5)",tapFocusBg:"#143d79"},box:{blueBorder:"#1b4b91",rowYellowBg:"rgba(235,196,64,0.1)",rowBlueBg:"rgba(108,182,255,0.1)",headerBlueBg:"#22272e",headerBlueBorder:"#444c56",borderInfo:"rgba(65,132,228,0.4)",bgInfo:"rgba(65,132,228,0.1)",borderWarning:"rgba(174,124,20,0.4)",bgWarning:"rgba(174,124,20,0.1)"},branchName:{text:"#adbac7",icon:"#909dab",bg:"rgba(83,155,245,0.1)",link:{text:"#539bf5",icon:"#539bf5",bg:"rgba(83,155,245,0.1)"}},markdown:{codeBg:"rgba(205,217,229,0.15)",frameBorder:"hsla(213,12%,35%,1)",blockquoteBorder:"hsla(213,12%,35%,1)",tableBorder:"hsla(213,12%,35%,1)",tableTrBorder:"hsla(213,12%,26%,1)"},menu:{headingText:"#768390",borderActive:"#f78166",bgActive:"#2d333b"},sidenav:{selectedBg:"#373e47",borderActive:"#f78166"},header:{text:"rgba(205,217,229,0.7)",bg:"#2d333b",logo:"#cdd9e5"},filterItem:{barBg:"hsla(214,13%,18%,1)"},hiddenTextExpander:{bg:"#373e47",bgHover:"#444c56"},dragAndDrop:{border:"hsla(213,12%,25%,1)"},uploadEnabled:{border:"hsla(213,12%,35%,1)",borderFocused:"hsla(216,61%,57%,1)"},previewableCommentForm:{border:"hsla(213,12%,25%,1)"},underlinenav:{border:"rgba(68,76,86,0)",borderHover:"#444c56",borderActive:"#f78166",text:"#768390",textHover:"#adbac7",textActive:"#adbac7",icon:"#636e7b",iconHover:"#adbac7",iconActive:"#adbac7",counterText:"#768390",counterBg:"rgba(118,131,144,0.2)"},verifiedBadge:{text:"#57ab5a",bg:"rgba(87,171,90,0.1)",border:"rgba(87,171,90,0.4)"},socialCount:{bg:"#373e47"},tooltip:{text:"#cdd9e5",bg:"#636e7b"},headerSearch:{bg:"#22272e",border:"#373e47"},searchKeyword:{hl:"rgba(174,124,20,0.4)"},diffstat:{neutralBg:"#444c56",neutralBorder:"rgba(205,217,229,0.1)",deletionBg:"#c93c37",deletionBorder:"#e5534b",additionBg:"#347d39",additionBorder:"#46954a"},mktg:{success:"rgba(61,137,66,1)",info:"rgba(56,119,213,1)",bgShadeGradient:{top:"rgba(28,33,40,0.065)",bottom:"rgba(28,33,40,0)"},btn:{bg:{top:"hsla(228,82%,66%,1)",bottom:"#4969ed"},bgOverlay:{top:"hsla(228,74%,59%,1)",bottom:"#3355e0"},text:"#cdd9e5",primary:{bg:{top:"hsla(137,56%,46%,1)",bottom:"#2ea44f"},bgOverlay:{top:"hsla(134,60%,38%,1)",bottom:"#22863a"},text:"#cdd9e5"},enterprise:{bg:{top:"hsla(249,100%,72%,1)",bottom:"#6f57ff"},bgOverlay:{top:"hsla(248,65%,63%,1)",bottom:"#614eda"},text:"#cdd9e5"},outline:{text:"#cdd9e5",border:"rgba(205,217,229,0.3)",hover:{text:"#cdd9e5",border:"rgba(205,217,229,0.5)"},focus:{border:"#cdd9e5",borderInset:"rgba(205,217,229,0.5)"}},dark:{text:"#cdd9e5",border:"rgba(205,217,229,0.3)",hover:{text:"#cdd9e5",border:"rgba(205,217,229,0.5)"},focus:{border:"#cdd9e5",borderInset:"rgba(205,217,229,0.5)"}}}},filesExplorerIcon:"#636e7b",hlAuthorBg:"#0f2d5c",hlAuthorBorder:"#1b4b91",logoSubdued:"#444c56",discussionBorder:"hsla(125,2%,29%,1)",discussionBgSuccess:"rgba(70,149,74,0.1)",actionsWorkflowTableStickyBg:"rgba(34,39,46,0.95)",repoLanguageColorBorder:"rgba(205,217,229,0.2)",codeSelectionBg:"rgba(108,182,255,0.3)",highlight:{text:"#ffd467",bg:"rgba(204,143,44,0.38)"},blob:{lineHighlightBg:"rgba(198,144,38,0.15)",lineHighlightBorder:"#daaa3f"},diff:{addition:{text:"#6bc46d",bg:"rgba(70,149,74,0.2)",border:"#2b6a30"},deletion:{text:"#e5534b",bg:"rgba(201,60,55,0.2)",border:"#ad2e2c"},change:{text:"#daaa3f",bg:"#452700",border:"#966600"}},diffBlob:{numText:"rgba(205,217,229,0.3)",numHoverText:"rgba(205,217,229,0.6)",addition:{numText:"#57ab5a",numHoverText:"#8ddb8c",numBg:"rgba(70,149,74,0.1)",lineBg:"rgba(70,149,74,0.2)",wordBg:"rgba(70,149,74,0.55)"},deletion:{numText:"#e5534b",numHoverText:"#ff938a",numBg:"rgba(201,60,55,0.1)",lineBg:"rgba(201,60,55,0.2)",wordBg:"rgba(201,60,55,0.5)"},hunk:{text:"#768390",numBg:"rgba(83,155,245,0.15)",lineBg:"rgba(83,155,245,0.1)"},emptyBlockBg:"#2d333b",selectedLineHighlightBg:"rgba(174,124,20,0.1)",selectedLineHighlightBorder:"#ae7c14",expander:{icon:"#768390",hoverIcon:"#cdd9e5",hoverBg:"#316dca"},commentButton:{icon:"#cdd9e5",bg:"#316dca",gradientBg:"hsla(216,61%,54%,1)"}},globalNav:{logo:"#cdd9e5",bg:"#2d333b",text:"#adbac7",icon:"#adbac7",inputBg:"#22272e",inputBorder:"#373e47",inputIcon:"#373e47",inputPlaceholder:"#545d68"},footerInvertocat:{octicon:"#444c56",octiconHover:"#636e7b"},prState:{draft:{text:"#768390",bg:"rgba(118,131,144,0.1)",border:"rgba(118,131,144,0.4)"},open:{text:"#57ab5a",bg:"rgba(87,171,90,0.1)",border:"rgba(87,171,90,0.4)"},merged:{text:"#986ee2",bg:"rgba(176,131,240,0.1)",border:"rgba(176,131,240,0.4)"},closed:{text:"#e5534b",bg:"rgba(201,60,55,0.1)",border:"rgba(201,60,55,0.4)"}},topicTag:{text:"#539bf5",bg:"rgba(65,132,228,0.1)",hoverBg:"rgba(65,132,228,0.2)",activeBg:"rgba(65,132,228,0.15)"},mergeBox:{successIconBg:"rgba(70,149,74,0.1)",successIconText:"#57ab5a",successIconBorder:"rgba(70,149,74,0.4)",successIndicatorBg:"#347d39",successIndicatorBorder:"#46954a",mergedIconBg:"rgba(176,131,240,0.1)",mergedIconText:"#986ee2",mergedIconBorder:"rgba(176,131,240,0.4)",mergedBoxBorder:"rgba(176,131,240,0.4)",neutralIconBg:"rgba(173,186,199,0.1)",neutralIconText:"#768390",neutralIconBorder:"rgba(173,186,199,0.4)",neutralIndicatorBg:"#545d68",neutralIndicatorBorder:"#636e7b",warningIconBg:"rgba(174,124,20,0.1)",warningIconText:"#daaa3f",warningIconBorder:"rgba(174,124,20,0.4)",warningBoxBorder:"rgba(174,124,20,0.4)",warningMergeHighlight:"rgba(174,124,20,0.1)",errorIconBg:"rgba(229,83,75,0.1)",errorIconText:"#e5534b",errorIconBorder:"rgba(229,83,75,0.4)",errorIndicatorBg:"#c93c37",errorIndicatorBorder:"#e5534b"},project:{cardBg:"#2d333b",headerBg:"#22272e",sidebarBg:"#2d333b",gradientIn:"#2d333b",gradientOut:"rgba(45,51,59,0)"},checks:{bg:"hsla(215,15%,14%,1)",textPrimary:"#adbac7",textSecondary:"#768390",textLink:"#539bf5",btnIcon:"#636e7b",btnHoverIcon:"#adbac7",btnHoverBg:"#444c56",inputText:"#768390",inputPlaceholderText:"#545d68",inputFocusText:"#adbac7",inputBg:"#22272e",donutError:"#e5534b",donutPending:"#c69026",donutSuccess:"#46954a",donutNeutral:"#768390",dropdownText:"#adbac7",dropdownBg:"rgba(50,57,65,1)",dropdownBorder:"#444c56",dropdownHoverText:"#cdd9e5",dropdownHoverBg:"#316dca",dropdownBtnHoverText:"#cdd9e5",dropdownBtnHoverBg:"#2d333b",scrollbarThumbBg:"#444c56",headerLabelText:"#768390",headerLabelOpenText:"#adbac7",headerBorder:"#373e47",headerIcon:"#636e7b",lineText:"#768390",lineNumText:"#768390",lineTimestampText:"#768390",lineHoverBg:"#2d333b",lineSelectedBg:"rgba(65,132,228,0.1)",lineSelectedNumText:"#539bf5",lineDtFmText:"#22272e",lineDtFmBg:"#c69026",gateBg:"rgba(128,84,0,0.15)",gateText:"#768390",gateWaitingText:"#daaa3f",stepHeaderOpenBg:"#2d333b",stepErrorText:"#e5534b",stepWarningText:"#daaa3f",loglineText:"#636e7b",loglineNumText:"#768390",loglineDebugText:"#b083f0",loglineErrorText:"#768390",loglineErrorNumText:"#768390",loglineErrorBg:"rgba(229,83,75,0.1)",loglineWarningText:"#768390",loglineWarningNumText:"#daaa3f",loglineWarningBg:"rgba(174,124,20,0.1)",loglineCommandText:"#539bf5",loglineSectionText:"#6bc46d",ansi:{black:"#22272e",blackBright:"#2d333b",white:"#909dab",whiteBright:"#909dab",gray:"#636e7b",red:"#f47067",redBright:"#ff938a",green:"#57ab5a",greenBright:"#6bc46d",yellow:"#c69026",yellowBright:"#daaa3f",blue:"#539bf5",blueBright:"#6cb6ff",magenta:"#b083f0",magentaBright:"#dcbdfb",cyan:"#76e3ea",cyanBright:"#b3f0ff"}},introShelf:{gradientLeft:"rgba(65,132,228,0.1)",gradientRight:"rgba(70,149,74,0.1)",gradientIn:"#22272e",gradientOut:"rgba(34,39,46,0)"},marketingIcon:{primary:"#6cb6ff",secondary:"#316dca"},prettylights:{syntax:{comment:"#768390",constant:"#6cb6ff",entity:"#dcbdfb",storageModifierImport:"#adbac7",entityTag:"#8ddb8c",keyword:"#f47067",string:"#96d0ff",variable:"#f69d50",brackethighlighterUnmatched:"#e5534b",invalidIllegalText:"#cdd9e5",invalidIllegalBg:"#922323",carriageReturnText:"#cdd9e5",carriageReturnBg:"#ad2e2c",stringRegexp:"#8ddb8c",markupList:"#eac55f",markupHeading:"#316dca",markupItalic:"#adbac7",markupBold:"#adbac7",markupDeletedText:"#ffd8d3",markupDeletedBg:"#78191b",markupInsertedText:"#b4f1b4",markupInsertedBg:"#1b4721",markupChangedText:"#ffddb0",markupChangedBg:"#682d0f",markupIgnoredText:"#adbac7",markupIgnoredBg:"#255ab2",metaDiffRange:"#dcbdfb",brackethighlighterAngle:"#768390",sublimelinterGutterMark:"#545d68",constantOtherReferenceLink:"#96d0ff"}},codemirror:{text:"#adbac7",bg:"#22272e",guttersBg:"#22272e",guttermarkerText:"#22272e",guttermarkerSubtleText:"#636e7b",linenumberText:"#768390",cursor:"#cdd9e5",selectionBg:"rgba(108,182,255,0.3)",activelineBg:"#2d333b",matchingbracketText:"#adbac7",linesBg:"#22272e",syntax:{comment:"#768390",constant:"#6cb6ff",entity:"#dcbdfb",keyword:"#f47067",storage:"#f47067",string:"#96d0ff",support:"#6cb6ff",variable:"#f69d50"}},ansi:{black:"#545d68",blackBright:"#636e7b",white:"#909dab",whiteBright:"#cdd9e5",gray:"#636e7b",red:"#f47067",redBright:"#ff938a",green:"#57ab5a",greenBright:"#6bc46d",yellow:"#c69026",yellowBright:"#daaa3f",blue:"#539bf5",blueBright:"#6cb6ff",magenta:"#b083f0",magentaBright:"#dcbdfb",cyan:"#39c5cf",cyanBright:"#56d4dd"}},shadows:{shadow:{small:"0 0 transparent",medium:"0 3px 6px #1c2128",large:"0 8px 24px #1c2128",extraLarge:"0 12px 48px #1c2128",highlight:"0 0 transparent",inset:"0 0 transparent"},state:{focus:{shadow:"0 0 0 3px #143d79"}},autocomplete:{shadow:"0 16px 32px rgba(28,33,40,0.85)"},btn:{shadow:"0 0 transparent",insetShadow:"0 0 transparent",focusShadow:"0 0 0 3px rgba(118,131,144,0.3)",shadowActive:"inset 0 0.15em 0.3em rgba(28,33,40,0.15)",shadowInputFocus:"0 0 0 0.2em rgba(49,109,202,0.3)",primary:{shadow:"0 0 transparent",insetShadow:"0 0 transparent",selectedShadow:"0 0 transparent",focusShadow:"0 0 0 3px rgba(46,164,79,0.4)"},outline:{hoverShadow:"0 1px 0 rgba(28,33,40,0.1)",hoverInsetShadow:"inset 0 1px 0 rgba(205,217,229,0.03)",selectedShadow:"0 0 transparent",focusShadow:"0 0 0 3px rgba(37,90,178,0.4)"},danger:{hoverShadow:"0 0 transparent",hoverInsetShadow:"0 0 transparent",selectedShadow:"0 0 transparent",focusShadow:"0 0 0 3px rgba(173,46,44,0.4)"}},dropdown:{shadow:"0 16px 32px rgba(28,33,40,0.85)"},overlay:{shadow:"0 0 0 1px #444c56, 0 16px 32px rgba(28,33,40,0.85)"},input:{shadow:"0 0 transparent"},avatar:{childShadow:"-2px -2px 0 #22272e"},toast:{shadow:"0 8px 24px #1c2128"},selectMenu:{shadow:"0 0 18px rgba(28,33,40,0.4)"},checks:{inputShadow:"0 0 0 1px #373e47"}}}}},c=Object.getOwnPropertySymbols,s=Object.prototype.hasOwnProperty,f=Object.prototype.propertyIsEnumerable;function b(e){if(null==e)throw new TypeError("Object.assign cannot be called with null or undefined");return Object(e)}var g=function(){try{if(!Object.assign)return!1;var e=new String("abc");if(e[5]="de","5"===Object.getOwnPropertyNames(e)[0])return!1;for(var t={},r=0;r<10;r++)t["_"+String.fromCharCode(r)]=r;if("0123456789"!==Object.getOwnPropertyNames(t).map((function(e){return t[e]})).join(""))return!1;var o={};return"abcdefghijklmnopqrst".split("").forEach((function(e){o[e]=e})),"abcdefghijklmnopqrst"===Object.keys(Object.assign({},o)).join("")}catch(e){return!1}}()?Object.assign:function(e,t){for(var r,o,a=b(e),n=1;n<arguments.length;n++){for(var i in r=Object(arguments[n]))s.call(r,i)&&(a[i]=r[i]);if(c){o=c(r);for(var d=0;d<o.length;d++)f.call(r,o[d])&&(a[o[d]]=r[o[d]])}}return a},u=function(e,t){var r=g({},e,t);for(var o in e){var a;e[o]&&"object"==typeof t[o]&&g(r,((a={})[o]=g(e[o],t[o]),a))}return r},p={breakpoints:[40,52,64].map((function(e){return e+"em"}))},m=function(e){return"@media screen and (min-width: "+e+")"},h=function(e,t){return x(t,e,e)},x=function(e,t,r,o,a){for(t=t&&t.split?t.split("."):[t],o=0;o<t.length;o++)e=e?e[t[o]]:a;return e===a?r:e},v=function e(t){var r={},o=function(e){var o,a,n={},i=!1,d=e.theme&&e.theme.disableStyledSystemCache;for(var l in e)if(t[l]){var c=t[l],s=e[l],f=x(e.theme,c.scale,c.defaults);if("object"!=typeof s)g(n,c(s,f,e));else{if(r.breakpoints=!d&&r.breakpoints||x(e.theme,"breakpoints",p.breakpoints),Array.isArray(s)){r.media=!d&&r.media||[null].concat(r.breakpoints.map(m)),n=u(n,y(r.media,c,f,s,e));continue}null!==s&&(n=u(n,w(r.breakpoints,c,f,s,e)),i=!0)}}return i&&(o=n,a={},Object.keys(o).sort((function(e,t){return e.localeCompare(t,void 0,{numeric:!0,sensitivity:"base"})})).forEach((function(e){a[e]=o[e]})),n=a),n};o.config=t,o.propNames=Object.keys(t),o.cache=r;var a=Object.keys(t).filter((function(e){return"config"!==e}));return a.length>1&&a.forEach((function(r){var a;o[r]=e(((a={})[r]=t[r],a))})),o},y=function(e,t,r,o,a){var n={};return o.slice(0,e.length).forEach((function(o,i){var d,l=e[i],c=t(o,r,a);l?g(n,((d={})[l]=g({},n[l],c),d)):g(n,c)})),n},w=function(e,t,r,o,a){var n={};for(var i in o){var d=e[i],l=t(o[i],r,a);if(d){var c,s=m(d);g(n,((c={})[s]=g({},n[s],l),c))}else g(n,l)}return n},B=function(e){var t=e.properties,r=e.property,o=e.scale,a=e.transform,n=void 0===a?h:a,i=e.defaultScale;t=t||[r];var d=function(e,r,o){var a={},i=n(e,r,o);if(null!==i)return t.forEach((function(e){a[e]=i})),a};return d.scale=o,d.defaults=i,d},$=function(e){void 0===e&&(e={});var t={};return Object.keys(e).forEach((function(r){var o=e[r];t[r]=!0!==o?"function"!=typeof o?B(o):o:B({property:r,scale:r})})),v(t)},k=function(){for(var e={},t=arguments.length,r=new Array(t),o=0;o<t;o++)r[o]=arguments[o];r.forEach((function(t){t&&t.config&&g(e,t.config)}));var a=v(e);return a},S=$({width:{property:"width",scale:"sizes",transform:function(e,t){return x(t,e,!function(e){return"number"==typeof e&&!isNaN(e)}(e)||e>1?e:100*e+"%")}},height:{property:"height",scale:"sizes"},minWidth:{property:"minWidth",scale:"sizes"},minHeight:{property:"minHeight",scale:"sizes"},maxWidth:{property:"maxWidth",scale:"sizes"},maxHeight:{property:"maxHeight",scale:"sizes"},size:{properties:["width","height"],scale:"sizes"},overflow:!0,overflowX:!0,overflowY:!0,display:!0,verticalAlign:!0}),T={color:{property:"color",scale:"colors"},backgroundColor:{property:"backgroundColor",scale:"colors"},opacity:!0};T.bg=T.backgroundColor;var E=$(T),C=$({fontFamily:{property:"fontFamily",scale:"fonts"},fontSize:{property:"fontSize",scale:"fontSizes",defaultScale:[12,14,16,20,24,32,48,64,72]},fontWeight:{property:"fontWeight",scale:"fontWeights"},lineHeight:{property:"lineHeight",scale:"lineHeights"},letterSpacing:{property:"letterSpacing",scale:"letterSpacings"},textAlign:!0,fontStyle:!0}),I=$({alignItems:!0,alignContent:!0,justifyItems:!0,justifyContent:!0,flexWrap:!0,flexDirection:!0,flex:!0,flexGrow:!0,flexShrink:!0,flexBasis:!0,justifySelf:!0,alignSelf:!0,order:!0}),N={space:[0,4,8,16,32,64,128,256,512]},A=$({gridGap:{property:"gridGap",scale:"space",defaultScale:N.space},gridColumnGap:{property:"gridColumnGap",scale:"space",defaultScale:N.space},gridRowGap:{property:"gridRowGap",scale:"space",defaultScale:N.space},gridColumn:!0,gridRow:!0,gridAutoFlow:!0,gridAutoColumns:!0,gridAutoRows:!0,gridTemplateColumns:!0,gridTemplateRows:!0,gridTemplateAreas:!0,gridArea:!0}),z={border:{property:"border",scale:"borders"},borderWidth:{property:"borderWidth",scale:"borderWidths"},borderStyle:{property:"borderStyle",scale:"borderStyles"},borderColor:{property:"borderColor",scale:"colors"},borderRadius:{property:"borderRadius",scale:"radii"},borderTop:{property:"borderTop",scale:"borders"},borderTopLeftRadius:{property:"borderTopLeftRadius",scale:"radii"},borderTopRightRadius:{property:"borderTopRightRadius",scale:"radii"},borderRight:{property:"borderRight",scale:"borders"},borderBottom:{property:"borderBottom",scale:"borders"},borderBottomLeftRadius:{property:"borderBottomLeftRadius",scale:"radii"},borderBottomRightRadius:{property:"borderBottomRightRadius",scale:"radii"},borderLeft:{property:"borderLeft",scale:"borders"},borderX:{properties:["borderLeft","borderRight"],scale:"borders"},borderY:{properties:["borderTop","borderBottom"],scale:"borders"},borderTopWidth:{property:"borderTopWidth",scale:"borderWidths"},borderTopColor:{property:"borderTopColor",scale:"colors"},borderTopStyle:{property:"borderTopStyle",scale:"borderStyles"}};z.borderTopLeftRadius={property:"borderTopLeftRadius",scale:"radii"},z.borderTopRightRadius={property:"borderTopRightRadius",scale:"radii"},z.borderBottomWidth={property:"borderBottomWidth",scale:"borderWidths"},z.borderBottomColor={property:"borderBottomColor",scale:"colors"},z.borderBottomStyle={property:"borderBottomStyle",scale:"borderStyles"},z.borderBottomLeftRadius={property:"borderBottomLeftRadius",scale:"radii"},z.borderBottomRightRadius={property:"borderBottomRightRadius",scale:"radii"},z.borderLeftWidth={property:"borderLeftWidth",scale:"borderWidths"},z.borderLeftColor={property:"borderLeftColor",scale:"colors"},z.borderLeftStyle={property:"borderLeftStyle",scale:"borderStyles"},z.borderRightWidth={property:"borderRightWidth",scale:"borderWidths"},z.borderRightColor={property:"borderRightColor",scale:"colors"},z.borderRightStyle={property:"borderRightStyle",scale:"borderStyles"};var R=$(z),L={background:!0,backgroundImage:!0,backgroundSize:!0,backgroundPosition:!0,backgroundRepeat:!0};L.bgImage=L.backgroundImage,L.bgSize=L.backgroundSize,L.bgPosition=L.backgroundPosition,L.bgRepeat=L.backgroundRepeat;var M=$(L),O={space:[0,4,8,16,32,64,128,256,512]},H=$({position:!0,zIndex:{property:"zIndex",scale:"zIndices"},top:{property:"top",scale:"space",defaultScale:O.space},right:{property:"right",scale:"space",defaultScale:O.space},bottom:{property:"bottom",scale:"space",defaultScale:O.space},left:{property:"left",scale:"space",defaultScale:O.space}}),P={space:[0,4,8,16,32,64,128,256,512]},D=function(e){return"number"==typeof e&&!isNaN(e)},W=function(e,t){if(!D(e))return x(t,e,e);var r=e<0,o=Math.abs(e),a=x(t,o,o);return D(a)?a*(r?-1:1):r?"-"+a:a},j={};j.margin={margin:{property:"margin",scale:"space",transform:W,defaultScale:P.space},marginTop:{property:"marginTop",scale:"space",transform:W,defaultScale:P.space},marginRight:{property:"marginRight",scale:"space",transform:W,defaultScale:P.space},marginBottom:{property:"marginBottom",scale:"space",transform:W,defaultScale:P.space},marginLeft:{property:"marginLeft",scale:"space",transform:W,defaultScale:P.space},marginX:{properties:["marginLeft","marginRight"],scale:"space",transform:W,defaultScale:P.space},marginY:{properties:["marginTop","marginBottom"],scale:"space",transform:W,defaultScale:P.space}},j.margin.m=j.margin.margin,j.margin.mt=j.margin.marginTop,j.margin.mr=j.margin.marginRight,j.margin.mb=j.margin.marginBottom,j.margin.ml=j.margin.marginLeft,j.margin.mx=j.margin.marginX,j.margin.my=j.margin.marginY,j.padding={padding:{property:"padding",scale:"space",defaultScale:P.space},paddingTop:{property:"paddingTop",scale:"space",defaultScale:P.space},paddingRight:{property:"paddingRight",scale:"space",defaultScale:P.space},paddingBottom:{property:"paddingBottom",scale:"space",defaultScale:P.space},paddingLeft:{property:"paddingLeft",scale:"space",defaultScale:P.space},paddingX:{properties:["paddingLeft","paddingRight"],scale:"space",defaultScale:P.space},paddingY:{properties:["paddingTop","paddingBottom"],scale:"space",defaultScale:P.space}},j.padding.p=j.padding.padding,j.padding.pt=j.padding.paddingTop,j.padding.pr=j.padding.paddingRight,j.padding.pb=j.padding.paddingBottom,j.padding.pl=j.padding.paddingLeft,j.padding.px=j.padding.paddingX,j.padding.py=j.padding.paddingY;var F=$(j.margin),U=$(j.padding),_=k(F,U),G=$({boxShadow:{property:"boxShadow",scale:"shadows"},textShadow:{property:"textShadow",scale:"shadows"}});function K(){return(K=Object.assign||function(e){for(var t=1;t<arguments.length;t++){var r=arguments[t];for(var o in r)Object.prototype.hasOwnProperty.call(r,o)&&(e[o]=r[o])}return e}).apply(this,arguments)}var X=function(e,t,r,o,a){for(t=t&&t.split?t.split("."):[t],o=0;o<t.length;o++)e=e?e[t[o]]:a;return e===a?r:e},V=[40,52,64].map((function(e){return e+"em"})),q={space:[0,4,8,16,32,64,128,256,512],fontSizes:[12,14,16,20,24,32,48,64,72]},Y={bg:"backgroundColor",m:"margin",mt:"marginTop",mr:"marginRight",mb:"marginBottom",ml:"marginLeft",mx:"marginX",my:"marginY",p:"padding",pt:"paddingTop",pr:"paddingRight",pb:"paddingBottom",pl:"paddingLeft",px:"paddingX",py:"paddingY"},J={marginX:["marginLeft","marginRight"],marginY:["marginTop","marginBottom"],paddingX:["paddingLeft","paddingRight"],paddingY:["paddingTop","paddingBottom"],size:["width","height"]},Z={color:"colors",backgroundColor:"colors",borderColor:"colors",margin:"space",marginTop:"space",marginRight:"space",marginBottom:"space",marginLeft:"space",marginX:"space",marginY:"space",padding:"space",paddingTop:"space",paddingRight:"space",paddingBottom:"space",paddingLeft:"space",paddingX:"space",paddingY:"space",top:"space",right:"space",bottom:"space",left:"space",gridGap:"space",gridColumnGap:"space",gridRowGap:"space",gap:"space",columnGap:"space",rowGap:"space",fontFamily:"fonts",fontSize:"fontSizes",fontWeight:"fontWeights",lineHeight:"lineHeights",letterSpacing:"letterSpacings",border:"borders",borderTop:"borders",borderRight:"borders",borderBottom:"borders",borderLeft:"borders",borderWidth:"borderWidths",borderStyle:"borderStyles",borderRadius:"radii",borderTopRightRadius:"radii",borderTopLeftRadius:"radii",borderBottomRightRadius:"radii",borderBottomLeftRadius:"radii",borderTopWidth:"borderWidths",borderTopColor:"colors",borderTopStyle:"borderStyles",borderBottomWidth:"borderWidths",borderBottomColor:"colors",borderBottomStyle:"borderStyles",borderLeftWidth:"borderWidths",borderLeftColor:"colors",borderLeftStyle:"borderStyles",borderRightWidth:"borderWidths",borderRightColor:"colors",borderRightStyle:"borderStyles",outlineColor:"colors",boxShadow:"shadows",textShadow:"shadows",zIndex:"zIndices",width:"sizes",minWidth:"sizes",maxWidth:"sizes",height:"sizes",minHeight:"sizes",maxHeight:"sizes",flexBasis:"sizes",size:"sizes",fill:"colors",stroke:"colors"},Q=function(e,t){if("number"!=typeof t||t>=0)return X(e,t,t);var r=Math.abs(t),o=X(e,r,r);return"string"==typeof o?"-"+o:-1*o},ee=["margin","marginTop","marginRight","marginBottom","marginLeft","marginX","marginY","top","bottom","left","right"].reduce((function(e,t){var r;return K({},e,((r={})[t]=Q,r))}),{}),te=function e(t){return function(r){void 0===r&&(r={});var o=K({},q,{},r.theme||r),a={},n=function(e){return function(t){var r={},o=X(t,"breakpoints",V),a=[null].concat(o.map((function(e){return"@media screen and (min-width: "+e+")"})));for(var n in e){var i="function"==typeof e[n]?e[n](t):e[n];if(null!=i)if(Array.isArray(i))for(var d=0;d<i.slice(0,a.length).length;d++){var l=a[d];l?(r[l]=r[l]||{},null!=i[d]&&(r[l][n]=i[d])):r[n]=i[d]}else r[n]=i}return r}}("function"==typeof t?t(o):t)(o);for(var i in n){var d=n[i],l="function"==typeof d?d(o):d;if("variant"!==i)if(l&&"object"==typeof l)a[i]=e(l)(o);else{var c=X(Y,i,i),s=X(Z,c),f=X(o,s,X(o,c,{})),b=X(ee,c,X)(f,l,l);if(J[c])for(var g=J[c],u=0;u<g.length;u++)a[g[u]]=b;else a[c]=b}else a=K({},a,{},e(X(o,l))(o))}return a}},re=function(e){var t,r,o=e.scale,a=e.prop,n=void 0===a?"variant":a,i=e.variants,d=void 0===i?{}:i,l=e.key;(r=Object.keys(d).length?function(e,t,r){return te(x(t,e,null))(r.theme)}:function(e,t){return x(t,e,null)}).scale=o||l,r.defaults=d;var c=((t={})[n]=r,t);return v(c)},oe=re({key:"buttons"}),ae=re({key:"textStyles",prop:"textStyle"}),ne=re({key:"colorStyles",prop:"colors"}),ie=S.width,de=S.height,le=S.minWidth,ce=S.minHeight,se=S.maxWidth,fe=S.maxHeight,be=S.size,ge=S.verticalAlign,ue=S.display,pe=S.overflow,me=S.overflowX,he=S.overflowY,xe=E.opacity,ve=C.fontSize,ye=C.fontFamily,we=C.fontWeight,Be=C.lineHeight,$e=C.textAlign,ke=C.fontStyle,Se=C.letterSpacing,Te=I.alignItems,Ee=I.alignContent,Ce=I.justifyItems,Ie=I.justifyContent,Ne=I.flexWrap,Ae=I.flexDirection,ze=I.flex,Re=I.flexGrow,Le=I.flexShrink,Me=I.flexBasis,Oe=I.justifySelf,He=I.alignSelf,Pe=I.order,De=A.gridGap,We=A.gridColumnGap,je=A.gridRowGap,Fe=A.gridColumn,Ue=A.gridRow,_e=A.gridAutoFlow,Ge=A.gridAutoColumns,Ke=A.gridAutoRows,Xe=A.gridTemplateColumns,Ve=A.gridTemplateRows,qe=A.gridTemplateAreas,Ye=A.gridArea,Je=R.borderWidth,Ze=R.borderStyle,Qe=R.borderColor,et=R.borderTop,tt=R.borderRight,rt=R.borderBottom,ot=R.borderLeft,at=R.borderRadius,nt=M.backgroundImage,it=M.backgroundSize,dt=M.backgroundPosition,lt=M.backgroundRepeat,ct=H.zIndex,st=H.top,ft=H.right,bt=H.bottom,gt=H.left,ut=function(e){var t=e.prop,r=e.cssProperty,o=e.alias,a=e.key,n=e.transformValue,i=e.scale,d=e.properties,l={};return l[t]=B({properties:d,property:r||t,scale:a,defaultScale:i,transform:n}),o&&(l[o]=l[t]),v(l)},pt=Object.freeze({__proto__:null,width:ie,height:de,minWidth:le,minHeight:ce,maxWidth:se,maxHeight:fe,size:be,verticalAlign:ge,display:ue,overflow:pe,overflowX:me,overflowY:he,opacity:xe,fontSize:ve,fontFamily:ye,fontWeight:we,lineHeight:Be,textAlign:$e,fontStyle:ke,letterSpacing:Se,alignItems:Te,alignContent:Ee,justifyItems:Ce,justifyContent:Ie,flexWrap:Ne,flexDirection:Ae,flex:ze,flexGrow:Re,flexShrink:Le,flexBasis:Me,justifySelf:Oe,alignSelf:He,order:Pe,gridGap:De,gridColumnGap:We,gridRowGap:je,gridColumn:Fe,gridRow:Ue,gridAutoFlow:_e,gridAutoColumns:Ge,gridAutoRows:Ke,gridTemplateColumns:Xe,gridTemplateRows:Ve,gridTemplateAreas:qe,gridArea:Ye,borderWidth:Je,borderStyle:Ze,borderColor:Qe,borderTop:et,borderRight:tt,borderBottom:rt,borderLeft:ot,borderRadius:at,backgroundImage:nt,backgroundSize:it,backgroundPosition:dt,backgroundRepeat:lt,zIndex:ct,top:st,right:ft,bottom:bt,left:gt,style:ut,get:x,createParser:v,createStyleFunction:B,compose:k,system:$,margin:F,padding:U,space:_,color:E,layout:S,typography:C,flexbox:I,border:R,background:M,position:H,grid:A,shadow:G,boxShadow:G,textShadow:G,variant:re,buttonStyle:oe,textStyle:ae,colorStyle:ne,borders:R});const{get:mt,compose:ht,system:xt}=pt,vt=e=>{return t=e,void 0===(r=mt(l,e))&&(r=null),function(e){return x(e.theme,t,r)};var t,r},yt=ht(_,E,ue),wt=ht(C,xt({whiteSpace:{property:"whiteSpace"}})),Bt=ht(R,G),$t=S,kt=H,St=I;function Tt(){return(Tt=Object.assign||function(e){for(var t=1;t<arguments.length;t++){var r=arguments[t];for(var o in r)Object.prototype.hasOwnProperty.call(r,o)&&(e[o]=r[o])}return e}).apply(this,arguments)}const Et=r.createGlobalStyle`
  * { box-sizing: border-box; }
  body { margin: 0; }
  table { border-collapse: collapse; }
  
  [role="button"]:focus:not(:focus-visible):not(.focus-visible),
  [role="tabpanel"][tabindex="0"]:focus:not(:focus-visible):not(.focus-visible),
  button:focus:not(:focus-visible):not(.focus-visible),
  summary:focus:not(:focus-visible):not(.focus-visible),
  a:focus:not(:focus-visible):not(.focus-visible) {
    outline: none;
    box-shadow: none;
  }

  [tabindex="0"]:focus:not(:focus-visible):not(.focus-visible),
  details-dialog:focus:not(:focus-visible):not(.focus-visible) {
    outline: none;
  }
`,Ct=i.default.div`
  ${wt};
  ${yt};
`;function It(e){const{children:t,...r}=e;return require("focus-visible"),n.default.createElement(Ct,Tt({},r,{"data-portal-root":!0}),n.default.createElement(Et,null),t)}It.displayName="BaseStyles",It.defaultProps={color:"text.primary",fontFamily:"normal",lineHeight:"default"};var Nt=function(e){return function(e){return!!e&&"object"==typeof e}(e)&&!function(e){var t=Object.prototype.toString.call(e);return"[object RegExp]"===t||"[object Date]"===t||function(e){return e.$$typeof===At}(e)}(e)};var At="function"==typeof Symbol&&Symbol.for?Symbol.for("react.element"):60103;function zt(e,t){return!1!==t.clone&&t.isMergeableObject(e)?Ht((r=e,Array.isArray(r)?[]:{}),e,t):e;var r}function Rt(e,t,r){return e.concat(t).map((function(e){return zt(e,r)}))}function Lt(e){return Object.keys(e).concat(function(e){return Object.getOwnPropertySymbols?Object.getOwnPropertySymbols(e).filter((function(t){return e.propertyIsEnumerable(t)})):[]}(e))}function Mt(e,t){try{return t in e}catch(e){return!1}}function Ot(e,t,r){var o={};return r.isMergeableObject(e)&&Lt(e).forEach((function(t){o[t]=zt(e[t],r)})),Lt(t).forEach((function(a){(function(e,t){return Mt(e,t)&&!(Object.hasOwnProperty.call(e,t)&&Object.propertyIsEnumerable.call(e,t))})(e,a)||(Mt(e,a)&&r.isMergeableObject(t[a])?o[a]=function(e,t){if(!t.customMerge)return Ht;var r=t.customMerge(e);return"function"==typeof r?r:Ht}(a,r)(e[a],t[a],r):o[a]=zt(t[a],r))})),o}function Ht(e,t,r){(r=r||{}).arrayMerge=r.arrayMerge||Rt,r.isMergeableObject=r.isMergeableObject||Nt,r.cloneUnlessOtherwiseSpecified=zt;var o=Array.isArray(t);return o===Array.isArray(e)?o?r.arrayMerge(e,t,r):Ot(e,t,r):zt(t,r)}Ht.all=function(e,t){if(!Array.isArray(e))throw new Error("first argument should be an array");return e.reduce((function(e,r){return Ht(e,r,t)}),{})};var Pt=Ht;const Dt="light",Wt="dark",jt=n.default.createContext({setColorMode:()=>null,setDayScheme:()=>null,setNightScheme:()=>null}),Ft=({children:e,...t})=>{var o,a,i,d,c,s,f,b;const{theme:g,colorMode:u,dayScheme:p,nightScheme:m}=Ut(),h=null!==(o=null!==(a=t.theme)&&void 0!==a?a:g)&&void 0!==o?o:l,[x,v]=n.default.useState(null!==(i=null!==(d=t.colorMode)&&void 0!==d?d:u)&&void 0!==i?i:"day"),[y,w]=n.default.useState(null!==(c=null!==(s=t.dayScheme)&&void 0!==s?s:p)&&void 0!==c?c:Dt),[B,$]=n.default.useState(null!==(f=null!==(b=t.nightScheme)&&void 0!==b?b:m)&&void 0!==f?f:Wt),k=function(e,t){switch(e){case"auto":return t;default:return e}}(x,function(){const[e,t]=n.default.useState(Gt);return n.default.useEffect((()=>{var e,r;const o=null===(e=window)||void 0===e||null===(r=e.matchMedia)||void 0===r?void 0:r.call(e,"(prefers-color-scheme: dark)");function a(e){const r=e.matches;t(r?"night":"day")}if(o){if(void 0!==o.addEventListener)return o.addEventListener("change",a),function(){o.removeEventListener("change",a)};if(void 0!==o.addListener)return o.addListener(a),function(){o.removeListener(a)}}}),[]),e}()),S=function(e,t,r){switch(e){case"day":return t;case"night":return r}}(k,y,B),T=n.default.useMemo((()=>function(e,t){if(!e.colorSchemes)return e;if(!e.colorSchemes[t]){console.error(`\`${t}\` scheme not defined in \`theme.colorSchemes\``);const r=Object.keys(e.colorSchemes)[0];return Pt(e,e.colorSchemes[r])}return Pt(e,e.colorSchemes[t])}(h,S)),[h,S]);return n.default.useEffect((()=>{var e,r;v(null!==(e=null!==(r=t.colorMode)&&void 0!==r?r:u)&&void 0!==e?e:"day")}),[t.colorMode,u]),n.default.useEffect((()=>{var e,r;w(null!==(e=null!==(r=t.dayScheme)&&void 0!==r?r:p)&&void 0!==e?e:Dt)}),[t.dayScheme,p]),n.default.useEffect((()=>{var e,r;$(null!==(e=null!==(r=t.nightScheme)&&void 0!==r?r:m)&&void 0!==e?e:Wt)}),[t.nightScheme,m]),n.default.createElement(jt.Provider,{value:{theme:T,colorScheme:S,colorMode:x,resolvedColorMode:k,dayScheme:y,nightScheme:B,setColorMode:v,setDayScheme:w,setNightScheme:$}},n.default.createElement(r.ThemeProvider,{theme:T},e))};function Ut(){return n.default.useContext(jt)}function _t(e,t){var r;const{colorScheme:o=""}=Ut();return null!==(r=e[o])&&void 0!==r?r:t}function Gt(){var e,t,r;return"undefined"!=typeof window&&null!==(e=(t=window).matchMedia)&&void 0!==e&&null!==(r=e.call(t,"(prefers-color-scheme: dark)"))&&void 0!==r&&r.matches?"night":"day"}Ft.displayName="ThemeProvider";const Kt=e=>te(e.sx),Xt=i.default.div(_,E,C,S,I,A,M,R,H,G,Kt),Vt=i.default(Xt)``;Vt.defaultProps={borderWidth:"1px",borderStyle:"solid",borderColor:"border.primary",borderRadius:2};const qt=i.default(Xt)``;qt.defaultProps={display:"flex"};const Yt=i.default(Xt)``;Yt.defaultProps={display:"grid"};const Jt=i.default(Xt)``,Zt=n.default.forwardRef(((e,t)=>n.default.createElement(Jt,Tt({},e,{position:"absolute",ref:t}))));Zt.displayName="Absolute";const Qt=n.default.forwardRef(((e,t)=>n.default.createElement(Jt,Tt({},e,{position:"fixed",ref:t}))));Qt.displayName="Fixed";const er=n.default.forwardRef(((e,t)=>n.default.createElement(Jt,Tt({},e,{position:"relative",ref:t}))));er.displayName="Relative";const tr=n.default.forwardRef(((e,t)=>n.default.createElement(Jt,Tt({},e,{position:"sticky",ref:t}))));function rr({ref:e,closeOnOutsideClick:r,defaultOpen:o,onClickOutside:a}){const[n,i]=t.useState(o),d=t.useRef(null),l=null!=e?e:d,c=t.useCallback((e=>{const{current:t}=l;e.target.closest("details")!==t&&(a&&a(e),e.defaultPrevented||i(!1))}),[l,i,a]);t.useEffect((()=>{if(n&&r)return document.addEventListener("click",c),()=>{document.removeEventListener("click",c)}}),[n,r,c]);const s=e=>{if(!e.defaultPrevented){const t=e.target;i(t.open)}};return{open:n,setOpen:i,getDetailsProps:()=>({onToggle:s,open:n,ref:l})}}tr.defaultProps={top:0,zIndex:1},tr.displayName="Sticky";const or=!0;function ar(e){if(!e.defaultPrevented)for(const t of Object.values(nr).reverse())if(t(e)===or||e.defaultPrevented)break}const nr={};let ir=0;const dr=({containerRef:e,ignoreClickRefs:r,onClickOutside:o})=>{const a=t.useMemo((()=>ir++),[]),n=t.useCallback((t=>{var a;return t instanceof MouseEvent&&t.button>0||null!==(a=e.current)&&void 0!==a&&a.contains(t.target)||r&&r.some((({current:e})=>null==e?void 0:e.contains(t.target)))?or:void o(t)}),[e,r,o]);t.useEffect((()=>(0===Object.keys(nr).length&&document.addEventListener("mousedown",ar,{capture:!0}),function(e,t){nr[e]=t}(a,n),()=>{!function(e){delete nr[e]}(a),0===Object.keys(nr).length&&document.removeEventListener("mousedown",ar,{capture:!0})})),[a,n])};function*lr(e,t={}){var r,o;const a=null!==(r=t.strict)&&void 0!==r&&r,n=null!==(o=t.onlyTabbable)&&void 0!==o&&o?sr:cr,i=document.createTreeWalker(e,NodeFilter.SHOW_ELEMENT,{acceptNode:e=>e instanceof HTMLElement&&n(e,a)?NodeFilter.FILTER_ACCEPT:NodeFilter.FILTER_SKIP});let d=null;if(!t.reverse&&n(e,a)&&(yield e),t.reverse){let e=i.lastChild();for(;e;)d=e,e=i.lastChild()}else d=i.firstChild();for(;d instanceof HTMLElement;)yield d,d=t.reverse?i.previousNode():i.nextNode();t.reverse&&n(e,a)&&(yield e)}function cr(e,t=!1){const r=["BUTTON","INPUT","SELECT","TEXTAREA","OPTGROUP","OPTION","FIELDSET"].includes(e.tagName)&&e.disabled,o=e.hidden,a=e instanceof HTMLInputElement&&"hidden"===e.type;if(r||o||a)return!1;if(t){const t=0===e.offsetWidth||0===e.offsetHeight,r=["hidden","collapse"].includes(getComputedStyle(e).visibility),o=0===e.getClientRects().length;if(t||r||o)return!1}return null!=e.getAttribute("tabindex")||!(e instanceof HTMLAnchorElement&&null==e.getAttribute("href"))&&-1!==e.tabIndex}function sr(e,t=!1){return cr(e,t)&&"-1"!==e.getAttribute("tabindex")}function fr({initialFocusRef:e,returnFocusRef:r,containerRef:o}){t.useEffect((()=>{const t=r.current;if(e&&e.current)e.current.focus();else if(o.current){const e=lr(o.current).next().value;null==e||e.focus()}return function(){null==t||t.focus()}}),[e,r,o])}const br=[];function gr(e){if("Escape"===e.key&&!e.defaultPrevented)for(let t=br.length-1;t>=0&&(br[t](e),!e.defaultPrevented);--t);}const ur=(e,r=[e])=>{const o=t.useCallback(e,r);t.useEffect((()=>(0===br.length&&document.addEventListener("keydown",gr),br.push(o),()=>{br.splice(br.findIndex((e=>e===o)),1),0===br.length&&document.removeEventListener("keydown",gr)})),[o])};function pr(e){const t=n.default.useRef(null);return null!=e?e:t}const mr=({overlayRef:e,returnFocusRef:t,initialFocusRef:r,onEscape:o,ignoreClickRefs:a,onClickOutside:n})=>{const i=pr(e);return fr({containerRef:i,returnFocusRef:t,initialFocusRef:r}),dr({containerRef:i,ignoreClickRefs:a,onClickOutside:n}),ur(o),{ref:i}};let hr,xr=!1;function vr(){}try{const e=Object.create({},{signal:{get(){xr=!0}}});window.addEventListener("test",vr,e),window.removeEventListener("test",vr,e)}catch(e){}function yr(){xr||(!function(){if("undefined"==typeof window)return;const e=EventTarget.prototype.addEventListener;EventTarget.prototype.addEventListener=function(t,r,o){return"object"==typeof o&&"signal"in o&&o.signal instanceof AbortSignal&&e.call(o.signal,"abort",(()=>{this.removeEventListener(t,r,o)})),e.call(this,t,r,o)}}(),xr=!0)}function wr(){return void 0===hr&&(hr=/^mac/i.test(window.navigator.platform)),hr}let Br,$r=1e4;function kr(){return"__primer_id_"+$r++}yr(),function(e){e[e.ArrowHorizontal=1]="ArrowHorizontal",e[e.ArrowVertical=2]="ArrowVertical",e[e.JK=4]="JK",e[e.HL=8]="HL",e[e.HomeAndEnd=16]="HomeAndEnd",e[e.PageUpDown=256]="PageUpDown",e[e.WS=32]="WS",e[e.AD=64]="AD",e[e.Tab=128]="Tab",e[e.ArrowAll=e.ArrowHorizontal|e.ArrowVertical]="ArrowAll",e[e.HJKL=e.HL|e.JK]="HJKL",e[e.WASD=e.WS|e.AD]="WASD",e[e.All=e.ArrowAll|e.HJKL|e.HomeAndEnd|e.PageUpDown|e.WASD|e.Tab]="All"}(Br||(Br={}));const Sr={ArrowLeft:Br.ArrowHorizontal,ArrowDown:Br.ArrowVertical,ArrowUp:Br.ArrowVertical,ArrowRight:Br.ArrowHorizontal,h:Br.HL,j:Br.JK,k:Br.JK,l:Br.HL,a:Br.AD,s:Br.WS,w:Br.WS,d:Br.AD,Tab:Br.Tab,Home:Br.HomeAndEnd,End:Br.HomeAndEnd,PageUp:Br.PageUpDown,PageDown:Br.PageUpDown},Tr={ArrowLeft:"previous",ArrowDown:"next",ArrowUp:"previous",ArrowRight:"next",h:"previous",j:"next",k:"previous",l:"next",a:"previous",s:"next",w:"previous",d:"next",Tab:"next",Home:"start",End:"end",PageUp:"start",PageDown:"end"};function Er(e){const t=Tr[e.key];if("Tab"===e.key&&e.shiftKey)return"previous";const r=wr();if(r&&e.metaKey||!r&&e.ctrlKey){if("ArrowLeft"===e.key||"ArrowUp"===e.key)return"start";if("ArrowRight"===e.key||"ArrowDown"===e.key)return"end"}return t}const Cr="data-is-active-descendant",Ir="activated-directly",Nr="activated-indirectly",Ar="data-has-active-descendant";function zr(e,t){var r,o,a,n;const i=[],d=new WeakMap,l=null!==(r=null==t?void 0:t.bindKeys)&&void 0!==r?r:(null!=t&&t.getNextFocusable?Br.ArrowAll:Br.ArrowVertical)|Br.HomeAndEnd,c=null!==(o=null==t?void 0:t.focusOutBehavior)&&void 0!==o?o:"stop",s=null!==(a=null==t?void 0:t.focusInStrategy)&&void 0!==a?a:"previous",f=null==t?void 0:t.activeDescendantControl,b=null==t?void 0:t.onActiveDescendantChanged;let g;function u(){return i[0]}function p(e,t=!1){const r=g;g=e,f?e&&document.activeElement===f?m(r,e,t):h():(r&&r!==e&&d.has(r)&&r.setAttribute("tabindex","-1"),null==e||e.setAttribute("tabindex","0"))}function m(t,r,o=!1){r.id||r.setAttribute("id",kr()),t&&t!==r&&t.removeAttribute(Cr),f&&(o||f.getAttribute("aria-activedescendant")!==r.id)&&(f.setAttribute("aria-activedescendant",r.id),e.setAttribute(Ar,r.id),r.setAttribute(Cr,o?Ir:Nr),null==b||b(r,t,o))}function h(t=g){"first"===s&&(g=void 0),null==f||f.removeAttribute("aria-activedescendant"),e.removeAttribute(Ar),null==t||t.removeAttribute(Cr),null==b||b(void 0,t,!1)}function x(...e){const r=e.filter((e=>{var r,o;return null===(r=null==t||null===(o=t.focusableElementFilter)||void 0===o?void 0:o.call(t,e))||void 0===r||r}));if(0===r.length)return;const o=i.findIndex((e=>(e.compareDocumentPosition(r[0])&Node.DOCUMENT_POSITION_PRECEDING)>0));i.splice(-1===o?i.length:o,0,...r);for(const e of r)d.has(e)||d.set(e,e.getAttribute("tabindex")),e.setAttribute("tabindex","-1");g||p(u())}function v(...e){for(const t of e){const e=i.indexOf(t);e>=0&&i.splice(e,1);const r=d.get(t);if(void 0!==r&&(null===r?t.removeAttribute("tabindex"):t.setAttribute("tabindex",r),d.delete(t)),t===g){p(u())}}}x(...lr(e)),p(u());new MutationObserver((e=>{for(const t of e)for(const e of t.removedNodes)e instanceof HTMLElement&&v(...lr(e));for(const t of e)for(const e of t.addedNodes)e instanceof HTMLElement&&x(...lr(e))})).observe(e,{subtree:!0,childList:!0});const y=new AbortController,w=null!==(n=null==t?void 0:t.abortSignal)&&void 0!==n?n:y.signal;let B;w.addEventListener("abort",(()=>{v(...i)})),e.addEventListener("mousedown",(e=>{e.target instanceof HTMLElement&&e.target!==document.activeElement&&(B=i.indexOf(e.target))}),{signal:w}),f?(e.addEventListener("focusin",(e=>{e.target instanceof HTMLElement&&i.includes(e.target)&&(f.focus(),p(e.target))})),e.addEventListener("mousemove",(({target:e})=>{if(!(e instanceof Node))return;const t=i.find((t=>t.contains(e)));t&&p(t)}),{signal:w,capture:!0}),f.addEventListener("focusin",(()=>{g?m(void 0,g):p(u())})),f.addEventListener("focusout",(()=>{h()}))):e.addEventListener("focusin",(t=>{if(t.target instanceof HTMLElement)if(void 0!==B)B>=0&&i[B]!==g&&p(i[B]),B=void 0;else if("previous"===s)p(t.target);else if("closest"===s||"first"===s){if(t.relatedTarget instanceof Element&&!e.contains(t.relatedTarget)){const e="previous"===k?i.length-1:0,t=i[e];return void(null==t||t.focus())}p(t.target)}else if("function"==typeof s)if(t.relatedTarget instanceof Element&&!e.contains(t.relatedTarget)){const e=s(t.relatedTarget);if((e?i.indexOf(e):-1)>=0&&e instanceof HTMLElement)return void e.focus();console.warn("Element requested is not a known focusable element.")}else p(t.target);k=void 0}),{signal:w});const $=null!=f?f:e;let k;return"closest"===s&&document.addEventListener("keydown",(e=>{"Tab"===e.key&&(k=Er(e))}),{signal:w,capture:!0}),$.addEventListener("keydown",(r=>{if(r.key in Tr){const a=Sr[r.key];if(!r.defaultPrevented&&(a&l)>0&&!function(e,t){const r=e.key,o=[...r].length,a=t instanceof HTMLInputElement&&"text"===t.type||t instanceof HTMLTextAreaElement;if(a&&(1===o||"Home"===r||"End"===r))return!0;if(t instanceof HTMLSelectElement){if(1===o)return!0;if("ArrowDown"===r&&wr()&&!e.metaKey)return!0;if("ArrowDown"===r&&!wr()&&e.altKey)return!0}if(t instanceof HTMLTextAreaElement&&("PageUp"===r||"PageDown"===r))return!0;if(a){const e=t,o=0===e.selectionStart&&0===e.selectionEnd,a=e.selectionStart===e.value.length&&e.selectionEnd===e.value.length;if("ArrowLeft"===r&&!o)return!0;if("ArrowRight"===r&&!a)return!0;if(e instanceof HTMLTextAreaElement){if("ArrowUp"===r&&!o)return!0;if("ArrowDown"===r&&!a)return!0}}return!1}(r,document.activeElement)){const a=Er(r);let n;var o;if(null!=t&&t.getNextFocusable)n=t.getNextFocusable(a,null!==(o=document.activeElement)&&void 0!==o?o:void 0,r);if(!n){const t=function(){if(!g)return 0;const t=i.indexOf(g);return-1!==t?t:g===e?-1:0}();let o=t;"previous"===a?o-=1:"start"===a?o=0:"next"===a?o+=1:o=i.length-1,o<0&&(o="wrap"===c&&"Tab"!==r.key?i.length-1:0),o>=i.length&&(o="wrap"===c&&"Tab"!==r.key?0:i.length-1),t!==o&&(n=i[o])}f?p(n||g,!0):n&&(k=a,n.focus()),("Tab"!==r.key||n)&&r.preventDefault()}}}),{signal:w}),y}var Rr=r.css(["position:relative;display:inline-block;padding:6px 16px;font-family:inherit;font-weight:",";line-height:20px;white-space:nowrap;vertical-align:middle;cursor:pointer;user-select:none;border-radius:",";appearance:none;text-decoration:none;text-align:center;&:hover{text-decoration:none;}&:focus{outline:none;}&:disabled{cursor:default;}&:disabled svg{opacity:0.6;}"],vt("fontWeights.bold"),vt("radii.2"));const Lr=k(ve,yt,$t),Mr=re({variants:{small:{p:"4px 12px",fontSize:0},medium:{fontSize:1},large:{fontSize:2,p:"10px 20px"}}}),Or=i.default.button.attrs((({disabled:e,onClick:t})=>({onClick:e?void 0:t})))`
  ${Rr}
  ${Mr}
`;Or.defaultProps={variant:"medium"};const Hr=i.default(Or)`
  color: ${vt("colors.btn.text")};
  background-color: ${vt("colors.btn.bg")};
  border: 1px solid ${vt("colors.btn.border")};
  box-shadow: ${vt("shadows.btn.shadow")}, ${vt("shadows.btn.insetShadow")}};

  &:hover {
    background-color: ${vt("colors.btn.hoverBg")};
    border-color: ${vt("colors.btn.hoverBorder")};
  }

  // focus must come before :active so that the active box shadow overrides
  &:focus {
    border-color: ${vt("colors.btn.focusBorder")};
    box-shadow: ${vt("shadows.btn.focusShadow")};
  }

  &:active {
    background-color: ${vt("colors.btn.selectedBg")};
    box-shadow: ${vt("shadows.btn.shadowActive")};
  }

  &:disabled {
    color: ${vt("colors.text.disabled")};
    background-color: ${vt("colors.btn.bg")};
    border-color: ${vt("colors.btn.border")};
  }

  ${Lr};
  ${Kt};
`,Pr=i.default(Or)`
  color: ${vt("colors.btn.danger.text")};
  border: 1px solid ${vt("colors.btn.border")};
  background-color: ${vt("colors.btn.bg")};
  box-shadow: ${vt("shadows.btn.shadow")};

  &:hover {
    color: ${vt("colors.btn.danger.hoverText")};
    background-color: ${vt("colors.btn.danger.hoverBg")};
    border-color: ${vt("colors.btn.danger.hoverBorder")};
    box-shadow: ${vt("shadows.btn.danger.hoverShadow")};
  }
  // focus must come before :active so that the active box shadow overrides
  &:focus {
    border-color: ${vt("colors.btn.danger.focusBorder")};
    box-shadow: ${vt("shadows.btn.danger.focusShadow")};
  }

  &:active {
    color: ${vt("colors.btn.danger.selectedText")};
    background-color: ${vt("colors.btn.danger.selectedBg")};
    box-shadow: ${vt("shadows.btn.danger.selectedShadow")};
    border-color: ${vt("colors.btn.danger.selectedBorder")};
  }

  &:disabled {
    color: ${vt("colors.btn.danger.disabledText")};
    background-color: ${vt("colors.btn.danger.disabledBg")};
    border-color: ${vt("colors.btn.danger.disabledBorder")};
  }

  ${Lr};
  ${Kt};
`,Dr=i.default(Xt)`
  vertical-align: middle;

  && > * {
    position: relative;
    border-right-width: 0;
    border-radius: 0;

    :first-child {
      border-top-left-radius: ${vt("radii.2")};
      border-bottom-left-radius: ${vt("radii.2")};
      margin-right: 0;
    }

    :not(:first-child) {
      margin-left: 0;
      margin-right: 0;
    }

    :last-child {
      border-right-width: 1px;
      border-top-right-radius: ${vt("radii.2")};
      border-bottom-right-radius: ${vt("radii.2")};
    }

    :focus,
    :active,
    :hover {
      border-right-width: 1px;
      + * {
        border-left-width: 0;
      }
    }

    :focus,
    :active {
      z-index: 1;
    }
  }

  ${Kt};
`;Dr.defaultProps={display:"inline-block"};const Wr=i.default(Or)`
  color: ${vt("colors.btn.outline.text")};
  border: 1px solid ${vt("colors.btn.border")};
  background-color: ${vt("colors.btn.bg")};
  box-shadow: ${vt("shadows.btn.shadow")};

  &:hover {
    color: ${vt("colors.btn.outline.hoverText")};
    background-color: ${vt("colors.btn.outline.hoverBg")};
    border-color: ${vt("colors.btn.outline.hoverBorder")};
    box-shadow: ${vt("shadows.btn.outline.hoverShadow")};
  }
  // focus must come before :active so that the active box shadow overrides
  &:focus {
    border-color: ${vt("colors.btn.outline.focusBorder")};
    box-shadow: ${vt("shadows.btn.outline.focusShadow")};
  }

  &:active {
    color: ${vt("colors.btn.outline.selectedText")};
    background-color: ${vt("colors.btn.outline.selectedBg")};
    box-shadow: ${vt("shadows.btn.outline.selectedShadow")};
    border-color: ${vt("colors.btn.outline.selectedBorder")};
  }

  &:disabled {
    color: ${vt("colors.btn.outline.disabledText")};
    background-color: ${vt("colors.btn.outline.disabledBg")};
    border-color: ${vt("colors.btn.border")};
  }

  ${Lr};
  ${Kt};
`,jr=i.default(Or)`
  color: ${vt("colors.btn.primary.text")};
  border: 1px solid ${vt("colors.btn.primary.border")};
  background-color: ${vt("colors.btn.primary.bg")};
  box-shadow: ${vt("shadows.btn.primary.shadow")};

  &:hover {
    color: ${vt("colors.btn.primary.hoverText")};
    background-color: ${vt("colors.btn.primary.hoverBg")};
    border-color: ${vt("colors.btn.primary.hoverBorder")};
    box-shadow: ${vt("shadows.btn.primary.hoverShadow")};
  }
  // focus must come before :active so that the active box shadow overrides
  &:focus {
    border-color: ${vt("colors.btn.primary.focusBorder")};
    box-shadow: ${vt("shadows.btn.primary.focusShadow")};
  }

  &:active {
    background-color: ${vt("colors.btn.primary.selectedBg")};
    box-shadow: ${vt("shadows.btn.primary.selectedShadow")};
  }

  &:disabled {
    color: ${vt("colors.btn.primary.disabledText")};
    background-color: ${vt("colors.btn.primary.disabledBg")};
    border-color: ${vt("colors.btn.primary.disabledBorder")};
  }

  ${Lr};
  ${Kt};
`,Fr=i.default(Or)`
  color: ${vt("colors.text.link")};
  background-color: transparent;
  border: 0;
  border-radius: ${vt("radii.2")};
  box-shadow: none;

  &:disabled {
    color: ${vt("colors.text.disabled")};
  }

  &:focus {
    box-shadow: ${vt("shadows.btn.focusShadow")};
  }

  ${Lr};
  ${Kt}
`,Ur=i.default.summary`
  display: inline-block;
  padding: 0;
  font-size: ${vt("fontSizes.1")};
  color: ${vt("colors.text.secondary")};
  text-decoration: none;
  white-space: nowrap;
  cursor: pointer;
  user-select: none;
  background-color: transparent;
  border: 0;
  appearance: none; // Corrects inability to style clickable input types in iOS.

  &:hover {
    text-decoration: underline;
  }

  &:disabled {
    &,
    &:hover {
      color: rgba(${vt("colors.text.secondary")}, 0.5);
      cursor: default;
    }
  }

  &:after {
    display: inline-block;
    margin-left: ${vt("space.1")};
    width: 0;
    height: 0;
    vertical-align: -2px;
    content: '';
    border: 4px solid transparent;
    border-top-color: currentcolor;
  }
  ${yt}
  ${wt}
  ${$t}
  ${Kt};
`;var _r={small:16,medium:32,large:64};function Gr(e){var t=e["aria-label"],r=e.className,o=e.fill,a=void 0===o?"currentColor":o,n=e.size,i=e.verticalAlign,d=e.svgDataByHeight,l=_r[n]||n,c=function(e,t){return e.map((function(e){return parseInt(e,10)})).reduce((function(e,r){return r<=t?r:e}),e[0])}(Object.keys(d),l),s=d[c].width;return{"aria-hidden":t?"false":"true","aria-label":t,role:"img",className:r,viewBox:"0 0 "+s+" "+c,width:l*(s/c),height:l,fill:a,style:{display:"inline-block",userSelect:"none",verticalAlign:i},dangerouslySetInnerHTML:{__html:d[c].path}}}var Kr=Object.assign||function(e){for(var t=1;t<arguments.length;t++){var r=arguments[t];for(var o in r)Object.prototype.hasOwnProperty.call(r,o)&&(e[o]=r[o])}return e};function Xr(e){return n.default.createElement("svg",Gr(Kr({},e,{svgDataByHeight:{16:{width:16,path:'<path fill-rule="evenodd" d="M13.78 4.22a.75.75 0 010 1.06l-7.25 7.25a.75.75 0 01-1.06 0L2.22 9.28a.75.75 0 011.06-1.06L6 10.94l6.72-6.72a.75.75 0 011.06 0z"></path>'},24:{width:24,path:'<path fill-rule="evenodd" d="M21.03 5.72a.75.75 0 010 1.06l-11.5 11.5a.75.75 0 01-1.072-.012l-5.5-5.75a.75.75 0 111.084-1.036l4.97 5.195L19.97 5.72a.75.75 0 011.06 0z"></path>'}}})))}function Vr(e){return n.default.createElement("svg",Gr(Kr({},e,{svgDataByHeight:{16:{width:16,path:'<path fill-rule="evenodd" d="M5 3.254V3.25v.005a.75.75 0 110-.005v.004zm.45 1.9a2.25 2.25 0 10-1.95.218v5.256a2.25 2.25 0 101.5 0V7.123A5.735 5.735 0 009.25 9h1.378a2.251 2.251 0 100-1.5H9.25a4.25 4.25 0 01-3.8-2.346zM12.75 9a.75.75 0 100-1.5.75.75 0 000 1.5zm-8.5 4.5a.75.75 0 100-1.5.75.75 0 000 1.5z"></path>'},24:{width:24,path:'<path fill-rule="evenodd" d="M5.75 21a1.75 1.75 0 110-3.5 1.75 1.75 0 010 3.5zM2.5 19.25a3.25 3.25 0 106.5 0 3.25 3.25 0 00-6.5 0zM5.75 6.5a1.75 1.75 0 110-3.5 1.75 1.75 0 010 3.5zM2.5 4.75a3.25 3.25 0 106.5 0 3.25 3.25 0 00-6.5 0zM18.25 15a1.75 1.75 0 110-3.5 1.75 1.75 0 010 3.5zM15 13.25a3.25 3.25 0 106.5 0 3.25 3.25 0 00-6.5 0z"></path><path fill-rule="evenodd" d="M6.5 7.25c0 2.9 2.35 5.25 5.25 5.25h4.5V14h-4.5A6.75 6.75 0 015 7.25h1.5z"></path><path fill-rule="evenodd" d="M5.75 16.75A.75.75 0 006.5 16V8A.75.75 0 005 8v8c0 .414.336.75.75.75z"></path>'}}})))}function qr(e){return n.default.createElement("svg",Gr(Kr({},e,{svgDataByHeight:{16:{width:16,path:'<path fill-rule="evenodd" d="M7.177 3.073L9.573.677A.25.25 0 0110 .854v4.792a.25.25 0 01-.427.177L7.177 3.427a.25.25 0 010-.354zM3.75 2.5a.75.75 0 100 1.5.75.75 0 000-1.5zm-2.25.75a2.25 2.25 0 113 2.122v5.256a2.251 2.251 0 11-1.5 0V5.372A2.25 2.25 0 011.5 3.25zM11 2.5h-1V4h1a1 1 0 011 1v5.628a2.251 2.251 0 101.5 0V5A2.5 2.5 0 0011 2.5zm1 10.25a.75.75 0 111.5 0 .75.75 0 01-1.5 0zM3.75 12a.75.75 0 100 1.5.75.75 0 000-1.5z"></path>'},24:{width:24,path:'<path fill-rule="evenodd" d="M4.75 3a1.75 1.75 0 100 3.5 1.75 1.75 0 000-3.5zM1.5 4.75a3.25 3.25 0 116.5 0 3.25 3.25 0 01-6.5 0zM4.75 17.5a1.75 1.75 0 100 3.5 1.75 1.75 0 000-3.5zM1.5 19.25a3.25 3.25 0 116.5 0 3.25 3.25 0 01-6.5 0zm17.75-1.75a1.75 1.75 0 100 3.5 1.75 1.75 0 000-3.5zM16 19.25a3.25 3.25 0 116.5 0 3.25 3.25 0 01-6.5 0z"></path><path fill-rule="evenodd" d="M4.75 7.25A.75.75 0 015.5 8v8A.75.75 0 014 16V8a.75.75 0 01.75-.75zm8.655-5.53a.75.75 0 010 1.06L12.185 4h4.065A3.75 3.75 0 0120 7.75v8.75a.75.75 0 01-1.5 0V7.75a2.25 2.25 0 00-2.25-2.25h-4.064l1.22 1.22a.75.75 0 01-1.061 1.06l-2.5-2.5a.75.75 0 010-1.06l2.5-2.5a.75.75 0 011.06 0z"></path>'}}})))}function Yr(e){return n.default.createElement("svg",Gr(Kr({},e,{svgDataByHeight:{16:{width:16,path:'<path fill-rule="evenodd" d="M1.5 8a6.5 6.5 0 0110.65-5.003.75.75 0 00.959-1.153 8 8 0 102.592 8.33.75.75 0 10-1.444-.407A6.5 6.5 0 011.5 8zM8 12a1 1 0 100-2 1 1 0 000 2zm0-8a.75.75 0 01.75.75v3.5a.75.75 0 11-1.5 0v-3.5A.75.75 0 018 4zm4.78 4.28l3-3a.75.75 0 00-1.06-1.06l-2.47 2.47-.97-.97a.749.749 0 10-1.06 1.06l1.5 1.5a.75.75 0 001.06 0z"></path>'},24:{width:24,path:'<path d="M2.5 12c0-5.24 4.288-9.5 9.593-9.5a9.608 9.608 0 017.197 3.219.75.75 0 001.12-.998A11.108 11.108 0 0012.093 1C5.973 1 1 5.919 1 12s4.973 11 11.093 11c5.403 0 9.91-3.832 10.893-8.915a.75.75 0 10-1.472-.285c-.848 4.381-4.74 7.7-9.421 7.7C6.788 21.5 2.5 17.24 2.5 12z"></path><path d="M12 17a1 1 0 100-2 1 1 0 000 2zm0-10a.75.75 0 01.75.75v4.5a.75.75 0 01-1.5 0v-4.5A.75.75 0 0112 7zm11.28.78a.75.75 0 00-1.06-1.06l-3.47 3.47-1.47-1.47a.75.75 0 10-1.06 1.06l2 2a.75.75 0 001.06 0l4-4z"></path>'}}})))}function Jr(e){return n.default.createElement("svg",Gr(Kr({},e,{svgDataByHeight:{16:{width:16,path:'<path fill-rule="evenodd" d="M8 1.5a6.5 6.5 0 100 13 6.5 6.5 0 000-13zM0 8a8 8 0 1116 0A8 8 0 010 8zm9 3a1 1 0 11-2 0 1 1 0 012 0zm-.25-6.25a.75.75 0 00-1.5 0v3.5a.75.75 0 001.5 0v-3.5z"></path>'},24:{width:24,path:'<path d="M12 7a.75.75 0 01.75.75v4.5a.75.75 0 01-1.5 0v-4.5A.75.75 0 0112 7zm1 9a1 1 0 11-2 0 1 1 0 012 0z"></path><path fill-rule="evenodd" d="M12 1C5.925 1 1 5.925 1 12s4.925 11 11 11 11-4.925 11-11S18.075 1 12 1zM2.5 12a9.5 9.5 0 1119 0 9.5 9.5 0 01-19 0z"></path>'}}})))}function Zr(e){return n.default.createElement("svg",Gr(Kr({},e,{svgDataByHeight:{16:{width:16,path:'<path fill-rule="evenodd" d="M8 1.5a6.5 6.5 0 100 13 6.5 6.5 0 000-13zM0 8a8 8 0 1116 0A8 8 0 010 8zm9 3a1 1 0 11-2 0 1 1 0 012 0zM6.92 6.085c.081-.16.19-.299.34-.398.145-.097.371-.187.74-.187.28 0 .553.087.738.225A.613.613 0 019 6.25c0 .177-.04.264-.077.318a.956.956 0 01-.277.245c-.076.051-.158.1-.258.161l-.007.004a7.728 7.728 0 00-.313.195 2.416 2.416 0 00-.692.661.75.75 0 001.248.832.956.956 0 01.276-.245 6.3 6.3 0 01.26-.16l.006-.004c.093-.057.204-.123.313-.195.222-.149.487-.355.692-.662.214-.32.329-.702.329-1.15 0-.76-.36-1.348-.863-1.725A2.76 2.76 0 008 4c-.631 0-1.155.16-1.572.438-.413.276-.68.638-.849.977a.75.75 0 101.342.67z"></path>'},24:{width:24,path:'<path d="M10.97 8.265a1.45 1.45 0 00-.487.57.75.75 0 01-1.341-.67c.2-.402.513-.826.997-1.148C10.627 6.69 11.244 6.5 12 6.5c.658 0 1.369.195 1.934.619a2.45 2.45 0 011.004 2.006c0 1.033-.513 1.72-1.027 2.215-.19.183-.399.358-.579.508l-.147.123a4.329 4.329 0 00-.435.409v1.37a.75.75 0 11-1.5 0v-1.473c0-.237.067-.504.247-.736.22-.28.486-.517.718-.714l.183-.153.001-.001c.172-.143.324-.27.47-.412.368-.355.569-.676.569-1.136a.953.953 0 00-.404-.806C12.766 8.118 12.384 8 12 8c-.494 0-.814.121-1.03.265zM13 17a1 1 0 11-2 0 1 1 0 012 0z"></path><path fill-rule="evenodd" d="M12 1C5.925 1 1 5.925 1 12s4.925 11 11 11 11-4.925 11-11S18.075 1 12 1zM2.5 12a9.5 9.5 0 1119 0 9.5 9.5 0 01-19 0z"></path>'}}})))}function Qr(e){return n.default.createElement("svg",Gr(Kr({},e,{svgDataByHeight:{16:{width:16,path:'<path d="M4.427 7.427l3.396 3.396a.25.25 0 00.354 0l3.396-3.396A.25.25 0 0011.396 7H4.604a.25.25 0 00-.177.427z"></path>'},24:{width:24,path:'<path d="M11.646 15.146L5.854 9.354a.5.5 0 01.353-.854h11.586a.5.5 0 01.353.854l-5.793 5.792a.5.5 0 01-.707 0z"></path>'}}})))}function eo(e){return n.default.createElement("svg",Gr(Kr({},e,{svgDataByHeight:{16:{width:16,path:'<path fill-rule="evenodd" d="M3.72 3.72a.75.75 0 011.06 0L8 6.94l3.22-3.22a.75.75 0 111.06 1.06L9.06 8l3.22 3.22a.75.75 0 11-1.06 1.06L8 9.06l-3.22 3.22a.75.75 0 01-1.06-1.06L6.94 8 3.72 4.78a.75.75 0 010-1.06z"></path>'},24:{width:24,path:'<path fill-rule="evenodd" d="M5.72 5.72a.75.75 0 011.06 0L12 10.94l5.22-5.22a.75.75 0 111.06 1.06L13.06 12l5.22 5.22a.75.75 0 11-1.06 1.06L12 13.06l-5.22 5.22a.75.75 0 01-1.06-1.06L10.94 12 5.72 6.78a.75.75 0 010-1.06z"></path>'}}})))}Xr.defaultProps={className:"octicon octicon-check",size:16,verticalAlign:"text-bottom"},Vr.defaultProps={className:"octicon octicon-git-merge",size:16,verticalAlign:"text-bottom"},qr.defaultProps={className:"octicon octicon-git-pull-request",size:16,verticalAlign:"text-bottom"},Yr.defaultProps={className:"octicon octicon-issue-closed",size:16,verticalAlign:"text-bottom"},Jr.defaultProps={className:"octicon octicon-issue-opened",size:16,verticalAlign:"text-bottom"},Zr.defaultProps={className:"octicon octicon-question",size:16,verticalAlign:"text-bottom"},Qr.defaultProps={className:"octicon octicon-triangle-down",size:16,verticalAlign:"text-bottom"},eo.defaultProps={className:"octicon octicon-x",size:16,verticalAlign:"text-bottom"};const to=i.default.button`
  border: none;
  padding: 0;
  background: transparent;
  outline: none;
  cursor: pointer;
  border-radius: ${vt("radii.2")};
  color: ${vt("colors.text.secondary")};

  &:focus {
    box-shadow: ${vt("shadows.btn.focusShadow")};
  }

  &:hover {
    color: ${vt("colors.text.link")};
  }
  ${yt};
  ${$t};
  ${Kt};
`,ro=t.forwardRef(((e,t)=>n.default.createElement(to,Tt({ref:t,"aria-label":"Close"},e),n.default.createElement(eo,null)))),oo={"outside-top":["outside-bottom","outside-right","outside-left","outside-bottom"],"outside-bottom":["outside-top","outside-right","outside-left","outside-bottom"],"outside-left":["outside-right","outside-bottom","outside-top","outside-bottom"],"outside-right":["outside-left","outside-bottom","outside-top","outside-bottom"]};function ao(e,t,r={}){const o=function(e){let t=e.parentNode;for(;null!==t;){if(t instanceof HTMLElement&&"static"!==getComputedStyle(t).position)return t;t=t.parentNode}return document.body}(e),a=function(e){let t=e;for(;null!==t&&t!==document.body;){if("visible"!==getComputedStyle(t).overflow)break;t=t.parentNode}const r=t!==document.body&&t instanceof HTMLElement?t:document.body,o=r.getBoundingClientRect(),a=getComputedStyle(r),[n,i,d,l]=[a.borderTopWidth,a.borderLeftWidth,a.borderRightWidth,a.borderBottomWidth].map((e=>parseInt(e,10)||0));return{top:o.top+n,left:o.left+i,width:o.width-d-i,height:Math.max(o.height-n-l,r===document.body?window.innerHeight:-1/0)}}(o),n=getComputedStyle(o),i=o.getBoundingClientRect(),[d,l]=[n.borderTopWidth,n.borderLeftWidth].map((e=>parseInt(e,10)||0));return function(e,t,r,o,{side:a,align:n,allowOutOfBounds:i,anchorOffset:d,alignmentOffset:l}){const c={top:e.top-t.top,left:e.left-t.left,width:e.width,height:e.height};let s=fo(r,o,a,n,d,l),f=a;if(s.top-=t.top,s.left-=t.left,!i){const i=oo[a];let b=0;if(i){let e=a;for(;b<i.length&&bo(e,s,c,r);){const a=i[b++];e=a,s=fo(r,o,a,n,d,l),s.top-=t.top,s.left-=t.left,f=a}}s.top<c.top&&(s.top=c.top),s.left<c.left&&(s.left=c.left),s.left+r.width>e.width+c.left&&(s.left=e.width+c.left-r.width),i&&b<i.length&&s.top+r.height>e.height+c.top&&(s.top=e.height+c.top-r.height)}return{...s,anchorSide:f}}(a,{top:i.top+d,left:i.left+l},e.getBoundingClientRect(),t instanceof Element?t.getBoundingClientRect():t,function(e={}){var t,r,o,a,n;const i=null!==(t=e.side)&&void 0!==t?t:no,d=null!==(r=e.align)&&void 0!==r?r:io;return{side:i,align:d,anchorOffset:null!==(o=e.anchorOffset)&&void 0!==o?o:"inside-center"===i?0:lo,alignmentOffset:null!==(a=e.alignmentOffset)&&void 0!==a?a:"center"!==d&&i.startsWith("inside")?co:0,allowOutOfBounds:null!==(n=e.allowOutOfBounds)&&void 0!==n?n:so}}(r))}const no="outside-bottom",io="start",lo=4,co=4,so=!1;function fo(e,t,r,o,a,n){const i=t.left+t.width,d=t.top+t.height;let l=-1,c=-1;return"outside-top"===r?l=t.top-a-e.height:"outside-bottom"===r?l=d+a:"outside-left"===r?c=t.left-a-e.width:"outside-right"===r&&(c=i+a),"outside-top"!==r&&"outside-bottom"!==r||(c="start"===o?t.left+n:"center"===o?t.left-(e.width-t.width)/2+n:i-e.width-n),"outside-left"!==r&&"outside-right"!==r||(l="start"===o?t.top+n:"center"===o?t.top-(e.height-t.height)/2+n:d-e.height-n),"inside-top"===r?l=t.top+a:"inside-bottom"===r?l=d-a-e.height:"inside-left"===r?c=t.left+a:"inside-right"===r?c=i-a-e.width:"inside-center"===r&&(c=(i+t.left)/2-e.width/2+a),"inside-top"===r||"inside-bottom"===r?c="start"===o?t.left+n:"center"===o?t.left-(e.width-t.width)/2+n:i-e.width-n:"inside-left"!==r&&"inside-right"!==r&&"inside-center"!==r||(l="start"===o?t.top+n:"center"===o?t.top-(e.height-t.height)/2+n:d-e.height-n),{top:l,left:c}}function bo(e,t,r,o){return"outside-top"===e||"outside-bottom"===e?t.top<r.top||t.top+o.height>r.height+r.top:t.left<r.left||t.left+o.width>r.width+r.left}function go(e,t=[]){const r=pr(null==e?void 0:e.floatingElementRef),o=pr(null==e?void 0:e.anchorElementRef),[a,i]=n.default.useState(void 0),d=n.default.useCallback((()=>{r.current instanceof Element&&o.current instanceof Element?i(ao(r.current,o.current,e)):i(void 0)}),[r,o,...t]);var l;return n.default.useLayoutEffect(d,[d]),l=d,n.default.useLayoutEffect((()=>{const e=new window.ResizeObserver((()=>l()));return e.observe(document.documentElement),()=>{e.disconnect()}}),[l]),{floatingElementRef:r,anchorElementRef:o,position:a}}yr();const uo=[];let po;function mo(e,t=!1){return lr(e,{reverse:t,strict:!0,onlyTabbable:!0}).next().value}function ho(e,t,r){const o=new AbortController,a=null!=r?r:o.signal;let n;function i(r){if(r instanceof HTMLElement&&document.contains(e)){if(e.contains(r))return void(n=r);if(n&&sr(n)&&e.contains(n))return void n.focus();if(t&&e.contains(t))return void t.focus();{const t=null===e.getAttribute("tabindex");return t&&e.setAttribute("tabindex","-1"),e.focus(),void(t&&e.addEventListener("blur",(()=>e.removeAttribute("tabindex")),{once:!0}))}}}e.setAttribute("data-focus-trap","active");const d=function(e){const t=new AbortController;return e.addEventListener("abort",(()=>{t.abort()})),t}(a);if(e.addEventListener("keydown",(t=>{if("Tab"!==t.key||t.defaultPrevented)return;const{target:r}=t,o=mo(e),a=mo(e,!0);r===o&&t.shiftKey?(t.preventDefault(),null==a||a.focus()):r!==a||t.shiftKey||(t.preventDefault(),null==o||o.focus())}),{signal:d.signal}),po){const e=po;po.container.setAttribute("data-focus-trap","suspended"),po.controller.abort(),uo.push(e)}d.signal.addEventListener("abort",(()=>{po=void 0})),a.addEventListener("abort",(()=>{e.removeAttribute("data-focus-trap");const t=uo.findIndex((t=>t.container===e));t>=0&&uo.splice(t,1),function(){const e=uo.pop();e&&ho(e.container,e.initialFocus,e.originalSignal)}()})),document.addEventListener("focus",(e=>{i(e.target)}),{signal:d.signal,capture:!0}),i(document.activeElement),po={container:e,controller:d,initialFocus:t,originalSignal:a};const l=uo.findIndex((t=>t.container===e));if(l>=0&&uo.splice(l,1),!r)return o}function xo(e,t=[]){const r=pr(null==e?void 0:e.containerRef),o=pr(null==e?void 0:e.initialFocusRef),a=null==e?void 0:e.disabled,i=n.default.useRef(),d=n.default.useRef(null);function l(){var t;null===(t=i.current)||void 0===t||t.abort(),null!=e&&e.restoreFocusOnCleanUp&&d.current instanceof HTMLElement&&(d.current.focus(),d.current=null)}return d.current||null!=e&&e.disabled||(d.current=document.activeElement),n.default.useEffect((()=>{if(r.current instanceof HTMLElement){var e;if(!a)return i.current=ho(r.current,null!==(e=o.current)&&void 0!==e?e:void 0),()=>{l()};l()}}),[r,o,a,...t]),{containerRef:r,initialFocusRef:o}}function vo({icon:e,...t}){return n.default.createElement(e,t)}vo.displayName="Octicon";const yo=i.default(vo)`
  ${yt}
  ${Kt}
`;function wo(e={},r=[]){const o=pr(e.containerRef),a=!!e.activeDescendantFocus,i=pr("boolean"!=typeof e.activeDescendantFocus&&e.activeDescendantFocus?e.activeDescendantFocus:void 0),d=e.disabled,l=n.default.useRef();return t.useEffect((()=>{if(o.current instanceof HTMLElement&&(!a||i.current instanceof HTMLElement)){if(!d){var t;const r={...e,activeDescendantControl:null!==(t=i.current)&&void 0!==t?t:void 0};return l.current=zr(o.current,r),()=>{var e;null===(e=l.current)||void 0===e||e.abort()}}var r;null===(r=l.current)||void 0===r||r.abort()}}),[d,...r]),{containerRef:o,activeDescendantControlRef:i}}yo.defaultProps={size:16};const Bo="__primerPortalRoot__",$o="__default__",ko={};function So(){const e=ko.__default__;if(!e||!document.body.contains(e)){let e=document.getElementById(Bo);if(!(e instanceof Element)){e=document.createElement("div"),e.setAttribute("id",Bo),e.style.position="absolute",e.style.top="0",e.style.left="0";const t=document.querySelector("[data-portal-root]");t?t.appendChild(e):document.body.appendChild(e)}!function(e,t="__default__"){ko[t]=e}(e)}}const To=({children:e,onMount:t,containerName:r})=>{const a=document.createElement("div");a.style.position="relative",a.style.zIndex="1";const i=n.default.useRef(a);return n.default.useLayoutEffect((()=>{let e=r;void 0===e&&(e=$o,So());const o=ko[e];if(!o)throw new Error(`Portal container '${r}' is not yet registered. Container must be registered with registerPortal before use.`);const a=i.current;return o.appendChild(a),null==t||t(),()=>{o.removeChild(a)}}),[i]),o.createPortal(e,i.current)};function Eo(...e){const r=t.useRef(null);return n.default.useLayoutEffect((()=>{function t(t=null){for(const r of e){if(!r)return;"function"==typeof r?r(t):r.current=t}}return t(r.current),()=>{t(r.current)}}),[...e,r.current]),r}const Co="200ms",Io=i.default("div")`
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: rgba(0, 0, 0, 0.4);
  animation: dialog-backdrop-appear ${Co} ${vt("animation.easeOutCubic")};

  @keyframes dialog-backdrop-appear {
    0% {
      opacity: 0;
    }
    100% {
      opacity: 1;
    }
  }
`,No={small:"480px",large:"640px",auto:"auto"},Ao={small:"296px",medium:"320px",large:"480px",xlarge:"640px"},zo=i.default.div`
  display: flex;
  flex-direction: column;
  background-color: ${vt("colors.bg.overlay")};
  box-shadow: ${vt("shadows.overlay.shadow")};
  min-width: 296px;
  max-width: calc(100vw - 64px);
  max-height: calc(100vh - 64px);
  width: ${e=>{var t;return Ao[null!==(t=e.width)&&void 0!==t?t:"xlarge"]}};
  height: ${e=>{var t;return No[null!==(t=e.height)&&void 0!==t?t:"auto"]}};
  border-radius: 12px;
  opacity: 1;
  animation: overlay--dialog-appear ${Co} ${vt("animation.easeOutCubic")};

  @keyframes overlay--dialog-appear {
    0% {
      opacity: 0;
      transform: scale(0.5);
    }
    100% {
      opacity: 1;
      transform: scale(1);
    }
  }

  ${yt};
  ${kt};
  ${Kt};
`,Ro=({dialogLabelId:e,title:r,subtitle:o,dialogDescriptionId:a,onClose:i})=>{const d=t.useCallback((()=>{i("close-button")}),[i]);return n.default.createElement(Go.Header,null,n.default.createElement(Xt,{display:"flex"},n.default.createElement(Xt,{display:"flex",px:2,py:"6px",flexDirection:"column",flexGrow:1},n.default.createElement(Go.Title,{id:e},null!=r?r:"Dialog"),o&&n.default.createElement(Go.Subtitle,{id:a},o)),n.default.createElement(Go.CloseButton,{onClose:d})))};Ro.displayName="DefaultHeader";const Lo=({children:e})=>n.default.createElement(Go.Body,null,e);Lo.displayName="DefaultBody";const Mo=({footerButtons:e})=>{const{containerRef:t}=wo({bindKeys:Br.ArrowHorizontal|Br.Tab,focusInStrategy:"closest"});return e?n.default.createElement(Go.Footer,{ref:t},n.default.createElement(Go.Buttons,{buttons:e})):null},Oo=n.default.forwardRef(((e,r)=>{const{title:o="Dialog",subtitle:a="",renderHeader:i,renderBody:d,renderFooter:l,onClose:c,role:s="dialog",width:f="xlarge",height:b="auto"}=e,g=kr(),u=kr(),p={...e,title:o,subtitle:a,role:s,dialogLabelId:g,dialogDescriptionId:u},m=t.useRef(null),h=Eo(m,r),x=t.useRef(null);xo({containerRef:m,restoreFocusOnCleanUp:!0}),ur((e=>{c("escape"),e.preventDefault()}),[c]);const v=(null!=i?i:Ro)(p),y=(null!=d?d:Lo)(p),w=(null!=l?l:Mo)(p);return n.default.createElement(n.default.Fragment,null,n.default.createElement(To,null,n.default.createElement(Io,{ref:x},n.default.createElement(zo,{width:f,height:b,ref:h,role:s,"aria-labelledby":g,"aria-describedby":u},v,y,w))))}));Oo.displayName="Dialog";const Ho=i.default(Xt).attrs({as:"header"})`
  box-shadow: 0 1px 0 ${vt("colors.border.overlay")};
  padding: ${vt("space.2")};
  z-index: 1;
  flex-shrink: 0;
`,Po=i.default(Xt)`
  font-size: ${vt("fontSizes.1")};
  font-weight: ${vt("fontWeights.bold")};
`,Do=i.default(Xt)`
  font-size: ${vt("fontSizes.0")};
  margin-top: ${vt("space.1")};
  color: ${vt("colors.text.tertiary")};
`,Wo=i.default(Xt)`
  flex-grow: 1;
  overflow: auto;
  padding: ${vt("space.3")};
`,jo=i.default(Xt).attrs({as:"footer"})`
  box-shadow: 0 -1px 0 ${vt("colors.border.overlay")};
  padding: ${vt("space.3")};
  display: flex;
  flex-flow: wrap;
  justify-content: flex-end;
  z-index: 1;
  flex-shrink: 0;

  button {
    margin-left: ${vt("space.1")};
    &:first-child {
      margin-left: 0;
    }
  }
`,Fo={normal:Hr,primary:jr,danger:Pr},Uo=i.default(Hr)`
  border-radius: 4px;
  background: transparent;
  border: 0;
  vertical-align: middle;
  color: ${vt("colors.text.secondary")};
  padding: ${vt("space.2")};
  align-self: flex-start;
  line-height: normal;
  box-shadow: none;
`,_o=({onClose:e})=>n.default.createElement(Uo,{"aria-label":"Close",onClick:e},n.default.createElement(yo,{icon:eo}));_o.displayName="CloseButton";const Go=Object.assign(Oo,{Header:Ho,Title:Po,Subtitle:Do,Body:Wo,Footer:jo,Buttons:({buttons:e})=>{const r=t.useRef(null);let o=0;const[a,i]=t.useState(0);return t.useEffect((()=>{var e;1===a?null===(e=r.current)||void 0===e||e.focus():i(a+1)}),[a]),n.default.createElement(n.default.Fragment,null,e.map(((e,t)=>{const{content:a,buttonType:i="normal",autoFocus:d=!1,...l}=e,c=Fo[i];return n.default.createElement(c,Tt({key:t},l,{ref:d&&0===o?(o++,r):null}),a)})))},CloseButton:_o}),Ko=i.default.header`
  padding: ${vt("space.2")};
  display: flex;
  flex-direction: row;
`,Xo=i.default(Xt)`
  font-size: ${vt("fontSizes.3")};
  font-weight: ${vt("fontWeights.bold")};
  padding: 6px ${vt("space.2")};
  flex-grow: 1;
`,Vo=({title:e,onClose:r,dialogLabelId:o})=>{const a=t.useCallback((()=>{r("close-button")}),[r]);return n.default.createElement(Ko,null,n.default.createElement(Xo,{id:o},e),n.default.createElement(Go.CloseButton,{onClose:a}))};Vo.displayName="ConfirmationHeader";const qo=i.default(Xt)`
  font-size: ${vt("fontSizes.1")};
  padding: 0 ${vt("space.3")} ${vt("space.3")} ${vt("space.3")};
  color: ${vt("colors.text.tertiary")};
  flex-grow: 1;
`,Yo=({children:e})=>n.default.createElement(qo,null,e);Yo.displayName="ConfirmationBody";const Jo=i.default(Xt)`
  display: grid;
  grid-auto-flow: column;
  grid-auto-columns: max-content;
  grid-gap: ${vt("space.2")};
  align-items: end;
  justify-content: end;
  padding: ${vt("space.1")} ${vt("space.3")} ${vt("space.3")};
`,Zo=({footerButtons:e})=>{const{containerRef:t}=wo({bindKeys:Br.ArrowHorizontal|Br.Tab,focusInStrategy:"closest"});return n.default.createElement(Jo,{ref:t},n.default.createElement(Go.Buttons,{buttons:null!=e?e:[]}))};Zo.displayName="ConfirmationFooter";const Qo=e=>{const{onClose:r,title:o,cancelButtonContent:a="Cancel",confirmButtonContent:i="OK",confirmButtonType:d="normal",children:l}=e,c=[{content:a,onClick:t.useCallback((()=>{r("cancel")}),[r]),autoFocus:!0},{content:i,buttonType:d,onClick:t.useCallback((()=>{r("confirm")}),[r])}];return n.default.createElement(Go,{onClose:r,title:o,footerButtons:c,role:"alertdialog",width:"medium",renderHeader:Vo,renderBody:Yo,renderFooter:Zo},l)};Qo.displayName="ConfirmationDialog";const ea=i.default.div`
   {
    /* 6px vertical padding + 20px line height = 32px total height
     *
     * TODO: When rem-based spacing on a 4px scale lands, replace
     * hardcoded '6px' with 'calc((${vt("space.s32")} - ${vt("space.20")}) / 2)'.
     */
  }
  padding: 6px ${vt("space.3")};
  font-size: ${vt("fontSizes.0")};
  font-weight: ${vt("fontWeights.bold")};
  color: ${vt("colors.text.secondary")};

  ${({variant:e})=>"filled"===e&&r.css(["background:",";margin:"," 0;border-top:1px solid ",";border-bottom:1px solid ",";&:first-child{margin-top:0;}"],vt("colors.bg.tertiary"),vt("space.2"),vt("colors.border.tertiary"),vt("colors.border.tertiary"))}

  ${Kt}
`;function ta({variant:e="subtle",title:t,auxiliaryText:r,children:o,...a}){return n.default.createElement(ea,Tt({role:"heading",variant:e},a),t,r&&n.default.createElement("span",null,"auxiliaryText"))}ta.displayName="Header";const ra=i.default.div`
  ${Kt}
`;function oa({header:e,items:t,...r}){return n.default.createElement(ra,r,e&&n.default.createElement(ta,e),t)}oa.displayName="Group";const aa=i.default.div`
  ${wt}
  ${yt}
  display: ${e=>e.inline?"inline-block":"inherit"};
  overflow: hidden;
  text-overflow: ellipsis;
  vertical-align: ${e=>e.inline?"top":"initial"};
  white-space: nowrap;
  ${se}
  ${e=>e.expandable?"&:hover { max-width: 10000px; }":""}
  ${Kt};
`;aa.defaultProps={expandable:!1,inline:!1,maxWidth:125};const na=i.default.div`
  height: 1px;
  background: ${vt("colors.selectMenu.borderSecondary")};
  margin-top: calc(${vt("space.2")} - 1px);
  margin-bottom: ${vt("space.2")};
`;function ia(){return n.default.createElement(na,null)}ia.displayName="Divider",ia.renderItem=ia;const da={default:{hover:{light:"rgba(46, 77, 108, 0.06)",dark:"rgba(201, 206, 212, 0.12)",dark_dimmed:"rgba(201, 206, 212, 0.12)"},focus:{light:"rgba(54, 77, 100, 0.16)",dark:"rgba(201, 206, 212, 0.24)",dark_dimmed:"rgba(201, 206, 212, 0.24)"}},danger:{hover:{light:"rgba(234, 74, 90, 0.08)",dark:"rgba(248, 81, 73, 0.16)",dark_dimmed:"rgba(248, 81, 73, 0.16)"},focus:{light:"rgba(234, 74, 90, 0.14)",dark:"rgba(248, 81, 73, 0.24)",dark_dimmed:"rgba(248, 81, 73, 0.24)"}}},la=(e="default",t)=>{if(t)return{color:vt("colors.text.disabled"),iconColor:vt("colors.text.disabled"),annotationColor:vt("colors.text.disabled"),hoverCursor:"default"};switch(e){case"danger":return{color:vt("colors.text.danger"),iconColor:vt("colors.icon.danger"),annotationColor:vt("colors.text.disabled"),hoverCursor:"pointer"};default:return{color:"inherit",iconColor:vt("colors.text.secondary"),annotationColor:vt("colors.text.secondary"),hoverCursor:"pointer"}}},ca=i.default.div`
  display: flex;
  min-width: 0;

  /* Required for dividers */
  position: relative;
  flex-grow: 1;
`,sa=i.default.div`
  align-items: baseline;
  display: flex;
  min-width: 0;
  flex-direction: var(--main-content-flex-direction);
  flex-grow: 1;
`,fa=i.default.div`
  /* 6px vertical padding + 20px line height = 32px total height
   *
   * TODO: When rem-based spacing on a 4px scale lands, replace
   * hardcoded '6px' with 'calc((${vt("space.s32")} - ${vt("space.20")}) / 2)'.
   */
  padding: 6px ${vt("space.2")};
  display: flex;
  border-radius: ${vt("radii.2")};
  color: ${({variant:e,item:t})=>la(e,null==t?void 0:t.disabled).color};
  // 2 frames on a 60hz monitor
  transition: background 33.333ms linear;

  @media (hover: hover) and (pointer: fine) {
    :hover {
      // allow override in case another item in the list is active/focused
      background: var(--item-hover-bg-override, ${({hoverBackground:e})=>e});
      cursor: ${({variant:e,item:t})=>la(e,null==t?void 0:t.disabled).hoverCursor};
    }
  }

  // Item dividers
  :not(:first-of-type):not(${na} + &):not(${ea} + &) {
    margin-top: ${({showDivider:e})=>e?"1px":"0"};

    ${ca}::before {
      content: ' ';
      display: block;
      position: absolute;
      width: 100%;
      top: -7px;
      // NB: This 'get' wont execute if its moved into the arrow function below.
      border: 0 solid ${vt("colors.selectMenu.borderSecondary")};
      border-top-width: ${({showDivider:e})=>e?"1px":"0"};
    }
  }

  // Item dividers should not be visible:
  // - above Hovered
  &:hover ${ca}::before,
  // - below Hovered
  // '*' instead of '&' because '&' maps to separate class names depending on 'variant'
  :hover + * ${ca}::before {
    // allow override in case another item in the list is active/focused
    border-color: var(--item-hover-divider-border-color-override, transparent) !important;
  }

  // - above Focused
  &:focus ${ca}::before,
  // - below Focused
  // '*' instead of '&' because '&' maps to separate class names depending on 'variant'
  :focus + * ${ca}::before,
  // - above Active Descendent
  &[${Cr}] ${ca}::before,
  // - below Active Descendent
  [${Cr}] + & ${ca}::before {
    // '!important' because all the ':not's above give higher specificity
    border-color: transparent !important;
  }

  // Active Descendant
  &[${Cr}='${Ir}'] {
    background: ${({focusBackground:e})=>e};
  }
  &[${Cr}='${Nr}'] {
    background: ${({hoverBackground:e})=>e};
  }

  &:focus {
    background: ${({focusBackground:e})=>e};
    outline: none;
  }

  &:active {
    background: ${({focusBackground:e})=>e};
  }

  ${Kt}
`,ba=i.default.span``,ga=i.default.div`
  // Match visual height to adjacent text line height.
  // TODO: When rem-based spacing on a 4px scale lands, replace
  // hardcoded '20px' with '${vt("space.s20")}'.
  height: 20px;
  width: ${vt("space.3")};
  margin-right: ${vt("space.2")};
`,ua=i.default(ga)`
  svg {
    fill: ${({variant:e,disabled:t})=>la(e,t).iconColor};
    font-size: ${vt("fontSizes.0")};
  }
`,pa=i.default(ua)`
  flex-shrink: 0;
  display: flex;
  flex-direction: column;
  justify-content: center;
`,ma=i.default(ua)`
  color: ${({variant:e,disabled:t})=>la(e,t).annotationColor}};
  margin-left: ${vt("space.2")};
  margin-right: 0;
  width: auto;
  div:nth-child(2) {
    margin-left: ${vt("space.2")};
  }
`,ha=i.default.span`
  color: ${vt("colors.text.secondary")};
  font-size: ${vt("fontSizes.0")};
  // TODO: When rem-based spacing on a 4px scale lands, replace
  // hardcoded '16px' with '${vt("lh-12")}'.
  line-height: 16px;
  margin-left: var(--description-container-margin-left);
  min-width: 0;
  flex-grow: 1;
  flex-basis: var(--description-container-flex-basis);
`,xa=i.default.input`
  pointer-events: none;
`;function va(e){const{text:r,description:o,descriptionVariant:a="inline",selected:i,selectionVariant:d,leadingVisual:l,trailingIcon:c,trailingText:s,variant:f="default",showDivider:b,disabled:g,onAction:u,onKeyPress:p,children:m,onClick:h,id:x,...v}=e,y=t.useMemo((()=>kr()),[]),w=t.useMemo((()=>kr()),[]),B=t.useCallback((t=>{if(g)return;null==p||p(t);t.target instanceof HTMLInputElement&&"checkbox"===t.target.type&&" "===t.key||!t.defaultPrevented&&[" ","Enter"].includes(t.key)&&(null==u||u(e,t))}),[u,g,e,p]),$=t.useCallback((t=>{g||(null==h||h(t),t.defaultPrevented||null==u||u(e,t))}),[u,g,e,h]),k=da[f],S=_t(k.hover,"inherit"),T=_t(k.focus,"inherit"),{theme:E}=Ut();return n.default.createElement(fa,Tt({tabIndex:g?void 0:-1,variant:f,showDivider:b,"aria-selected":i,"aria-labelledby":r?y:void 0,"aria-describedby":o?w:void 0},v,{"data-id":x,onKeyPress:B,onClick:$,hoverBackground:g?"inherit":S,focusBackground:g?"inherit":T}),!!i===i&&n.default.createElement(ga,null,"multiple"===d?n.default.createElement(n.default.Fragment,null,n.default.createElement(xa,{disabled:g,tabIndex:-1,type:"checkbox",checked:i,"aria-label":r,readOnly:!0,"aria-readonly":"false"})):i&&n.default.createElement(Xr,{fill:null==E?void 0:E.colors.text.primary})),l&&n.default.createElement(pa,{variant:f,disabled:g},n.default.createElement(l,null)),n.default.createElement(ca,null,n.default.createElement(sa,{style:{"--main-content-flex-direction":"inline"===a?"row":"column"}},m,r?n.default.createElement(ba,{id:y},r):null,o?n.default.createElement(ha,{id:w,style:{"--description-container-margin-left":"inline"===a?vt("space.2")(E):0,"--description-container-flex-basis":"inline"===a?0:"auto"}},"block"===a?o:n.default.createElement(aa,{title:o,inline:!0,maxWidth:"100%"},o)):null),c||s?n.default.createElement(ma,{variant:f,disabled:g},s,c&&n.default.createElement(c,null)):null))}va.displayName="Item";const ya=i.default.div`
  font-size: ${vt("fontSizes.1")};
  /* 14px font-size * 1.428571429 = 20px line height
   *
   * TODO: When rem-based spacing on a 4px scale lands, replace
   * hardcoded '20px'
   */
  line-height: 20px;

  &[${Ar}], &:focus-within {
    --item-hover-bg-override: none;
    --item-hover-divider-border-color-override: ${vt("colors.selectMenu.borderSecondary")};
  }
`;const wa=n.default.forwardRef(((e,t)=>{const{firstGroupStyle:r,lastGroupStyle:o,headerStyle:a,itemStyle:i}=function(e="inset"){switch(e){case"full":return{headerStyle:{paddingX:vt("space.2")},itemStyle:{borderRadius:0}};default:return{firstGroupStyle:{marginTop:vt("space.2")},lastGroupStyle:{marginBottom:vt("space.2")},itemStyle:{marginX:vt("space.2")}}}}(e.variant),d=(t,r,o)=>{var a,d,l;const c="renderItem"in t&&t.renderItem||e.renderItem||va,s=null!==(a=null!==(d=t.key)&&void 0!==d?d:null===(l=t.id)||void 0===l?void 0:l.toString())&&void 0!==a?a:o.toString();return n.default.createElement(c,Tt({showDivider:e.showItemDividers,selectionVariant:e.selectionVariant},t,{key:s,sx:{...i,...t.sx},item:r}))};let l=[];if(function(e){return"groupMetadata"in e}(e)){const t=e.groupMetadata.reduce(((e,t)=>e.set(t.groupId,t)),new Map);for(const r of e.items){var c,s,f;const e=t.get(r.groupId),o=null!==(c=null==e||null===(s=e.items)||void 0===s?void 0:s.length)&&void 0!==c?c:0;t.set(r.groupId,{...e,items:[...null!==(f=null==e?void 0:e.items)&&void 0!==f?f:[],d({showDivider:null==e?void 0:e.showItemDividers,...e&&"renderItem"in e&&{renderItem:e.renderItem},...r},r,o)]})}l=[...t.values()]}else l=[{items:e.items.map(((e,t)=>d(e,e,t))),groupId:"0"}];return n.default.createElement(ya,Tt({},e,{ref:t}),l.map((({header:t,...i},d)=>{const c="filled"===(null==t?void 0:t.variant),s=d>0&&!c;return n.default.createElement(n.default.Fragment,{key:i.groupId},s?n.default.createElement(ia,{key:`${i.groupId}-divider`}):null,(t=>{var r;const o=(null!==(r="renderGroup"in t&&t.renderGroup)&&void 0!==r?r:e.renderGroup)||oa;return n.default.createElement(o,Tt({},t,{key:t.groupId}))})({sx:{...0===d&&r,...d===l.length-1&&o,...d>0&&!s&&{mt:2}},...t&&{header:{...t,sx:{...a,...t.sx}}},...i}))})))}));wa.displayName="ActionList";const Ba=Object.assign(wa,{Group:oa,Item:va,Divider:ia}),$a={xsmall:"192px",small:"256px",medium:"320px",large:"432px",xlarge:"600px",auto:"auto",initial:"auto"},ka={small:"256px",medium:"320px",large:"480px",xlarge:"640px",xxlarge:"960px",auto:"auto"};const Sa=i.default.div`
  background-color: ${vt("colors.bg.overlay")};
  box-shadow: ${vt("shadows.overlay.shadow")};
  position: absolute;
  min-width: 192px;
  max-width: 640px;
  height: ${e=>$a[e.height||"auto"]};
  max-height: ${e=>e.maxHeight&&$a[e.maxHeight]};
  width: ${e=>ka[e.width||"auto"]};
  border-radius: 12px;
  overflow: hidden;
  animation: overlay-appear ${200}ms ${vt("animation.easeOutCubic")};

  @keyframes overlay-appear {
    0% {
      opacity: 0;
    }
    100% {
      opacity: 1;
    }
  }
  visibility: var(--styled-overlay-visibility);
  :focus {
    outline: none;
  }
  ${yt};
  ${Kt};
`,Ta=n.default.forwardRef((({onClickOutside:e,role:r="none",initialFocusRef:o,returnFocusRef:a,ignoreClickRefs:i,onEscape:d,visibility:l="visible",height:c,top:s,left:f,anchorSide:b,portalContainerName:g,...u},p)=>{const m=t.useRef(null),h=Eo(m,p),{theme:x}=Ut(),v=parseInt(vt("space.2")(x).replace("px","")),y=vt("animation.easeOutCubic")(x);return mr({overlayRef:m,returnFocusRef:a,onEscape:d,ignoreClickRefs:i,onClickOutside:e,initialFocusRef:o}),t.useEffect((()=>{var e;"initial"===c&&null!==(e=h.current)&&void 0!==e&&e.clientHeight&&(h.current.style.height=`${h.current.clientHeight}px`)}),[c,h]),t.useLayoutEffect((()=>{var e;const{x:t,y:r}=function(e){return null!=e&&e.endsWith("bottom")?{x:0,y:-1}:null!=e&&e.endsWith("top")?{x:0,y:1}:null!=e&&e.endsWith("right")?{x:-1,y:0}:null!=e&&e.endsWith("left")?{x:1,y:0}:{x:0,y:0}}(b);(t||r)&&null!==(e=m.current)&&void 0!==e&&e.animate&&"hidden"!==l&&m.current.animate({transform:[`translate(${v*t}px, ${v*r}px)`,"translate(0, 0)"]},{duration:200,easing:y})}),[b,v,y,l]),n.default.createElement(To,{containerName:g},n.default.createElement(Sa,Tt({height:c,role:r},u,{ref:h,style:{top:`${s||0}px`,left:`${f||0}px`,...u.style,"--styled-overlay-visibility":l}})))}));Ta.defaultProps={height:"auto",width:"auto"};const Ea=({renderAnchor:e,anchorRef:r,children:o,open:a,onOpen:i,onClose:d,height:l,width:c,overlayProps:s,focusTrapSettings:f,focusZoneSettings:b})=>{const g=pr(r),[u,p]=function(){const[e,r]=t.useState(null),o=t.useRef(null);o.current=e;const a=t.useCallback((e=>{o.current=e,r(e)}),[o]);return[o,a]}(),m=t.useMemo(kr,[]),h=t.useCallback((()=>null==d?void 0:d("click-outside")),[d]),x=t.useCallback((()=>null==d?void 0:d("escape")),[d]),v=t.useCallback((e=>{e.defaultPrevented||!a&&["ArrowDown","ArrowUp"," ","Enter"].includes(e.key)&&(null==i||i("anchor-key-press"),e.preventDefault())}),[a,i]),y=t.useCallback((e=>{e.defaultPrevented||0!==e.button||(a?null==d||d("anchor-click"):null==i||i("anchor-click"))}),[a,i,d]),{position:w}=go({anchorElementRef:g,floatingElementRef:u},[u.current]);return t.useEffect((()=>{!a&&u.current&&p(null)}),[a,u,p]),wo({containerRef:u,disabled:!a||!w,...b}),xo({containerRef:u,disabled:!a||!w,...f}),n.default.createElement(n.default.Fragment,null,e&&e({ref:g,id:m,"aria-labelledby":m,"aria-haspopup":"true",tabIndex:0,onClick:y,onKeyDown:v}),a?n.default.createElement(Ta,Tt({returnFocusRef:g,onClickOutside:h,ignoreClickRefs:[g],onEscape:x,ref:p,role:"none",visibility:w?"visible":"hidden",height:l,width:c,top:(null==w?void 0:w.top)||0,left:(null==w?void 0:w.left)||0,anchorSide:null==w?void 0:w.anchorSide},s),o):null)};function Ca(e,r,o){const[a,n]=t.useState(o);return[null!=e?e:a,t.useCallback((e=>{n(e),r&&r(e)}),[r])]}Ea.displayName="AnchoredOverlay";const Ia=e=>n.default.createElement(va,Tt({role:"menuitem"},e));Ia.displayName="ActionMenuItem",Ia.displayName="ActionMenu.Item";const Na=({anchorContent:e,renderAnchor:r=(e=>n.default.createElement(Hr,e)),anchorRef:o,onAction:a,open:i,setOpen:d,overlayProps:l,items:c,...s})=>{const[f,b]=Ca(i,d,!1),g=pr(o),u=t.useCallback((()=>b(!0)),[b]),p=t.useCallback((()=>b(!1)),[b]),m=t.useMemo((()=>null===r?null:t=>r({"aria-label":"menu",children:e,...t})),[e,r]),h=t.useMemo((()=>c.map((e=>({...e,role:"menuitem",onAction:(t,r)=>{var o;const n=null!==(o=e.onAction)&&void 0!==o?o:a;null==n||n(t,r),r.defaultPrevented||p()}})))),[c,a,p]);return n.default.createElement(Ea,{renderAnchor:m,anchorRef:g,open:f,onOpen:u,onClose:p,overlayProps:l},n.default.createElement(wa,Tt({},s,{role:"menu",items:h})))};Na.displayName="ActionMenuBase",Na.displayName="ActionMenu";const Aa=Object.assign(Na,{Divider:ia,Item:Ia});const za=i.default.img.attrs((e=>({height:e.size,width:e.size})))`
  display: inline-block;
  overflow: hidden; // Ensure page layout in Firefox should images fail to load
  line-height: ${vt("lineHeights.condensedUltra")};
  vertical-align: middle;
  border-radius: ${e=>function({size:e,square:t}){return t?e&&e<=24?"4px":"6px":"50%"}(e)};
  ${yt};
  ${Kt}
`;za.defaultProps={size:20,alt:"",square:!1};const Ra=i.default(za)`
  position: absolute;
  right: -15%;
  bottom: -9%;
  box-shadow: ${vt("shadows.avatar.childShadow")};
`,La=({children:e,...t})=>{const r=n.default.Children.map(e,((e,t)=>n.default.isValidElement(e)?0===t?n.default.cloneElement(e,{size:40}):n.default.createElement(Ra,Tt({bg:"bg.canvas"},e.props,{size:20})):e));return n.default.createElement(Xt,Tt({position:"relative",display:"inline-flex"},t),r)};La.displayName="AvatarPair",La.displayName="AvatarPair";var Ma={exports:{}};
/*!
    Copyright (c) 2018 Jed Watson.
    Licensed under the MIT License (MIT), see
    http://jedwatson.github.io/classnames
  */!function(e){!function(){var t={}.hasOwnProperty;function r(){for(var e=[],o=0;o<arguments.length;o++){var a=arguments[o];if(a){var n=typeof a;if("string"===n||"number"===n)e.push(a);else if(Array.isArray(a)){if(a.length){var i=r.apply(null,a);i&&e.push(i)}}else if("object"===n)if(a.toString===Object.prototype.toString)for(var d in a)t.call(a,d)&&a[d]&&e.push(d);else e.push(a.toString())}}return e.join(" ")}e.exports?(r.default=r,e.exports=r):window.classNames=r}()}(Ma);var Oa=Ma.exports;const Ha=i.default.span`
  display: flex;
  position: relative;
  height: 20px;
  min-width: ${e=>1===e.count?"20px":2===e.count?"30px":"38px"};

  .pc-AvatarItem {
    flex-shrink: 0;
    height: 20px;
    width: 20px;
    box-shadow: 0 0 0 1px ${vt("colors.bg.canvas")};
    margin-left: -11px;
    position: relative;
    overflow: hidden;
    transition: margin 0.2s ease-in-out, opacity 0.2s ease-in-out, visibility 0.2s ease-in-out,
      box-shadow 0.1s ease-in-out;

    &:first-child {
      margin-left: 0;
    }
    &:nth-child(n + 4) {
      display: none;
    }
  }

  &.pc-AvatarStack--two {
    min-width: 30px;
    .pc-AvatarItem {
      &:nth-child(n + 3) {
        display: none;
      }
    }
  }

  &.pc-AvatarStack--three-plus {
    min-width: 38px;
    .pc-AvatarItem {
      &:nth-child(3) {
        opacity: ${55}%;
        margin-left: -17px;
      }
      &:nth-child(4) {
        opacity: ${40}%;
        margin-left: -17px;
      }
      &:nth-child(5) {
        opacity: ${25}%;
        margin-left: -17px;
      }
      &:nth-child(n + 4) {
        display: block;
      }
      &:nth-child(n + 6) {
        opacity: 0;
        visibility: hidden;
      }
    }
  }

  &.pc-AvatarStack--right {
    justify-content: flex-end;
    .pc-AvatarItem {
      margin-left: 0 !important;
      margin-right: -11px;

      &:first-child {
        margin-right: 0;
      }
    }

    .pc-AvatarStackBody {
      flex-direction: row-reverse;

      &:hover {
        .pc-AvatarItem {
          margin-right: ${vt("space.1")}!important;
          margin-left: 0 !important;

          &:first-child {
            margin-right: 0 !important;
          }
        }
      }
    }
  }

  &.pc-AvatarStack--three-plus.pc-AvatarStack--right {
    .pc-AvatarItem {
      &:nth-child(3) {
        margin-right: -17px;
      }
      &:nth-child(4) {
        margin-right: -17px;
      }
      &:nth-child(5) {
        margin-right: -17px;
      }
    }
  }

  .pc-AvatarStackBody:hover {
    width: auto;

    .pc-AvatarItem {
      margin-left: ${vt("space.1")};
      opacity: 100%;
      visibility: visible;
      box-shadow: 0 0 0 4px ${vt("colors.bg.canvas")};
      &:first-child {
        margin-left: 0;
      }
    }
  }

  ${yt}
  ${Kt};
`,Pa=({children:e,alignRight:t,...r})=>{const o=n.default.Children.count(e),a=Oa({"pc-AvatarStack--two":2===o,"pc-AvatarStack--three-plus":o>2,"pc-AvatarStack--right":t});return n.default.createElement(Ha,Tt({count:o,className:a},r),n.default.createElement(Xt,{position:"absolute",display:"flex",width:"38px",className:"pc-AvatarStackBody"},(e=>n.default.Children.map(e,((e,t)=>n.default.isValidElement(e)?n.default.cloneElement(e,{className:Oa(e.props.className,"pc-AvatarItem"),sx:{zIndex:10-t,...e.props.sx}}):e)))(e)))};Pa.displayName="AvatarStack";const Da=i.default.a`
  display: inline-block;
  padding: 2px 6px;
  font-size: ${vt("fontSizes.0")};
  font-family: ${vt("fonts.mono")};
  color: ${vt("colors.branchName.text")};
  background-color: ${vt("colors.branchName.bg")};
  border-radius: ${vt("radii.2")};
  ${yt};
  ${Kt};
`,Wa=i.default.li`
  display: inline-block;
  white-space: nowrap;
  list-style: none;
  &::after {
    padding-right: 0.5em;
    padding-left: 0.5em;
    color: ${vt("colors.text.disabled")};
    font-size: ${vt("fontSizes.1")};
    content: '/';
  }
  &:first-child {
    margin-left: 0;
  }
  &:last-child {
    &::after {
      content: none;
    }
  }
`,ja=i.default.nav`
  display: flex;
  justify-content: space-between;
  ${yt};
  ${St};
  ${Kt};
`;function Fa({className:e,children:t,theme:r,...o}){const a=Oa(e,"Breadcrumb"),i=n.default.Children.map(t,(e=>n.default.createElement(Wa,{theme:r},e)));return n.default.createElement(ja,Tt({className:a,"aria-label":"breadcrumb",theme:r},o),n.default.createElement(Xt,{as:"ol",my:0,pl:0},i))}Fa.displayName="Breadcrumb";const Ua=i.default.a.attrs((e=>({activeClassName:"string"==typeof e.to?"selected":"",className:Oa(e.selected&&"selected",e.className),"aria-current":e.selected?"page":null})))`
  color: ${vt("colors.text.link")};
  display: inline-block;
  font-size: ${vt("fontSizes.1")};
  text-decoration: none;
  &:hover {
    text-decoration: underline;
  }
  &.selected {
    color: ${vt("colors.text.primary")};
    pointer-events: none;
  }
  ${yt}
  ${Kt};
`;Fa.displayName="Breadcrumb",Ua.displayName="Breadcrumb.Item";var _a=Object.assign(Fa,{Item:Ua});const Ga={top:"Bottom",right:"Left",bottom:"Top",left:"Right"},Ka={top:"Left",right:"Top",bottom:"Left",left:"Top"};function Xa(e,t,r){const o=Ga[e].toLowerCase(),a=Ka[e].toLowerCase();return{[o]:"100%",[t||a]:t?r:"50%"}}const Va=ut({prop:"bg",key:"colors"}),qa=ut({prop:"borderColor",key:"colors"}),Ya=ut({prop:"borderWidth",key:"borderWidths",scale:[0,1]});function Ja(e){var t;const o=n.default.useContext(r.ThemeContext),a={...e,theme:null!==(t=e.theme)&&void 0!==t?t:o},{bg:i}=Va(a),{borderColor:d}=qa(a),{borderWidth:l}=Ya(a),{size:c=8,location:s="bottom"}=e,[f,b]=function(e){const[t,r]=e.split("-");return[t,r]}(s),g=Ka[f],u=[-c,0],p=[0,c],m=[c,0],h=`M${u}L${p}L${m}L${u}Z`,x=`M${u}L${p}L${m}`,v={top:`translate(${[c,2*c]}) rotate(180)`,right:`translate(${[0,c]}) rotate(-90)`,bottom:`translate(${[c,0]})`,left:`translate(${[2*c,c]}) rotate(90)`}[f];return n.default.createElement("svg",{width:2*c,height:2*c,style:{pointerEvents:"none",position:"absolute",...Xa(f,b,c),[`margin${g}`]:b?null:-c}},n.default.createElement("g",{transform:v},n.default.createElement("path",{d:h,fill:i}),n.default.createElement("path",{d:x,fill:"none",stroke:d,strokeWidth:l})))}Ja.displayName="Caret",Ja.locations=["top","top-left","top-right","right","right-top","right-bottom","bottom","bottom-left","bottom-right","left","left-top","left-bottom"],Ja.defaultProps={bg:"bg.canvas",borderColor:"border.primary",borderWidth:1};const Za={small:56,medium:96,large:128},Qa=i.default.div`
  display: ${e=>e.inline?"inline-flex":"flex"};
  align-items: center;
  justify-content: center;
  background-color: ${vt("colors.bg.canvas")};
  border-radius: 50%;
  box-shadow: ${vt("shadows.shadow.medium")};
  ${yt};
  ${({size:e,variant:t="medium"})=>{const r=(o=e,!isNaN(parseFloat(o))&&isFinite(o)?e:Za[t]);var o;return{width:r,height:r}}};
  ${Kt};
`,en=i.default(yo)`
  height: auto;
  max-width: 60%;
  max-height: 55%;
`;Qa.defaultProps={inline:!1},en.displayName="CircleBadge.Icon";var tn=Object.assign(Qa,{Icon:en});function rn(e){const{size:t,as:r}=e,{icon:o,bg:a,...i}=e;return n.default.createElement(Xt,{as:r,bg:a,overflow:"hidden",borderWidth:0,size:t,borderRadius:"50%",borderStyle:"solid",borderColor:"border.primary"},n.default.createElement(Xt,Tt({display:"flex"},i,{alignItems:"center",justifyContent:"center"}),n.default.createElement(o,{size:t})))}rn.displayName="CircleOcticon",rn.defaultProps={...Xt.defaultProps,size:32};const on=i.default.span`
  display: inline-block;
  padding: 2px 5px;
  font-size: ${vt("fontSizes.0")};
  font-weight: ${vt("fontWeights.bold")};
  line-height: ${vt("lineHeights.condensedUltra")};
  border-radius: 20px;
  ${({scheme:e,...t})=>({color:"secondary"===e?vt("colors.counter.text")(t):"primary"===e?vt("colors.counter.primary.text")(t):vt("colors.counter.text")(t)})};
  ${({scheme:e,...t})=>({backgroundColor:"secondary"===e?vt("colors.counter.bg")(t):"primary"===e?vt("colors.counter.primary.bg")(t):vt("colors.counter.bg")(t)})};
  ${yt};

  &:empty {
    display: none;
  }

  ${Kt};
`,an=i.default.details`
  & > summary {
    list-style: none;
  }
  & > summary::-webkit-details-marker {
    display: none;
  }

  ${yt}
  ${Kt};
`;an.displayName="Details";const nn=()=>null;function dn(e){const t=e;return t.tabIndex>=0&&!t.disabled&&function(e){return!e.hidden&&(!e.type||"hidden"!==e.type)&&(e.offsetWidth>0||e.offsetHeight>0)}(t)}const ln=i.default.span`
  ${wt};
  ${yt};
  ${Kt};
`,cn=()=>null,sn=i.default.div`
  box-shadow: ${vt("shadows.shadow.large")};
  border-radius: ${vt("radii.2")};
  position: fixed;
  top: 0;
  left: 50%;
  transform: translateX(-50%);
  max-height: 80vh;
  z-index: 999;
  margin: 10vh auto;
  background-color: ${vt("colors.bg.primary")};
  width: ${e=>e.narrow?"320px":e.wide?"640px":"440px"};
  outline: none;

  @media screen and (max-width: 750px) {
    width: 100vw;
    margin: 0;
    border-radius: 0;
    height: 100vh;
  }

  ${$t};
  ${yt};
  ${Kt};
`,fn=i.default(Xt)`
  border-radius: ${vt("radii.2")} ${vt("radii.2")} 0px 0px;
  border-bottom: 1px solid ${vt("colors.border.primary")};
  display: flex;

  @media screen and (max-width: 750px) {
    border-radius: 0px;
  }

  ${Kt};
`;function bn({theme:e,children:t,backgroundColor:r="gray.1",...o}){return n.default.Children.toArray(t).every((e=>"string"==typeof e))&&(t=n.default.createElement(ln,{theme:e,color:"text.primary",fontSize:1,fontWeight:"bold",fontFamily:"sans-serif"},t)),n.default.createElement(fn,Tt({theme:e,p:3,backgroundColor:r},o),t)}bn.displayName="DialogHeader";const gn=i.default.span`
  &:before {
    position: fixed;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    display: block;
    cursor: default;
    content: ' ';
    background: transparent;
    z-index: 99;
    background: ${vt("colors.fade.black50")};
  }
`,un=t.forwardRef((({children:e,onDismiss:r=cn,isOpen:o,initialFocusRef:a,returnFocusRef:i,...d},l)=>{const c=t.useRef(null),s=Eo(l),f=t.useRef(null),b=()=>{r(),i&&i.current&&i.current.focus()},{getDialogProps:g}=function({modalRef:e,overlayRef:r,isOpen:o,onDismiss:a=nn,initialFocusRef:n,closeButtonRef:i}){const d=t.useCallback((t=>{e.current&&r.current&&!e.current.contains(t.target)&&r.current.contains(t.target)&&a()}),[a,e,r]);t.useEffect((()=>{if(o)return document.addEventListener("click",d),()=>{document.removeEventListener("click",d)}}),[o,d]),t.useEffect((()=>{o&&(n&&n.current?n.current.focus():i&&i.current&&i.current.focus())}),[o,n,i]);const l=t.useCallback(((t,r)=>{if(e.current){const o=Array.from(e.current.querySelectorAll("*")).filter(dn);if(0===o.length)return;t.preventDefault();const a=document.activeElement;if(!a)return;const n=o.indexOf(a)+r,i=1===r?0:o.length-1;return o[n]||o[i]}}),[e]),c=t.useCallback((e=>{const t=e.shiftKey?-1:1,r=l(e,t);r&&r.focus()}),[l]),s=t.useCallback((e=>{switch(e.key){case"Tab":c(e);break;case"Escape":a(),e.stopPropagation()}}),[c,a]);return{getDialogProps:()=>({onKeyDown:s})}}({modalRef:s,onDismiss:b,isOpen:o,initialFocusRef:a,closeButtonRef:f,returnFocusRef:i,overlayRef:c});return o?n.default.createElement(n.default.Fragment,null,n.default.createElement(gn,{ref:c}),n.default.createElement(sn,Tt({tabIndex:-1,ref:s,role:"dialog","aria-modal":"true"},d,g()),n.default.createElement(ro,{ref:f,onClick:b,sx:{position:"absolute",top:"16px",right:"16px"}}),e)):null}));bn.defaultProps={backgroundColor:"bg.tertiary"},bn.propTypes={...Xt.propTypes},bn.displayName="Dialog.Header",un.displayName="Dialog";var pn=Object.assign(un,{Header:bn});const mn=i.default(an)`
  position: relative;
  display: inline-block;
`,hn=({children:e,className:t,...r})=>{const{getDetailsProps:o}=rr({closeOnOutsideClick:!0});return n.default.createElement(mn,Tt({className:t},o(),r),e)};hn.displayName="Dropdown";const xn=({children:e,...t})=>n.default.createElement(Hr,Tt({as:"summary","aria-haspopup":"true"},t),e,n.default.createElement(vn,null));xn.displayName="DropdownButton";const vn=i.default.div`
  border: 4px solid transparent;
  margin-left: 12px;
  border-top-color: currentcolor;
  border-bottom-width: 0;
  content: '';
  display: inline-block;
  height: 0;
  vertical-align: middle;
  width: 0;
  ${yt};
  ${Kt};
`,yn=i.default.ul`
  background-clip: padding-box;
  background-color: ${vt("colors.bg.overlay")};
  border: 1px solid ${vt("colors.border.overlay")};
  border-radius: ${vt("radii.2")};
  box-shadow: ${vt("shadows.dropdown.shadow")};
  left: 0;
  list-style: none;
  margin-top: 2px;
  padding: 5px 0 5px 0 !important; //TODO: fix this override on our markdown styles
  position: absolute;
  top: 100%;
  width: 160px;
  z-index: 100;

  &::before {
    position: absolute;
    display: inline-block;
    content: '';
  }

  &::after {
    position: absolute;
    display: inline-block;
    content: '';
  }

  &::before {
    border: 8px solid transparent;
    border-bottom-color: ${vt("colors.bg.overlay")};
  }

  &::after {
    border: 7px solid transparent;
    border-bottom-color: ${vt("colors.bg.overlay")};
  }

  // stylelint-disable-next-line selector-max-type
  > ul {
    list-style: none;
  }
  ${e=>e.direction?((e,t)=>({w:`\n      top: 0;\n      right: 100%;\n      left: auto;\n      width: auto;\n      margin-top: 0;\n      margin-right: 10px;\n\n      &::before {\n        top: 10px;\n        right: -16px;\n        left: auto;\n        border-color: transparent;\n        border-left-color: ${vt("colors.border.overlay")(e)};\n      }\n\n      &::after {\n        top: 11px;\n        right: -14px;\n        left: auto;\n        border-color: transparent;\n        border-left-color: ${vt("colors.border.overlay")(e)};\n      }\n    `,e:`\n      top: 0;\n      left: 100%;\n      width: auto;\n      margin-top: 0;\n      margin-left: 10px;\n\n      &::before {\n        top: 10px;\n        left: -16px;\n        border-color: transparent;\n        border-right-color: ${vt("colors.border.overlay")(e)};\n      }\n\n      &::after {\n        top: 11px;\n        left: -14px;\n        border-color: transparent;\n        border-right-color: ${vt("colors.border.overlay")(e)};\n      }\n    `,ne:`\n      top: auto;\n      bottom: 100%;\n      left: 0;\n      margin-bottom: 3px;\n\n      &::before,\n      &::after {\n        top: auto;\n        right: auto;\n      }\n\n      &::before {\n        bottom: -8px;\n        left: 9px;\n        border-top: 8px solid ${vt("colors.border.overlay")(e)};\n        border-bottom: 0;\n        border-left: 8px solid transparent;\n      }\n\n      &::after {\n        bottom: -7px;\n        left: 10px;\n        border-top: 7px solid ${vt("colors.border.overlay")(e)};\n        border-right: 7px solid transparent;\n        border-bottom: 0;\n        border-left: 7px solid transparent;\n      }\n    `,s:"\n      right: 50%;\n      left: auto;\n      transform: translateX(50%);\n\n      &::before {\n        top: -16px;\n        right: 50%;\n        transform: translateX(50%);\n      }\n\n      &::after {\n        top: -14px;\n        right: 50%;\n        transform: translateX(50%);\n      }\n    ",sw:"\n      right: 0;\n      left: auto;\n\n      &::before {\n        top: -16px;\n        right: 9px;\n        left: auto;\n      }\n\n      &::after {\n        top: -14px;\n        right: 10px;\n        left: auto;\n      }\n    ",se:"\n      &::before {\n        top: -16px;\n        left: 9px;\n      }\n\n      &::after {\n        top: -14px;\n        left: 10px;\n      }\n    "}[t]))(e.theme,e.direction):""};
  ${yt};
  ${Kt};
`,wn=i.default.li`
  display: block;
  padding: ${vt("space.1")} 10px ${vt("space.1")} 15px;
  overflow: hidden;
  color: ${vt("colors.text.primary")};
  text-overflow: ellipsis;
  white-space: nowrap;
  a {
    color: ${vt("colors.text.primary")};
    text-decoration: none;
    display: block;
    overflow: hidden;
    color: ${vt("colors.text.primary")};
    text-overflow: ellipsis;
    white-space: nowrap;
  }

  &:focus,
  a:focus {
    color: ${vt("colors.state.hover.primaryText")};
    text-decoration: none;
    background-color: ${vt("colors.state.hover.primaryBg")};
  }

  &:hover,
  &:hover a {
    color: ${vt("colors.state.hover.primaryText")};
    text-decoration: none;
    background-color: ${vt("colors.state.hover.primaryBg")};
    outline: none;
  }
  ${yt};
  ${Kt};
`;yn.defaultProps={direction:"sw"},yn.displayName="Dropdown.Menu",wn.displayName="Dropdown.Item",xn.defaultProps=Hr.defaultProps,xn.displayName="Dropdown.Button",vn.displayName="Dropdown.Caret",hn.defaultProps=an.defaultProps;var Bn=Object.assign(hn,{Caret:vn,Menu:yn,Item:wn,Button:xn});const $n=n.default.forwardRef((({children:e,...t},r)=>n.default.createElement(Hr,Tt({ref:r,type:"button"},t),e,n.default.createElement(yo,{icon:Qr,ml:1}))));function kn({renderAnchor:e=(e=>n.default.createElement($n,e)),anchorRef:r,placeholder:o,selectedItem:a,onChange:i,overlayProps:d,items:l,open:c,onOpenChange:s,...f}){const[b,g]=Ca(c,s,!1),u=t.useCallback((()=>g(!0)),[g]),p=t.useCallback((()=>g(!1)),[g]),m=pr(r),h=t.useMemo((()=>null===e?null:t=>{var r;return e({...t,children:null!==(r=null==a?void 0:a.text)&&void 0!==r?r:o})}),[o,e,null==a?void 0:a.text]),x=t.useMemo((()=>l.map((e=>({...e,role:"option",selected:e===a,onAction:(t,r)=>{var o;null===(o=e.onAction)||void 0===o||o.call(e,t,r),r.defaultPrevented||(p(),null==i||i(e===a?void 0:e))}})))),[l,i,p,a]);return n.default.createElement(Ea,{renderAnchor:h,anchorRef:m,open:b,onOpen:u,onClose:p,overlayProps:d},n.default.createElement(wa,Tt({},f,{role:"listbox",items:x})))}kn.displayName="DropdownMenu",kn.displayName="DropdownMenu";const Sn=i.default.div`
  ${yt};
  display: flex;
  align-items: stretch;

  summary {
    border-radius: 0;
    border-top-left-radius: ${vt("radii.2")};
    border-bottom-left-radius: ${vt("radii.2")};
    border-right: 0;
  }
  .TextInput-wrapper {
    border-radius: 0;
    border-top-right-radius: ${vt("radii.2")};
    border-bottom-right-radius: ${vt("radii.2")};
    z-index: 1; // Allows the focus outline to show on top of the dropdown.
  }

  ${Kt}
`,Tn=i.default.ul`
  list-style-type: none;
  ${yt};
  ${Kt};
`,En=({children:e,...t})=>{const r=n.default.Children.map(e,(e=>n.default.createElement("li",null,e)));return n.default.createElement(Tn,t,r)};En.displayName="FilterList";const Cn=i.default.a`
  position: relative;
  display: block;
  padding: ${e=>e.small?`${vt("space.1")(e)} 10px`:`${vt("space.2")(e)} 11px`};
  margin: ${e=>e.small?"0 0 2px":"0 0 5px 0"};
  overflow: hidden;
  font-size: ${vt("fontSizes.1")};
  color: ${e=>e.selected?vt("colors.state.selected.primaryText"):vt("colors.text.secondary")};
  background-color: ${e=>e.selected?vt("colors.state.selected.primaryBg"):""}!important;
  text-decoration: none;
  text-overflow: ellipsis;
  white-space: nowrap;
  cursor: pointer;
  border-radius: ${vt("radii.1")};
  &:hover {
    text-decoration: none;
    background-color: ${vt("colors.bg.tertiary")};
  }
  &:active {
    color: ${vt("colors.state.selected.primaryText")};
    background-color: ${vt("colors.state.selected.primaryBg")};
  }
  .count {
    float: right;
    font-weight: ${vt("fontWeights.bold")};
  }
  ${yt};
  ${Kt};
`;function In({children:e,count:t,...r}){return n.default.createElement(Cn,r,t&&n.default.createElement("span",{title:"results",className:"count"},t),e)}In.displayName="FilterListItem",En.defaultProps={m:0,p:0},In.displayName="FilterList.Item";var Nn=Object.assign(En,{Item:In});const An=re({variants:{default:{color:"alert.info.text",backgroundColor:"alert.info.bg",borderColor:"alert.info.border",svg:{color:"alert.info.icon"}},success:{color:"alert.success.text",backgroundColor:"alert.success.bg",borderColor:"alert.success.border",svg:{color:"alert.success.icon"}},danger:{color:"alert.error.text",backgroundColor:"alert.error.bg",borderColor:"alert.error.border",svg:{color:"alert.error.icon"}},warning:{color:"alert.warn.text",backgroundColor:"alert.warn.bg",borderColor:"alert.warn.border",svg:{color:"alert.warn.icon"}}}}),zn=i.default.div`
  position: relative;
  color: ${vt("colors.text.primary")};
  padding: ${vt("space.3")};
  border-style: solid;
  border-width: ${e=>e.full?"1px 0px":"1px"};
  border-radius: ${e=>e.full?"0":vt("radii.2")};
  margin-top: ${e=>e.full?"-1px":"0"};

  p:last-child {
    margin-bottom: 0;
  }

  svg {
    margin-right: ${vt("space.2")};
  }

  ${yt};
  ${An};
  ${Kt};
`;zn.defaultProps={variant:"default"};const Rn=i.default.div`
  margin: ${vt("space.3")} 0;
  font-weight: ${vt("fontWeights.normal")};
  ${yt};
  ${Kt};
`,Ln=i.default.label`
  display: block;
  margin: 0 0 ${vt("space.2")};
  font-size: ${vt("fontSizes.1")};
  font-weight: ${vt("fontWeights.bold")};
  ${wt};
  ${yt};
  ${Kt};
`;Ln.displayName="FormGroup.Label";var Mn=Object.assign(Rn,{Label:Ln});const On=i.default.div`
  z-index: 32;
  display: flex;
  padding: ${vt("space.3")};
  font-size: ${vt("fontSizes.1")};
  line-height: ${vt("lineHeights.default")};
  color: ${vt("colors.header.text")};
  background-color: ${vt("colors.header.bg")};
  align-items: center;
  flex-wrap: nowrap;

  ${yt}
  ${Bt}
  ${Kt};
`,Hn=i.default.div`
  display: flex;
  margin-right: ${vt("space.3")};
  align-self: stretch;
  align-items: center;
  flex-wrap: nowrap;

  ${({full:e})=>e&&r.css(["flex:auto;"])};

  ${yt};
  ${Bt};
  ${Kt};
`;Hn.displayName="Header.Item";const Pn=i.default.a.attrs((({to:e})=>"string"==typeof e?{"aria-current":"page"}:{}))`
  font-weight: ${vt("fontWeights.bold")};
  color: ${vt("colors.header.logo")};
  white-space: nowrap;
  cursor: pointer;
  text-decoration: none;
  display: flex;
  align-items: center;

  &:hover,
  &:focus {
    color: ${vt("colors.header.text")};
  }

  ${yt};
  ${Bt};
  ${wt};
  ${Kt};
`;Pn.displayName="Header.Link";var Dn=Object.assign(On,{Link:Pn,Item:Hn});const Wn=i.default.h2`
  font-weight: ${vt("fontWeights.bold")};
  font-size: ${vt("fontSizes.5")};
  margin: 0;
  ${wt};
  ${yt};
  ${Kt};
`;Wn.defaultProps={theme:l};const jn=i.default.span`
  ${yt}
  & * {
    margin-right: ${vt("space.1")};
  }
  & *:last-child {
    margin-right: 0;
  }
  ${Kt};
`,Fn=r.css(["margin-top:-1px;margin-bottom:-1px;color:",";border:"," solid ",";box-shadow:none;",";",";background-color:transparent;"],vt("colors.label.secondary.text"),vt("borderWidths.1"),vt("colors.label.border"),Qe,yt),Un=re({variants:{small:{fontSize:0,lineHeight:"16px",padding:"0px 8px"},medium:{fontSize:0,lineHeight:"20px",padding:"0 8px"},large:{fontSize:0,lineHeight:"24px",padding:"0 12px"},xl:{fontSize:1,lineHeight:"16px",padding:"8px 12px"}}}),_n=i.default.span`
  display: inline-block;
  font-weight: ${vt("fontWeights.semibold")};
  color: ${vt("colors.text.inverse")};
  border-radius: ${vt("radii.3")};

  &:hover {
    text-decoration: none;
  }

  ${Un}
  ${yt}
  ${e=>e.dropshadow?"box-shadow: inset 0 -1px 0 rgba(27, 31, 35, 0.12)":""}
  ${e=>e.outline?Fn:""} // must be last to override other values
  ${Kt}
`;_n.defaultProps={bg:"label.primary.border",variant:"medium"};const Gn=$({hoverColor:{property:"color",scale:"colors"}}),Kn=i.default.a`
  color: ${e=>e.muted?vt("colors.text.secondary")(e):vt("colors.text.link")(e)};
  text-decoration: ${e=>e.underline?"underline":"none"};
  &:hover {
    text-decoration: ${e=>e.muted?"none":"underline"};
    ${e=>e.hoverColor?Gn:e.muted?`color: ${vt("colors.text.link")(e)}`:""};
  }
  &:is(button) {
    display: inline-block;
    padding: 0;
    font-size: inherit;
    white-space: nowrap;
    cursor: pointer;
    user-select: none;
    background-color: transparent;
    border: 0;
    appearance: none;
  }
  ${wt};
  ${yt};
  ${Kt};
`,Xn=i.default.div`
  position: relative;
  padding-top: ${vt("space.4")};
  padding-bottom: ${vt("space.4")};
  margin-bottom: ${vt("space.4")};
  border-bottom: 1px solid ${vt("colors.border.primary")};
  ${yt};
  ${Kt};
`;const Vn=i.default.a`
  display: inline-block;
  min-width: 32px;
  padding: 5px 10px;
  font-style: normal;
  line-height: 20px;
  color: ${vt("colors.text.primary")};
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  cursor: pointer;
  user-select: none;
  text-decoration: none;

  margin-right: ${vt("space.1")};

  &:last-child {
    margin-right: 0;
  }

  border: ${vt("borderWidths.1")} solid transparent;
  border-radius: ${vt("radii.2")};
  transition: border-color 0.2s cubic-bezier(0.3, 0, 0.5, 1);

  &:hover,
  &:focus {
    text-decoration: none;
    border-color: ${vt("colors.border.primary")};
    outline: 0;
    transition-duration: 0.1s;
  }

  &:active {
    border-color: ${vt("colors.border.secondary")};
  }

  &[rel='prev'],
  &[rel='next'] {
    color: ${vt("colors.text.link")};
  }

  &[aria-current],
  &[aria-current]:hover {
    color: ${vt("colors.state.selected.primaryText")};
    background-color: ${vt("colors.state.selected.primaryBg")};
    border-color: transparent;
  }

  &[aria-disabled],
  &[aria-disabled]:hover {
    color: ${vt("colors.text.disabled")}; // check
    cursor: default;
    border-color: transparent;
  }

  @supports (clip-path: polygon(50% 0, 100% 50%, 50% 100%)) {
    &[rel='prev']::before,
    &[rel='next']::after {
      display: inline-block;
      width: 16px;
      height: 16px;
      vertical-align: text-bottom;
      content: '';
      background-color: currentColor;
    }

    // chevron-left
    &[rel='prev']::before {
      margin-right: ${vt("space.1")};
      clip-path: polygon(
        9.8px 12.8px,
        8.7px 12.8px,
        4.5px 8.5px,
        4.5px 7.5px,
        8.7px 3.2px,
        9.8px 4.3px,
        6.1px 8px,
        9.8px 11.7px,
        9.8px 12.8px
      );
    }

    // chevron-right
    &[rel='next']::after {
      margin-left: ${vt("space.1")};
      clip-path: polygon(
        6.2px 3.2px,
        7.3px 3.2px,
        11.5px 7.5px,
        11.5px 8.5px,
        7.3px 12.8px,
        6.2px 11.7px,
        9.9px 8px,
        6.2px 4.3px,
        6.2px 3.2px
      );
    }
  }

  ${yt};
`;function qn({theme:e,pageCount:t,currentPage:r,onPageChange:o,hrefBuilder:a,marginPageCount:i,showPages:d,surroundingPageCount:l}){const c=n.default.useCallback((e=>t=>o(t,e)),[o]),s=n.default.useMemo((()=>function(e,t,r,o,a){const n=[];if(r){const r=[],i=t=>{t>=1&&t<=e&&r.push(t)};let d=t-a,l=t+a;if(d<1&&l>e)d=1,l=e;else if(d<1)for(;d<1;)d++,l++;else if(l>e)for(;l>e;)d--,l--;for(let t=1;t<=o;t++){const r=t,o=e-(t-1);r>=d?l++:i(r),o<=l?d--:i(o)}for(let e=d;e<=l;e++)i(e);const c=r.slice().sort(((e,t)=>e-t)).filter(((e,t,r)=>!t||e!==r[t-1]));for(let e=0;e<c.length;e++){const r=c[e],o=r===t;0===e?(1!==r&&n.push({type:"BREAK",num:1}),n.push({type:"NUM",num:r,selected:o})):(1==r-c[e-1]||n.push({type:"BREAK",num:r-1}),n.push({type:"NUM",num:r,selected:o}))}const s=n[n.length-1];"NUM"===s.type&&s.num!==e&&n.push({type:"BREAK",num:e})}const i={type:"NEXT",num:t+1,disabled:t===e};return[{type:"PREV",num:t-1,disabled:1===t},...n,i]}(t,r,!!d,i,l)),[t,r,d,i,l]);return n.default.useMemo((()=>s.map((t=>{const{props:r,key:o,content:i}=function(e,t,r){const o={};let a="",n="";switch(e.type){case"PREV":n="page-prev",a="Previous",e.disabled?Object.assign(o,{as:"span","aria-disabled":"true"}):Object.assign(o,{rel:"prev",href:t(e.num),"aria-label":"Previous Page",onClick:r});break;case"NEXT":n="page-next",a="Next",e.disabled?Object.assign(o,{as:"span","aria-disabled":"true"}):Object.assign(o,{rel:"next",href:t(e.num),"aria-label":"Next Page",onClick:r});break;case"NUM":n=`page-${e.num}`,a=String(e.num),e.selected?Object.assign(o,{as:"em","aria-current":"page"}):Object.assign(o,{href:t(e.num),"aria-label":`Page ${e.num}`,onClick:r});break;case"BREAK":n=`page-${e.num}-break`,a="",Object.assign(o,{as:"span","aria-disabled":!0})}return{props:o,key:n,content:a}}(t,a,c(t.num));return n.default.createElement(Vn,Tt({},r,{key:o,theme:e}),i)}))),[s,a,c,e])}const Yn=i.default.nav`
  margin-top: 20px;
  margin-bottom: 15px;
  text-align: center;
  ${Kt};
`;function Jn({theme:e,pageCount:t,currentPage:r,onPageChange:o=Qn,hrefBuilder:a=Zn,marginPageCount:i=1,showPages:d=!0,surroundingPageCount:l=2,...c}){const s=qn({theme:e,pageCount:t,currentPage:r,onPageChange:o,hrefBuilder:a,marginPageCount:i,showPages:d,surroundingPageCount:l});return n.default.createElement(Yn,Tt({"aria-label":"Pagination"},c,{theme:e}),n.default.createElement(Xt,{display:"inline-block",theme:e},s))}function Zn(e){return`#${e}`}function Qn(){}function ei(e){const{bg:t,border:r,borderColor:o,theme:a}=e,{caret:i,children:d,...l}=e,c={bg:t,borderColor:o,borderWidth:r,location:i,theme:a};return n.default.createElement(Vt,Tt({sx:{position:"relative"}},l),d,n.default.createElement(Ja,c))}Jn.displayName="Pagination",Jn.defaultProps={hrefBuilder:Zn,marginPageCount:1,onPageChange:Qn,showPages:!0,surroundingPageCount:2},ei.displayName="PointerBox";const ti=i.default.div.attrs((({className:e,caret:t})=>({className:Oa(e,`caret-pos--${t}`)})))`
  position: ${e=>e.relative?"relative":"absolute"};
  z-index: 100;
  display: ${e=>e.open?"block":"none"};

  ${yt};
  ${$t};
  ${kt};
  ${Kt};
`,ri=i.default(Xt)`
  border: 1px solid ${vt("colors.border.primary")};
  border-radius: ${vt("radii.2")};
  position: relative;
  width: 232px;
  margin-right: auto;
  margin-left: auto;
  padding: ${vt("space.4")};
  background-color: ${vt("colors.bg.overlay")};

  ${yt};
  ${$t};

  // Carets
  &::before,
  &::after {
    position: absolute;
    left: 50%;
    display: inline-block;
    content: '';
  }

  &::before {
    top: -${vt("space.3")};
    margin-left: -9px;
    border: ${vt("space.2")} solid transparent; // TODO: solid?
    border-bottom-color: ${vt("colors.border.overlay")};
  }

  &::after {
    top: -14px;
    margin-left: -${vt("space.2")};
    border: 7px solid transparent; // todo: solid
    border-bottom-color: ${vt("colors.bg.overlay")};
  }

  // Bottom-oriented carets
  ${ti}.caret-pos--bottom & ,
  ${ti}.caret-pos--bottom-right & ,
  ${ti}.caret-pos--bottom-left & {
    &::before,
    &::after {
      top: auto;
      border-bottom-color: transparent;
    }

    &::before {
      bottom: -${vt("space.3")};
      border-top-color: ${vt("colors.border.overlay")};
    }

    &::after {
      bottom: -14px;
      // stylelint-disable-next-line primer/borders
      border-top-color: ${vt("colors.bg.overlay")};
    }
  }

  // Top & Bottom: Right-oriented carets
  ${ti}.caret-pos--top-right & ,
  ${ti}.caret-pos--bottom-right & {
    right: -9px;
    margin-right: 0;

    &::before,
    &::after {
      left: auto;
      margin-left: 0;
    }

    &::before {
      right: 20px;
    }

    &::after {
      right: 21px;
    }
  }

  // Top & Bottom: Left-oriented carets
  ${ti}.caret-pos--top-left & ,
  ${ti}.caret-pos--bottom-left & {
    left: -9px;
    margin-left: 0;

    &::before,
    &::after {
      left: ${vt("space.4")};
      margin-left: 0;
    }

    &::after {
      left: calc(${vt("space.4")} + 1px);
    }
  }

  // Right- & Left-oriented carets
  ${ti}.caret-pos--right & ,
  ${ti}.caret-pos--right-top & ,
  ${ti}.caret-pos--right-bottom & ,
  ${ti}.caret-pos--left & ,
  ${ti}.caret-pos--left-top & ,
  ${ti}.caret-pos--left-bottom & {
    &::before,
    &::after {
      top: 50%;
      left: auto;
      margin-left: 0;
      border-bottom-color: transparent;
    }

    &::before {
      // stylelint-disable-next-line primer/spacing
      margin-top: calc((${vt("space.2")} + 1px) * -1);
    }

    &::after {
      margin-top: -${vt("space.2")};
    }
  }

  // Right-oriented carets
  ${ti}.caret-pos--right & ,
  ${ti}.caret-pos--right-top & ,
  ${ti}.caret-pos--right-bottom & {
    &::before {
      right: -${vt("space.3")};
      border-left-color: ${vt("colors.border.overlay")};
    }

    &::after {
      right: -14px;
      // stylelint-disable-next-line primer/borders
      border-left-color: ${vt("colors.bg.overlay")};
    }
  }

  // Left-oriented carets
  ${ti}.caret-pos--left & ,
  ${ti}.caret-pos--left-top & ,
  ${ti}.caret-pos--left-bottom & {
    &::before {
      left: -${vt("space.3")};
      border-right-color: ${vt("colors.border.overlay")};
    }

    &::after {
      left: -14px;
      // stylelint-disable-next-line primer/borders
      border-right-color: ${vt("colors.bg.overlay")};
    }
  }

  // Right & Left: Top-oriented carets
  ${ti}.caret-pos--right-top & ,
  ${ti}.caret-pos--left-top & {
    &::before,
    &::after {
      top: ${vt("space.4")};
    }
  }

  // Right & Left: Bottom-oriented carets
  ${ti}.caret-pos--right-bottom & ,
  ${ti}.caret-pos--left-bottom & {
    &::before,
    &::after {
      top: auto;
    }

    &::before {
      bottom: ${vt("space.3")};
    }

    &::after {
      bottom: calc(${vt("space.3")} + 1px);
    }
  }

  ${Kt};
`;ti.defaultProps={caret:"top"},ri.displayName="Popover.Content";var oi=Object.assign(ti,{Content:ri});const ai=i.default.span`
  width: ${e=>e.progress?`${e.progress}%`:0};
  ${yt}
`,ni={small:"5px",large:"10px",default:"8px"},ii=i.default.span`
  display: ${e=>e.inline?"inline-flex":"flex"};
  overflow: hidden;
  background-color: ${vt("colors.border.primary")};
  border-radius: ${vt("radii.1")};
  height: ${e=>ni[e.barSize||"default"]};
  ${yt}
  ${ie}
  ${Kt};
`;function di({progress:e,bg:t,theme:r,...o}){return n.default.createElement(ii,Tt({theme:r},o),n.default.createElement(ai,{progress:e,bg:t,theme:r}))}function li(e,r,o){const a=t.useCallback((t=>{const a=()=>{o(!0)},n=t=>{const r=Array.from(e.current.querySelectorAll('[role^="menuitem"]:not([hidden]):not([disabled]):not([aria-disabled="true"])')),o=document.activeElement,a=r.indexOf(o),n=t?r[a+1]:r[a-1],i=t?r[0]:r[r.length-1];return n||i};if(!(t instanceof KeyboardEvent))return;const i=t.target instanceof Element&&"SUMMARY"===t.target.tagName;switch(t.key){case"Escape":r&&((()=>{o(!1);const t=e.current.querySelector("summary");t&&t.focus()})(),t.preventDefault(),t.stopPropagation());break;case"ArrowDown":{i&&!r&&a();const e=n(!0);e&&e.focus(),t.preventDefault()}break;case"ArrowUp":{i&&!r&&a();const e=n(!1);e&&e.focus(),t.preventDefault()}break;case" ":case"Enter":{const r=document.activeElement;r&&(e=>{const t=e.getAttribute("role");return"menuitem"===t||"menuitemcheckbox"===t||"menuitemradio"===t})(r)&&r.closest("details")===e&&(t.preventDefault(),t.stopPropagation(),r.click())}}}),[e,r,o]);t.useEffect((()=>{const t=e.current;if(t)return t.addEventListener("keydown",a),()=>{t.removeEventListener("keydown",a)}}),[e,a])}di.displayName="ProgressBar",di.defaultProps={bg:"bg.successInverse",barSize:"default"};const ci=t.createContext({}),si=r.css(["padding:"," ",";margin:0;font-size:",";font-weight:",";color:",";background-color:",";border-bottom:"," solid ",";"],vt("space.1"),vt("space.3"),vt("fontSizes.0"),vt("fontWeights.bold"),vt("colors.text.tertiary"),vt("colors.bg.tertiary"),vt("borderWidths.1"),vt("colors.selectMenu.borderSecondary")),fi=i.default.div`
  ${si}
  ${yt}
  ${Kt};
`;fi.displayName="SelectMenu.Divider";var bi=k(_,C,E,S,I,R,M,H,A,G,oe,ae,ne),gi=new RegExp("^("+bi.propNames.join("|")+")$");const ui=re({variants:{small:{minHeight:"28px",px:2,py:"3px",fontSize:0,lineHeight:"20px"},large:{px:2,py:"10px",fontSize:3}}}),pi=i.default.input`
  border: 0;
  font-size: inherit;
  font-family: inherit;
  background-color: transparent;
  -webkit-appearance: none;
  color: inherit;
  width: 100%;
  &:focus {
    outline: 0;
  }
`,mi=i.default.span`
  display: inline-flex;
  align-items: stretch;
  min-height: 34px;
  font-size: ${vt("fontSizes.1")};
  line-height: 20px;
  color: ${vt("colors.text.primary")};
  vertical-align: middle;
  background-repeat: no-repeat; // Repeat and position set for form states (success, error, etc)
  background-position: right 8px center; // For form validation. This keeps images 8px from right and centered vertically.
  border: 1px solid ${vt("colors.border.primary")};
  border-radius: ${vt("radii.2")};
  outline: none;
  box-shadow: ${vt("shadows.shadow.inset")};

  ${e=>e.hasIcon?r.css(["padding:0;"]):r.css(["padding:6px 12px;"])}

  .TextInput-icon {
    align-self: center;
    color: ${vt("colors.icon.tertiary")};
    margin: 0 ${vt("space.2")};
    flex-shrink: 0;
  }

  &:focus-within {
    border-color: ${vt("colors.state.focus.border")};
    box-shadow: ${vt("shadows.state.focus.shadow")};
  }

  ${e=>e.contrast&&r.css(["background-color:",";"],vt("colors.input.contrastBg"))}

  ${e=>e.disabled&&r.css(["color:",";background-color:",";border-color:",";"],vt("colors.text.secondary"),vt("colors.input.disabledBg"),vt("colors.input.disabledBorder"))}

  ${e=>e.block&&r.css(["display:block;width:100%;"])}

  // Ensures inputs don't zoom on mobile but are body-font size on desktop
  @media (min-width: ${vt("breakpoints.1")}) {
    font-size: ${vt("fontSizes.1")};
  }
  ${yt}
  ${ie}
  ${le}
  ${se}
  ${ui}
  ${Kt};
`,hi=n.default.forwardRef((({icon:e,contrast:t,className:r,block:o,disabled:a,theme:i,sx:d,...l},c)=>{const s=Oa(r,"TextInput-wrapper"),f=function(e){var t={};for(var r in e)gi.test(r)&&(t[r]=e[r]);return t}(l),b=function(e){var t={};for(var r in e)gi.test(r)||(t[r]=e[r]);return t}(l);return n.default.createElement(mi,Tt({className:s,hasIcon:!!e,block:o,theme:i,disabled:a,contrast:t,sx:d},f),e&&n.default.createElement(e,{className:"TextInput-icon"}),n.default.createElement(pi,Tt({ref:c,disabled:a},b)))}));hi.defaultProps={type:"text"},hi.displayName="TextInput";const xi=i.default.form`
  padding: ${vt("space.3")};
  margin: 0;
  border-bottom: ${vt("borderWidths.1")} solid ${vt("colors.selectMenu.borderSecondary")};
  background-color: ${vt("colors.bg.overlay")};
  ${yt};

  @media (min-width: ${vt("breakpoints.0")}) {
    padding: ${vt("space.2")};
  }

  ${Kt};
`,vi=t.forwardRef((({theme:e,value:r,sx:o,...a},i)=>{const d=t.useRef(null),l=null!=i?i:d,{open:c}=t.useContext(ci);return t.useEffect((()=>{var e;c&&(null===(e=d.current)||void 0===e||e.focus())}),[c]),n.default.createElement(xi,{theme:e,sx:o},n.default.createElement(hi,Tt({theme:e,ref:l,width:"100%",block:!0,value:r,contrast:!0},a)))}));vi.displayName="SelectMenu.Filter";const yi=r.css(["margin-top:-1px;padding:"," ",";font-size:",";color:",";text-align:center;border-top:"," solid ",";@media (min-width:","){padding:"," ",";}"],vt("space.2"),vt("space.3"),vt("fontSizes.0"),vt("colors.text.tertiary"),vt("borderWidths.1"),vt("colors.selectMenu.borderSecondary"),vt("breakpoints.0"),vt("space.1"),vt("space.2")),wi=i.default.footer`
  ${yi}
  ${yt}
  ${Kt};
`;wi.displayName="SelectMenu.Footer";const Bi=i.default.h3`
  color: ${vt("colors.text.primary")};
  flex: auto;
  font-size: ${vt("fontSizes.1")};
  font-weight: ${vt("fontWeights.bold")};
  margin: 0;

  @media (min-width: ${vt("breakpoints.0")}) {
    font-size: inherit;
  }
`,$i=i.default.header`
  display: flex;
  flex: none; // fixes header from getting squeezed in Safari iOS
  padding: ${vt("space.3")};
  border-bottom: ${vt("borderWidths")} solid ${vt("colors.selectMenu.borderSecondary")};
  ${yt}
  ${wt}

  @media (min-width: ${vt("breakpoints.0")}) {
    padding-top: ${vt("space.2")};
    padding-bottom: ${vt("space.2")};
  }

  ${Kt};
`,ki=({children:e,theme:t,...r})=>n.default.createElement($i,Tt({theme:t},r),n.default.createElement(Bi,{theme:t},e));ki.displayName="SelectMenuHeader",ki.displayName="SelectMenu.Header";const Si=r.css(["display:flex;align-items:center;padding:",";overflow:hidden;text-align:left;cursor:pointer;background-color:",";border:0;border-bottom:"," solid ",";color:",";text-decoration:none;font-size:",";font-family:inherit;width:100%;&:hover{text-decoration:none;}&:focus{outline:none;}&[hidden]{display:none !important;}@media (min-width:","){padding-top:",";padding-bottom:",";}.SelectMenu-icon{width:",";margin-right:",";flex-shrink:0;}.SelectMenu-selected-icon{visibility:hidden;transition:transform 0.12s cubic-bezier(0.5,0.1,1,0.5),visibility 0s 0.12s linear;transform:scale(0);}&[aria-checked='true']{font-weight:500;color:",";.SelectMenu-selected-icon{visibility:visible;transition:transform 0.12s cubic-bezier(0,0,0.2,1),visibility 0s linear;transform:scale(1);}}@media (hover:hover){&:hover,&:active,&:focus{background-color:",";}}@media (hover:none){&:focus,&:active{background-color:",";}-webkit-tap-highlight-color:",";}"],vt("space.3"),vt("colors.bg.overlay"),vt("borderWidths.1"),vt("colors.selectMenu.borderSecondary"),vt("colors.text.secondary"),vt("fontSizes.0"),vt("breakpoints.0"),vt("space.2"),vt("space.2"),vt("space.3"),vt("space.2"),vt("colors.text.primary"),vt("colors.state.hover.secondaryBg"),vt("colors.bg.secondary"),vt("colors.selectMenu.tapHighlight")),Ti=i.default.a.attrs((()=>({role:"menuitemcheckbox"})))`
  ${Si}
  ${yt}
  ${Kt};
`,Ei=t.forwardRef((({children:e,selected:r,theme:o,onClick:a,...i},d)=>{const l=t.useContext(ci),c=t.useRef(null),s=null!=d?d:c;return n.default.createElement(Ti,Tt({ref:s},i,{theme:o,onClick:e=>{var t;(a&&a(e),e.defaultPrevented)||(null===(t=l.setOpen)||void 0===t||t.call(l,!1))},"aria-checked":r}),n.default.createElement(yo,{theme:o,className:"SelectMenu-icon SelectMenu-selected-icon",icon:Xr}),e)}));Ei.defaultProps={selected:!1},Ei.displayName="SelectMenu.Item";const Ci=r.css(["position:relative;padding:0;margin:0;flex:auto;overflow-x:hidden;overflow-y:auto;background-color:",";-webkit-overflow-scrolling:touch;@media (hover:hover){.SelectMenuTab:focus{background-color:",";}.SelectMenuTab:not([aria-checked='true']):hover{color:",";background-color:",";}.SelectMenuTab:not([aria-checked='true']):active{color:",";background-color:",";}}"],vt("colors.bg.overlay"),vt("colors.selectMenu.tapFocusBg"),vt("colors.text.primary"),vt("colors.bg.tertiary"),vt("colors.text.primary"),vt("colors.bg.tertiary")),Ii=i.default.div`
  ${Ci}
  ${yt}
  ${Kt};
`;Ii.displayName="SelectMenu.List";const Ni={small:"16px",medium:"32px",large:"64px"};function Ai({size:e="medium",...t}){const r=Ni[e];return n.default.createElement("svg",Tt({height:r,width:r,viewBox:"0 0 16 16",fill:"none"},t),n.default.createElement("circle",{cx:"8",cy:"8",r:"7",stroke:"currentColor",strokeOpacity:"0.25",strokeWidth:"2",vectorEffect:"non-scaling-stroke"}),n.default.createElement("path",{d:"M15 8a7.002 7.002 0 00-7-7",stroke:"currentColor",strokeWidth:"2",strokeLinecap:"round",vectorEffect:"non-scaling-stroke"}))}Ai.displayName="Spinner";const zi=i.default(Ai)`
  @keyframes rotate-keyframes {
    100% {
      transform: rotate(360deg);
    }
  }

  animation: rotate-keyframes 1s linear infinite;

  ${yt}
  ${Kt}
`;zi.displayName="Spinner";const Ri=i.default.div`
  padding: ${vt("space.6")} ${vt("space.4")};
  text-align: center;
  background-color: ${vt("colors.bg.overlay")};
  ${yt}
`,Li=e=>n.default.createElement(Ri,e,n.default.createElement(zi,null));Li.displayName="SelectMenuLoadingAnimation";const Mi=r.keyframes(["0%{opacity:0;transform:scale(0.9);}"]),Oi=r.css(["position:relative;z-index:99;display:flex;",";max-height:",";margin:auto 0;",";overflow:hidden;pointer-events:auto;flex-direction:column;background-color:",";border-radius:",";box-shadow:",";animation:"," 0.12s cubic-bezier(0,0.1,0.1,1) backwards;@media (min-width:","){height:auto;max-height:350px;margin:"," 0 "," 0;font-size:",";border:"," solid ",";border-radius:",";box-shadow:",";}"],(e=>e.filter?"height: 80%":""),(e=>e.filter?"none":"66%"),(e=>e.filter?"margin-top: 0":""),vt("colors.bg.overlay"),vt("radii.2"),vt("shadows.shadow.small"),Mi,vt("breakpoints.0"),vt("space.1"),vt("space.3"),vt("fontSizes.0"),vt("borderWidths.1"),vt("colors.border.overlay"),vt("radii.2"),vt("shadows.shadow.small")),Hi=r.css(["position:fixed;top:0;right:0;bottom:0;left:0;z-index:99;display:flex;padding:",";pointer-events:none;flex-direction:column;&::before{position:absolute;top:0;right:0;bottom:0;left:0;pointer-events:none;content:'';background-color:",";@media (min-width:","){display:none;}}@media (min-width:","){position:absolute;top:auto;right:",";bottom:auto;left:auto;padding:0;}"],vt("space.3"),vt("colors.selectMenu.backdropBg"),vt("breakpoints.0"),vt("breakpoints.0"),(e=>"right"===e.align?"0":"auto")),Pi=i.default.div`
  ${Oi}
  ${ie}
`,Di=i.default.div`
  ${Hi}
  ${yt}
  ${Kt};
`,Wi=n.default.forwardRef((({children:e,theme:t,width:r,...o},a)=>n.default.createElement(Di,Tt({theme:t},o,{role:"menu",ref:a}),n.default.createElement(Pi,{theme:t,width:r},e))));Wi.defaultProps={align:"left",width:"300px"},Wi.displayName="SelectMenu.Modal";const ji=r.css(["flex:1;padding:"," ",";font-size:",";font-weight:500;color:",";text-align:center;background-color:transparent;border:0;box-shadow:inset 0 -1px 0 ",";@media (min-width:","){flex:none;padding:"," ",";border:"," solid transparent;border-bottom-width:0;border-top-left-radius:",";border-top-right-radius:",";}&[aria-selected='true']{z-index:1;color:",";background-color:",";box-shadow:0 0 0 1px ",";@media (min-width:","){border-color:",";box-shadow:none;}}&:focus{background-color:",";}"],vt("space.2"),vt("space.3"),vt("fontSizes.0"),vt("colors.text.tertiary"),vt("colors.selectMenu.borderSecondary"),vt("breakpoints.0"),vt("space.1"),vt("space.3"),vt("borderWidths.1"),vt("radii.2"),vt("radii.2"),vt("colors.text-primary"),vt("colors.bg.overlay"),vt("colors.selectMenu.borderSecondary"),vt("breakpoints.0"),vt("colors.selectMenu.borderSecondary"),vt("colors.state.hover.secondaryBg")),Fi=i.default.button`
  ${ji}
  ${yt}
  ${Kt};
`,Ui=({tabName:e="",index:r,className:o,onClick:a,...i})=>{const d=t.useContext(ci);t.useEffect((()=>{var t;d.selectedTab||0!==r||(null===(t=d.setSelectedTab)||void 0===t||t.call(d,e))}),[r,d,e]);const l=d.selectedTab===e;return n.default.createElement(Fi,Tt({role:"tab",className:Oa("SelectMenuTab",o),"aria-selected":l,onClick:t=>{var r;(a&&a(t),t.defaultPrevented)||(null===(r=d.setSelectedTab)||void 0===r||r.call(d,e))}},i),e)};Ui.displayName="SelectMenuTab",Ui.displayName="SelectMenu.Tab";const _i=i.default.div`
  border-top: ${vt("borderWidths.1")} solid ${vt("colors.selectMenu.borderSecondary")};
  ${yt}
  ${Kt};
`,Gi=({tabName:e,className:r,children:o,...a})=>{const i=t.useContext(ci);return n.default.createElement(_i,Tt({role:"tabpanel",className:r,hidden:i.selectedTab!==e},a),n.default.createElement(Ii,null,o))};Gi.displayName="TabPanel",Gi.displayName="SelectMenu.TabPanel";const Ki=r.css(["display:flex;flex-shrink:0;margin-bottom:-1px;-webkit-overflow-scrolling:touch;overflow-x:auto;overflow-y:hidden;&::-webkit-scrollbar{display:none;}@media (min-width:","){padding:0 ",";margin-top:",";}"],vt("breakpoints.0"),vt("space.2"),vt("space.3")),Xi=i.default.div`
  ${Ki}
  ${yt}
  ${Kt};
`,Vi=({children:e,...t})=>n.default.createElement(Xi,Tt({role:"tablist"},t),e);Vi.displayName="SelectMenuTabs",Vi.displayName="SelectMenu.Tabs";const qi=i.default.details`
  ${"\n  // Remove marker added by the display: list-item browser default\n  > summary {\n    list-style: none;\n  }\n  // Remove marker added by details polyfill\n  > summary::before {\n    display: none;\n  }\n  // Remove marker added by Chrome\n  > summary::-webkit-details-marker {\n    display: none;\n  }\n"}
  ${yt}
  ${Kt};
`,Yi=n.default.forwardRef((({children:e,initialTab:r="",as:o,...a},i)=>{const d=t.useRef(null),l=null!=i?i:d,[c,s]=t.useState(r),[f,b]=t.useState(!1),g={selectedTab:c,setSelectedTab:s,setOpen:b,open:f,initialTab:r},u=t.useCallback((e=>{"current"in l&&l.current&&!l.current.contains(e.target)&&(e.defaultPrevented||b(!1))}),[l,b]);return t.useEffect((()=>{if(f)return document.addEventListener("click",u),()=>{document.removeEventListener("click",u)}}),[f,u]),li(l,f,b),n.default.createElement(ci.Provider,{value:g},n.default.createElement(qi,Tt({ref:l},a,{open:f,onToggle:function(e){b(e.target.open)}}),e))}));Yi.displayName="SelectMenu";var Ji=Object.assign(Yi,{MenuContext:ci,List:Ii,Divider:fi,Filter:vi,Footer:wi,Item:Ei,Modal:Wi,Tabs:Vi,Tab:Ui,TabPanel:Gi,Header:ki,LoadingAnimation:Li});function Zi({variant:e,className:t,bordered:r,children:o,...a}){const i=Oa(t,`variant-${"lightweight"===e?"lightweight":"normal"}`);return r||(a={...a,borderWidth:0}),n.default.createElement(Xt,Tt({borderWidth:"1px",borderStyle:"solid",borderColor:"border.primary",borderRadius:2,as:"nav",className:i},a),o)}Zi.displayName="SideNavBase";const Qi=i.default(Zi)`
  background-color: ${vt("colors.bg.secondary")};

  ${e=>e.bordered&&r.css(["& > &{border-left:0;border-right:0;border-bottom:0;}"])}

  ${yt};
  ${Kt};
`,ed=r.css(["background-color:",";outline:none;text-decoration:none;"],vt("colors.state.hover.secondaryBg")),td=r.css(["color:",";text-decoration:none;outline:none;"],vt("colors.text.primary")),rd=i.default(Kn).attrs((e=>"string"==typeof e.to||e.selected?{"aria-current":"page"}:{}))`
  position: relative;
  display: block;
  ${e=>"full"===e.variant&&r.css(["display:flex;align-items:center;justify-content:space-between;"])}
  width: 100%;
  text-align: left;
  font-size: ${vt("fontSizes.1")};

  & > ${Qi} {
    border-bottom: none;
  }

  ${Qi}.variant-normal > & {
    color: ${vt("colors.text.primary")};
    padding: ${vt("space.3")};
    border: 0;
    border-top: ${vt("borderWidths.1")} solid ${vt("colors.border.secondary")};

    &:first-child {
      border-top: 0;
      border-top-right-radius: ${vt("radii.2")};
      border-top-left-radius: ${vt("radii.2")};
    }

    &:last-child {
      border-bottom-right-radius: ${vt("radii.2")};
      border-bottom-left-radius: ${vt("radii.2")};
    }

    // Bar on the left
    &::before {
      position: absolute;
      top: 0;
      bottom: 0;
      left: 0;
      z-index: 1;
      width: 3px;
      pointer-events: none;
      content: '';
    }

    &:hover {
      ${ed}
    }

    &:focus {
      ${ed}
      box-shadow: ${vt("shadows.state.focus.shadow")};
      z-index: 1;
    }

    &[aria-current='page'],
    &[aria-selected='true'] {
      background-color: ${vt("colors.sidenav.selectedBg")};

      // Bar on the left
      &::before {
        background-color: ${vt("colors.sidenav.borderActive")};
      }
    }
  }

  ${Qi}.variant-lightweight > & {
    padding: ${vt("space.1")} 0;
    color: ${vt("colors.text.link")};

    &:hover {
      ${td}
    }

    &:focus {
      ${td}
      box-shadow: ${vt("shadows.state.focus.shadow")};
      z-index: 1;
    }

    &[aria-current='page'],
    &[aria-selected='true'] {
      color: ${vt("colors.text.primary")};
      font-weight: ${vt("fontWeights.semibold")};
    }
  }

  ${Kt};
`;Qi.defaultProps={variant:"normal"},rd.defaultProps={variant:"normal"},rd.displayName="SideNav.Link";var od=Object.assign(Qi,{Link:rd});const ad={issueOpened:Jr,pullOpened:qr,issueClosed:Yr,pullClosed:qr,pullMerged:Vr,draft:qr},nd=re({prop:"status",variants:{issueClosed:{backgroundColor:"prState.closed.bg",color:"prState.closed.text",borderColor:"prState.closed.border"},pullClosed:{backgroundColor:"prState.closed.bg",color:"prState.closed.text",borderColor:"prState.closed.border"},pullMerged:{backgroundColor:"prState.merged.bg",color:"prState.merged.text",borderColor:"prState.merged.border"},issueOpened:{backgroundColor:"prState.open.bg",color:"prState.open.text",borderColor:"prState.open.border"},pullOpened:{backgroundColor:"prState.open.bg",color:"prState.open.text",borderColor:"prState.open.border"},draft:{backgroundColor:"prState.draft.bg",color:"prState.draft.text",borderColor:"prState.draft.border"}}}),id=re({prop:"variant",variants:{small:{paddingX:2,paddingY:1,fontSize:0},normal:{paddingX:"12px",paddingY:2,fontSize:1}}}),dd=i.default.span`
  display: inline-flex;
  align-items: center;
  font-weight: ${vt("fontWeights.bold")};
  line-height: 16px;
  color: ${vt("colors.bg.primary")};
  text-align: center;
  border-radius: ${vt("radii.3")};
  border-width: 1px;
  border-style: solid;
  ${nd};
  ${id};
  ${yt};
  ${Kt};
`;function ld({children:e,status:t,variant:r,...o}){const a="small"===r?{width:"1em"}:{};return n.default.createElement(dd,Tt({},o,{variant:r,status:t}),t&&n.default.createElement(yo,Tt({mr:1},a,{icon:ad[t]||Zr})),e)}ld.displayName="StateLabel",ld.defaultProps={variant:"normal"};const cd=i.default.nav`
  display: flex;
  justify-content: space-between;

  .SubNav-body {
    display: flex;
    margin-bottom: -1px;

    > * {
      margin-left: ${vt("space.2")};
    }

    > *:first-child {
      margin-left: 0;
    }
  }

  .SubNav-actions {
    align-self: center;
  }

  ${yt};
  ${St};
  ${Kt};
`;function sd({actions:e,className:t,children:r,label:o,...a}){const i=Oa(t,"SubNav");return n.default.createElement(cd,Tt({className:i,"aria-label":o},a),n.default.createElement("div",{className:"SubNav-body"},r),e&&n.default.createElement("div",{className:"SubNav-actions"},e))}function fd(e){return n.default.createElement(Xt,Tt({display:"flex"},e))}sd.displayName="SubNav",fd.displayName="SubNavLinks";const bd=i.default.a.attrs((e=>({activeClassName:"string"==typeof e.to?"selected":"",className:Oa("SubNav-item",e.selected&&"selected",e.className)})))`
  padding-left: ${vt("space.3")};
  padding-right: ${vt("space.3")};
  font-weight: ${vt("fontWeights.semibold")};
  font-size: ${vt("fontSizes.1")};
  line-height: 20px; //custom value for SubNav
  min-height: 34px; //custom value for SubNav
  color: ${vt("colors.text.primary")};
  text-align: center;
  text-decoration: none;
  border-top: 1px solid ${vt("colors.border.primary")};
  border-bottom: 1px solid ${vt("colors.border.primary")};
  border-right: 1px solid ${vt("colors.border.primary")};
  display: flex;
  align-items: center;

  &:first-of-type {
    border-top-left-radius: ${vt("radii.2")};
    border-bottom-left-radius: ${vt("radii.2")};
    border-left: 1px solid ${vt("colors.border.primary")};
  }

  &:last-of-type {
    border-top-right-radius: ${vt("radii.2")};
    border-bottom-right-radius: ${vt("radii.2")};
  }

  &:hover,
  &:focus {
    text-decoration: none;
    background-color: ${vt("colors.bg.tertiary")};
    transition: 0.2s ease;

    .SubNav-octicon {
      color: ${vt("colors.icon.secondary")};
    }
  }

  &.selected {
    color: ${vt("colors.state.selected.primaryText")};
    background-color: ${vt("colors.state.selected.primaryBg")};
    border-color: ${vt("colors.state.selected.primaryBorder")};
    .SubNav-octicon {
      color: ${vt("colors.state.selected.primaryText")};
    }
  }

  ${yt};
  ${Kt};
`;bd.displayName="SubNav.Link",fd.displayName="SubNav.Links";var gd=Object.assign(sd,{Link:bd,Links:fd});const ud=i.default.div`
  margin-top: 0;
  border-bottom: 1px solid ${vt("colors.border.primary")};
  ${yt}
  ${Kt}
`,pd=i.default.nav`
  display: flex;
  margin-bottom: -1px;
  overflow: auto;
`;function md({children:e,"aria-label":t,...r}){return n.default.createElement(ud,r,n.default.createElement(pd,{"aria-label":t},e))}md.displayName="TabNav";const hd=i.default.a.attrs((e=>({activeClassName:"string"==typeof e.to?"selected":"",className:Oa("TabNav-item",e.selected&&"selected",e.className)})))`
  padding: 8px 12px;
  font-size: ${vt("fontSizes.1")};
  line-height: 20px;
  color: ${vt("colors.text.primary")};
  text-decoration: none;
  background-color: transparent;
  border: 1px solid transparent;
  border-bottom: 0;

  &:hover,
  &:focus {
    color: ${vt("colors.text.primary")};
    text-decoration: none;
  }

  &.selected {
    color: ${vt("colors.text.primary")};
    border-color: ${vt("colors.border.primary")};
    border-top-right-radius: ${vt("radii.2")};
    border-top-left-radius: ${vt("radii.2")};
    background-color: ${vt("colors.bg.canvas")};
  }

  ${yt};
  ${Kt};
`;hd.displayName="TabNav.Link";var xd=Object.assign(md,{Link:hd});const vd=i.default(Xt).withConfig({displayName:"Timeline",componentId:"c4dq2e-0"})(["display:flex;flex-direction:column;"," ",";"],(e=>e.clipSidebar&&r.css([".Timeline-Item:first-child{padding-top:0;}.Timeline-Item:last-child{padding-bottom:0;}"])),Kt),yd=i.default(Xt).attrs((e=>({className:Oa("Timeline-Item",e.className)}))).withConfig({displayName:"Timeline__TimelineItem",componentId:"c4dq2e-1"})(["position:relative;padding:"," 0;margin-left:",";&::before{position:absolute;top:0;bottom:0;left:0;display:block;width:2px;content:'';background-color:",";}"," ",";",";"],vt("space.3"),vt("space.3"),vt("colors.border.secondary"),(e=>e.condensed&&r.css(["padding-top:",";padding-bottom:0;&:last-child{padding-bottom:",";}.TimelineItem-Badge{height:16px;margin-top:",";margin-bottom:",";color:",";background-color:",";border:0;}"],vt("space.1"),vt("space.3"),vt("space.2"),vt("space.2"),vt("colors.icon.tertiary"),vt("colors.bg.canvas"))),yt,Kt);var wd=i.default(Xt).withConfig({displayName:"Timeline___StyledBox",componentId:"c4dq2e-2"})(["border-radius:50%;border:2px solid ",";"],(e=>e._css));const Bd=e=>n.default.createElement(Xt,{position:"relative",zIndex:1},n.default.createElement(wd,Tt({display:"flex",className:Oa(e.className,"TimelineItem-Badge"),flexShrink:0,overflow:"hidden",color:"icon.secondary",bg:"timeline.badgeBg",width:"32px",height:"32px",mr:2,ml:"-15px",alignItems:"center",justifyContent:"center"},e,{_css:vt("colors.bg.canvas")}),e.children));Bd.displayName="TimelineBadge";const $d=i.default(Xt).withConfig({displayName:"Timeline__TimelineBody",componentId:"c4dq2e-3"})(["min-width:0;max-width:100%;margin-top:",";color:",";flex:auto;font-size:",";",";"],vt("space.1"),vt("colors.timeline.text"),vt("fontSizes.1"),Kt),kd=i.default(Xt).withConfig({displayName:"Timeline__TimelineBreak",componentId:"c4dq2e-4"})(["position:relative z-index:1;height:24px;margin:0;margin-bottom:-",";margin-left:0;background-color:",";border:0;border-top:"," solid ",";",";"],vt("space.3"),vt("colors.bg.canvas"),vt("space.1"),vt("colors.border.primary"),Kt);yd.displayName="Timeline.Item",Bd.displayName="Timeline.Badge",$d.displayName="Timeline.Body",kd.displayName="Timeline.Break";var Sd=Object.assign(vd,{Item:yd,Badge:Bd,Body:$d,Break:kd});const Td=i.default.span`
  position: relative;

  &::before {
    position: absolute;
    z-index: 1000001;
    display: none;
    width: 0px;
    height: 0px;
    color: ${vt("colors.tooltip.bg")};
    pointer-events: none;
    content: '';
    border: 6px solid transparent;
    opacity: 0;
  }

  &::after {
    position: absolute;
    z-index: 1000000;
    display: none;
    padding: 0.5em 0.75em;
    font: normal normal 11px/1.5 ${vt("fonts.normal")};
    -webkit-font-smoothing: subpixel-antialiased;
    color: ${vt("colors.tooltip.text")};
    text-align: center;
    text-decoration: none;
    text-shadow: none;
    text-transform: none;
    letter-spacing: normal;
    word-wrap: break-word;
    white-space: pre;
    pointer-events: none;
    content: attr(aria-label);
    background: ${vt("colors.tooltip.bg")};
    border-radius: ${vt("radii.1")};
    opacity: 0;
  }

  // delay animation for tooltip
  @keyframes tooltip-appear {
    from {
      opacity: 0;
    }

    to {
      opacity: 1;
    }
  }

  &:hover,
  &:active,
  &:focus {
    &::before,
    &::after {
      display: inline-block;
      text-decoration: none;
      animation-name: tooltip-appear;
      animation-duration: 0.1s;
      animation-fill-mode: forwards;
      animation-timing-function: ease-in;
      animation-delay: 0.4s;
    }
  }

  &.tooltipped-no-delay:hover,
  &.tooltipped-no-delay:active,
  &.tooltipped-no-delay:focus {
    &::before,
    &::after {
      animation-delay: 0s;
    }
  }

  &.tooltipped-multiline:hover,
  &.tooltipped-multiline:active,
  &.tooltipped-multiline:focus {
    &::after {
      display: table-cell;
    }
  }

  // Tooltipped south
  &.tooltipped-s,
  &.tooltipped-se,
  &.tooltipped-sw {
    &::after {
      top: 100%;
      right: 50%;
      margin-top: 6px;
    }

    &::before {
      top: auto;
      right: 50%;
      bottom: -7px;
      margin-right: -6px;
      border-bottom-color: ${vt("colors.tooltip.bg")};
    }
  }

  &.tooltipped-se {
    &::after {
      right: auto;
      left: 50%;
      margin-left: -${vt("space.3")};
    }
  }

  &.tooltipped-sw::after {
    margin-right: -${vt("space.3")};
  }

  // Tooltips above the object
  &.tooltipped-n,
  &.tooltipped-ne,
  &.tooltipped-nw {
    &::after {
      right: 50%;
      bottom: 100%;
      margin-bottom: 6px;
    }

    &::before {
      top: -7px;
      right: 50%;
      bottom: auto;
      margin-right: -6px;
      border-top-color: ${vt("colors.tooltip.bg")};
    }
  }

  &.tooltipped-ne {
    &::after {
      right: auto;
      left: 50%;
      margin-left: -${vt("space.3")};
    }
  }

  &.tooltipped-nw::after {
    margin-right: -${vt("space.3")};
  }

  // Move the tooltip body to the center of the object.
  &.tooltipped-s::after,
  &.tooltipped-n::after {
    transform: translateX(50%);
  }

  // Tooltipped to the left
  &.tooltipped-w {
    &::after {
      right: 100%;
      bottom: 50%;
      margin-right: 6px;
      transform: translateY(50%);
    }

    &::before {
      top: 50%;
      bottom: 50%;
      left: -7px;
      margin-top: -6px;
      border-left-color: ${vt("colors.tooltip.bg")};
    }
  }

  // tooltipped to the right
  &.tooltipped-e {
    &::after {
      bottom: 50%;
      left: 100%;
      margin-left: 6px;
      transform: translateY(50%);
    }

    &::before {
      top: 50%;
      right: -7px;
      bottom: 50%;
      margin-top: -6px;
      border-right-color: ${vt("colors.tooltip.bg")};
    }
  }

  &.tooltipped-multiline {
    &::after {
      width: max-content;
      max-width: 250px;
      word-wrap: break-word;
      white-space: pre-line;
      border-collapse: separate;
    }

    &.tooltipped-s::after,
    &.tooltipped-n::after {
      right: auto;
      left: 50%;
      transform: translateX(-50%);
    }

    &.tooltipped-w::after,
    &.tooltipped-e::after {
      right: 100%;
    }
  }

  &.tooltipped-align-right-2::after {
    right: 0;
    margin-right: 0;
  }

  &.tooltipped-align-right-2::before {
    right: 15px;
  }

  &.tooltipped-align-left-2::after {
    left: 0;
    margin-left: 0;
  }

  &.tooltipped-align-left-2::before {
    left: 10px;
  }
  ${yt};
  ${Kt};
`;function Ed({direction:e="n",children:t,className:r,text:o,noDelay:a,align:i,wrap:d,...l}){const c=Oa(r,`tooltipped-${e}`,i&&`tooltipped-align-${i}-2`,a&&"tooltipped-no-delay",d&&"tooltipped-multiline");return n.default.createElement(Td,Tt({role:"tooltip","aria-label":o},l,{className:c}),t)}Ed.displayName="Tooltip",Ed.alignments=["left","right"],Ed.directions=["n","ne","e","se","s","sw","w","nw"];const Cd=i.default.nav`
  display: flex;
  justify-content: space-between;
  border-bottom: 1px solid ${vt("colors.border.secondary")};
  &.UnderlineNav--right {
    justify-content: flex-end;

    .UnderlineNav-item {
      margin-right: 0;
      margin-left: ${vt("space.3")};
    }

    .UnderlineNav-actions {
      flex: 1 1 auto;
    }
  }
  &.UnderlineNav--full {
    display: block;
  }

  .UnderlineNav-body {
    display: flex;
    margin-bottom: -1px;
  }

  .UnderlineNav-actions {
    align-self: center;
  }

  ${yt};
  ${Kt};
`;function Id({actions:e,className:t,align:r,children:o,full:a,label:i,theme:d,...l}){const c=Oa(t,"UnderlineNav",r&&`UnderlineNav--${r}`,a&&"UnderlineNav--full");return n.default.createElement(Cd,Tt({className:c,"aria-label":i,theme:d},l),n.default.createElement("div",{className:"UnderlineNav-body"},o),e&&n.default.createElement("div",{className:"UnderlineNav-actions"},e))}Id.displayName="UnderlineNav";const Nd=i.default.a.attrs((e=>({activeClassName:"string"==typeof e.to?"selected":"",className:Oa("UnderlineNav-item",e.selected&&"selected",e.className)})))`
  padding: ${vt("space.3")} ${vt("space.2")};
  margin-right: ${vt("space.3")};
  font-size: ${vt("fontSizes.1")};
  line-height: ${vt("lineHeights.default")};
  color: ${vt("colors.underlinenav.text")};
  text-align: center;
  border-bottom: 2px solid transparent;
  text-decoration: none;

  &:hover,
  &:focus {
    color: ${vt("colors.underlinenav.textHover")};
    text-decoration: none;
    border-bottom-color: ${vt("colors.border.tertiary")};
    transition: 0.2s ease;

    .UnderlineNav-octicon {
      color: ${vt("colors.text.tertiary")};
    }
  }

  &.selected {
    color: ${vt("colors.underlinenav.textActive")};
    border-bottom-color: ${vt("colors.underlinenav.borderActive")};

    .UnderlineNav-octicon {
      color: ${vt("colors.underlinenav.iconActive")};
    }
  }

  ${yt};
  ${Kt};
`;Nd.displayName="UnderlineNav.Link";var Ad=Object.assign(Id,{Link:Nd});e.Absolute=Zt,e.ActionList=Ba,e.ActionMenu=Aa,e.Avatar=za,e.AvatarPair=La,e.AvatarStack=Pa,e.BaseStyles=It,e.BorderBox=Vt,e.Box=Xt,e.BranchName=Da,e.Breadcrumb=_a,e.Button=Hr,e.ButtonClose=ro,e.ButtonDanger=Pr,e.ButtonGroup=Dr,e.ButtonInvisible=Fr,e.ButtonOutline=Wr,e.ButtonPrimary=jr,e.ButtonTableList=Ur,e.Caret=Ja,e.CircleBadge=tn,e.CircleOcticon=rn,e.ConfirmationDialog=Qo,e.CounterLabel=on,e.Details=an,e.Dialog=pn,e.Dropdown=Bn,e.DropdownButton=$n,e.DropdownMenu=kn,e.FilterList=Nn,e.FilteredSearch=Sn,e.Fixed=Qt,e.Flash=zn,e.Flex=qt,e.FormGroup=Mn,e.Grid=Yt,e.Header=Dn,e.Heading=Wn,e.Label=_n,e.LabelGroup=jn,e.Link=Kn,e.Overlay=Ta,e.Pagehead=Xn,e.Pagination=Jn,e.PointerBox=ei,e.Popover=oi,e.Position=Jt,e.ProgressBar=di,e.Relative=er,e.SelectMenu=Ji,e.SideNav=od,e.Spinner=zi,e.StateLabel=ld,e.Sticky=tr,e.StyledOcticon=yo,e.SubNav=gd,e.TabNav=xd,e.Text=ln,e.TextInput=hi,e.ThemeProvider=Ft,e.Timeline=Sd,e.Tooltip=Ed,e.Truncate=aa,e.UnderlineNav=Ad,e.theme=l,e.themeGet=vt,e.useColorSchemeVar=_t,e.useConfirm=function(){const{theme:e,colorMode:r,dayScheme:o,nightScheme:a}=Ut();return t.useCallback((t=>async function(e,t){const{content:r,...o}=t;return new Promise((t=>{const a=document.createElement("div");d.default.render(n.default.createElement(Ft,e,n.default.createElement(Qo,Tt({},o,{onClose:e=>{d.default.unmountComponentAtNode(a),t("confirm"===e)}}),r)),a)}))}({theme:e,colorMode:r,dayScheme:o,nightScheme:a},t)),[e,r,o,a])},e.useDetails=rr,e.useOnEscapePress=ur,e.useOnOutsideClick=dr,e.useOpenAndCloseFocus=fr,e.useOverlay=mr,e.useSafeTimeout=function(){const e=t.useRef(new Set),r=t.useCallback(((t,r,...o)=>{const a=window.setTimeout(t,r,...o);return e.current.add(a),a}),[]),o=t.useCallback((t=>{clearTimeout(t),e.current.delete(t)}),[]);return t.useEffect((()=>()=>{for(const t of e.current)clearTimeout(t)}),[]),{safeSetTimeout:r,safeClearTimeout:o}},e.useTheme=Ut,Object.defineProperty(e,"__esModule",{value:!0})}));
