// @primer/components/dist/browser.esm.js
import e,{useState as r,useRef as t,useCallback as o,useEffect as a,useMemo as n,forwardRef as i,useLayoutEffect as d,createContext as l,useContext as c}from"react";import s,{createGlobalStyle as f,ThemeProvider as b,css as g,ThemeContext as p,keyframes as u}from"styled-components";import m,{createPortal as h}from"react-dom";var x={animation:{easeOutCubic:"cubic-bezier(0.33, 1, 0.68, 1)"},borderWidths:[0,"1px"],breakpoints:["544px","768px","1012px","1280px"],fonts:{normal:'-apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji"',mono:'SFMono-Regular, Consolas, "Liberation Mono", Menlo, Courier, monospace'},fontSizes:["12px","14px","16px","20px","24px","32px","40px","48px"],fontWeights:{light:300,normal:400,semibold:500,bold:600},lineHeights:{condensedUltra:1,condensed:1.25,default:1.5},radii:["0","3px","6px","100px"],sizes:{small:"544px",medium:"768px",large:"1012px",xlarge:"1280px"},space:["0","4px","8px","16px","24px","32px","40px","48px","64px","80px","96px","112px","128px"],colorSchemes:{light:{colors:{auto:{black:"#1b1f23",white:"#ffffff",gray:["#fafbfc","#f6f8fa","#e1e4e8","#d1d5da","#959da5","#6a737d","#586069","#444d56","#2f363d","#24292e"],blue:["#f1f8ff","#dbedff","#c8e1ff","#79b8ff","#2188ff","#0366d6","#005cc5","#044289","#032f62","#05264c"],green:["#f0fff4","#dcffe4","#bef5cb","#85e89d","#34d058","#28a745","#22863a","#176f2c","#165c26","#144620"],yellow:["#fffdef","#fffbdd","#fff5b1","#ffea7f","#ffdf5d","#ffd33d","#f9c513","#dbab09","#b08800","#735c0f"],orange:["#fff8f2","#ffebda","#ffd1ac","#ffab70","#fb8532","#f66a0a","#e36209","#d15704","#c24e00","#a04100"],red:["#ffeef0","#ffdce0","#fdaeb7","#f97583","#ea4a5a","#d73a49","#cb2431","#b31d28","#9e1c23","#86181d"],purple:["#f5f0ff","#e6dcfd","#d1bcf9","#b392f0","#8a63d2","#6f42c1","#5a32a3","#4c2889","#3a1d6e","#29134e"],pink:["#ffeef8","#fedbf0","#f9b3dd","#f692ce","#ec6cb9","#ea4aaa","#d03592","#b93a86","#99306f","#6d224f"]},text:{primary:"#24292e",secondary:"#586069",tertiary:"#6a737d",placeholder:"#6a737d",disabled:"#959da5",inverse:"#ffffff",link:"#0366d6",danger:"#cb2431",success:"#22863a",warning:"#b08800",white:"#ffffff"},icon:{primary:"#24292e",secondary:"#586069",tertiary:"#959da5",info:"#0366d6",danger:"#d73a49",success:"#22863a",warning:"#b08800"},border:{primary:"#e1e4e8",secondary:"hsla(214,13%,93%,1)",tertiary:"#d1d5da",overlay:"#e1e4e8",inverse:"#ffffff",info:"#0366d6",danger:"#d73a49",success:"#34d058",warning:"#f9c513"},bg:{canvas:"#ffffff",canvasMobile:"#ffffff",canvasInverse:"#24292e",canvasInset:"#f6f8fa",primary:"#ffffff",secondary:"#fafbfc",tertiary:"#f6f8fa",overlay:"#ffffff",backdrop:"rgba(27,31,35,0.5)",info:"#f1f8ff",infoInverse:"#0366d6",danger:"#ffeef0",dangerInverse:"#d73a49",success:"#dcffe4",successInverse:"#28a745",warning:"#fff5b1",warningInverse:"#ffd33d"},state:{hover:{primaryBg:"#0366d6",primaryBorder:"#0366d6",primaryText:"#ffffff",primaryIcon:"#ffffff",secondaryBg:"#f6f8fa",secondaryBorder:"#f6f8fa"},selected:{primaryBg:"#0366d6",primaryBorder:"#0366d6",primaryText:"#ffffff",primaryIcon:"#ffffff"},focus:{border:"#0366d6"}},fade:{fg10:"rgba(27,31,35,0.1)",fg15:"rgba(27,31,35,0.15)",fg30:"rgba(27,31,35,0.3)",fg50:"rgba(27,31,35,0.5)",fg70:"rgba(27,31,35,0.7)",fg85:"rgba(27,31,35,0.85)",black10:"rgba(27,31,35,0.1)",black15:"rgba(27,31,35,0.15)",black30:"rgba(27,31,35,0.3)",black50:"rgba(27,31,35,0.5)",black70:"rgba(27,31,35,0.7)",black85:"rgba(27,31,35,0.85)",white10:"rgba(255,255,255,0.1)",white15:"rgba(255,255,255,0.15)",white30:"rgba(255,255,255,0.3)",white50:"rgba(255,255,255,0.5)",white70:"rgba(255,255,255,0.7)",white85:"rgba(255,255,255,0.85)"},alert:{info:{text:"#24292e",icon:"rgba(4,66,137,0.6)",bg:"#dbedff",border:"rgba(4,66,137,0.2)"},warn:{text:"#24292e",icon:"#b08800",bg:"#fffbdd",border:"rgba(176,136,0,0.2)"},error:{text:"#24292e",icon:"rgba(158,28,35,0.6)",bg:"#ffe3e6",border:"rgba(158,28,35,0.2)"},success:{text:"#24292e",icon:"rgba(23,111,44,0.8)",bg:"#dcffe4",border:"rgba(23,111,44,0.2)"}},autocomplete:{rowBorder:"hsla(214,13%,93%,1)"},blankslate:{icon:"hsla(212,8%,50%,1)"},btn:{text:"#24292e",bg:"#fafbfc",border:"rgba(27,31,35,0.15)",hoverBg:"#f3f4f6",hoverBorder:"rgba(27,31,35,0.15)",activeBg:"hsla(220,14%,93%,1)",activeBorder:"rgba(27,31,35,0.1)",selectedBg:"hsla(220,14%,94%,1)",focusBg:"#fafbfc",focusBorder:"rgba(27,31,35,0.15)",counterBg:"rgba(27,31,35,0.08)",primary:{text:"#ffffff",bg:"#2ea44f",border:"rgba(27,31,35,0.15)",hoverBg:"#2c974b",hoverBorder:"rgba(27,31,35,0.15)",selectedBg:"hsla(137,55%,36%,1)",disabledText:"rgba(255,255,255,0.8)",disabledBg:"#94d3a2",disabledBorder:"rgba(27,31,35,0.1)",focusBg:"#2ea44f",focusBorder:"rgba(27,31,35,0.15)",icon:"rgba(255,255,255,0.8)",counterBg:"rgba(255,255,255,0.2)"},outline:{text:"#0366d6",hoverText:"#ffffff",hoverBg:"#0366d6",hoverBorder:"rgba(27,31,35,0.15)",hoverCounterBg:"rgba(255,255,255,0.2)",selectedText:"#ffffff",selectedBg:"hsla(212,97%,40%,1)",selectedBorder:"rgba(27,31,35,0.15)",disabledText:"rgba(3,102,214,0.5)",disabledBg:"#fafbfc",disabledCounterBg:"rgba(3,102,214,0.05)",focusBorder:"rgba(27,31,35,0.15)",counterBg:"rgba(3,102,214,0.1)"},danger:{text:"#d73a49",hoverText:"#ffffff",hoverBg:"#cb2431",hoverBorder:"rgba(27,31,35,0.15)",hoverCounterBg:"rgba(255,255,255,0.2)",selectedText:"#ffffff",selectedBg:"hsla(354,66%,51%,1)",selectedBorder:"rgba(27,31,35,0.15)",disabledText:"rgba(215,58,73,0.5)",disabledBg:"#fafbfc",disabledCounterBg:"rgba(215,58,73,0.05)",focusBorder:"rgba(27,31,35,0.15)",counterBg:"rgba(215,58,73,0.1)",icon:"#d73a49",hoverIcon:"#ffffff"}},counter:{text:"#24292e",bg:"rgba(209,213,218,0.5)",primary:{text:"#ffffff",bg:"#6a737d"},secondary:{text:"#6a737d",bg:"rgba(209,213,218,0.5)"}},label:{border:"#e1e4e8",primary:{text:"#24292e",border:"#6a737d"},secondary:{text:"#586069",border:"#e1e4e8"},info:{text:"#0366d6",border:"#0366d6"},success:{text:"#22863a",border:"#28a745"},warning:{text:"#735c0f",border:"#b08800"},danger:{text:"#d73a49",border:"#cb2431"},orange:{text:"#c24e00",border:"#f66a0a"}},input:{bg:"#ffffff",contrastBg:"#fafbfc",border:"#e1e4e8",disabledBg:"#f6f8fa",disabledBorder:"#e1e4e8",warningBorder:"#f9c513",errorBorder:"#cb2431",tooltip:{success:{text:"#144620",bg:"#dcffe4",border:"#34d058"},warning:{text:"#735c0f",bg:"#fff5b1",border:"#f9c513"},error:{text:"#86181d",bg:"#ffeef0",border:"#f97583"}}},avatar:{bg:"#ffffff",border:"rgba(0,0,0,0)",stackFade:"#d1d5da",stackFadeMore:"#e1e4e8"},toast:{text:"#24292e",bg:"#ffffff",border:"#e1e4e8",icon:"#ffffff",iconBg:"#0366d6",iconBorder:"rgba(0,0,0,0)",success:{text:"#24292e",border:"#e1e4e8",icon:"#ffffff",iconBg:"#28a745",iconBorder:"rgba(0,0,0,0)"},warning:{text:"#24292e",border:"#e1e4e8",icon:"#24292e",iconBg:"#ffd33d",iconBorder:"rgba(0,0,0,0)"},danger:{text:"#24292e",border:"#e1e4e8",icon:"#ffffff",iconBg:"#d73a49",iconBorder:"rgba(0,0,0,0)"},loading:{text:"#24292e",border:"#e1e4e8",icon:"#ffffff",iconBg:"#586069",iconBorder:"rgba(0,0,0,0)"}},timeline:{text:"#444d56",badgeBg:"#e1e4e8",badgeSuccessBorder:"rgba(0,0,0,0)",targetBadgeBorder:"#2188ff",targetBadgeShadow:"#c8e1ff"},selectMenu:{borderSecondary:"hsla(214,13%,93%,1)",backdropBg:"rgba(27,31,35,0.5)",backdropBorder:"rgba(0,0,0,0)",tapHighlight:"rgba(209,213,218,0.5)",tapFocusBg:"#dbedff"},box:{blueBorder:"#c8e1ff",rowYellowBg:"#fffbdd",rowBlueBg:"#f1f8ff",headerBlueBg:"#f1f8ff",headerBlueBorder:"#c8e1ff",borderInfo:"rgba(3,102,214,0.2)",bgInfo:"#f1f8ff",borderWarning:"rgba(255,211,61,0.4)",bgWarning:"#fffdef"},branchName:{text:"#586069",icon:"hsla(212,30%,74%,1)",bg:"hsla(210,100%,96%,1)",link:{text:"#0366d6",icon:"hsla(212,30%,74%,1)",bg:"hsla(210,100%,96%,1)"}},markdown:{codeBg:"rgba(27,31,35,0.05)",frameBorder:"hsla(213,11%,89%,1)",blockquoteBorder:"hsla(213,11%,89%,1)",tableBorder:"hsla(213,11%,89%,1)",tableTrBorder:"hsla(213,11%,80%,1)"},menu:{headingText:"#24292e",borderActive:"#f9826c",bgActive:"rgba(0,0,0,0)"},sidenav:{selectedBg:"#ffffff",borderActive:"#f9826c"},header:{text:"rgba(255,255,255,0.7)",bg:"#24292e",logo:"#ffffff"},filterItem:{barBg:"hsla(210,29%,95%,1)"},hiddenTextExpander:{bg:"hsla(213,11%,89%,1)",bgHover:"hsla(213,11%,80%,1)"},dragAndDrop:{border:"hsla(213,11%,79%,1)"},uploadEnabled:{border:"hsla(213,11%,89%,1)",borderFocused:"hsla(212,100%,64%,1)"},previewableCommentForm:{border:"hsla(213,11%,79%,1)"},underlinenav:{border:"rgba(209,213,218,0)",borderHover:"#d1d5da",borderActive:"#f9826c",text:"#24292e",textHover:"#24292e",textActive:"#24292e",icon:"#959da5",iconHover:"#959da5",iconActive:"#24292e",counterText:"#24292e",counterBg:"rgba(209,213,218,0.5)"},verifiedBadge:{text:"#22863a",bg:"#ffffff",border:"#e1e4e8"},socialCount:{bg:"#ffffff"},tooltip:{text:"#ffffff",bg:"#24292e"},headerSearch:{bg:"#24292e",border:"#444d56"},searchKeyword:{hl:"#fffbdd"},diffstat:{neutralBg:"#d1d5da",neutralBorder:"#d1d5da",deletionBg:"#d73a49",deletionBorder:"#d73a49",additionBg:"#28a745",additionBorder:"#28a745"},mktg:{success:"rgba(46,188,79,1)",info:"rgba(16,116,231,1)",bgShadeGradient:{top:"rgba(27,31,35,0.065)",bottom:"rgba(27,31,35,0)"},btn:{bg:{top:"hsla(228,82%,66%,1)",bottom:"#4969ed"},bgOverlay:{top:"hsla(228,74%,59%,1)",bottom:"#3355e0"},text:"#ffffff",primary:{bg:{top:"hsla(137,56%,46%,1)",bottom:"#2ea44f"},bgOverlay:{top:"hsla(134,60%,38%,1)",bottom:"#22863a"},text:"#ffffff"},enterprise:{bg:{top:"hsla(249,100%,72%,1)",bottom:"#6f57ff"},bgOverlay:{top:"hsla(248,65%,63%,1)",bottom:"#614eda"},text:"#ffffff"},outline:{text:"#4969ed",border:"rgba(73,105,237,0.3)",hover:{text:"#3355e0",border:"rgba(51,85,224,0.5)"},focus:{border:"#4969ed",borderInset:"rgba(73,105,237,0.5)"}},dark:{text:"#ffffff",border:"rgba(255,255,255,0.3)",hover:{text:"#ffffff",border:"rgba(255,255,255,0.5)"},focus:{border:"#ffffff",borderInset:"rgba(255,255,255,0.5)"}}}},filesExplorerIcon:"#79b8ff",hlAuthorBg:"#f1f8ff",hlAuthorBorder:"#c8e1ff",logoSubdued:"#d1d5da",discussionBorder:"hsla(135,28%,72%,1)",discussionBgSuccess:"#28a745",actionsWorkflowTableStickyBg:"rgba(255,255,255,0.95)",repoLanguageColorBorder:"rgba(27,31,35,0.1)",codeSelectionBg:"#c8e1ff",highlight:{text:"#442c12",bg:"#fff0bb"},blob:{lineHighlightBg:"#fffbdd",lineHighlightBorder:"rgba(0,0,0,0)"},diff:{addition:{text:"#22863a",bg:"hsla(136,100%,95%,1)",border:"#34d058"},deletion:{text:"#cb2431",bg:"#ffeef0",border:"#d73a49"},change:{text:"#b08800",bg:"#fff5b1",border:"#f9c513"}},diffBlob:{numText:"rgba(27,31,35,0.3)",numHoverText:"rgba(27,31,35,0.6)",addition:{numText:"rgba(27,31,35,0.3)",numHoverText:"rgba(27,31,35,0.6)",numBg:"hsla(134,100%,90%,1)",lineBg:"hsla(136,100%,95%,1)",wordBg:"hsla(134,73%,81%,1)"},deletion:{numText:"rgba(27,31,35,0.3)",numHoverText:"rgba(27,31,35,0.6)",numBg:"#ffdce0",lineBg:"#ffeef0",wordBg:"hsla(353,95%,86%,1)"},hunk:{text:"rgba(27,31,35,0.7)",numBg:"#dbedff",lineBg:"#f1f8ff"},emptyBlockBg:"#fafbfc",selectedLineHighlightBg:"rgba(255,223,93,0.2)",selectedLineHighlightBorder:"#ffd33d",expander:{icon:"#586069",hoverIcon:"#ffffff",hoverBg:"#0366d6"},commentButton:{icon:"#ffffff",bg:"#0366d6",gradientBg:"hsla(212,97%,48%,1)"}},globalNav:{logo:"#ffffff",bg:"#24292e",text:"#ffffff",icon:"#ffffff",inputBg:"#fafbfc",inputBorder:"#fafbfc",inputIcon:"#d1d5da",inputPlaceholder:"#959da5"},footerInvertocat:{octicon:"#d1d5da",octiconHover:"#6a737d"},prState:{draft:{text:"#ffffff",bg:"#6a737d",border:"rgba(0,0,0,0)"},open:{text:"#ffffff",bg:"#28a745",border:"rgba(0,0,0,0)"},merged:{text:"#ffffff",bg:"#6f42c1",border:"rgba(0,0,0,0)"},closed:{text:"#ffffff",bg:"#d73a49",border:"rgba(0,0,0,0)"}},topicTag:{text:"#0366d6",bg:"#f1f8ff",hoverBg:"hsla(210,100%,93%,1)",activeBg:"hsla(210,100%,95%,1)"},mergeBox:{successIconBg:"#28a745",successIconText:"#ffffff",successIconBorder:"rgba(0,0,0,0)",successIndicatorBg:"#28a745",successIndicatorBorder:"rgba(0,0,0,0)",mergedIconBg:"#6f42c1",mergedIconText:"#ffffff",mergedIconBorder:"rgba(0,0,0,0)",mergedBoxBorder:"#6f42c1",neutralIconBg:"#6a737d",neutralIconText:"#ffffff",neutralIconBorder:"rgba(0,0,0,0)",neutralIndicatorBg:"#6a737d",neutralIndicatorBorder:"rgba(0,0,0,0)",warningIconBg:"#dbab09",warningIconText:"#ffffff",warningIconBorder:"rgba(0,0,0,0)",warningBoxBorder:"#ffd33d",warningMergeHighlight:"rgba(0,0,0,0)",errorIconBg:"#d73a49",errorIconText:"#ffffff",errorIconBorder:"rgba(0,0,0,0)",errorIndicatorBg:"#d73a49",errorIndicatorBorder:"rgba(0,0,0,0)"},project:{cardBg:"#ffffff",headerBg:"#24292e",sidebarBg:"#ffffff",gradientIn:"#ffffff",gradientOut:"rgba(255,255,255,0)"},checks:{bg:"#24292e",textPrimary:"#fafbfc",textSecondary:"#959da5",textLink:"#79b8ff",btnIcon:"#d1d5da",btnHoverIcon:"#fafbfc",btnHoverBg:"rgba(255,255,255,0.125)",inputText:"#f6f8fa",inputPlaceholderText:"#959da5",inputFocusText:"#959da5",inputBg:"#2f363d",donutError:"#cb2431",donutPending:"#dbab09",donutSuccess:"#28a745",donutNeutral:"#959da5",dropdownText:"#d1d5da",dropdownBg:"#2f363d",dropdownBorder:"#444d56",dropdownHoverText:"#fafbfc",dropdownHoverBg:"#444d56",dropdownBtnHoverText:"#fafbfc",dropdownBtnHoverBg:"#2f363d",scrollbarThumbBg:"#586069",headerLabelText:"#e1e4e8",headerLabelOpenText:"#fafbfc",headerBorder:"#2f363d",headerIcon:"#959da5",lineText:"#e1e4e8",lineNumText:"rgba(149,157,165,0.75)",lineTimestampText:"#959da5",lineHoverBg:"#2f363d",lineSelectedBg:"rgba(33,136,255,0.15)",lineSelectedNumText:"#79b8ff",lineDtFmText:"#24292e",lineDtFmBg:"#ffd33d",gateBg:"rgba(249,197,19,0.15)",gateText:"#e1e4e8",gateWaitingText:"#d1d5da",stepHeaderOpenBg:"#2f363d",stepErrorText:"#f97583",stepWarningText:"#ffea7f",loglineText:"#959da5",loglineNumText:"rgba(149,157,165,0.75)",loglineDebugText:"#b392f0",loglineErrorText:"#e1e4e8",loglineErrorNumText:"#f97583",loglineErrorBg:"rgba(203,36,49,0.15)",loglineWarningText:"#e1e4e8",loglineWarningNumText:"#ffea7f",loglineWarningBg:"rgba(249,197,19,0.15)",loglineCommandText:"#79b8ff",loglineSectionText:"#85e89d",ansi:{black:"#24292e",blackBright:"#2f363d",white:"#e1e4e8",whiteBright:"#e1e4e8",gray:"#959da5",red:"#f97583",redBright:"#fdaeb7",green:"#85e89d",greenBright:"#bef5cb",yellow:"#ffea7f",yellowBright:"#fff5b1",blue:"#79b8ff",blueBright:"#c8e1ff",magenta:"#b392f0",magentaBright:"#d1bcf9",cyan:"#76e3ea",cyanBright:"#b3f0ff"}},introShelf:{gradientLeft:"#f1f8ff",gradientRight:"#dcffe4",gradientIn:"#ffffff",gradientOut:"rgba(255,255,255,0)"},marketingIcon:{primary:"#2188ff",secondary:"#79b8ff"},prettylights:{syntax:{comment:"#6a737d",constant:"#005cc5",entity:"#6f42c1",storageModifierImport:"#24292e",entityTag:"#22863a",keyword:"#d73a49",string:"#032f62",variable:"#e36209",brackethighlighterUnmatched:"#b31d28",invalidIllegalText:"#fafbfc",invalidIllegalBg:"#b31d28",carriageReturnText:"#fafbfc",carriageReturnBg:"#d73a49",stringRegexp:"#22863a",markupList:"#735c0f",markupHeading:"#005cc5",markupItalic:"#24292e",markupBold:"#24292e",markupDeletedText:"#b31d28",markupDeletedBg:"#ffeef0",markupInsertedText:"#22863a",markupInsertedBg:"#f0fff4",markupChangedText:"#e36209",markupChangedBg:"#ffebda",markupIgnoredText:"#f6f8fa",markupIgnoredBg:"#005cc5",metaDiffRange:"#6f42c1",brackethighlighterAngle:"#586069",sublimelinterGutterMark:"#959da5",constantOtherReferenceLink:"#032f62"}},codemirror:{text:"#24292e",bg:"#ffffff",guttersBg:"#ffffff",guttermarkerText:"#ffffff",guttermarkerSubtleText:"#d1d5da",linenumberText:"#959da5",cursor:"#24292e",selectionBg:"#c8e1ff",activelineBg:"#fafbfc",matchingbracketText:"#24292e",linesBg:"#ffffff",syntax:{comment:"#6a737d",constant:"#005cc5",entity:"#6f42c1",keyword:"#d73a49",storage:"#d73a49",string:"#032f62",support:"#005cc5",variable:"#e36209"}},ansi:{black:"#24292e",blackBright:"#586069",white:"#6a737d",whiteBright:"#959da5",gray:"#6a737d",red:"#d73a49",redBright:"#cb2431",green:"#22863a",greenBright:"#28a745",yellow:"#b08800",yellowBright:"#dbab09",blue:"#0366d6",blueBright:"#2188ff",magenta:"#6f42c1",magentaBright:"#8a63d2",cyan:"#1b7c83",cyanBright:"#3192aa"}},shadows:{shadow:{small:"0 1px 0 rgba(27,31,35,0.04)",medium:"0 3px 6px rgba(149,157,165,0.15)",large:"0 8px 24px rgba(149,157,165,0.2)",extraLarge:"0 12px 48px rgba(149,157,165,0.3)",highlight:"inset 0 1px 0 rgba(255,255,255,0.25)",inset:"inset 0 1px 0 rgba(225,228,232,0.2)"},state:{focus:{shadow:"0 0 0 3px rgba(3,102,214,0.3)"}},autocomplete:{shadow:"0 3px 6px rgba(149,157,165,0.15)"},btn:{shadow:"0 1px 0 rgba(27,31,35,0.04)",insetShadow:"inset 0 1px 0 rgba(255,255,255,0.25)",focusShadow:"0 0 0 3px rgba(3,102,214,0.3)",shadowActive:"inset 0 0.15em 0.3em rgba(27,31,35,0.15)",shadowInputFocus:"0 0 0 0.2em rgba(3,102,214,0.3)",primary:{shadow:"0 1px 0 rgba(27,31,35,0.1)",insetShadow:"inset 0 1px 0 rgba(255,255,255,0.03)",selectedShadow:"inset 0 1px 0 rgba(20,70,32,0.2)",focusShadow:"0 0 0 3px rgba(46,164,79,0.4)"},outline:{hoverShadow:"0 1px 0 rgba(27,31,35,0.1)",hoverInsetShadow:"inset 0 1px 0 rgba(255,255,255,0.03)",selectedShadow:"inset 0 1px 0 rgba(5,38,76,0.2)",focusShadow:"0 0 0 3px rgba(0,92,197,0.4)"},danger:{hoverShadow:"0 1px 0 rgba(27,31,35,0.1)",hoverInsetShadow:"inset 0 1px 0 rgba(255,255,255,0.03)",selectedShadow:"inset 0 1px 0 rgba(134,24,29,0.2)",focusShadow:"0 0 0 3px rgba(203,36,49,0.4)"}},dropdown:{shadow:"0 8px 24px rgba(149,157,165,0.2)"},overlay:{shadow:"0 1px 3px rgba(27,31,35,0.12), 0 8px 24px rgba(68,77,86,0.12)"},input:{shadow:"inset 0 1px 2px rgba(27,31,35,0.075)"},avatar:{childShadow:"-2px -2px 0 rgba(255,255,255,0.8)"},toast:{shadow:"0 8px 24px rgba(149,157,165,0.2)"},selectMenu:{shadow:"0 0 18px rgba(27,31,35,0.4)"}}},dark:{colors:{auto:{black:"#f0f6fc",white:"#010409",gray:["#0d1117","#161b22","#21262d","#30363d","#484f58","#6e7681","#8b949e","#b1bac4","#c9d1d9","#f0f6fc"],blue:["#051d4d","#0c2d6b","#0d419d","#1158c7","#1f6feb","#388bfd","#58a6ff","#79c0ff","#a5d6ff","#cae8ff"],green:["#04260f","#033a16","#0f5323","#196c2e","#238636","#2ea043","#3fb950","#56d364","#7ee787","#aff5b4"],yellow:["#341a00","#4b2900","#693e00","#845306","#9e6a03","#bb8009","#d29922","#e3b341","#f2cc60","#f8e3a1"],orange:["#3d1300","#5a1e02","#762d0a","#9b4215","#bd561d","#db6d28","#f0883e","#ffa657","#ffc680","#ffdfb6"],red:["#490202","#67060c","#8e1519","#b62324","#da3633","#f85149","#ff7b72","#ffa198","#ffc1ba","#ffdcd7"],purple:["#271052","#3c1e70","#553098","#6e40c9","#8957e5","#a371f7","#bc8cff","#d2a8ff","#e2c5ff","#eddeff"],pink:["#42062a","#5e103e","#7d2457","#9e3670","#bf4b8a","#db61a2","#f778ba","#ff9bce","#ffbedd","#ffdaec"]},text:{primary:"#c9d1d9",secondary:"#8b949e",tertiary:"#8b949e",placeholder:"#484f58",disabled:"#484f58",inverse:"#0d1117",link:"#58a6ff",danger:"#f85149",success:"#56d364",warning:"#e3b341",white:"#f0f6fc"},icon:{primary:"#c9d1d9",secondary:"#6e7681",tertiary:"#484f58",info:"#58a6ff",danger:"#f85149",success:"#56d364",warning:"#e3b341"},border:{primary:"#30363d",secondary:"#21262d",tertiary:"#6e7681",overlay:"#30363d",inverse:"#f0f6fc",info:"rgba(56,139,253,0.4)",danger:"rgba(248,81,73,0.4)",success:"rgba(63,185,80,0.4)",warning:"rgba(187,128,9,0.4)"},bg:{canvas:"#0d1117",canvasMobile:"#010409",canvasInverse:"#f0f6fc",canvasInset:"hsla(216,28%,5%,1)",primary:"#0d1117",secondary:"#0d1117",tertiary:"#161b22",overlay:"rgba(28,33,40,1)",backdrop:"rgba(1,4,9,0.8)",info:"rgba(56,139,253,0.1)",infoInverse:"#388bfd",danger:"rgba(248,81,73,0.1)",dangerInverse:"#da3633",success:"rgba(46,160,67,0.1)",successInverse:"#2ea043",warning:"rgba(187,128,9,0.1)",warningInverse:"#bb8009"},state:{hover:{primaryBg:"#1f6feb",primaryBorder:"#388bfd",primaryText:"#f0f6fc",primaryIcon:"#f0f6fc",secondaryBg:"#161b22",secondaryBorder:"#161b22"},selected:{primaryBg:"#1f6feb",primaryBorder:"#388bfd",primaryText:"#f0f6fc",primaryIcon:"#f0f6fc"},focus:{border:"#388bfd"}},fade:{fg10:"rgba(240,246,252,0.1)",fg15:"rgba(240,246,252,0.15)",fg30:"rgba(240,246,252,0.3)",fg50:"rgba(240,246,252,0.5)",fg70:"rgba(240,246,252,0.7)",fg85:"rgba(240,246,252,0.85)",black10:"rgba(1,4,9,0.1)",black15:"rgba(1,4,9,0.15)",black30:"rgba(1,4,9,0.3)",black50:"rgba(1,4,9,0.5)",black70:"rgba(1,4,9,0.7)",black85:"rgba(1,4,9,0.85)",white10:"rgba(240,246,252,0.1)",white15:"rgba(240,246,252,0.15)",white30:"rgba(240,246,252,0.3)",white50:"rgba(240,246,252,0.5)",white70:"rgba(240,246,252,0.7)",white85:"rgba(240,246,252,0.85)"},alert:{info:{text:"#79c0ff",icon:"#79c0ff",bg:"rgba(56,139,253,0.1)",border:"rgba(56,139,253,0.4)"},warn:{text:"#e3b341",icon:"#e3b341",bg:"rgba(187,128,9,0.1)",border:"rgba(187,128,9,0.4)"},error:{text:"#ff7b72",icon:"#ff7b72",bg:"rgba(248,81,73,0.1)",border:"rgba(248,81,73,0.4)"},success:{text:"#56d364",icon:"#56d364",bg:"rgba(46,160,67,0.1)",border:"rgba(46,160,67,0.4)"}},autocomplete:{rowBorder:"#30363d"},blankslate:{icon:"hsla(214,10%,36%,1)"},btn:{text:"#c9d1d9",bg:"#21262d",border:"#30363d",hoverBg:"#30363d",hoverBorder:"#8b949e",activeBg:"hsla(212,12%,18%,1)",activeBorder:"#6e7681",selectedBg:"#161b22",focusBg:"#21262d",focusBorder:"#8b949e",primary:{text:"#ffffff",bg:"#238636",border:"#2ea043",hoverBg:"#2ea043",hoverBorder:"#3fb950",selectedBg:"#238636",disabledText:"rgba(240,246,252,0.5)",disabledBg:"rgba(35,134,54,0.6)",disabledBorder:"rgba(0,0,0,0)",focusBg:"#238636",focusBorder:"#3fb950",icon:"#f0f6fc",counterBg:"rgba(240,246,252,0.2)"},outline:{text:"#58a6ff",hoverText:"#58a6ff",hoverBg:"#30363d",hoverBorder:"#58a6ff",hoverCounterBg:"rgba(240,246,252,0.2)",selectedText:"#f0f6fc",selectedBg:"#0d419d",selectedBorder:"rgba(240,246,252,0.1)",disabledText:"rgba(88,166,255,0.5)",disabledBg:"#0d1117",disabledCounterBg:"rgba(31,111,235,0.05)",focusBorder:"#58a6ff",counterBg:"rgba(31,111,235,0.1)"},danger:{text:"#f85149",hoverText:"#ffffff",hoverBg:"#da3633",hoverBorder:"#f85149",hoverCounterBg:"rgba(255,255,255,0.2)",selectedText:"#ffffff",selectedBg:"#b62324",selectedBorder:"rgba(240,246,252,0.1)",disabledText:"rgba(248,81,73,0.5)",disabledBg:"#0d1117",disabledCounterBg:"rgba(218,54,51,0.05)",focusBorder:"#f85149",counterBg:"rgba(218,54,51,0.1)",icon:"#f85149",hoverIcon:"#f0f6fc"},counterBg:"#30363d"},counter:{text:"#c9d1d9",bg:"#30363d",primary:{text:"#c9d1d9",bg:"#6e7681"},secondary:{text:"#8b949e",bg:"rgba(139,148,158,0.2)"}},label:{border:"#30363d",primary:{text:"#b1bac4",border:"#6e7681"},secondary:{text:"#8b949e",border:"#30363d"},info:{text:"#388bfd",border:"rgba(56,139,253,0.4)"},success:{text:"#3fb950",border:"rgba(46,160,67,0.4)"},warning:{text:"#e3b341",border:"rgba(242,211,91,0.4)"},danger:{text:"#f85149",border:"rgba(248,81,73,0.4)"},orange:{text:"#db6d28",border:"rgba(219,109,40,0.4)"}},input:{bg:"#0d1117",contrastBg:"rgba(1,4,9,0.5)",border:"#21262d",disabledBg:"#161b22",disabledBorder:"#30363d",warningBorder:"#d29922",errorBorder:"#f85149",tooltip:{success:{text:"#56d364",bg:"rgba(16,31,27,1)",border:"rgba(28,83,43,1)"},warning:{text:"#e3b341",bg:"rgba(30,28,22,1)",border:"rgba(93,68,17,1)"},error:{text:"#ff7b72",bg:"rgba(37,23,28,1)",border:"rgba(121,46,46,1)"}}},avatar:{bg:"rgba(240,246,252,0.1)",border:"rgba(240,246,252,0.1)",stackFade:"#30363d",stackFadeMore:"#21262d"},toast:{text:"#c9d1d9",bg:"#30363d",border:"#6e7681",icon:"#f0f6fc",iconBg:"#1f6feb",iconBorder:"#388bfd",success:{text:"#c9d1d9",border:"#6e7681",icon:"#f0f6fc",iconBg:"#2ea043",iconBorder:"#3fb950"},warning:{text:"#c9d1d9",border:"#6e7681",icon:"#f0f6fc",iconBg:"#bb8009",iconBorder:"#d29922"},danger:{text:"#c9d1d9",border:"#6e7681",icon:"#f0f6fc",iconBg:"#da3633",iconBorder:"#f85149"},loading:{text:"#c9d1d9",border:"#6e7681",icon:"#f0f6fc",iconBg:"#6e7681",iconBorder:"#8b949e"}},timeline:{text:"#b1bac4",badgeBg:"#0d1117",badgeSuccessBorder:"#2ea043",targetBadgeBorder:"#1f6feb",targetBadgeShadow:"#0d419d"},selectMenu:{borderSecondary:"#30363d",backdropBg:"rgba(1,4,9,0.5)",backdropBorder:"#484f58",tapHighlight:"rgba(48,54,61,0.5)",tapFocusBg:"#0c2d6b"},box:{blueBorder:"#0d419d",rowYellowBg:"rgba(235,196,64,0.1)",rowBlueBg:"rgba(121,192,255,0.1)",headerBlueBg:"#0d1117",headerBlueBorder:"#30363d",borderInfo:"rgba(56,139,253,0.4)",bgInfo:"rgba(56,139,253,0.1)",borderWarning:"rgba(187,128,9,0.4)",bgWarning:"rgba(187,128,9,0.1)"},branchName:{text:"#c9d1d9",icon:"#b1bac4",bg:"rgba(88,166,255,0.1)",link:{text:"#58a6ff",icon:"#58a6ff",bg:"rgba(88,166,255,0.1)"}},markdown:{codeBg:"rgba(240,246,252,0.15)",frameBorder:"hsla(212,12%,26%,1)",blockquoteBorder:"hsla(212,12%,26%,1)",tableBorder:"hsla(212,12%,26%,1)",tableTrBorder:"hsla(212,12%,17%,1)"},menu:{headingText:"#8b949e",borderActive:"#f78166",bgActive:"#161b22"},sidenav:{selectedBg:"#21262d",borderActive:"#f78166"},header:{text:"rgba(240,246,252,0.7)",bg:"#161b22",logo:"#f0f6fc"},filterItem:{barBg:"hsla(215,21%,9%,1)"},hiddenTextExpander:{bg:"#21262d",bgHover:"#30363d"},dragAndDrop:{border:"hsla(212,12%,16%,1)"},uploadEnabled:{border:"hsla(212,12%,26%,1)",borderFocused:"hsla(216,84%,60%,1)"},previewableCommentForm:{border:"hsla(212,12%,16%,1)"},underlinenav:{border:"rgba(48,54,61,0)",borderHover:"#30363d",borderActive:"#f78166",text:"#8b949e",textHover:"#c9d1d9",textActive:"#c9d1d9",icon:"#6e7681",iconHover:"#c9d1d9",iconActive:"#c9d1d9",counterText:"#8b949e",counterBg:"rgba(139,148,158,0.2)"},verifiedBadge:{text:"#3fb950",bg:"rgba(63,185,80,0.1)",border:"rgba(63,185,80,0.4)"},socialCount:{bg:"#21262d"},tooltip:{text:"#f0f6fc",bg:"#6e7681"},headerSearch:{bg:"#0d1117",border:"#21262d"},searchKeyword:{hl:"rgba(187,128,9,0.4)"},diffstat:{neutralBg:"#30363d",neutralBorder:"rgba(240,246,252,0.1)",deletionBg:"#da3633",deletionBorder:"#f85149",additionBg:"#238636",additionBorder:"#2ea043"},mktg:{success:"rgba(41,147,61,1)",info:"rgba(42,123,243,1)",bgShadeGradient:{top:"rgba(1,4,9,0.065)",bottom:"rgba(1,4,9,0)"},btn:{bg:{top:"hsla(228,82%,66%,1)",bottom:"#4969ed"},bgOverlay:{top:"hsla(228,74%,59%,1)",bottom:"#3355e0"},text:"#f0f6fc",primary:{bg:{top:"hsla(137,56%,46%,1)",bottom:"#2ea44f"},bgOverlay:{top:"hsla(134,60%,38%,1)",bottom:"#22863a"},text:"#f0f6fc"},enterprise:{bg:{top:"hsla(249,100%,72%,1)",bottom:"#6f57ff"},bgOverlay:{top:"hsla(248,65%,63%,1)",bottom:"#614eda"},text:"#f0f6fc"},outline:{text:"#f0f6fc",border:"rgba(240,246,252,0.3)",hover:{text:"#f0f6fc",border:"rgba(240,246,252,0.5)"},focus:{border:"#f0f6fc",borderInset:"rgba(240,246,252,0.5)"}},dark:{text:"#f0f6fc",border:"rgba(240,246,252,0.3)",hover:{text:"#f0f6fc",border:"rgba(240,246,252,0.5)"},focus:{border:"#f0f6fc",borderInset:"rgba(240,246,252,0.5)"}}}},filesExplorerIcon:"#6e7681",hlAuthorBg:"#051d4d",hlAuthorBorder:"#0d419d",logoSubdued:"#30363d",discussionBorder:"hsla(135,22%,26%,1)",discussionBgSuccess:"rgba(46,160,67,0.1)",actionsWorkflowTableStickyBg:"rgba(13,17,23,0.95)",repoLanguageColorBorder:"rgba(240,246,252,0.2)",codeSelectionBg:"rgba(121,192,255,0.3)",highlight:{text:"#ffd467",bg:"rgba(204,143,44,0.38)"},blob:{lineHighlightBg:"rgba(210,153,34,0.15)",lineHighlightBorder:"#e3b341"},diff:{addition:{text:"#56d364",bg:"rgba(46,160,67,0.2)",border:"#196c2e"},deletion:{text:"#f85149",bg:"rgba(218,54,51,0.2)",border:"#b62324"},change:{text:"#e3b341",bg:"#341a00",border:"#9e6a03"}},diffBlob:{numText:"rgba(240,246,252,0.3)",numHoverText:"rgba(240,246,252,0.6)",addition:{numText:"#3fb950",numHoverText:"#7ee787",numBg:"rgba(46,160,67,0.1)",lineBg:"rgba(46,160,67,0.2)",wordBg:"rgba(46,160,67,0.55)"},deletion:{numText:"#f85149",numHoverText:"#ffa198",numBg:"rgba(218,54,51,0.1)",lineBg:"rgba(218,54,51,0.2)",wordBg:"rgba(218,54,51,0.5)"},hunk:{text:"#8b949e",numBg:"rgba(88,166,255,0.15)",lineBg:"rgba(88,166,255,0.1)"},emptyBlockBg:"#161b22",selectedLineHighlightBg:"rgba(187,128,9,0.1)",selectedLineHighlightBorder:"#bb8009",expander:{icon:"#8b949e",hoverIcon:"#f0f6fc",hoverBg:"#1f6feb"},commentButton:{icon:"#f0f6fc",bg:"#1f6feb",gradientBg:"hsla(216,84%,57%,1)"}},globalNav:{logo:"#f0f6fc",bg:"#161b22",text:"#c9d1d9",icon:"#c9d1d9",inputBg:"#0d1117",inputBorder:"#21262d",inputIcon:"#21262d",inputPlaceholder:"#484f58"},footerInvertocat:{octicon:"#30363d",octiconHover:"#6e7681"},prState:{draft:{text:"#8b949e",bg:"rgba(139,148,158,0.1)",border:"rgba(139,148,158,0.4)"},open:{text:"#3fb950",bg:"rgba(63,185,80,0.1)",border:"rgba(63,185,80,0.4)"},merged:{text:"#a371f7",bg:"rgba(188,140,255,0.1)",border:"rgba(188,140,255,0.4)"},closed:{text:"#f85149",bg:"rgba(218,54,51,0.1)",border:"rgba(218,54,51,0.4)"}},topicTag:{text:"#58a6ff",bg:"rgba(56,139,253,0.1)",hoverBg:"rgba(56,139,253,0.2)",activeBg:"rgba(56,139,253,0.15)"},mergeBox:{successIconBg:"rgba(46,160,67,0.1)",successIconText:"#3fb950",successIconBorder:"rgba(46,160,67,0.4)",successIndicatorBg:"#238636",successIndicatorBorder:"#2ea043",mergedIconBg:"rgba(188,140,255,0.1)",mergedIconText:"#a371f7",mergedIconBorder:"rgba(188,140,255,0.4)",mergedBoxBorder:"rgba(188,140,255,0.4)",neutralIconBg:"rgba(201,209,217,0.1)",neutralIconText:"#8b949e",neutralIconBorder:"rgba(201,209,217,0.4)",neutralIndicatorBg:"#484f58",neutralIndicatorBorder:"#6e7681",warningIconBg:"rgba(187,128,9,0.1)",warningIconText:"#e3b341",warningIconBorder:"rgba(187,128,9,0.4)",warningBoxBorder:"rgba(187,128,9,0.4)",warningMergeHighlight:"rgba(187,128,9,0.1)",errorIconBg:"rgba(248,81,73,0.1)",errorIconText:"#f85149",errorIconBorder:"rgba(248,81,73,0.4)",errorIndicatorBg:"#da3633",errorIndicatorBorder:"#f85149"},project:{cardBg:"#161b22",headerBg:"#0d1117",sidebarBg:"#161b22",gradientIn:"#161b22",gradientOut:"rgba(22,27,34,0)"},checks:{bg:"hsla(216,28%,5%,1)",textPrimary:"#c9d1d9",textSecondary:"#8b949e",textLink:"#58a6ff",btnIcon:"#6e7681",btnHoverIcon:"#c9d1d9",btnHoverBg:"#30363d",inputText:"#8b949e",inputPlaceholderText:"#484f58",inputFocusText:"#c9d1d9",inputBg:"#0d1117",donutError:"#f85149",donutPending:"#d29922",donutSuccess:"#2ea043",donutNeutral:"#8b949e",dropdownText:"#c9d1d9",dropdownBg:"rgba(28,33,40,1)",dropdownBorder:"#30363d",dropdownHoverText:"#f0f6fc",dropdownHoverBg:"#1f6feb",dropdownBtnHoverText:"#f0f6fc",dropdownBtnHoverBg:"#161b22",scrollbarThumbBg:"#30363d",headerLabelText:"#8b949e",headerLabelOpenText:"#c9d1d9",headerBorder:"#21262d",headerIcon:"#6e7681",lineText:"#8b949e",lineNumText:"#8b949e",lineTimestampText:"#8b949e",lineHoverBg:"#161b22",lineSelectedBg:"rgba(56,139,253,0.1)",lineSelectedNumText:"#58a6ff",lineDtFmText:"#0d1117",lineDtFmBg:"#d29922",gateBg:"rgba(132,83,6,0.15)",gateText:"#8b949e",gateWaitingText:"#e3b341",stepHeaderOpenBg:"#161b22",stepErrorText:"#f85149",stepWarningText:"#e3b341",loglineText:"#6e7681",loglineNumText:"#8b949e",loglineDebugText:"#bc8cff",loglineErrorText:"#8b949e",loglineErrorNumText:"#8b949e",loglineErrorBg:"rgba(248,81,73,0.1)",loglineWarningText:"#8b949e",loglineWarningNumText:"#e3b341",loglineWarningBg:"rgba(187,128,9,0.1)",loglineCommandText:"#58a6ff",loglineSectionText:"#56d364",ansi:{black:"#0d1117",blackBright:"#161b22",white:"#b1bac4",whiteBright:"#b1bac4",gray:"#6e7681",red:"#ff7b72",redBright:"#ffa198",green:"#3fb950",greenBright:"#56d364",yellow:"#d29922",yellowBright:"#e3b341",blue:"#58a6ff",blueBright:"#79c0ff",magenta:"#bc8cff",magentaBright:"#d2a8ff",cyan:"#76e3ea",cyanBright:"#b3f0ff"}},introShelf:{gradientLeft:"rgba(56,139,253,0.1)",gradientRight:"rgba(46,160,67,0.1)",gradientIn:"#0d1117",gradientOut:"rgba(13,17,23,0)"},marketingIcon:{primary:"#79c0ff",secondary:"#1f6feb"},prettylights:{syntax:{comment:"#8b949e",constant:"#79c0ff",entity:"#d2a8ff",storageModifierImport:"#c9d1d9",entityTag:"#7ee787",keyword:"#ff7b72",string:"#a5d6ff",variable:"#ffa657",brackethighlighterUnmatched:"#f85149",invalidIllegalText:"#f0f6fc",invalidIllegalBg:"#8e1519",carriageReturnText:"#f0f6fc",carriageReturnBg:"#b62324",stringRegexp:"#7ee787",markupList:"#f2cc60",markupHeading:"#1f6feb",markupItalic:"#c9d1d9",markupBold:"#c9d1d9",markupDeletedText:"#ffdcd7",markupDeletedBg:"#67060c",markupInsertedText:"#aff5b4",markupInsertedBg:"#033a16",markupChangedText:"#ffdfb6",markupChangedBg:"#5a1e02",markupIgnoredText:"#c9d1d9",markupIgnoredBg:"#1158c7",metaDiffRange:"#d2a8ff",brackethighlighterAngle:"#8b949e",sublimelinterGutterMark:"#484f58",constantOtherReferenceLink:"#a5d6ff"}},codemirror:{text:"#c9d1d9",bg:"#0d1117",guttersBg:"#0d1117",guttermarkerText:"#0d1117",guttermarkerSubtleText:"#6e7681",linenumberText:"#8b949e",cursor:"#f0f6fc",selectionBg:"rgba(121,192,255,0.3)",activelineBg:"#161b22",matchingbracketText:"#c9d1d9",linesBg:"#0d1117",syntax:{comment:"#8b949e",constant:"#79c0ff",entity:"#d2a8ff",keyword:"#ff7b72",storage:"#ff7b72",string:"#a5d6ff",support:"#79c0ff",variable:"#ffa657"}},ansi:{black:"#484f58",blackBright:"#6e7681",white:"#b1bac4",whiteBright:"#f0f6fc",gray:"#6e7681",red:"#ff7b72",redBright:"#ffa198",green:"#3fb950",greenBright:"#56d364",yellow:"#d29922",yellowBright:"#e3b341",blue:"#58a6ff",blueBright:"#79c0ff",magenta:"#bc8cff",magentaBright:"#d2a8ff",cyan:"#39c5cf",cyanBright:"#56d4dd"}},shadows:{shadow:{small:"0 0 transparent",medium:"0 3px 6px #010409",large:"0 8px 24px #010409",extraLarge:"0 12px 48px #010409",highlight:"0 0 transparent",inset:"0 0 transparent"},state:{focus:{shadow:"0 0 0 3px #0c2d6b"}},autocomplete:{shadow:"0 16px 32px rgba(1,4,9,0.85)"},btn:{shadow:"0 0 transparent",insetShadow:"0 0 transparent",focusShadow:"0 0 0 3px rgba(139,148,158,0.3)",shadowActive:"inset 0 0.15em 0.3em rgba(1,4,9,0.15)",shadowInputFocus:"0 0 0 0.2em rgba(31,111,235,0.3)",primary:{shadow:"0 0 transparent",insetShadow:"0 0 transparent",selectedShadow:"0 0 transparent",focusShadow:"0 0 0 3px rgba(46,164,79,0.4)"},outline:{hoverShadow:"0 1px 0 rgba(1,4,9,0.1)",hoverInsetShadow:"inset 0 1px 0 rgba(240,246,252,0.03)",selectedShadow:"0 0 transparent",focusShadow:"0 0 0 3px rgba(17,88,199,0.4)"},danger:{hoverShadow:"0 0 transparent",hoverInsetShadow:"0 0 transparent",selectedShadow:"0 0 transparent",focusShadow:"0 0 0 3px rgba(182,35,36,0.4)"}},dropdown:{shadow:"0 16px 32px rgba(1,4,9,0.85)"},overlay:{shadow:"0 0 0 1px #30363d, 0 16px 32px rgba(1,4,9,0.85)"},input:{shadow:"0 0 transparent"},avatar:{childShadow:"-2px -2px 0 #0d1117"},toast:{shadow:"0 8px 24px #010409"},selectMenu:{shadow:"0 0 18px rgba(1,4,9,0.4)"},checks:{inputShadow:"0 0 0 1px #21262d"}}},dark_dimmed:{colors:{auto:{black:"#cdd9e5",white:"#1c2128",gray:["#22272e","#2d333b","#373e47","#444c56","#545d68","#636e7b","#768390","#909dab","#adbac7","#cdd9e5"],blue:["#0f2d5c","#143d79","#1b4b91","#255ab2","#316dca","#4184e4","#539bf5","#6cb6ff","#96d0ff","#c6e6ff"],green:["#113417","#1b4721","#245829","#2b6a30","#347d39","#46954a","#57ab5a","#6bc46d","#8ddb8c","#b4f1b4"],yellow:["#452700","#593600","#6c4400","#805400","#966600","#ae7c14","#c69026","#daaa3f","#eac55f","#fbe090"],orange:["#4d210c","#682d0f","#7f3913","#94471b","#ae5622","#cc6b2c","#e0823d","#f69d50","#ffbc6f","#ffddb0"],red:["#78191b","#78191b","#922323","#ad2e2c","#c93c37","#e5534b","#f47067","#ff938a","#ffb8b0","#ffd8d3"],purple:["#352160","#472c82","#5936a2","#6b44bc","#8256d0","#986ee2","#b083f0","#dcbdfb","#dcbdfb","#eedcff"],pink:["#551639","#69264a","#7e325a","#983b6e","#ae4c82","#c96198","#e275ad","#fc8dc7","#ffb3d8","#ffd7eb"]},text:{primary:"#adbac7",secondary:"#768390",tertiary:"#768390",placeholder:"#545d68",disabled:"#545d68",inverse:"#22272e",link:"#539bf5",danger:"#e5534b",success:"#6bc46d",warning:"#daaa3f",white:"#cdd9e5"},icon:{primary:"#adbac7",secondary:"#636e7b",tertiary:"#545d68",info:"#539bf5",danger:"#e5534b",success:"#6bc46d",warning:"#daaa3f"},border:{primary:"#444c56",secondary:"#373e47",tertiary:"#636e7b",overlay:"#444c56",inverse:"#cdd9e5",info:"rgba(65,132,228,0.4)",danger:"rgba(229,83,75,0.4)",success:"rgba(87,171,90,0.4)",warning:"rgba(174,124,20,0.4)"},bg:{canvas:"#22272e",canvasMobile:"#1c2128",canvasInverse:"#cdd9e5",canvasInset:"hsla(215,15%,14%,1)",primary:"#22272e",secondary:"#22272e",tertiary:"#2d333b",overlay:"rgba(50,57,65,1)",backdrop:"rgba(28,33,40,0.8)",info:"rgba(65,132,228,0.1)",infoInverse:"#4184e4",danger:"rgba(229,83,75,0.1)",dangerInverse:"#c93c37",success:"rgba(70,149,74,0.1)",successInverse:"#46954a",warning:"rgba(174,124,20,0.1)",warningInverse:"#ae7c14"},state:{hover:{primaryBg:"#316dca",primaryBorder:"#4184e4",primaryText:"#cdd9e5",primaryIcon:"#cdd9e5",secondaryBg:"#2d333b",secondaryBorder:"#2d333b"},selected:{primaryBg:"#316dca",primaryBorder:"#4184e4",primaryText:"#cdd9e5",primaryIcon:"#cdd9e5"},focus:{border:"#4184e4"}},fade:{fg10:"rgba(205,217,229,0.1)",fg15:"rgba(205,217,229,0.15)",fg30:"rgba(205,217,229,0.3)",fg50:"rgba(205,217,229,0.5)",fg70:"rgba(205,217,229,0.7)",fg85:"rgba(205,217,229,0.85)",black10:"rgba(28,33,40,0.1)",black15:"rgba(28,33,40,0.15)",black30:"rgba(28,33,40,0.3)",black50:"rgba(28,33,40,0.5)",black70:"rgba(28,33,40,0.7)",black85:"rgba(28,33,40,0.85)",white10:"rgba(205,217,229,0.1)",white15:"rgba(205,217,229,0.15)",white30:"rgba(205,217,229,0.3)",white50:"rgba(205,217,229,0.5)",white70:"rgba(205,217,229,0.7)",white85:"rgba(205,217,229,0.85)"},alert:{info:{text:"#6cb6ff",icon:"#6cb6ff",bg:"rgba(65,132,228,0.1)",border:"rgba(65,132,228,0.4)"},warn:{text:"#daaa3f",icon:"#daaa3f",bg:"rgba(174,124,20,0.1)",border:"rgba(174,124,20,0.4)"},error:{text:"#f47067",icon:"#f47067",bg:"rgba(229,83,75,0.1)",border:"rgba(229,83,75,0.4)"},success:{text:"#6bc46d",icon:"#6bc46d",bg:"rgba(70,149,74,0.1)",border:"rgba(70,149,74,0.4)"}},autocomplete:{rowBorder:"#444c56"},blankslate:{icon:"hsla(213,11%,42%,1)"},btn:{text:"#adbac7",bg:"#373e47",border:"#444c56",hoverBg:"#444c56",hoverBorder:"#768390",activeBg:"hsla(213,12%,27%,1)",activeBorder:"#636e7b",selectedBg:"#2d333b",focusBg:"#373e47",focusBorder:"#768390",primary:{text:"#ffffff",bg:"#347d39",border:"#46954a",hoverBg:"#46954a",hoverBorder:"#57ab5a",selectedBg:"#347d39",disabledText:"rgba(205,217,229,0.5)",disabledBg:"rgba(52,125,57,0.6)",disabledBorder:"rgba(0,0,0,0)",focusBg:"#347d39",focusBorder:"#57ab5a",icon:"#cdd9e5",counterBg:"rgba(205,217,229,0.2)"},outline:{text:"#539bf5",hoverText:"#539bf5",hoverBg:"#444c56",hoverBorder:"#539bf5",hoverCounterBg:"rgba(205,217,229,0.2)",selectedText:"#cdd9e5",selectedBg:"#1b4b91",selectedBorder:"rgba(205,217,229,0.1)",disabledText:"rgba(83,155,245,0.5)",disabledBg:"#22272e",disabledCounterBg:"rgba(49,109,202,0.05)",focusBorder:"#539bf5",counterBg:"rgba(49,109,202,0.1)"},danger:{text:"#e5534b",hoverText:"#ffffff",hoverBg:"#c93c37",hoverBorder:"#e5534b",hoverCounterBg:"rgba(255,255,255,0.2)",selectedText:"#ffffff",selectedBg:"#ad2e2c",selectedBorder:"rgba(205,217,229,0.1)",disabledText:"rgba(229,83,75,0.5)",disabledBg:"#22272e",disabledCounterBg:"rgba(201,60,55,0.05)",focusBorder:"#e5534b",counterBg:"rgba(201,60,55,0.1)",icon:"#e5534b",hoverIcon:"#cdd9e5"},counterBg:"#444c56"},counter:{text:"#adbac7",bg:"#444c56",primary:{text:"#adbac7",bg:"#636e7b"},secondary:{text:"#768390",bg:"rgba(118,131,144,0.2)"}},label:{border:"#444c56",primary:{text:"#909dab",border:"#636e7b"},secondary:{text:"#768390",border:"#444c56"},info:{text:"#4184e4",border:"rgba(65,132,228,0.4)"},success:{text:"#57ab5a",border:"rgba(70,149,74,0.4)"},warning:{text:"#daaa3f",border:"rgba(242,211,91,0.4)"},danger:{text:"#e5534b",border:"rgba(229,83,75,0.4)"},orange:{text:"#cc6b2c",border:"rgba(204,107,44,0.4)"}},input:{bg:"#22272e",contrastBg:"rgba(28,33,40,0.5)",border:"#373e47",disabledBg:"#2d333b",disabledBorder:"#444c56",warningBorder:"#c69026",errorBorder:"#e5534b",tooltip:{success:{text:"#6bc46d",bg:"rgba(38,50,49,1)",border:"rgba(51,90,59,1)"},warning:{text:"#daaa3f",bg:"rgba(48,48,43,1)",border:"rgba(98,78,34,1)"},error:{text:"#f47067",bg:"rgba(53,43,49,1)",border:"rgba(124,59,59,1)"}}},avatar:{bg:"rgba(205,217,229,0.1)",border:"rgba(205,217,229,0.1)",stackFade:"#444c56",stackFadeMore:"#373e47"},toast:{text:"#adbac7",bg:"#444c56",border:"#636e7b",icon:"#cdd9e5",iconBg:"#316dca",iconBorder:"#4184e4",success:{text:"#adbac7",border:"#636e7b",icon:"#cdd9e5",iconBg:"#46954a",iconBorder:"#57ab5a"},warning:{text:"#adbac7",border:"#636e7b",icon:"#cdd9e5",iconBg:"#ae7c14",iconBorder:"#c69026"},danger:{text:"#adbac7",border:"#636e7b",icon:"#cdd9e5",iconBg:"#c93c37",iconBorder:"#e5534b"},loading:{text:"#adbac7",border:"#636e7b",icon:"#cdd9e5",iconBg:"#636e7b",iconBorder:"#768390"}},timeline:{text:"#909dab",badgeBg:"#22272e",badgeSuccessBorder:"#46954a",targetBadgeBorder:"#316dca",targetBadgeShadow:"#1b4b91"},selectMenu:{borderSecondary:"#444c56",backdropBg:"rgba(28,33,40,0.5)",backdropBorder:"#545d68",tapHighlight:"rgba(68,76,86,0.5)",tapFocusBg:"#143d79"},box:{blueBorder:"#1b4b91",rowYellowBg:"rgba(235,196,64,0.1)",rowBlueBg:"rgba(108,182,255,0.1)",headerBlueBg:"#22272e",headerBlueBorder:"#444c56",borderInfo:"rgba(65,132,228,0.4)",bgInfo:"rgba(65,132,228,0.1)",borderWarning:"rgba(174,124,20,0.4)",bgWarning:"rgba(174,124,20,0.1)"},branchName:{text:"#adbac7",icon:"#909dab",bg:"rgba(83,155,245,0.1)",link:{text:"#539bf5",icon:"#539bf5",bg:"rgba(83,155,245,0.1)"}},markdown:{codeBg:"rgba(205,217,229,0.15)",frameBorder:"hsla(213,12%,35%,1)",blockquoteBorder:"hsla(213,12%,35%,1)",tableBorder:"hsla(213,12%,35%,1)",tableTrBorder:"hsla(213,12%,26%,1)"},menu:{headingText:"#768390",borderActive:"#f78166",bgActive:"#2d333b"},sidenav:{selectedBg:"#373e47",borderActive:"#f78166"},header:{text:"rgba(205,217,229,0.7)",bg:"#2d333b",logo:"#cdd9e5"},filterItem:{barBg:"hsla(214,13%,18%,1)"},hiddenTextExpander:{bg:"#373e47",bgHover:"#444c56"},dragAndDrop:{border:"hsla(213,12%,25%,1)"},uploadEnabled:{border:"hsla(213,12%,35%,1)",borderFocused:"hsla(216,61%,57%,1)"},previewableCommentForm:{border:"hsla(213,12%,25%,1)"},underlinenav:{border:"rgba(68,76,86,0)",borderHover:"#444c56",borderActive:"#f78166",text:"#768390",textHover:"#adbac7",textActive:"#adbac7",icon:"#636e7b",iconHover:"#adbac7",iconActive:"#adbac7",counterText:"#768390",counterBg:"rgba(118,131,144,0.2)"},verifiedBadge:{text:"#57ab5a",bg:"rgba(87,171,90,0.1)",border:"rgba(87,171,90,0.4)"},socialCount:{bg:"#373e47"},tooltip:{text:"#cdd9e5",bg:"#636e7b"},headerSearch:{bg:"#22272e",border:"#373e47"},searchKeyword:{hl:"rgba(174,124,20,0.4)"},diffstat:{neutralBg:"#444c56",neutralBorder:"rgba(205,217,229,0.1)",deletionBg:"#c93c37",deletionBorder:"#e5534b",additionBg:"#347d39",additionBorder:"#46954a"},mktg:{success:"rgba(61,137,66,1)",info:"rgba(56,119,213,1)",bgShadeGradient:{top:"rgba(28,33,40,0.065)",bottom:"rgba(28,33,40,0)"},btn:{bg:{top:"hsla(228,82%,66%,1)",bottom:"#4969ed"},bgOverlay:{top:"hsla(228,74%,59%,1)",bottom:"#3355e0"},text:"#cdd9e5",primary:{bg:{top:"hsla(137,56%,46%,1)",bottom:"#2ea44f"},bgOverlay:{top:"hsla(134,60%,38%,1)",bottom:"#22863a"},text:"#cdd9e5"},enterprise:{bg:{top:"hsla(249,100%,72%,1)",bottom:"#6f57ff"},bgOverlay:{top:"hsla(248,65%,63%,1)",bottom:"#614eda"},text:"#cdd9e5"},outline:{text:"#cdd9e5",border:"rgba(205,217,229,0.3)",hover:{text:"#cdd9e5",border:"rgba(205,217,229,0.5)"},focus:{border:"#cdd9e5",borderInset:"rgba(205,217,229,0.5)"}},dark:{text:"#cdd9e5",border:"rgba(205,217,229,0.3)",hover:{text:"#cdd9e5",border:"rgba(205,217,229,0.5)"},focus:{border:"#cdd9e5",borderInset:"rgba(205,217,229,0.5)"}}}},filesExplorerIcon:"#636e7b",hlAuthorBg:"#0f2d5c",hlAuthorBorder:"#1b4b91",logoSubdued:"#444c56",discussionBorder:"hsla(125,2%,29%,1)",discussionBgSuccess:"rgba(70,149,74,0.1)",actionsWorkflowTableStickyBg:"rgba(34,39,46,0.95)",repoLanguageColorBorder:"rgba(205,217,229,0.2)",codeSelectionBg:"rgba(108,182,255,0.3)",highlight:{text:"#ffd467",bg:"rgba(204,143,44,0.38)"},blob:{lineHighlightBg:"rgba(198,144,38,0.15)",lineHighlightBorder:"#daaa3f"},diff:{addition:{text:"#6bc46d",bg:"rgba(70,149,74,0.2)",border:"#2b6a30"},deletion:{text:"#e5534b",bg:"rgba(201,60,55,0.2)",border:"#ad2e2c"},change:{text:"#daaa3f",bg:"#452700",border:"#966600"}},diffBlob:{numText:"rgba(205,217,229,0.3)",numHoverText:"rgba(205,217,229,0.6)",addition:{numText:"#57ab5a",numHoverText:"#8ddb8c",numBg:"rgba(70,149,74,0.1)",lineBg:"rgba(70,149,74,0.2)",wordBg:"rgba(70,149,74,0.55)"},deletion:{numText:"#e5534b",numHoverText:"#ff938a",numBg:"rgba(201,60,55,0.1)",lineBg:"rgba(201,60,55,0.2)",wordBg:"rgba(201,60,55,0.5)"},hunk:{text:"#768390",numBg:"rgba(83,155,245,0.15)",lineBg:"rgba(83,155,245,0.1)"},emptyBlockBg:"#2d333b",selectedLineHighlightBg:"rgba(174,124,20,0.1)",selectedLineHighlightBorder:"#ae7c14",expander:{icon:"#768390",hoverIcon:"#cdd9e5",hoverBg:"#316dca"},commentButton:{icon:"#cdd9e5",bg:"#316dca",gradientBg:"hsla(216,61%,54%,1)"}},globalNav:{logo:"#cdd9e5",bg:"#2d333b",text:"#adbac7",icon:"#adbac7",inputBg:"#22272e",inputBorder:"#373e47",inputIcon:"#373e47",inputPlaceholder:"#545d68"},footerInvertocat:{octicon:"#444c56",octiconHover:"#636e7b"},prState:{draft:{text:"#768390",bg:"rgba(118,131,144,0.1)",border:"rgba(118,131,144,0.4)"},open:{text:"#57ab5a",bg:"rgba(87,171,90,0.1)",border:"rgba(87,171,90,0.4)"},merged:{text:"#986ee2",bg:"rgba(176,131,240,0.1)",border:"rgba(176,131,240,0.4)"},closed:{text:"#e5534b",bg:"rgba(201,60,55,0.1)",border:"rgba(201,60,55,0.4)"}},topicTag:{text:"#539bf5",bg:"rgba(65,132,228,0.1)",hoverBg:"rgba(65,132,228,0.2)",activeBg:"rgba(65,132,228,0.15)"},mergeBox:{successIconBg:"rgba(70,149,74,0.1)",successIconText:"#57ab5a",successIconBorder:"rgba(70,149,74,0.4)",successIndicatorBg:"#347d39",successIndicatorBorder:"#46954a",mergedIconBg:"rgba(176,131,240,0.1)",mergedIconText:"#986ee2",mergedIconBorder:"rgba(176,131,240,0.4)",mergedBoxBorder:"rgba(176,131,240,0.4)",neutralIconBg:"rgba(173,186,199,0.1)",neutralIconText:"#768390",neutralIconBorder:"rgba(173,186,199,0.4)",neutralIndicatorBg:"#545d68",neutralIndicatorBorder:"#636e7b",warningIconBg:"rgba(174,124,20,0.1)",warningIconText:"#daaa3f",warningIconBorder:"rgba(174,124,20,0.4)",warningBoxBorder:"rgba(174,124,20,0.4)",warningMergeHighlight:"rgba(174,124,20,0.1)",errorIconBg:"rgba(229,83,75,0.1)",errorIconText:"#e5534b",errorIconBorder:"rgba(229,83,75,0.4)",errorIndicatorBg:"#c93c37",errorIndicatorBorder:"#e5534b"},project:{cardBg:"#2d333b",headerBg:"#22272e",sidebarBg:"#2d333b",gradientIn:"#2d333b",gradientOut:"rgba(45,51,59,0)"},checks:{bg:"hsla(215,15%,14%,1)",textPrimary:"#adbac7",textSecondary:"#768390",textLink:"#539bf5",btnIcon:"#636e7b",btnHoverIcon:"#adbac7",btnHoverBg:"#444c56",inputText:"#768390",inputPlaceholderText:"#545d68",inputFocusText:"#adbac7",inputBg:"#22272e",donutError:"#e5534b",donutPending:"#c69026",donutSuccess:"#46954a",donutNeutral:"#768390",dropdownText:"#adbac7",dropdownBg:"rgba(50,57,65,1)",dropdownBorder:"#444c56",dropdownHoverText:"#cdd9e5",dropdownHoverBg:"#316dca",dropdownBtnHoverText:"#cdd9e5",dropdownBtnHoverBg:"#2d333b",scrollbarThumbBg:"#444c56",headerLabelText:"#768390",headerLabelOpenText:"#adbac7",headerBorder:"#373e47",headerIcon:"#636e7b",lineText:"#768390",lineNumText:"#768390",lineTimestampText:"#768390",lineHoverBg:"#2d333b",lineSelectedBg:"rgba(65,132,228,0.1)",lineSelectedNumText:"#539bf5",lineDtFmText:"#22272e",lineDtFmBg:"#c69026",gateBg:"rgba(128,84,0,0.15)",gateText:"#768390",gateWaitingText:"#daaa3f",stepHeaderOpenBg:"#2d333b",stepErrorText:"#e5534b",stepWarningText:"#daaa3f",loglineText:"#636e7b",loglineNumText:"#768390",loglineDebugText:"#b083f0",loglineErrorText:"#768390",loglineErrorNumText:"#768390",loglineErrorBg:"rgba(229,83,75,0.1)",loglineWarningText:"#768390",loglineWarningNumText:"#daaa3f",loglineWarningBg:"rgba(174,124,20,0.1)",loglineCommandText:"#539bf5",loglineSectionText:"#6bc46d",ansi:{black:"#22272e",blackBright:"#2d333b",white:"#909dab",whiteBright:"#909dab",gray:"#636e7b",red:"#f47067",redBright:"#ff938a",green:"#57ab5a",greenBright:"#6bc46d",yellow:"#c69026",yellowBright:"#daaa3f",blue:"#539bf5",blueBright:"#6cb6ff",magenta:"#b083f0",magentaBright:"#dcbdfb",cyan:"#76e3ea",cyanBright:"#b3f0ff"}},introShelf:{gradientLeft:"rgba(65,132,228,0.1)",gradientRight:"rgba(70,149,74,0.1)",gradientIn:"#22272e",gradientOut:"rgba(34,39,46,0)"},marketingIcon:{primary:"#6cb6ff",secondary:"#316dca"},prettylights:{syntax:{comment:"#768390",constant:"#6cb6ff",entity:"#dcbdfb",storageModifierImport:"#adbac7",entityTag:"#8ddb8c",keyword:"#f47067",string:"#96d0ff",variable:"#f69d50",brackethighlighterUnmatched:"#e5534b",invalidIllegalText:"#cdd9e5",invalidIllegalBg:"#922323",carriageReturnText:"#cdd9e5",carriageReturnBg:"#ad2e2c",stringRegexp:"#8ddb8c",markupList:"#eac55f",markupHeading:"#316dca",markupItalic:"#adbac7",markupBold:"#adbac7",markupDeletedText:"#ffd8d3",markupDeletedBg:"#78191b",markupInsertedText:"#b4f1b4",markupInsertedBg:"#1b4721",markupChangedText:"#ffddb0",markupChangedBg:"#682d0f",markupIgnoredText:"#adbac7",markupIgnoredBg:"#255ab2",metaDiffRange:"#dcbdfb",brackethighlighterAngle:"#768390",sublimelinterGutterMark:"#545d68",constantOtherReferenceLink:"#96d0ff"}},codemirror:{text:"#adbac7",bg:"#22272e",guttersBg:"#22272e",guttermarkerText:"#22272e",guttermarkerSubtleText:"#636e7b",linenumberText:"#768390",cursor:"#cdd9e5",selectionBg:"rgba(108,182,255,0.3)",activelineBg:"#2d333b",matchingbracketText:"#adbac7",linesBg:"#22272e",syntax:{comment:"#768390",constant:"#6cb6ff",entity:"#dcbdfb",keyword:"#f47067",storage:"#f47067",string:"#96d0ff",support:"#6cb6ff",variable:"#f69d50"}},ansi:{black:"#545d68",blackBright:"#636e7b",white:"#909dab",whiteBright:"#cdd9e5",gray:"#636e7b",red:"#f47067",redBright:"#ff938a",green:"#57ab5a",greenBright:"#6bc46d",yellow:"#c69026",yellowBright:"#daaa3f",blue:"#539bf5",blueBright:"#6cb6ff",magenta:"#b083f0",magentaBright:"#dcbdfb",cyan:"#39c5cf",cyanBright:"#56d4dd"}},shadows:{shadow:{small:"0 0 transparent",medium:"0 3px 6px #1c2128",large:"0 8px 24px #1c2128",extraLarge:"0 12px 48px #1c2128",highlight:"0 0 transparent",inset:"0 0 transparent"},state:{focus:{shadow:"0 0 0 3px #143d79"}},autocomplete:{shadow:"0 16px 32px rgba(28,33,40,0.85)"},btn:{shadow:"0 0 transparent",insetShadow:"0 0 transparent",focusShadow:"0 0 0 3px rgba(118,131,144,0.3)",shadowActive:"inset 0 0.15em 0.3em rgba(28,33,40,0.15)",shadowInputFocus:"0 0 0 0.2em rgba(49,109,202,0.3)",primary:{shadow:"0 0 transparent",insetShadow:"0 0 transparent",selectedShadow:"0 0 transparent",focusShadow:"0 0 0 3px rgba(46,164,79,0.4)"},outline:{hoverShadow:"0 1px 0 rgba(28,33,40,0.1)",hoverInsetShadow:"inset 0 1px 0 rgba(205,217,229,0.03)",selectedShadow:"0 0 transparent",focusShadow:"0 0 0 3px rgba(37,90,178,0.4)"},danger:{hoverShadow:"0 0 transparent",hoverInsetShadow:"0 0 transparent",selectedShadow:"0 0 transparent",focusShadow:"0 0 0 3px rgba(173,46,44,0.4)"}},dropdown:{shadow:"0 16px 32px rgba(28,33,40,0.85)"},overlay:{shadow:"0 0 0 1px #444c56, 0 16px 32px rgba(28,33,40,0.85)"},input:{shadow:"0 0 transparent"},avatar:{childShadow:"-2px -2px 0 #22272e"},toast:{shadow:"0 8px 24px #1c2128"},selectMenu:{shadow:"0 0 18px rgba(28,33,40,0.4)"},checks:{inputShadow:"0 0 0 1px #373e47"}}}}},v=Object.getOwnPropertySymbols,y=Object.prototype.hasOwnProperty,w=Object.prototype.propertyIsEnumerable;function B(e){if(null==e)throw new TypeError("Object.assign cannot be called with null or undefined");return Object(e)}var $=function(){try{if(!Object.assign)return!1;var e=new String("abc");if(e[5]="de","5"===Object.getOwnPropertyNames(e)[0])return!1;for(var r={},t=0;t<10;t++)r["_"+String.fromCharCode(t)]=t;if("0123456789"!==Object.getOwnPropertyNames(r).map((function(e){return r[e]})).join(""))return!1;var o={};return"abcdefghijklmnopqrst".split("").forEach((function(e){o[e]=e})),"abcdefghijklmnopqrst"===Object.keys(Object.assign({},o)).join("")}catch(e){return!1}}()?Object.assign:function(e,r){for(var t,o,a=B(e),n=1;n<arguments.length;n++){for(var i in t=Object(arguments[n]))y.call(t,i)&&(a[i]=t[i]);if(v){o=v(t);for(var d=0;d<o.length;d++)w.call(t,o[d])&&(a[o[d]]=t[o[d]])}}return a},k=function(e,r){var t=$({},e,r);for(var o in e){var a;e[o]&&"object"==typeof r[o]&&$(t,((a={})[o]=$(e[o],r[o]),a))}return t},S={breakpoints:[40,52,64].map((function(e){return e+"em"}))},T=function(e){return"@media screen and (min-width: "+e+")"},E=function(e,r){return I(r,e,e)},I=function(e,r,t,o,a){for(r=r&&r.split?r.split("."):[r],o=0;o<r.length;o++)e=e?e[r[o]]:a;return e===a?t:e},C=function e(r){var t={},o=function(e){var o,a,n={},i=!1,d=e.theme&&e.theme.disableStyledSystemCache;for(var l in e)if(r[l]){var c=r[l],s=e[l],f=I(e.theme,c.scale,c.defaults);if("object"!=typeof s)$(n,c(s,f,e));else{if(t.breakpoints=!d&&t.breakpoints||I(e.theme,"breakpoints",S.breakpoints),Array.isArray(s)){t.media=!d&&t.media||[null].concat(t.breakpoints.map(T)),n=k(n,N(t.media,c,f,s,e));continue}null!==s&&(n=k(n,A(t.breakpoints,c,f,s,e)),i=!0)}}return i&&(o=n,a={},Object.keys(o).sort((function(e,r){return e.localeCompare(r,void 0,{numeric:!0,sensitivity:"base"})})).forEach((function(e){a[e]=o[e]})),n=a),n};o.config=r,o.propNames=Object.keys(r),o.cache=t;var a=Object.keys(r).filter((function(e){return"config"!==e}));return a.length>1&&a.forEach((function(t){var a;o[t]=e(((a={})[t]=r[t],a))})),o},N=function(e,r,t,o,a){var n={};return o.slice(0,e.length).forEach((function(o,i){var d,l=e[i],c=r(o,t,a);l?$(n,((d={})[l]=$({},n[l],c),d)):$(n,c)})),n},A=function(e,r,t,o,a){var n={};for(var i in o){var d=e[i],l=r(o[i],t,a);if(d){var c,s=T(d);$(n,((c={})[s]=$({},n[s],l),c))}else $(n,l)}return n},z=function(e){var r=e.properties,t=e.property,o=e.scale,a=e.transform,n=void 0===a?E:a,i=e.defaultScale;r=r||[t];var d=function(e,t,o){var a={},i=n(e,t,o);if(null!==i)return r.forEach((function(e){a[e]=i})),a};return d.scale=o,d.defaults=i,d},R=function(e){void 0===e&&(e={});var r={};return Object.keys(e).forEach((function(t){var o=e[t];r[t]=!0!==o?"function"!=typeof o?z(o):o:z({property:t,scale:t})})),C(r)},L=function(){for(var e={},r=arguments.length,t=new Array(r),o=0;o<r;o++)t[o]=arguments[o];t.forEach((function(r){r&&r.config&&$(e,r.config)}));var a=C(e);return a},M=R({width:{property:"width",scale:"sizes",transform:function(e,r){return I(r,e,!function(e){return"number"==typeof e&&!isNaN(e)}(e)||e>1?e:100*e+"%")}},height:{property:"height",scale:"sizes"},minWidth:{property:"minWidth",scale:"sizes"},minHeight:{property:"minHeight",scale:"sizes"},maxWidth:{property:"maxWidth",scale:"sizes"},maxHeight:{property:"maxHeight",scale:"sizes"},size:{properties:["width","height"],scale:"sizes"},overflow:!0,overflowX:!0,overflowY:!0,display:!0,verticalAlign:!0}),O={color:{property:"color",scale:"colors"},backgroundColor:{property:"backgroundColor",scale:"colors"},opacity:!0};O.bg=O.backgroundColor;var H=R(O),P=R({fontFamily:{property:"fontFamily",scale:"fonts"},fontSize:{property:"fontSize",scale:"fontSizes",defaultScale:[12,14,16,20,24,32,48,64,72]},fontWeight:{property:"fontWeight",scale:"fontWeights"},lineHeight:{property:"lineHeight",scale:"lineHeights"},letterSpacing:{property:"letterSpacing",scale:"letterSpacings"},textAlign:!0,fontStyle:!0}),W=R({alignItems:!0,alignContent:!0,justifyItems:!0,justifyContent:!0,flexWrap:!0,flexDirection:!0,flex:!0,flexGrow:!0,flexShrink:!0,flexBasis:!0,justifySelf:!0,alignSelf:!0,order:!0}),D={space:[0,4,8,16,32,64,128,256,512]},j=R({gridGap:{property:"gridGap",scale:"space",defaultScale:D.space},gridColumnGap:{property:"gridColumnGap",scale:"space",defaultScale:D.space},gridRowGap:{property:"gridRowGap",scale:"space",defaultScale:D.space},gridColumn:!0,gridRow:!0,gridAutoFlow:!0,gridAutoColumns:!0,gridAutoRows:!0,gridTemplateColumns:!0,gridTemplateRows:!0,gridTemplateAreas:!0,gridArea:!0}),F={border:{property:"border",scale:"borders"},borderWidth:{property:"borderWidth",scale:"borderWidths"},borderStyle:{property:"borderStyle",scale:"borderStyles"},borderColor:{property:"borderColor",scale:"colors"},borderRadius:{property:"borderRadius",scale:"radii"},borderTop:{property:"borderTop",scale:"borders"},borderTopLeftRadius:{property:"borderTopLeftRadius",scale:"radii"},borderTopRightRadius:{property:"borderTopRightRadius",scale:"radii"},borderRight:{property:"borderRight",scale:"borders"},borderBottom:{property:"borderBottom",scale:"borders"},borderBottomLeftRadius:{property:"borderBottomLeftRadius",scale:"radii"},borderBottomRightRadius:{property:"borderBottomRightRadius",scale:"radii"},borderLeft:{property:"borderLeft",scale:"borders"},borderX:{properties:["borderLeft","borderRight"],scale:"borders"},borderY:{properties:["borderTop","borderBottom"],scale:"borders"},borderTopWidth:{property:"borderTopWidth",scale:"borderWidths"},borderTopColor:{property:"borderTopColor",scale:"colors"},borderTopStyle:{property:"borderTopStyle",scale:"borderStyles"}};F.borderTopLeftRadius={property:"borderTopLeftRadius",scale:"radii"},F.borderTopRightRadius={property:"borderTopRightRadius",scale:"radii"},F.borderBottomWidth={property:"borderBottomWidth",scale:"borderWidths"},F.borderBottomColor={property:"borderBottomColor",scale:"colors"},F.borderBottomStyle={property:"borderBottomStyle",scale:"borderStyles"},F.borderBottomLeftRadius={property:"borderBottomLeftRadius",scale:"radii"},F.borderBottomRightRadius={property:"borderBottomRightRadius",scale:"radii"},F.borderLeftWidth={property:"borderLeftWidth",scale:"borderWidths"},F.borderLeftColor={property:"borderLeftColor",scale:"colors"},F.borderLeftStyle={property:"borderLeftStyle",scale:"borderStyles"},F.borderRightWidth={property:"borderRightWidth",scale:"borderWidths"},F.borderRightColor={property:"borderRightColor",scale:"colors"},F.borderRightStyle={property:"borderRightStyle",scale:"borderStyles"};var _=R(F),U={background:!0,backgroundImage:!0,backgroundSize:!0,backgroundPosition:!0,backgroundRepeat:!0};U.bgImage=U.backgroundImage,U.bgSize=U.backgroundSize,U.bgPosition=U.backgroundPosition,U.bgRepeat=U.backgroundRepeat;var G=R(U),K={space:[0,4,8,16,32,64,128,256,512]},X=R({position:!0,zIndex:{property:"zIndex",scale:"zIndices"},top:{property:"top",scale:"space",defaultScale:K.space},right:{property:"right",scale:"space",defaultScale:K.space},bottom:{property:"bottom",scale:"space",defaultScale:K.space},left:{property:"left",scale:"space",defaultScale:K.space}}),V={space:[0,4,8,16,32,64,128,256,512]},Y=function(e){return"number"==typeof e&&!isNaN(e)},q=function(e,r){if(!Y(e))return I(r,e,e);var t=e<0,o=Math.abs(e),a=I(r,o,o);return Y(a)?a*(t?-1:1):t?"-"+a:a},J={};J.margin={margin:{property:"margin",scale:"space",transform:q,defaultScale:V.space},marginTop:{property:"marginTop",scale:"space",transform:q,defaultScale:V.space},marginRight:{property:"marginRight",scale:"space",transform:q,defaultScale:V.space},marginBottom:{property:"marginBottom",scale:"space",transform:q,defaultScale:V.space},marginLeft:{property:"marginLeft",scale:"space",transform:q,defaultScale:V.space},marginX:{properties:["marginLeft","marginRight"],scale:"space",transform:q,defaultScale:V.space},marginY:{properties:["marginTop","marginBottom"],scale:"space",transform:q,defaultScale:V.space}},J.margin.m=J.margin.margin,J.margin.mt=J.margin.marginTop,J.margin.mr=J.margin.marginRight,J.margin.mb=J.margin.marginBottom,J.margin.ml=J.margin.marginLeft,J.margin.mx=J.margin.marginX,J.margin.my=J.margin.marginY,J.padding={padding:{property:"padding",scale:"space",defaultScale:V.space},paddingTop:{property:"paddingTop",scale:"space",defaultScale:V.space},paddingRight:{property:"paddingRight",scale:"space",defaultScale:V.space},paddingBottom:{property:"paddingBottom",scale:"space",defaultScale:V.space},paddingLeft:{property:"paddingLeft",scale:"space",defaultScale:V.space},paddingX:{properties:["paddingLeft","paddingRight"],scale:"space",defaultScale:V.space},paddingY:{properties:["paddingTop","paddingBottom"],scale:"space",defaultScale:V.space}},J.padding.p=J.padding.padding,J.padding.pt=J.padding.paddingTop,J.padding.pr=J.padding.paddingRight,J.padding.pb=J.padding.paddingBottom,J.padding.pl=J.padding.paddingLeft,J.padding.px=J.padding.paddingX,J.padding.py=J.padding.paddingY;var Z=R(J.margin),Q=R(J.padding),ee=L(Z,Q),re=R({boxShadow:{property:"boxShadow",scale:"shadows"},textShadow:{property:"textShadow",scale:"shadows"}});function te(){return(te=Object.assign||function(e){for(var r=1;r<arguments.length;r++){var t=arguments[r];for(var o in t)Object.prototype.hasOwnProperty.call(t,o)&&(e[o]=t[o])}return e}).apply(this,arguments)}var oe=function(e,r,t,o,a){for(r=r&&r.split?r.split("."):[r],o=0;o<r.length;o++)e=e?e[r[o]]:a;return e===a?t:e},ae=[40,52,64].map((function(e){return e+"em"})),ne={space:[0,4,8,16,32,64,128,256,512],fontSizes:[12,14,16,20,24,32,48,64,72]},ie={bg:"backgroundColor",m:"margin",mt:"marginTop",mr:"marginRight",mb:"marginBottom",ml:"marginLeft",mx:"marginX",my:"marginY",p:"padding",pt:"paddingTop",pr:"paddingRight",pb:"paddingBottom",pl:"paddingLeft",px:"paddingX",py:"paddingY"},de={marginX:["marginLeft","marginRight"],marginY:["marginTop","marginBottom"],paddingX:["paddingLeft","paddingRight"],paddingY:["paddingTop","paddingBottom"],size:["width","height"]},le={color:"colors",backgroundColor:"colors",borderColor:"colors",margin:"space",marginTop:"space",marginRight:"space",marginBottom:"space",marginLeft:"space",marginX:"space",marginY:"space",padding:"space",paddingTop:"space",paddingRight:"space",paddingBottom:"space",paddingLeft:"space",paddingX:"space",paddingY:"space",top:"space",right:"space",bottom:"space",left:"space",gridGap:"space",gridColumnGap:"space",gridRowGap:"space",gap:"space",columnGap:"space",rowGap:"space",fontFamily:"fonts",fontSize:"fontSizes",fontWeight:"fontWeights",lineHeight:"lineHeights",letterSpacing:"letterSpacings",border:"borders",borderTop:"borders",borderRight:"borders",borderBottom:"borders",borderLeft:"borders",borderWidth:"borderWidths",borderStyle:"borderStyles",borderRadius:"radii",borderTopRightRadius:"radii",borderTopLeftRadius:"radii",borderBottomRightRadius:"radii",borderBottomLeftRadius:"radii",borderTopWidth:"borderWidths",borderTopColor:"colors",borderTopStyle:"borderStyles",borderBottomWidth:"borderWidths",borderBottomColor:"colors",borderBottomStyle:"borderStyles",borderLeftWidth:"borderWidths",borderLeftColor:"colors",borderLeftStyle:"borderStyles",borderRightWidth:"borderWidths",borderRightColor:"colors",borderRightStyle:"borderStyles",outlineColor:"colors",boxShadow:"shadows",textShadow:"shadows",zIndex:"zIndices",width:"sizes",minWidth:"sizes",maxWidth:"sizes",height:"sizes",minHeight:"sizes",maxHeight:"sizes",flexBasis:"sizes",size:"sizes",fill:"colors",stroke:"colors"},ce=function(e,r){if("number"!=typeof r||r>=0)return oe(e,r,r);var t=Math.abs(r),o=oe(e,t,t);return"string"==typeof o?"-"+o:-1*o},se=["margin","marginTop","marginRight","marginBottom","marginLeft","marginX","marginY","top","bottom","left","right"].reduce((function(e,r){var t;return te({},e,((t={})[r]=ce,t))}),{}),fe=function e(r){return function(t){void 0===t&&(t={});var o=te({},ne,{},t.theme||t),a={},n=function(e){return function(r){var t={},o=oe(r,"breakpoints",ae),a=[null].concat(o.map((function(e){return"@media screen and (min-width: "+e+")"})));for(var n in e){var i="function"==typeof e[n]?e[n](r):e[n];if(null!=i)if(Array.isArray(i))for(var d=0;d<i.slice(0,a.length).length;d++){var l=a[d];l?(t[l]=t[l]||{},null!=i[d]&&(t[l][n]=i[d])):t[n]=i[d]}else t[n]=i}return t}}("function"==typeof r?r(o):r)(o);for(var i in n){var d=n[i],l="function"==typeof d?d(o):d;if("variant"!==i)if(l&&"object"==typeof l)a[i]=e(l)(o);else{var c=oe(ie,i,i),s=oe(le,c),f=oe(o,s,oe(o,c,{})),b=oe(se,c,oe)(f,l,l);if(de[c])for(var g=de[c],p=0;p<g.length;p++)a[g[p]]=b;else a[c]=b}else a=te({},a,{},e(oe(o,l))(o))}return a}},be=function(e){var r,t,o=e.scale,a=e.prop,n=void 0===a?"variant":a,i=e.variants,d=void 0===i?{}:i,l=e.key;(t=Object.keys(d).length?function(e,r,t){return fe(I(r,e,null))(t.theme)}:function(e,r){return I(r,e,null)}).scale=o||l,t.defaults=d;var c=((r={})[n]=t,r);return C(c)},ge=be({key:"buttons"}),pe=be({key:"textStyles",prop:"textStyle"}),ue=be({key:"colorStyles",prop:"colors"}),me=M.width,he=M.height,xe=M.minWidth,ve=M.minHeight,ye=M.maxWidth,we=M.maxHeight,Be=M.size,$e=M.verticalAlign,ke=M.display,Se=M.overflow,Te=M.overflowX,Ee=M.overflowY,Ie=H.opacity,Ce=P.fontSize,Ne=P.fontFamily,Ae=P.fontWeight,ze=P.lineHeight,Re=P.textAlign,Le=P.fontStyle,Me=P.letterSpacing,Oe=W.alignItems,He=W.alignContent,Pe=W.justifyItems,We=W.justifyContent,De=W.flexWrap,je=W.flexDirection,Fe=W.flex,_e=W.flexGrow,Ue=W.flexShrink,Ge=W.flexBasis,Ke=W.justifySelf,Xe=W.alignSelf,Ve=W.order,Ye=j.gridGap,qe=j.gridColumnGap,Je=j.gridRowGap,Ze=j.gridColumn,Qe=j.gridRow,er=j.gridAutoFlow,rr=j.gridAutoColumns,tr=j.gridAutoRows,or=j.gridTemplateColumns,ar=j.gridTemplateRows,nr=j.gridTemplateAreas,ir=j.gridArea,dr=_.borderWidth,lr=_.borderStyle,cr=_.borderColor,sr=_.borderTop,fr=_.borderRight,br=_.borderBottom,gr=_.borderLeft,pr=_.borderRadius,ur=G.backgroundImage,mr=G.backgroundSize,hr=G.backgroundPosition,xr=G.backgroundRepeat,vr=X.zIndex,yr=X.top,wr=X.right,Br=X.bottom,$r=X.left,kr=function(e){var r=e.prop,t=e.cssProperty,o=e.alias,a=e.key,n=e.transformValue,i=e.scale,d=e.properties,l={};return l[r]=z({properties:d,property:t||r,scale:a,defaultScale:i,transform:n}),o&&(l[o]=l[r]),C(l)},Sr=Object.freeze({__proto__:null,width:me,height:he,minWidth:xe,minHeight:ve,maxWidth:ye,maxHeight:we,size:Be,verticalAlign:$e,display:ke,overflow:Se,overflowX:Te,overflowY:Ee,opacity:Ie,fontSize:Ce,fontFamily:Ne,fontWeight:Ae,lineHeight:ze,textAlign:Re,fontStyle:Le,letterSpacing:Me,alignItems:Oe,alignContent:He,justifyItems:Pe,justifyContent:We,flexWrap:De,flexDirection:je,flex:Fe,flexGrow:_e,flexShrink:Ue,flexBasis:Ge,justifySelf:Ke,alignSelf:Xe,order:Ve,gridGap:Ye,gridColumnGap:qe,gridRowGap:Je,gridColumn:Ze,gridRow:Qe,gridAutoFlow:er,gridAutoColumns:rr,gridAutoRows:tr,gridTemplateColumns:or,gridTemplateRows:ar,gridTemplateAreas:nr,gridArea:ir,borderWidth:dr,borderStyle:lr,borderColor:cr,borderTop:sr,borderRight:fr,borderBottom:br,borderLeft:gr,borderRadius:pr,backgroundImage:ur,backgroundSize:mr,backgroundPosition:hr,backgroundRepeat:xr,zIndex:vr,top:yr,right:wr,bottom:Br,left:$r,style:kr,get:I,createParser:C,createStyleFunction:z,compose:L,system:R,margin:Z,padding:Q,space:ee,color:H,layout:M,typography:P,flexbox:W,border:_,background:G,position:X,grid:j,shadow:re,boxShadow:re,textShadow:re,variant:be,buttonStyle:ge,textStyle:pe,colorStyle:ue,borders:_});const{get:Tr,compose:Er,system:Ir}=Sr,Cr=e=>{return r=e,void 0===(t=Tr(x,e))&&(t=null),function(e){return I(e.theme,r,t)};var r,t},Nr=Er(ee,H,ke),Ar=Er(P,Ir({whiteSpace:{property:"whiteSpace"}})),zr=Er(_,re),Rr=M,Lr=X,Mr=W;function Or(){return(Or=Object.assign||function(e){for(var r=1;r<arguments.length;r++){var t=arguments[r];for(var o in t)Object.prototype.hasOwnProperty.call(t,o)&&(e[o]=t[o])}return e}).apply(this,arguments)}const Hr=f`
  * { box-sizing: border-box; }
  body { margin: 0; }
  table { border-collapse: collapse; }
  
  [role="button"]:focus:not(:focus-visible):not(.focus-visible),
  [role="tabpanel"][tabindex="0"]:focus:not(:focus-visible):not(.focus-visible),
  button:focus:not(:focus-visible):not(.focus-visible),
  summary:focus:not(:focus-visible):not(.focus-visible),
  a:focus:not(:focus-visible):not(.focus-visible) {
    outline: none;
    box-shadow: none;
  }

  [tabindex="0"]:focus:not(:focus-visible):not(.focus-visible),
  details-dialog:focus:not(:focus-visible):not(.focus-visible) {
    outline: none;
  }
`,Pr=s.div`
  ${Ar};
  ${Nr};
`;function Wr(r){const{children:t,...o}=r;return require("focus-visible"),e.createElement(Pr,Or({},o,{"data-portal-root":!0}),e.createElement(Hr,null),t)}Wr.displayName="BaseStyles",Wr.defaultProps={color:"text.primary",fontFamily:"normal",lineHeight:"default"};var Dr=function(e){return function(e){return!!e&&"object"==typeof e}(e)&&!function(e){var r=Object.prototype.toString.call(e);return"[object RegExp]"===r||"[object Date]"===r||function(e){return e.$$typeof===jr}(e)}(e)};var jr="function"==typeof Symbol&&Symbol.for?Symbol.for("react.element"):60103;function Fr(e,r){return!1!==r.clone&&r.isMergeableObject(e)?Xr((t=e,Array.isArray(t)?[]:{}),e,r):e;var t}function _r(e,r,t){return e.concat(r).map((function(e){return Fr(e,t)}))}function Ur(e){return Object.keys(e).concat(function(e){return Object.getOwnPropertySymbols?Object.getOwnPropertySymbols(e).filter((function(r){return e.propertyIsEnumerable(r)})):[]}(e))}function Gr(e,r){try{return r in e}catch(e){return!1}}function Kr(e,r,t){var o={};return t.isMergeableObject(e)&&Ur(e).forEach((function(r){o[r]=Fr(e[r],t)})),Ur(r).forEach((function(a){(function(e,r){return Gr(e,r)&&!(Object.hasOwnProperty.call(e,r)&&Object.propertyIsEnumerable.call(e,r))})(e,a)||(Gr(e,a)&&t.isMergeableObject(r[a])?o[a]=function(e,r){if(!r.customMerge)return Xr;var t=r.customMerge(e);return"function"==typeof t?t:Xr}(a,t)(e[a],r[a],t):o[a]=Fr(r[a],t))})),o}function Xr(e,r,t){(t=t||{}).arrayMerge=t.arrayMerge||_r,t.isMergeableObject=t.isMergeableObject||Dr,t.cloneUnlessOtherwiseSpecified=Fr;var o=Array.isArray(r);return o===Array.isArray(e)?o?t.arrayMerge(e,r,t):Kr(e,r,t):Fr(r,t)}Xr.all=function(e,r){if(!Array.isArray(e))throw new Error("first argument should be an array");return e.reduce((function(e,t){return Xr(e,t,r)}),{})};var Vr=Xr;const Yr=e.createContext({setColorMode:()=>null,setDayScheme:()=>null,setNightScheme:()=>null}),qr=({children:r,...t})=>{var o,a,n,i,d,l,c,s;const{theme:f,colorMode:g,dayScheme:p,nightScheme:u}=Jr(),m=null!==(o=null!==(a=t.theme)&&void 0!==a?a:f)&&void 0!==o?o:x,[h,v]=e.useState(null!==(n=null!==(i=t.colorMode)&&void 0!==i?i:g)&&void 0!==n?n:"day"),[y,w]=e.useState(null!==(d=null!==(l=t.dayScheme)&&void 0!==l?l:p)&&void 0!==d?d:"light"),[B,$]=e.useState(null!==(c=null!==(s=t.nightScheme)&&void 0!==s?s:u)&&void 0!==c?c:"dark"),k=function(e,r){switch(e){case"auto":return r;default:return e}}(h,function(){const[r,t]=e.useState(Qr);return e.useEffect((()=>{var e,r;const o=null===(e=window)||void 0===e||null===(r=e.matchMedia)||void 0===r?void 0:r.call(e,"(prefers-color-scheme: dark)");function a(e){const r=e.matches;t(r?"night":"day")}if(o){if(void 0!==o.addEventListener)return o.addEventListener("change",a),function(){o.removeEventListener("change",a)};if(void 0!==o.addListener)return o.addListener(a),function(){o.removeListener(a)}}}),[]),r}()),S=function(e,r,t){switch(e){case"day":return r;case"night":return t}}(k,y,B),T=e.useMemo((()=>function(e,r){if(!e.colorSchemes)return e;if(!e.colorSchemes[r]){console.error(`\`${r}\` scheme not defined in \`theme.colorSchemes\``);const t=Object.keys(e.colorSchemes)[0];return Vr(e,e.colorSchemes[t])}return Vr(e,e.colorSchemes[r])}(m,S)),[m,S]);return e.useEffect((()=>{var e,r;v(null!==(e=null!==(r=t.colorMode)&&void 0!==r?r:g)&&void 0!==e?e:"day")}),[t.colorMode,g]),e.useEffect((()=>{var e,r;w(null!==(e=null!==(r=t.dayScheme)&&void 0!==r?r:p)&&void 0!==e?e:"light")}),[t.dayScheme,p]),e.useEffect((()=>{var e,r;$(null!==(e=null!==(r=t.nightScheme)&&void 0!==r?r:u)&&void 0!==e?e:"dark")}),[t.nightScheme,u]),e.createElement(Yr.Provider,{value:{theme:T,colorScheme:S,colorMode:h,resolvedColorMode:k,dayScheme:y,nightScheme:B,setColorMode:v,setDayScheme:w,setNightScheme:$}},e.createElement(b,{theme:T},r))};function Jr(){return e.useContext(Yr)}function Zr(e,r){var t;const{colorScheme:o=""}=Jr();return null!==(t=e[o])&&void 0!==t?t:r}function Qr(){var e,r,t;return"undefined"!=typeof window&&null!==(e=(r=window).matchMedia)&&void 0!==e&&null!==(t=e.call(r,"(prefers-color-scheme: dark)"))&&void 0!==t&&t.matches?"night":"day"}qr.displayName="ThemeProvider";const et=e=>fe(e.sx),rt=s.div(ee,H,P,M,W,j,G,_,X,re,et),tt=s(rt)``;tt.defaultProps={borderWidth:"1px",borderStyle:"solid",borderColor:"border.primary",borderRadius:2};const ot=s(rt)``;ot.defaultProps={display:"flex"};const at=s(rt)``;at.defaultProps={display:"grid"};const nt=s(rt)``,it=e.forwardRef(((r,t)=>e.createElement(nt,Or({},r,{position:"absolute",ref:t}))));it.displayName="Absolute";const dt=e.forwardRef(((r,t)=>e.createElement(nt,Or({},r,{position:"fixed",ref:t}))));dt.displayName="Fixed";const lt=e.forwardRef(((r,t)=>e.createElement(nt,Or({},r,{position:"relative",ref:t}))));lt.displayName="Relative";const ct=e.forwardRef(((r,t)=>e.createElement(nt,Or({},r,{position:"sticky",ref:t}))));function st({ref:e,closeOnOutsideClick:n,defaultOpen:i,onClickOutside:d}){const[l,c]=r(i),s=t(null),f=null!=e?e:s,b=o((e=>{const{current:r}=f;e.target.closest("details")!==r&&(d&&d(e),e.defaultPrevented||c(!1))}),[f,c,d]);a((()=>{if(l&&n)return document.addEventListener("click",b),()=>{document.removeEventListener("click",b)}}),[l,n,b]);const g=e=>{if(!e.defaultPrevented){const r=e.target;c(r.open)}};return{open:l,setOpen:c,getDetailsProps:()=>({onToggle:g,open:l,ref:f})}}function ft(){const e=t(new Set),r=o(((r,t,...o)=>{const a=window.setTimeout(r,t,...o);return e.current.add(a),a}),[]),n=o((r=>{clearTimeout(r),e.current.delete(r)}),[]);return a((()=>()=>{for(const r of e.current)clearTimeout(r)}),[]),{safeSetTimeout:r,safeClearTimeout:n}}ct.defaultProps={top:0,zIndex:1},ct.displayName="Sticky";function bt(e){if(!e.defaultPrevented)for(const r of Object.values(gt).reverse())if(true===r(e)||e.defaultPrevented)break}const gt={};let pt=0;const ut=({containerRef:e,ignoreClickRefs:r,onClickOutside:t})=>{const i=n((()=>pt++),[]),d=o((o=>{var a;return o instanceof MouseEvent&&o.button>0||(!(null===(a=e.current)||void 0===a||!a.contains(o.target))||(!(!r||!r.some((({current:e})=>null==e?void 0:e.contains(o.target))))||void t(o)))}),[e,r,t]);a((()=>(0===Object.keys(gt).length&&document.addEventListener("mousedown",bt,{capture:!0}),function(e,r){gt[e]=r}(i,d),()=>{!function(e){delete gt[e]}(i),0===Object.keys(gt).length&&document.removeEventListener("mousedown",bt,{capture:!0})})),[i,d])};function*mt(e,r={}){var t,o;const a=null!==(t=r.strict)&&void 0!==t&&t,n=null!==(o=r.onlyTabbable)&&void 0!==o&&o?xt:ht,i=document.createTreeWalker(e,NodeFilter.SHOW_ELEMENT,{acceptNode:e=>e instanceof HTMLElement&&n(e,a)?NodeFilter.FILTER_ACCEPT:NodeFilter.FILTER_SKIP});let d=null;if(!r.reverse&&n(e,a)&&(yield e),r.reverse){let e=i.lastChild();for(;e;)d=e,e=i.lastChild()}else d=i.firstChild();for(;d instanceof HTMLElement;)yield d,d=r.reverse?i.previousNode():i.nextNode();r.reverse&&n(e,a)&&(yield e)}function ht(e,r=!1){const t=["BUTTON","INPUT","SELECT","TEXTAREA","OPTGROUP","OPTION","FIELDSET"].includes(e.tagName)&&e.disabled,o=e.hidden,a=e instanceof HTMLInputElement&&"hidden"===e.type;if(t||o||a)return!1;if(r){const r=0===e.offsetWidth||0===e.offsetHeight,t=["hidden","collapse"].includes(getComputedStyle(e).visibility),o=0===e.getClientRects().length;if(r||t||o)return!1}return null!=e.getAttribute("tabindex")||!(e instanceof HTMLAnchorElement&&null==e.getAttribute("href"))&&-1!==e.tabIndex}function xt(e,r=!1){return ht(e,r)&&"-1"!==e.getAttribute("tabindex")}function vt({initialFocusRef:e,returnFocusRef:r,containerRef:t}){a((()=>{const o=r.current;if(e&&e.current)e.current.focus();else if(t.current){const e=mt(t.current).next().value;null==e||e.focus()}return function(){null==o||o.focus()}}),[e,r,t])}const yt=[];function wt(e){if("Escape"===e.key&&!e.defaultPrevented)for(let r=yt.length-1;r>=0&&(yt[r](e),!e.defaultPrevented);--r);}const Bt=(e,r=[e])=>{const t=o(e,r);a((()=>(0===yt.length&&document.addEventListener("keydown",wt),yt.push(t),()=>{yt.splice(yt.findIndex((e=>e===t)),1),0===yt.length&&document.removeEventListener("keydown",wt)})),[t])};function $t(r){const t=e.useRef(null);return null!=r?r:t}const kt=({overlayRef:e,returnFocusRef:r,initialFocusRef:t,onEscape:o,ignoreClickRefs:a,onClickOutside:n})=>{const i=$t(e);return vt({containerRef:i,returnFocusRef:r,initialFocusRef:t}),ut({containerRef:i,ignoreClickRefs:a,onClickOutside:n}),Bt(o),{ref:i}};let St,Tt=!1;function Et(){}try{const e=Object.create({},{signal:{get(){Tt=!0}}});window.addEventListener("test",Et,e),window.removeEventListener("test",Et,e)}catch(e){}function It(){Tt||(!function(){if("undefined"==typeof window)return;const e=EventTarget.prototype.addEventListener;EventTarget.prototype.addEventListener=function(r,t,o){return"object"==typeof o&&"signal"in o&&o.signal instanceof AbortSignal&&e.call(o.signal,"abort",(()=>{this.removeEventListener(r,t,o)})),e.call(this,r,t,o)}}(),Tt=!0)}function Ct(){return void 0===St&&(St=/^mac/i.test(window.navigator.platform)),St}let Nt,At=1e4;function zt(){return"__primer_id_"+At++}It(),function(e){e[e.ArrowHorizontal=1]="ArrowHorizontal",e[e.ArrowVertical=2]="ArrowVertical",e[e.JK=4]="JK",e[e.HL=8]="HL",e[e.HomeAndEnd=16]="HomeAndEnd",e[e.PageUpDown=256]="PageUpDown",e[e.WS=32]="WS",e[e.AD=64]="AD",e[e.Tab=128]="Tab",e[e.ArrowAll=e.ArrowHorizontal|e.ArrowVertical]="ArrowAll",e[e.HJKL=e.HL|e.JK]="HJKL",e[e.WASD=e.WS|e.AD]="WASD",e[e.All=e.ArrowAll|e.HJKL|e.HomeAndEnd|e.PageUpDown|e.WASD|e.Tab]="All"}(Nt||(Nt={}));const Rt={ArrowLeft:Nt.ArrowHorizontal,ArrowDown:Nt.ArrowVertical,ArrowUp:Nt.ArrowVertical,ArrowRight:Nt.ArrowHorizontal,h:Nt.HL,j:Nt.JK,k:Nt.JK,l:Nt.HL,a:Nt.AD,s:Nt.WS,w:Nt.WS,d:Nt.AD,Tab:Nt.Tab,Home:Nt.HomeAndEnd,End:Nt.HomeAndEnd,PageUp:Nt.PageUpDown,PageDown:Nt.PageUpDown},Lt={ArrowLeft:"previous",ArrowDown:"next",ArrowUp:"previous",ArrowRight:"next",h:"previous",j:"next",k:"previous",l:"next",a:"previous",s:"next",w:"previous",d:"next",Tab:"next",Home:"start",End:"end",PageUp:"start",PageDown:"end"};function Mt(e){const r=Lt[e.key];if("Tab"===e.key&&e.shiftKey)return"previous";const t=Ct();if(t&&e.metaKey||!t&&e.ctrlKey){if("ArrowLeft"===e.key||"ArrowUp"===e.key)return"start";if("ArrowRight"===e.key||"ArrowDown"===e.key)return"end"}return r}function Ot(e,r){var t,o,a,n;const i=[],d=new WeakMap,l=null!==(t=null==r?void 0:r.bindKeys)&&void 0!==t?t:(null!=r&&r.getNextFocusable?Nt.ArrowAll:Nt.ArrowVertical)|Nt.HomeAndEnd,c=null!==(o=null==r?void 0:r.focusOutBehavior)&&void 0!==o?o:"stop",s=null!==(a=null==r?void 0:r.focusInStrategy)&&void 0!==a?a:"previous",f=null==r?void 0:r.activeDescendantControl,b=null==r?void 0:r.onActiveDescendantChanged;let g;function p(){return i[0]}function u(e,r=!1){const t=g;g=e,f?e&&document.activeElement===f?m(t,e,r):h():(t&&t!==e&&d.has(t)&&t.setAttribute("tabindex","-1"),null==e||e.setAttribute("tabindex","0"))}function m(r,t,o=!1){t.id||t.setAttribute("id",zt()),r&&r!==t&&r.removeAttribute("data-is-active-descendant"),f&&(o||f.getAttribute("aria-activedescendant")!==t.id)&&(f.setAttribute("aria-activedescendant",t.id),e.setAttribute("data-has-active-descendant",t.id),t.setAttribute("data-is-active-descendant",o?"activated-directly":"activated-indirectly"),null==b||b(t,r,o))}function h(r=g){"first"===s&&(g=void 0),null==f||f.removeAttribute("aria-activedescendant"),e.removeAttribute("data-has-active-descendant"),null==r||r.removeAttribute("data-is-active-descendant"),null==b||b(void 0,r,!1)}function x(...e){const t=e.filter((e=>{var t,o;return null===(t=null==r||null===(o=r.focusableElementFilter)||void 0===o?void 0:o.call(r,e))||void 0===t||t}));if(0===t.length)return;const o=i.findIndex((e=>(e.compareDocumentPosition(t[0])&Node.DOCUMENT_POSITION_PRECEDING)>0));i.splice(-1===o?i.length:o,0,...t);for(const e of t)d.has(e)||d.set(e,e.getAttribute("tabindex")),e.setAttribute("tabindex","-1");g||u(p())}function v(...e){for(const r of e){const e=i.indexOf(r);e>=0&&i.splice(e,1);const t=d.get(r);if(void 0!==t&&(null===t?r.removeAttribute("tabindex"):r.setAttribute("tabindex",t),d.delete(r)),r===g){u(p())}}}x(...mt(e)),u(p());new MutationObserver((e=>{for(const r of e)for(const e of r.removedNodes)e instanceof HTMLElement&&v(...mt(e));for(const r of e)for(const e of r.addedNodes)e instanceof HTMLElement&&x(...mt(e))})).observe(e,{subtree:!0,childList:!0});const y=new AbortController,w=null!==(n=null==r?void 0:r.abortSignal)&&void 0!==n?n:y.signal;let B;w.addEventListener("abort",(()=>{v(...i)})),e.addEventListener("mousedown",(e=>{e.target instanceof HTMLElement&&e.target!==document.activeElement&&(B=i.indexOf(e.target))}),{signal:w}),f?(e.addEventListener("focusin",(e=>{e.target instanceof HTMLElement&&i.includes(e.target)&&(f.focus(),u(e.target))})),e.addEventListener("mousemove",(({target:e})=>{if(!(e instanceof Node))return;const r=i.find((r=>r.contains(e)));r&&u(r)}),{signal:w,capture:!0}),f.addEventListener("focusin",(()=>{g?m(void 0,g):u(p())})),f.addEventListener("focusout",(()=>{h()}))):e.addEventListener("focusin",(r=>{if(r.target instanceof HTMLElement)if(void 0!==B)B>=0&&i[B]!==g&&u(i[B]),B=void 0;else if("previous"===s)u(r.target);else if("closest"===s||"first"===s){if(r.relatedTarget instanceof Element&&!e.contains(r.relatedTarget)){const e="previous"===k?i.length-1:0,r=i[e];return void(null==r||r.focus())}u(r.target)}else if("function"==typeof s)if(r.relatedTarget instanceof Element&&!e.contains(r.relatedTarget)){const e=s(r.relatedTarget);if((e?i.indexOf(e):-1)>=0&&e instanceof HTMLElement)return void e.focus();console.warn("Element requested is not a known focusable element.")}else u(r.target);k=void 0}),{signal:w});const $=null!=f?f:e;let k;return"closest"===s&&document.addEventListener("keydown",(e=>{"Tab"===e.key&&(k=Mt(e))}),{signal:w,capture:!0}),$.addEventListener("keydown",(t=>{if(t.key in Lt){const a=Rt[t.key];if(!t.defaultPrevented&&(a&l)>0&&!function(e,r){const t=e.key,o=[...t].length,a=r instanceof HTMLInputElement&&"text"===r.type||r instanceof HTMLTextAreaElement;if(a&&(1===o||"Home"===t||"End"===t))return!0;if(r instanceof HTMLSelectElement){if(1===o)return!0;if("ArrowDown"===t&&Ct()&&!e.metaKey)return!0;if("ArrowDown"===t&&!Ct()&&e.altKey)return!0}if(r instanceof HTMLTextAreaElement&&("PageUp"===t||"PageDown"===t))return!0;if(a){const e=r,o=0===e.selectionStart&&0===e.selectionEnd,a=e.selectionStart===e.value.length&&e.selectionEnd===e.value.length;if("ArrowLeft"===t&&!o)return!0;if("ArrowRight"===t&&!a)return!0;if(e instanceof HTMLTextAreaElement){if("ArrowUp"===t&&!o)return!0;if("ArrowDown"===t&&!a)return!0}}return!1}(t,document.activeElement)){const a=Mt(t);let n;var o;if(null!=r&&r.getNextFocusable)n=r.getNextFocusable(a,null!==(o=document.activeElement)&&void 0!==o?o:void 0,t);if(!n){const r=function(){if(!g)return 0;const r=i.indexOf(g);return-1!==r?r:g===e?-1:0}();let o=r;"previous"===a?o-=1:"start"===a?o=0:"next"===a?o+=1:o=i.length-1,o<0&&(o="wrap"===c&&"Tab"!==t.key?i.length-1:0),o>=i.length&&(o="wrap"===c&&"Tab"!==t.key?0:i.length-1),r!==o&&(n=i[o])}f?u(n||g,!0):n&&(k=a,n.focus()),("Tab"!==t.key||n)&&t.preventDefault()}}}),{signal:w}),y}var Ht=g(["position:relative;display:inline-block;padding:6px 16px;font-family:inherit;font-weight:",";line-height:20px;white-space:nowrap;vertical-align:middle;cursor:pointer;user-select:none;border-radius:",";appearance:none;text-decoration:none;text-align:center;&:hover{text-decoration:none;}&:focus{outline:none;}&:disabled{cursor:default;}&:disabled svg{opacity:0.6;}"],Cr("fontWeights.bold"),Cr("radii.2"));const Pt=L(Ce,Nr,Rr),Wt=be({variants:{small:{p:"4px 12px",fontSize:0},medium:{fontSize:1},large:{fontSize:2,p:"10px 20px"}}}),Dt=s.button.attrs((({disabled:e,onClick:r})=>({onClick:e?void 0:r})))`
  ${Ht}
  ${Wt}
`;Dt.defaultProps={variant:"medium"};const jt=s(Dt)`
  color: ${Cr("colors.btn.text")};
  background-color: ${Cr("colors.btn.bg")};
  border: 1px solid ${Cr("colors.btn.border")};
  box-shadow: ${Cr("shadows.btn.shadow")}, ${Cr("shadows.btn.insetShadow")}};

  &:hover {
    background-color: ${Cr("colors.btn.hoverBg")};
    border-color: ${Cr("colors.btn.hoverBorder")};
  }

  // focus must come before :active so that the active box shadow overrides
  &:focus {
    border-color: ${Cr("colors.btn.focusBorder")};
    box-shadow: ${Cr("shadows.btn.focusShadow")};
  }

  &:active {
    background-color: ${Cr("colors.btn.selectedBg")};
    box-shadow: ${Cr("shadows.btn.shadowActive")};
  }

  &:disabled {
    color: ${Cr("colors.text.disabled")};
    background-color: ${Cr("colors.btn.bg")};
    border-color: ${Cr("colors.btn.border")};
  }

  ${Pt};
  ${et};
`,Ft=s(Dt)`
  color: ${Cr("colors.btn.danger.text")};
  border: 1px solid ${Cr("colors.btn.border")};
  background-color: ${Cr("colors.btn.bg")};
  box-shadow: ${Cr("shadows.btn.shadow")};

  &:hover {
    color: ${Cr("colors.btn.danger.hoverText")};
    background-color: ${Cr("colors.btn.danger.hoverBg")};
    border-color: ${Cr("colors.btn.danger.hoverBorder")};
    box-shadow: ${Cr("shadows.btn.danger.hoverShadow")};
  }
  // focus must come before :active so that the active box shadow overrides
  &:focus {
    border-color: ${Cr("colors.btn.danger.focusBorder")};
    box-shadow: ${Cr("shadows.btn.danger.focusShadow")};
  }

  &:active {
    color: ${Cr("colors.btn.danger.selectedText")};
    background-color: ${Cr("colors.btn.danger.selectedBg")};
    box-shadow: ${Cr("shadows.btn.danger.selectedShadow")};
    border-color: ${Cr("colors.btn.danger.selectedBorder")};
  }

  &:disabled {
    color: ${Cr("colors.btn.danger.disabledText")};
    background-color: ${Cr("colors.btn.danger.disabledBg")};
    border-color: ${Cr("colors.btn.danger.disabledBorder")};
  }

  ${Pt};
  ${et};
`,_t=s(rt)`
  vertical-align: middle;

  && > * {
    position: relative;
    border-right-width: 0;
    border-radius: 0;

    :first-child {
      border-top-left-radius: ${Cr("radii.2")};
      border-bottom-left-radius: ${Cr("radii.2")};
      margin-right: 0;
    }

    :not(:first-child) {
      margin-left: 0;
      margin-right: 0;
    }

    :last-child {
      border-right-width: 1px;
      border-top-right-radius: ${Cr("radii.2")};
      border-bottom-right-radius: ${Cr("radii.2")};
    }

    :focus,
    :active,
    :hover {
      border-right-width: 1px;
      + * {
        border-left-width: 0;
      }
    }

    :focus,
    :active {
      z-index: 1;
    }
  }

  ${et};
`;_t.defaultProps={display:"inline-block"};const Ut=s(Dt)`
  color: ${Cr("colors.btn.outline.text")};
  border: 1px solid ${Cr("colors.btn.border")};
  background-color: ${Cr("colors.btn.bg")};
  box-shadow: ${Cr("shadows.btn.shadow")};

  &:hover {
    color: ${Cr("colors.btn.outline.hoverText")};
    background-color: ${Cr("colors.btn.outline.hoverBg")};
    border-color: ${Cr("colors.btn.outline.hoverBorder")};
    box-shadow: ${Cr("shadows.btn.outline.hoverShadow")};
  }
  // focus must come before :active so that the active box shadow overrides
  &:focus {
    border-color: ${Cr("colors.btn.outline.focusBorder")};
    box-shadow: ${Cr("shadows.btn.outline.focusShadow")};
  }

  &:active {
    color: ${Cr("colors.btn.outline.selectedText")};
    background-color: ${Cr("colors.btn.outline.selectedBg")};
    box-shadow: ${Cr("shadows.btn.outline.selectedShadow")};
    border-color: ${Cr("colors.btn.outline.selectedBorder")};
  }

  &:disabled {
    color: ${Cr("colors.btn.outline.disabledText")};
    background-color: ${Cr("colors.btn.outline.disabledBg")};
    border-color: ${Cr("colors.btn.border")};
  }

  ${Pt};
  ${et};
`,Gt=s(Dt)`
  color: ${Cr("colors.btn.primary.text")};
  border: 1px solid ${Cr("colors.btn.primary.border")};
  background-color: ${Cr("colors.btn.primary.bg")};
  box-shadow: ${Cr("shadows.btn.primary.shadow")};

  &:hover {
    color: ${Cr("colors.btn.primary.hoverText")};
    background-color: ${Cr("colors.btn.primary.hoverBg")};
    border-color: ${Cr("colors.btn.primary.hoverBorder")};
    box-shadow: ${Cr("shadows.btn.primary.hoverShadow")};
  }
  // focus must come before :active so that the active box shadow overrides
  &:focus {
    border-color: ${Cr("colors.btn.primary.focusBorder")};
    box-shadow: ${Cr("shadows.btn.primary.focusShadow")};
  }

  &:active {
    background-color: ${Cr("colors.btn.primary.selectedBg")};
    box-shadow: ${Cr("shadows.btn.primary.selectedShadow")};
  }

  &:disabled {
    color: ${Cr("colors.btn.primary.disabledText")};
    background-color: ${Cr("colors.btn.primary.disabledBg")};
    border-color: ${Cr("colors.btn.primary.disabledBorder")};
  }

  ${Pt};
  ${et};
`,Kt=s(Dt)`
  color: ${Cr("colors.text.link")};
  background-color: transparent;
  border: 0;
  border-radius: ${Cr("radii.2")};
  box-shadow: none;

  &:disabled {
    color: ${Cr("colors.text.disabled")};
  }

  &:focus {
    box-shadow: ${Cr("shadows.btn.focusShadow")};
  }

  ${Pt};
  ${et}
`,Xt=s.summary`
  display: inline-block;
  padding: 0;
  font-size: ${Cr("fontSizes.1")};
  color: ${Cr("colors.text.secondary")};
  text-decoration: none;
  white-space: nowrap;
  cursor: pointer;
  user-select: none;
  background-color: transparent;
  border: 0;
  appearance: none; // Corrects inability to style clickable input types in iOS.

  &:hover {
    text-decoration: underline;
  }

  &:disabled {
    &,
    &:hover {
      color: rgba(${Cr("colors.text.secondary")}, 0.5);
      cursor: default;
    }
  }

  &:after {
    display: inline-block;
    margin-left: ${Cr("space.1")};
    width: 0;
    height: 0;
    vertical-align: -2px;
    content: '';
    border: 4px solid transparent;
    border-top-color: currentcolor;
  }
  ${Nr}
  ${Ar}
  ${Rr}
  ${et};
`;var Vt={small:16,medium:32,large:64};function Yt(e){var r=e["aria-label"],t=e.className,o=e.fill,a=void 0===o?"currentColor":o,n=e.size,i=e.verticalAlign,d=e.svgDataByHeight,l=Vt[n]||n,c=function(e,r){return e.map((function(e){return parseInt(e,10)})).reduce((function(e,t){return t<=r?t:e}),e[0])}(Object.keys(d),l),s=d[c].width;return{"aria-hidden":r?"false":"true","aria-label":r,role:"img",className:t,viewBox:"0 0 "+s+" "+c,width:l*(s/c),height:l,fill:a,style:{display:"inline-block",userSelect:"none",verticalAlign:i},dangerouslySetInnerHTML:{__html:d[c].path}}}var qt=Object.assign||function(e){for(var r=1;r<arguments.length;r++){var t=arguments[r];for(var o in t)Object.prototype.hasOwnProperty.call(t,o)&&(e[o]=t[o])}return e};function Jt(r){return e.createElement("svg",Yt(qt({},r,{svgDataByHeight:{16:{width:16,path:'<path fill-rule="evenodd" d="M13.78 4.22a.75.75 0 010 1.06l-7.25 7.25a.75.75 0 01-1.06 0L2.22 9.28a.75.75 0 011.06-1.06L6 10.94l6.72-6.72a.75.75 0 011.06 0z"></path>'},24:{width:24,path:'<path fill-rule="evenodd" d="M21.03 5.72a.75.75 0 010 1.06l-11.5 11.5a.75.75 0 01-1.072-.012l-5.5-5.75a.75.75 0 111.084-1.036l4.97 5.195L19.97 5.72a.75.75 0 011.06 0z"></path>'}}})))}function Zt(r){return e.createElement("svg",Yt(qt({},r,{svgDataByHeight:{16:{width:16,path:'<path fill-rule="evenodd" d="M5 3.254V3.25v.005a.75.75 0 110-.005v.004zm.45 1.9a2.25 2.25 0 10-1.95.218v5.256a2.25 2.25 0 101.5 0V7.123A5.735 5.735 0 009.25 9h1.378a2.251 2.251 0 100-1.5H9.25a4.25 4.25 0 01-3.8-2.346zM12.75 9a.75.75 0 100-1.5.75.75 0 000 1.5zm-8.5 4.5a.75.75 0 100-1.5.75.75 0 000 1.5z"></path>'},24:{width:24,path:'<path fill-rule="evenodd" d="M5.75 21a1.75 1.75 0 110-3.5 1.75 1.75 0 010 3.5zM2.5 19.25a3.25 3.25 0 106.5 0 3.25 3.25 0 00-6.5 0zM5.75 6.5a1.75 1.75 0 110-3.5 1.75 1.75 0 010 3.5zM2.5 4.75a3.25 3.25 0 106.5 0 3.25 3.25 0 00-6.5 0zM18.25 15a1.75 1.75 0 110-3.5 1.75 1.75 0 010 3.5zM15 13.25a3.25 3.25 0 106.5 0 3.25 3.25 0 00-6.5 0z"></path><path fill-rule="evenodd" d="M6.5 7.25c0 2.9 2.35 5.25 5.25 5.25h4.5V14h-4.5A6.75 6.75 0 015 7.25h1.5z"></path><path fill-rule="evenodd" d="M5.75 16.75A.75.75 0 006.5 16V8A.75.75 0 005 8v8c0 .414.336.75.75.75z"></path>'}}})))}function Qt(r){return e.createElement("svg",Yt(qt({},r,{svgDataByHeight:{16:{width:16,path:'<path fill-rule="evenodd" d="M7.177 3.073L9.573.677A.25.25 0 0110 .854v4.792a.25.25 0 01-.427.177L7.177 3.427a.25.25 0 010-.354zM3.75 2.5a.75.75 0 100 1.5.75.75 0 000-1.5zm-2.25.75a2.25 2.25 0 113 2.122v5.256a2.251 2.251 0 11-1.5 0V5.372A2.25 2.25 0 011.5 3.25zM11 2.5h-1V4h1a1 1 0 011 1v5.628a2.251 2.251 0 101.5 0V5A2.5 2.5 0 0011 2.5zm1 10.25a.75.75 0 111.5 0 .75.75 0 01-1.5 0zM3.75 12a.75.75 0 100 1.5.75.75 0 000-1.5z"></path>'},24:{width:24,path:'<path fill-rule="evenodd" d="M4.75 3a1.75 1.75 0 100 3.5 1.75 1.75 0 000-3.5zM1.5 4.75a3.25 3.25 0 116.5 0 3.25 3.25 0 01-6.5 0zM4.75 17.5a1.75 1.75 0 100 3.5 1.75 1.75 0 000-3.5zM1.5 19.25a3.25 3.25 0 116.5 0 3.25 3.25 0 01-6.5 0zm17.75-1.75a1.75 1.75 0 100 3.5 1.75 1.75 0 000-3.5zM16 19.25a3.25 3.25 0 116.5 0 3.25 3.25 0 01-6.5 0z"></path><path fill-rule="evenodd" d="M4.75 7.25A.75.75 0 015.5 8v8A.75.75 0 014 16V8a.75.75 0 01.75-.75zm8.655-5.53a.75.75 0 010 1.06L12.185 4h4.065A3.75 3.75 0 0120 7.75v8.75a.75.75 0 01-1.5 0V7.75a2.25 2.25 0 00-2.25-2.25h-4.064l1.22 1.22a.75.75 0 01-1.061 1.06l-2.5-2.5a.75.75 0 010-1.06l2.5-2.5a.75.75 0 011.06 0z"></path>'}}})))}function eo(r){return e.createElement("svg",Yt(qt({},r,{svgDataByHeight:{16:{width:16,path:'<path fill-rule="evenodd" d="M1.5 8a6.5 6.5 0 0110.65-5.003.75.75 0 00.959-1.153 8 8 0 102.592 8.33.75.75 0 10-1.444-.407A6.5 6.5 0 011.5 8zM8 12a1 1 0 100-2 1 1 0 000 2zm0-8a.75.75 0 01.75.75v3.5a.75.75 0 11-1.5 0v-3.5A.75.75 0 018 4zm4.78 4.28l3-3a.75.75 0 00-1.06-1.06l-2.47 2.47-.97-.97a.749.749 0 10-1.06 1.06l1.5 1.5a.75.75 0 001.06 0z"></path>'},24:{width:24,path:'<path d="M2.5 12c0-5.24 4.288-9.5 9.593-9.5a9.608 9.608 0 017.197 3.219.75.75 0 001.12-.998A11.108 11.108 0 0012.093 1C5.973 1 1 5.919 1 12s4.973 11 11.093 11c5.403 0 9.91-3.832 10.893-8.915a.75.75 0 10-1.472-.285c-.848 4.381-4.74 7.7-9.421 7.7C6.788 21.5 2.5 17.24 2.5 12z"></path><path d="M12 17a1 1 0 100-2 1 1 0 000 2zm0-10a.75.75 0 01.75.75v4.5a.75.75 0 01-1.5 0v-4.5A.75.75 0 0112 7zm11.28.78a.75.75 0 00-1.06-1.06l-3.47 3.47-1.47-1.47a.75.75 0 10-1.06 1.06l2 2a.75.75 0 001.06 0l4-4z"></path>'}}})))}function ro(r){return e.createElement("svg",Yt(qt({},r,{svgDataByHeight:{16:{width:16,path:'<path fill-rule="evenodd" d="M8 1.5a6.5 6.5 0 100 13 6.5 6.5 0 000-13zM0 8a8 8 0 1116 0A8 8 0 010 8zm9 3a1 1 0 11-2 0 1 1 0 012 0zm-.25-6.25a.75.75 0 00-1.5 0v3.5a.75.75 0 001.5 0v-3.5z"></path>'},24:{width:24,path:'<path d="M12 7a.75.75 0 01.75.75v4.5a.75.75 0 01-1.5 0v-4.5A.75.75 0 0112 7zm1 9a1 1 0 11-2 0 1 1 0 012 0z"></path><path fill-rule="evenodd" d="M12 1C5.925 1 1 5.925 1 12s4.925 11 11 11 11-4.925 11-11S18.075 1 12 1zM2.5 12a9.5 9.5 0 1119 0 9.5 9.5 0 01-19 0z"></path>'}}})))}function to(r){return e.createElement("svg",Yt(qt({},r,{svgDataByHeight:{16:{width:16,path:'<path fill-rule="evenodd" d="M8 1.5a6.5 6.5 0 100 13 6.5 6.5 0 000-13zM0 8a8 8 0 1116 0A8 8 0 010 8zm9 3a1 1 0 11-2 0 1 1 0 012 0zM6.92 6.085c.081-.16.19-.299.34-.398.145-.097.371-.187.74-.187.28 0 .553.087.738.225A.613.613 0 019 6.25c0 .177-.04.264-.077.318a.956.956 0 01-.277.245c-.076.051-.158.1-.258.161l-.007.004a7.728 7.728 0 00-.313.195 2.416 2.416 0 00-.692.661.75.75 0 001.248.832.956.956 0 01.276-.245 6.3 6.3 0 01.26-.16l.006-.004c.093-.057.204-.123.313-.195.222-.149.487-.355.692-.662.214-.32.329-.702.329-1.15 0-.76-.36-1.348-.863-1.725A2.76 2.76 0 008 4c-.631 0-1.155.16-1.572.438-.413.276-.68.638-.849.977a.75.75 0 101.342.67z"></path>'},24:{width:24,path:'<path d="M10.97 8.265a1.45 1.45 0 00-.487.57.75.75 0 01-1.341-.67c.2-.402.513-.826.997-1.148C10.627 6.69 11.244 6.5 12 6.5c.658 0 1.369.195 1.934.619a2.45 2.45 0 011.004 2.006c0 1.033-.513 1.72-1.027 2.215-.19.183-.399.358-.579.508l-.147.123a4.329 4.329 0 00-.435.409v1.37a.75.75 0 11-1.5 0v-1.473c0-.237.067-.504.247-.736.22-.28.486-.517.718-.714l.183-.153.001-.001c.172-.143.324-.27.47-.412.368-.355.569-.676.569-1.136a.953.953 0 00-.404-.806C12.766 8.118 12.384 8 12 8c-.494 0-.814.121-1.03.265zM13 17a1 1 0 11-2 0 1 1 0 012 0z"></path><path fill-rule="evenodd" d="M12 1C5.925 1 1 5.925 1 12s4.925 11 11 11 11-4.925 11-11S18.075 1 12 1zM2.5 12a9.5 9.5 0 1119 0 9.5 9.5 0 01-19 0z"></path>'}}})))}function oo(r){return e.createElement("svg",Yt(qt({},r,{svgDataByHeight:{16:{width:16,path:'<path d="M4.427 7.427l3.396 3.396a.25.25 0 00.354 0l3.396-3.396A.25.25 0 0011.396 7H4.604a.25.25 0 00-.177.427z"></path>'},24:{width:24,path:'<path d="M11.646 15.146L5.854 9.354a.5.5 0 01.353-.854h11.586a.5.5 0 01.353.854l-5.793 5.792a.5.5 0 01-.707 0z"></path>'}}})))}function ao(r){return e.createElement("svg",Yt(qt({},r,{svgDataByHeight:{16:{width:16,path:'<path fill-rule="evenodd" d="M3.72 3.72a.75.75 0 011.06 0L8 6.94l3.22-3.22a.75.75 0 111.06 1.06L9.06 8l3.22 3.22a.75.75 0 11-1.06 1.06L8 9.06l-3.22 3.22a.75.75 0 01-1.06-1.06L6.94 8 3.72 4.78a.75.75 0 010-1.06z"></path>'},24:{width:24,path:'<path fill-rule="evenodd" d="M5.72 5.72a.75.75 0 011.06 0L12 10.94l5.22-5.22a.75.75 0 111.06 1.06L13.06 12l5.22 5.22a.75.75 0 11-1.06 1.06L12 13.06l-5.22 5.22a.75.75 0 01-1.06-1.06L10.94 12 5.72 6.78a.75.75 0 010-1.06z"></path>'}}})))}Jt.defaultProps={className:"octicon octicon-check",size:16,verticalAlign:"text-bottom"},Zt.defaultProps={className:"octicon octicon-git-merge",size:16,verticalAlign:"text-bottom"},Qt.defaultProps={className:"octicon octicon-git-pull-request",size:16,verticalAlign:"text-bottom"},eo.defaultProps={className:"octicon octicon-issue-closed",size:16,verticalAlign:"text-bottom"},ro.defaultProps={className:"octicon octicon-issue-opened",size:16,verticalAlign:"text-bottom"},to.defaultProps={className:"octicon octicon-question",size:16,verticalAlign:"text-bottom"},oo.defaultProps={className:"octicon octicon-triangle-down",size:16,verticalAlign:"text-bottom"},ao.defaultProps={className:"octicon octicon-x",size:16,verticalAlign:"text-bottom"};const no=s.button`
  border: none;
  padding: 0;
  background: transparent;
  outline: none;
  cursor: pointer;
  border-radius: ${Cr("radii.2")};
  color: ${Cr("colors.text.secondary")};

  &:focus {
    box-shadow: ${Cr("shadows.btn.focusShadow")};
  }

  &:hover {
    color: ${Cr("colors.text.link")};
  }
  ${Nr};
  ${Rr};
  ${et};
`,io=i(((r,t)=>e.createElement(no,Or({ref:t,"aria-label":"Close"},r),e.createElement(ao,null)))),lo={"outside-top":["outside-bottom","outside-right","outside-left","outside-bottom"],"outside-bottom":["outside-top","outside-right","outside-left","outside-bottom"],"outside-left":["outside-right","outside-bottom","outside-top","outside-bottom"],"outside-right":["outside-left","outside-bottom","outside-top","outside-bottom"]};function co(e,r,t={}){const o=function(e){let r=e.parentNode;for(;null!==r;){if(r instanceof HTMLElement&&"static"!==getComputedStyle(r).position)return r;r=r.parentNode}return document.body}(e),a=function(e){let r=e;for(;null!==r&&r!==document.body;){if("visible"!==getComputedStyle(r).overflow)break;r=r.parentNode}const t=r!==document.body&&r instanceof HTMLElement?r:document.body,o=t.getBoundingClientRect(),a=getComputedStyle(t),[n,i,d,l]=[a.borderTopWidth,a.borderLeftWidth,a.borderRightWidth,a.borderBottomWidth].map((e=>parseInt(e,10)||0));return{top:o.top+n,left:o.left+i,width:o.width-d-i,height:Math.max(o.height-n-l,t===document.body?window.innerHeight:-1/0)}}(o),n=getComputedStyle(o),i=o.getBoundingClientRect(),[d,l]=[n.borderTopWidth,n.borderLeftWidth].map((e=>parseInt(e,10)||0));return function(e,r,t,o,{side:a,align:n,allowOutOfBounds:i,anchorOffset:d,alignmentOffset:l}){const c={top:e.top-r.top,left:e.left-r.left,width:e.width,height:e.height};let s=uo(t,o,a,n,d,l),f=a;if(s.top-=r.top,s.left-=r.left,!i){const i=lo[a];let b=0;if(i){let e=a;for(;b<i.length&&mo(e,s,c,t);){const a=i[b++];e=a,s=uo(t,o,a,n,d,l),s.top-=r.top,s.left-=r.left,f=a}}s.top<c.top&&(s.top=c.top),s.left<c.left&&(s.left=c.left),s.left+t.width>e.width+c.left&&(s.left=e.width+c.left-t.width),i&&b<i.length&&s.top+t.height>e.height+c.top&&(s.top=e.height+c.top-t.height)}return{...s,anchorSide:f}}(a,{top:i.top+d,left:i.left+l},e.getBoundingClientRect(),r instanceof Element?r.getBoundingClientRect():r,function(e={}){var r,t,o,a,n;const i=null!==(r=e.side)&&void 0!==r?r:so,d=null!==(t=e.align)&&void 0!==t?t:fo;return{side:i,align:d,anchorOffset:null!==(o=e.anchorOffset)&&void 0!==o?o:"inside-center"===i?0:bo,alignmentOffset:null!==(a=e.alignmentOffset)&&void 0!==a?a:"center"!==d&&i.startsWith("inside")?go:0,allowOutOfBounds:null!==(n=e.allowOutOfBounds)&&void 0!==n?n:po}}(t))}const so="outside-bottom",fo="start",bo=4,go=4,po=!1;function uo(e,r,t,o,a,n){const i=r.left+r.width,d=r.top+r.height;let l=-1,c=-1;return"outside-top"===t?l=r.top-a-e.height:"outside-bottom"===t?l=d+a:"outside-left"===t?c=r.left-a-e.width:"outside-right"===t&&(c=i+a),"outside-top"!==t&&"outside-bottom"!==t||(c="start"===o?r.left+n:"center"===o?r.left-(e.width-r.width)/2+n:i-e.width-n),"outside-left"!==t&&"outside-right"!==t||(l="start"===o?r.top+n:"center"===o?r.top-(e.height-r.height)/2+n:d-e.height-n),"inside-top"===t?l=r.top+a:"inside-bottom"===t?l=d-a-e.height:"inside-left"===t?c=r.left+a:"inside-right"===t?c=i-a-e.width:"inside-center"===t&&(c=(i+r.left)/2-e.width/2+a),"inside-top"===t||"inside-bottom"===t?c="start"===o?r.left+n:"center"===o?r.left-(e.width-r.width)/2+n:i-e.width-n:"inside-left"!==t&&"inside-right"!==t&&"inside-center"!==t||(l="start"===o?r.top+n:"center"===o?r.top-(e.height-r.height)/2+n:d-e.height-n),{top:l,left:c}}function mo(e,r,t,o){return"outside-top"===e||"outside-bottom"===e?r.top<t.top||r.top+o.height>t.height+t.top:r.left<t.left||r.left+o.width>t.width+t.left}function ho(r,t=[]){const o=$t(null==r?void 0:r.floatingElementRef),a=$t(null==r?void 0:r.anchorElementRef),[n,i]=e.useState(void 0),d=e.useCallback((()=>{o.current instanceof Element&&a.current instanceof Element?i(co(o.current,a.current,r)):i(void 0)}),[o,a,...t]);var l;return e.useLayoutEffect(d,[d]),l=d,e.useLayoutEffect((()=>{const e=new window.ResizeObserver((()=>l()));return e.observe(document.documentElement),()=>{e.disconnect()}}),[l]),{floatingElementRef:o,anchorElementRef:a,position:n}}It();const xo=[];let vo;function yo(e,r=!1){return mt(e,{reverse:r,strict:!0,onlyTabbable:!0}).next().value}function wo(e,r,t){const o=new AbortController,a=null!=t?t:o.signal;let n;function i(t){if(t instanceof HTMLElement&&document.contains(e)){if(e.contains(t))return void(n=t);if(n&&xt(n)&&e.contains(n))return void n.focus();if(r&&e.contains(r))return void r.focus();{const r=null===e.getAttribute("tabindex");return r&&e.setAttribute("tabindex","-1"),e.focus(),void(r&&e.addEventListener("blur",(()=>e.removeAttribute("tabindex")),{once:!0}))}}}e.setAttribute("data-focus-trap","active");const d=function(e){const r=new AbortController;return e.addEventListener("abort",(()=>{r.abort()})),r}(a);if(e.addEventListener("keydown",(r=>{if("Tab"!==r.key||r.defaultPrevented)return;const{target:t}=r,o=yo(e),a=yo(e,!0);t===o&&r.shiftKey?(r.preventDefault(),null==a||a.focus()):t!==a||r.shiftKey||(r.preventDefault(),null==o||o.focus())}),{signal:d.signal}),vo){const e=vo;vo.container.setAttribute("data-focus-trap","suspended"),vo.controller.abort(),xo.push(e)}d.signal.addEventListener("abort",(()=>{vo=void 0})),a.addEventListener("abort",(()=>{e.removeAttribute("data-focus-trap");const r=xo.findIndex((r=>r.container===e));r>=0&&xo.splice(r,1),function(){const e=xo.pop();e&&wo(e.container,e.initialFocus,e.originalSignal)}()})),document.addEventListener("focus",(e=>{i(e.target)}),{signal:d.signal,capture:!0}),i(document.activeElement),vo={container:e,controller:d,initialFocus:r,originalSignal:a};const l=xo.findIndex((r=>r.container===e));if(l>=0&&xo.splice(l,1),!t)return o}function Bo(r,t=[]){const o=$t(null==r?void 0:r.containerRef),a=$t(null==r?void 0:r.initialFocusRef),n=null==r?void 0:r.disabled,i=e.useRef(),d=e.useRef(null);function l(){var e;null===(e=i.current)||void 0===e||e.abort(),null!=r&&r.restoreFocusOnCleanUp&&d.current instanceof HTMLElement&&(d.current.focus(),d.current=null)}return d.current||null!=r&&r.disabled||(d.current=document.activeElement),e.useEffect((()=>{if(o.current instanceof HTMLElement){var e;if(!n)return i.current=wo(o.current,null!==(e=a.current)&&void 0!==e?e:void 0),()=>{l()};l()}}),[o,a,n,...t]),{containerRef:o,initialFocusRef:a}}function $o({icon:r,...t}){return e.createElement(r,t)}$o.displayName="Octicon";const ko=s($o)`
  ${Nr}
  ${et}
`;function So(r={},t=[]){const o=$t(r.containerRef),n=!!r.activeDescendantFocus,i=$t("boolean"!=typeof r.activeDescendantFocus&&r.activeDescendantFocus?r.activeDescendantFocus:void 0),d=r.disabled,l=e.useRef();return a((()=>{if(o.current instanceof HTMLElement&&(!n||i.current instanceof HTMLElement)){if(!d){var e;const t={...r,activeDescendantControl:null!==(e=i.current)&&void 0!==e?e:void 0};return l.current=Ot(o.current,t),()=>{var e;null===(e=l.current)||void 0===e||e.abort()}}var t;null===(t=l.current)||void 0===t||t.abort()}}),[d,...t]),{containerRef:o,activeDescendantControlRef:i}}ko.defaultProps={size:16};const To={};function Eo(){const e=To.__default__;if(!e||!document.body.contains(e)){let e=document.getElementById("__primerPortalRoot__");if(!(e instanceof Element)){e=document.createElement("div"),e.setAttribute("id","__primerPortalRoot__"),e.style.position="absolute",e.style.top="0",e.style.left="0";const r=document.querySelector("[data-portal-root]");r?r.appendChild(e):document.body.appendChild(e)}!function(e,r="__default__"){To[r]=e}(e)}}const Io=({children:r,onMount:t,containerName:o})=>{const a=document.createElement("div");a.style.position="relative",a.style.zIndex="1";const n=e.useRef(a);return e.useLayoutEffect((()=>{let e=o;void 0===e&&(e="__default__",Eo());const r=To[e];if(!r)throw new Error(`Portal container '${o}' is not yet registered. Container must be registered with registerPortal before use.`);const a=n.current;return r.appendChild(a),null==t||t(),()=>{r.removeChild(a)}}),[n]),h(r,n.current)};function Co(...r){const o=t(null);return e.useLayoutEffect((()=>{function e(e=null){for(const t of r){if(!t)return;"function"==typeof t?t(e):t.current=e}}return e(o.current),()=>{e(o.current)}}),[...r,o.current]),o}const No=s("div")`
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: rgba(0, 0, 0, 0.4);
  animation: dialog-backdrop-appear ${"200ms"} ${Cr("animation.easeOutCubic")};

  @keyframes dialog-backdrop-appear {
    0% {
      opacity: 0;
    }
    100% {
      opacity: 1;
    }
  }
`,Ao={small:"480px",large:"640px",auto:"auto"},zo={small:"296px",medium:"320px",large:"480px",xlarge:"640px"},Ro=s.div`
  display: flex;
  flex-direction: column;
  background-color: ${Cr("colors.bg.overlay")};
  box-shadow: ${Cr("shadows.overlay.shadow")};
  min-width: 296px;
  max-width: calc(100vw - 64px);
  max-height: calc(100vh - 64px);
  width: ${e=>{var r;return zo[null!==(r=e.width)&&void 0!==r?r:"xlarge"]}};
  height: ${e=>{var r;return Ao[null!==(r=e.height)&&void 0!==r?r:"auto"]}};
  border-radius: 12px;
  opacity: 1;
  animation: overlay--dialog-appear ${"200ms"} ${Cr("animation.easeOutCubic")};

  @keyframes overlay--dialog-appear {
    0% {
      opacity: 0;
      transform: scale(0.5);
    }
    100% {
      opacity: 1;
      transform: scale(1);
    }
  }

  ${Nr};
  ${Lr};
  ${et};
`,Lo=({dialogLabelId:r,title:t,subtitle:a,dialogDescriptionId:n,onClose:i})=>{const d=o((()=>{i("close-button")}),[i]);return e.createElement(Ko.Header,null,e.createElement(rt,{display:"flex"},e.createElement(rt,{display:"flex",px:2,py:"6px",flexDirection:"column",flexGrow:1},e.createElement(Ko.Title,{id:r},null!=t?t:"Dialog"),a&&e.createElement(Ko.Subtitle,{id:n},a)),e.createElement(Ko.CloseButton,{onClose:d})))};Lo.displayName="DefaultHeader";const Mo=({children:r})=>e.createElement(Ko.Body,null,r);Mo.displayName="DefaultBody";const Oo=({footerButtons:r})=>{const{containerRef:t}=So({bindKeys:Nt.ArrowHorizontal|Nt.Tab,focusInStrategy:"closest"});return r?e.createElement(Ko.Footer,{ref:t},e.createElement(Ko.Buttons,{buttons:r})):null},Ho=e.forwardRef(((r,o)=>{const{title:a="Dialog",subtitle:n="",renderHeader:i,renderBody:d,renderFooter:l,onClose:c,role:s="dialog",width:f="xlarge",height:b="auto"}=r,g=zt(),p=zt(),u={...r,title:a,subtitle:n,role:s,dialogLabelId:g,dialogDescriptionId:p},m=t(null),h=Co(m,o),x=t(null);Bo({containerRef:m,restoreFocusOnCleanUp:!0}),Bt((e=>{c("escape"),e.preventDefault()}),[c]);const v=(null!=i?i:Lo)(u),y=(null!=d?d:Mo)(u),w=(null!=l?l:Oo)(u);return e.createElement(e.Fragment,null,e.createElement(Io,null,e.createElement(No,{ref:x},e.createElement(Ro,{width:f,height:b,ref:h,role:s,"aria-labelledby":g,"aria-describedby":p},v,y,w))))}));Ho.displayName="Dialog";const Po=s(rt).attrs({as:"header"})`
  box-shadow: 0 1px 0 ${Cr("colors.border.overlay")};
  padding: ${Cr("space.2")};
  z-index: 1;
  flex-shrink: 0;
`,Wo=s(rt)`
  font-size: ${Cr("fontSizes.1")};
  font-weight: ${Cr("fontWeights.bold")};
`,Do=s(rt)`
  font-size: ${Cr("fontSizes.0")};
  margin-top: ${Cr("space.1")};
  color: ${Cr("colors.text.tertiary")};
`,jo=s(rt)`
  flex-grow: 1;
  overflow: auto;
  padding: ${Cr("space.3")};
`,Fo=s(rt).attrs({as:"footer"})`
  box-shadow: 0 -1px 0 ${Cr("colors.border.overlay")};
  padding: ${Cr("space.3")};
  display: flex;
  flex-flow: wrap;
  justify-content: flex-end;
  z-index: 1;
  flex-shrink: 0;

  button {
    margin-left: ${Cr("space.1")};
    &:first-child {
      margin-left: 0;
    }
  }
`,_o={normal:jt,primary:Gt,danger:Ft},Uo=s(jt)`
  border-radius: 4px;
  background: transparent;
  border: 0;
  vertical-align: middle;
  color: ${Cr("colors.text.secondary")};
  padding: ${Cr("space.2")};
  align-self: flex-start;
  line-height: normal;
  box-shadow: none;
`,Go=({onClose:r})=>e.createElement(Uo,{"aria-label":"Close",onClick:r},e.createElement(ko,{icon:ao}));Go.displayName="CloseButton";const Ko=Object.assign(Ho,{Header:Po,Title:Wo,Subtitle:Do,Body:jo,Footer:Fo,Buttons:({buttons:o})=>{const n=t(null);let i=0;const[d,l]=r(0);return a((()=>{var e;1===d?null===(e=n.current)||void 0===e||e.focus():l(d+1)}),[d]),e.createElement(e.Fragment,null,o.map(((r,t)=>{const{content:o,buttonType:a="normal",autoFocus:d=!1,...l}=r,c=_o[a];return e.createElement(c,Or({key:t},l,{ref:d&&0===i?(i++,n):null}),o)})))},CloseButton:Go}),Xo=s.header`
  padding: ${Cr("space.2")};
  display: flex;
  flex-direction: row;
`,Vo=s(rt)`
  font-size: ${Cr("fontSizes.3")};
  font-weight: ${Cr("fontWeights.bold")};
  padding: 6px ${Cr("space.2")};
  flex-grow: 1;
`,Yo=({title:r,onClose:t,dialogLabelId:a})=>{const n=o((()=>{t("close-button")}),[t]);return e.createElement(Xo,null,e.createElement(Vo,{id:a},r),e.createElement(Ko.CloseButton,{onClose:n}))};Yo.displayName="ConfirmationHeader";const qo=s(rt)`
  font-size: ${Cr("fontSizes.1")};
  padding: 0 ${Cr("space.3")} ${Cr("space.3")} ${Cr("space.3")};
  color: ${Cr("colors.text.tertiary")};
  flex-grow: 1;
`,Jo=({children:r})=>e.createElement(qo,null,r);Jo.displayName="ConfirmationBody";const Zo=s(rt)`
  display: grid;
  grid-auto-flow: column;
  grid-auto-columns: max-content;
  grid-gap: ${Cr("space.2")};
  align-items: end;
  justify-content: end;
  padding: ${Cr("space.1")} ${Cr("space.3")} ${Cr("space.3")};
`,Qo=({footerButtons:r})=>{const{containerRef:t}=So({bindKeys:Nt.ArrowHorizontal|Nt.Tab,focusInStrategy:"closest"});return e.createElement(Zo,{ref:t},e.createElement(Ko.Buttons,{buttons:null!=r?r:[]}))};Qo.displayName="ConfirmationFooter";const ea=r=>{const{onClose:t,title:a,cancelButtonContent:n="Cancel",confirmButtonContent:i="OK",confirmButtonType:d="normal",children:l}=r,c=[{content:n,onClick:o((()=>{t("cancel")}),[t]),autoFocus:!0},{content:i,buttonType:d,onClick:o((()=>{t("confirm")}),[t])}];return e.createElement(Ko,{onClose:t,title:a,footerButtons:c,role:"alertdialog",width:"medium",renderHeader:Yo,renderBody:Jo,renderFooter:Qo},l)};function ra(){const{theme:r,colorMode:t,dayScheme:a,nightScheme:n}=Jr();return o((o=>async function(r,t){const{content:o,...a}=t;return new Promise((t=>{const n=document.createElement("div");m.render(e.createElement(qr,r,e.createElement(ea,Or({},a,{onClose:e=>{m.unmountComponentAtNode(n),t("confirm"===e)}}),o)),n)}))}({theme:r,colorMode:t,dayScheme:a,nightScheme:n},o)),[r,t,a,n])}ea.displayName="ConfirmationDialog";const ta=s.div`
   {
    /* 6px vertical padding + 20px line height = 32px total height
     *
     * TODO: When rem-based spacing on a 4px scale lands, replace
     * hardcoded '6px' with 'calc((${Cr("space.s32")} - ${Cr("space.20")}) / 2)'.
     */
  }
  padding: 6px ${Cr("space.3")};
  font-size: ${Cr("fontSizes.0")};
  font-weight: ${Cr("fontWeights.bold")};
  color: ${Cr("colors.text.secondary")};

  ${({variant:e})=>"filled"===e&&g(["background:",";margin:"," 0;border-top:1px solid ",";border-bottom:1px solid ",";&:first-child{margin-top:0;}"],Cr("colors.bg.tertiary"),Cr("space.2"),Cr("colors.border.tertiary"),Cr("colors.border.tertiary"))}

  ${et}
`;function oa({variant:r="subtle",title:t,auxiliaryText:o,children:a,...n}){return e.createElement(ta,Or({role:"heading",variant:r},n),t,o&&e.createElement("span",null,"auxiliaryText"))}oa.displayName="Header";const aa=s.div`
  ${et}
`;function na({header:r,items:t,...o}){return e.createElement(aa,o,r&&e.createElement(oa,r),t)}na.displayName="Group";const ia=s.div`
  ${Ar}
  ${Nr}
  display: ${e=>e.inline?"inline-block":"inherit"};
  overflow: hidden;
  text-overflow: ellipsis;
  vertical-align: ${e=>e.inline?"top":"initial"};
  white-space: nowrap;
  ${ye}
  ${e=>e.expandable?"&:hover { max-width: 10000px; }":""}
  ${et};
`;ia.defaultProps={expandable:!1,inline:!1,maxWidth:125};const da=s.div`
  height: 1px;
  background: ${Cr("colors.selectMenu.borderSecondary")};
  margin-top: calc(${Cr("space.2")} - 1px);
  margin-bottom: ${Cr("space.2")};
`;function la(){return e.createElement(da,null)}la.displayName="Divider",la.renderItem=la;const ca={default:{hover:{light:"rgba(46, 77, 108, 0.06)",dark:"rgba(201, 206, 212, 0.12)",dark_dimmed:"rgba(201, 206, 212, 0.12)"},focus:{light:"rgba(54, 77, 100, 0.16)",dark:"rgba(201, 206, 212, 0.24)",dark_dimmed:"rgba(201, 206, 212, 0.24)"}},danger:{hover:{light:"rgba(234, 74, 90, 0.08)",dark:"rgba(248, 81, 73, 0.16)",dark_dimmed:"rgba(248, 81, 73, 0.16)"},focus:{light:"rgba(234, 74, 90, 0.14)",dark:"rgba(248, 81, 73, 0.24)",dark_dimmed:"rgba(248, 81, 73, 0.24)"}}},sa=(e="default",r)=>{if(r)return{color:Cr("colors.text.disabled"),iconColor:Cr("colors.text.disabled"),annotationColor:Cr("colors.text.disabled"),hoverCursor:"default"};switch(e){case"danger":return{color:Cr("colors.text.danger"),iconColor:Cr("colors.icon.danger"),annotationColor:Cr("colors.text.disabled"),hoverCursor:"pointer"};default:return{color:"inherit",iconColor:Cr("colors.text.secondary"),annotationColor:Cr("colors.text.secondary"),hoverCursor:"pointer"}}},fa=s.div`
  display: flex;
  min-width: 0;

  /* Required for dividers */
  position: relative;
  flex-grow: 1;
`,ba=s.div`
  align-items: baseline;
  display: flex;
  min-width: 0;
  flex-direction: var(--main-content-flex-direction);
  flex-grow: 1;
`,ga=s.div`
  /* 6px vertical padding + 20px line height = 32px total height
   *
   * TODO: When rem-based spacing on a 4px scale lands, replace
   * hardcoded '6px' with 'calc((${Cr("space.s32")} - ${Cr("space.20")}) / 2)'.
   */
  padding: 6px ${Cr("space.2")};
  display: flex;
  border-radius: ${Cr("radii.2")};
  color: ${({variant:e,item:r})=>sa(e,null==r?void 0:r.disabled).color};
  // 2 frames on a 60hz monitor
  transition: background 33.333ms linear;

  @media (hover: hover) and (pointer: fine) {
    :hover {
      // allow override in case another item in the list is active/focused
      background: var(--item-hover-bg-override, ${({hoverBackground:e})=>e});
      cursor: ${({variant:e,item:r})=>sa(e,null==r?void 0:r.disabled).hoverCursor};
    }
  }

  // Item dividers
  :not(:first-of-type):not(${da} + &):not(${ta} + &) {
    margin-top: ${({showDivider:e})=>e?"1px":"0"};

    ${fa}::before {
      content: ' ';
      display: block;
      position: absolute;
      width: 100%;
      top: -7px;
      // NB: This 'get' wont execute if its moved into the arrow function below.
      border: 0 solid ${Cr("colors.selectMenu.borderSecondary")};
      border-top-width: ${({showDivider:e})=>e?"1px":"0"};
    }
  }

  // Item dividers should not be visible:
  // - above Hovered
  &:hover ${fa}::before,
  // - below Hovered
  // '*' instead of '&' because '&' maps to separate class names depending on 'variant'
  :hover + * ${fa}::before {
    // allow override in case another item in the list is active/focused
    border-color: var(--item-hover-divider-border-color-override, transparent) !important;
  }

  // - above Focused
  &:focus ${fa}::before,
  // - below Focused
  // '*' instead of '&' because '&' maps to separate class names depending on 'variant'
  :focus + * ${fa}::before,
  // - above Active Descendent
  &[${"data-is-active-descendant"}] ${fa}::before,
  // - below Active Descendent
  [${"data-is-active-descendant"}] + & ${fa}::before {
    // '!important' because all the ':not's above give higher specificity
    border-color: transparent !important;
  }

  // Active Descendant
  &[${"data-is-active-descendant"}='${"activated-directly"}'] {
    background: ${({focusBackground:e})=>e};
  }
  &[${"data-is-active-descendant"}='${"activated-indirectly"}'] {
    background: ${({hoverBackground:e})=>e};
  }

  &:focus {
    background: ${({focusBackground:e})=>e};
    outline: none;
  }

  &:active {
    background: ${({focusBackground:e})=>e};
  }

  ${et}
`,pa=s.span``,ua=s.div`
  // Match visual height to adjacent text line height.
  // TODO: When rem-based spacing on a 4px scale lands, replace
  // hardcoded '20px' with '${Cr("space.s20")}'.
  height: 20px;
  width: ${Cr("space.3")};
  margin-right: ${Cr("space.2")};
`,ma=s(ua)`
  svg {
    fill: ${({variant:e,disabled:r})=>sa(e,r).iconColor};
    font-size: ${Cr("fontSizes.0")};
  }
`,ha=s(ma)`
  flex-shrink: 0;
  display: flex;
  flex-direction: column;
  justify-content: center;
`,xa=s(ma)`
  color: ${({variant:e,disabled:r})=>sa(e,r).annotationColor}};
  margin-left: ${Cr("space.2")};
  margin-right: 0;
  width: auto;
  div:nth-child(2) {
    margin-left: ${Cr("space.2")};
  }
`,va=s.span`
  color: ${Cr("colors.text.secondary")};
  font-size: ${Cr("fontSizes.0")};
  // TODO: When rem-based spacing on a 4px scale lands, replace
  // hardcoded '16px' with '${Cr("lh-12")}'.
  line-height: 16px;
  margin-left: var(--description-container-margin-left);
  min-width: 0;
  flex-grow: 1;
  flex-basis: var(--description-container-flex-basis);
`,ya=s.input`
  pointer-events: none;
`;function wa(r){const{text:t,description:a,descriptionVariant:i="inline",selected:d,selectionVariant:l,leadingVisual:c,trailingIcon:s,trailingText:f,variant:b="default",showDivider:g,disabled:p,onAction:u,onKeyPress:m,children:h,onClick:x,id:v,...y}=r,w=n((()=>zt()),[]),B=n((()=>zt()),[]),$=o((e=>{if(p)return;null==m||m(e);e.target instanceof HTMLInputElement&&"checkbox"===e.target.type&&" "===e.key||!e.defaultPrevented&&[" ","Enter"].includes(e.key)&&(null==u||u(r,e))}),[u,p,r,m]),k=o((e=>{p||(null==x||x(e),e.defaultPrevented||null==u||u(r,e))}),[u,p,r,x]),S=ca[b],T=Zr(S.hover,"inherit"),E=Zr(S.focus,"inherit"),{theme:I}=Jr();return e.createElement(ga,Or({tabIndex:p?void 0:-1,variant:b,showDivider:g,"aria-selected":d,"aria-labelledby":t?w:void 0,"aria-describedby":a?B:void 0},y,{"data-id":v,onKeyPress:$,onClick:k,hoverBackground:p?"inherit":T,focusBackground:p?"inherit":E}),!!d===d&&e.createElement(ua,null,"multiple"===l?e.createElement(e.Fragment,null,e.createElement(ya,{disabled:p,tabIndex:-1,type:"checkbox",checked:d,"aria-label":t,readOnly:!0,"aria-readonly":"false"})):d&&e.createElement(Jt,{fill:null==I?void 0:I.colors.text.primary})),c&&e.createElement(ha,{variant:b,disabled:p},e.createElement(c,null)),e.createElement(fa,null,e.createElement(ba,{style:{"--main-content-flex-direction":"inline"===i?"row":"column"}},h,t?e.createElement(pa,{id:w},t):null,a?e.createElement(va,{id:B,style:{"--description-container-margin-left":"inline"===i?Cr("space.2")(I):0,"--description-container-flex-basis":"inline"===i?0:"auto"}},"block"===i?a:e.createElement(ia,{title:a,inline:!0,maxWidth:"100%"},a)):null),s||f?e.createElement(xa,{variant:b,disabled:p},f,s&&e.createElement(s,null)):null))}wa.displayName="Item";const Ba=s.div`
  font-size: ${Cr("fontSizes.1")};
  /* 14px font-size * 1.428571429 = 20px line height
   *
   * TODO: When rem-based spacing on a 4px scale lands, replace
   * hardcoded '20px'
   */
  line-height: 20px;

  &[${"data-has-active-descendant"}], &:focus-within {
    --item-hover-bg-override: none;
    --item-hover-divider-border-color-override: ${Cr("colors.selectMenu.borderSecondary")};
  }
`;const $a=e.forwardRef(((r,t)=>{const{firstGroupStyle:o,lastGroupStyle:a,headerStyle:n,itemStyle:i}=function(e="inset"){switch(e){case"full":return{headerStyle:{paddingX:Cr("space.2")},itemStyle:{borderRadius:0}};default:return{firstGroupStyle:{marginTop:Cr("space.2")},lastGroupStyle:{marginBottom:Cr("space.2")},itemStyle:{marginX:Cr("space.2")}}}}(r.variant),d=(t,o,a)=>{var n,d,l;const c="renderItem"in t&&t.renderItem||r.renderItem||wa,s=null!==(n=null!==(d=t.key)&&void 0!==d?d:null===(l=t.id)||void 0===l?void 0:l.toString())&&void 0!==n?n:a.toString();return e.createElement(c,Or({showDivider:r.showItemDividers,selectionVariant:r.selectionVariant},t,{key:s,sx:{...i,...t.sx},item:o}))};let l=[];if(function(e){return"groupMetadata"in e}(r)){const e=r.groupMetadata.reduce(((e,r)=>e.set(r.groupId,r)),new Map);for(const t of r.items){var c,s,f;const r=e.get(t.groupId),o=null!==(c=null==r||null===(s=r.items)||void 0===s?void 0:s.length)&&void 0!==c?c:0;e.set(t.groupId,{...r,items:[...null!==(f=null==r?void 0:r.items)&&void 0!==f?f:[],d({showDivider:null==r?void 0:r.showItemDividers,...r&&"renderItem"in r&&{renderItem:r.renderItem},...t},t,o)]})}l=[...e.values()]}else l=[{items:r.items.map(((e,r)=>d(e,e,r))),groupId:"0"}];return e.createElement(Ba,Or({},r,{ref:t}),l.map((({header:t,...i},d)=>{const c="filled"===(null==t?void 0:t.variant),s=d>0&&!c;return e.createElement(e.Fragment,{key:i.groupId},s?e.createElement(la,{key:`${i.groupId}-divider`}):null,(t=>{var o;const a=(null!==(o="renderGroup"in t&&t.renderGroup)&&void 0!==o?o:r.renderGroup)||na;return e.createElement(a,Or({},t,{key:t.groupId}))})({sx:{...0===d&&o,...d===l.length-1&&a,...d>0&&!s&&{mt:2}},...t&&{header:{...t,sx:{...n,...t.sx}}},...i}))})))}));$a.displayName="ActionList";const ka=Object.assign($a,{Group:na,Item:wa,Divider:la}),Sa={xsmall:"192px",small:"256px",medium:"320px",large:"432px",xlarge:"600px",auto:"auto",initial:"auto"},Ta={small:"256px",medium:"320px",large:"480px",xlarge:"640px",xxlarge:"960px",auto:"auto"};const Ea=s.div`
  background-color: ${Cr("colors.bg.overlay")};
  box-shadow: ${Cr("shadows.overlay.shadow")};
  position: absolute;
  min-width: 192px;
  max-width: 640px;
  height: ${e=>Sa[e.height||"auto"]};
  max-height: ${e=>e.maxHeight&&Sa[e.maxHeight]};
  width: ${e=>Ta[e.width||"auto"]};
  border-radius: 12px;
  overflow: hidden;
  animation: overlay-appear ${200}ms ${Cr("animation.easeOutCubic")};

  @keyframes overlay-appear {
    0% {
      opacity: 0;
    }
    100% {
      opacity: 1;
    }
  }
  visibility: var(--styled-overlay-visibility);
  :focus {
    outline: none;
  }
  ${Nr};
  ${et};
`,Ia=e.forwardRef((({onClickOutside:r,role:o="none",initialFocusRef:n,returnFocusRef:i,ignoreClickRefs:l,onEscape:c,visibility:s="visible",height:f,top:b,left:g,anchorSide:p,portalContainerName:u,...m},h)=>{const x=t(null),v=Co(x,h),{theme:y}=Jr(),w=parseInt(Cr("space.2")(y).replace("px","")),B=Cr("animation.easeOutCubic")(y);return kt({overlayRef:x,returnFocusRef:i,onEscape:c,ignoreClickRefs:l,onClickOutside:r,initialFocusRef:n}),a((()=>{var e;"initial"===f&&null!==(e=v.current)&&void 0!==e&&e.clientHeight&&(v.current.style.height=`${v.current.clientHeight}px`)}),[f,v]),d((()=>{var e;const{x:r,y:t}=function(e){return null!=e&&e.endsWith("bottom")?{x:0,y:-1}:null!=e&&e.endsWith("top")?{x:0,y:1}:null!=e&&e.endsWith("right")?{x:-1,y:0}:null!=e&&e.endsWith("left")?{x:1,y:0}:{x:0,y:0}}(p);(r||t)&&null!==(e=x.current)&&void 0!==e&&e.animate&&"hidden"!==s&&x.current.animate({transform:[`translate(${w*r}px, ${w*t}px)`,"translate(0, 0)"]},{duration:200,easing:B})}),[p,w,B,s]),e.createElement(Io,{containerName:u},e.createElement(Ea,Or({height:f,role:o},m,{ref:v,style:{top:`${b||0}px`,left:`${g||0}px`,...m.style,"--styled-overlay-visibility":s}})))}));Ia.defaultProps={height:"auto",width:"auto"};const Ca=({renderAnchor:i,anchorRef:d,children:l,open:c,onOpen:s,onClose:f,height:b,width:g,overlayProps:p,focusTrapSettings:u,focusZoneSettings:m})=>{const h=$t(d),[x,v]=function(){const[e,a]=r(null),n=t(null);n.current=e;const i=o((e=>{n.current=e,a(e)}),[n]);return[n,i]}(),y=n(zt,[]),w=o((()=>null==f?void 0:f("click-outside")),[f]),B=o((()=>null==f?void 0:f("escape")),[f]),$=o((e=>{e.defaultPrevented||!c&&["ArrowDown","ArrowUp"," ","Enter"].includes(e.key)&&(null==s||s("anchor-key-press"),e.preventDefault())}),[c,s]),k=o((e=>{e.defaultPrevented||0!==e.button||(c?null==f||f("anchor-click"):null==s||s("anchor-click"))}),[c,s,f]),{position:S}=ho({anchorElementRef:h,floatingElementRef:x},[x.current]);return a((()=>{!c&&x.current&&v(null)}),[c,x,v]),So({containerRef:x,disabled:!c||!S,...m}),Bo({containerRef:x,disabled:!c||!S,...u}),e.createElement(e.Fragment,null,i&&i({ref:h,id:y,"aria-labelledby":y,"aria-haspopup":"true",tabIndex:0,onClick:k,onKeyDown:$}),c?e.createElement(Ia,Or({returnFocusRef:h,onClickOutside:w,ignoreClickRefs:[h],onEscape:B,ref:v,role:"none",visibility:S?"visible":"hidden",height:b,width:g,top:(null==S?void 0:S.top)||0,left:(null==S?void 0:S.left)||0,anchorSide:null==S?void 0:S.anchorSide},p),l):null)};function Na(e,t,a){const[n,i]=r(a);return[null!=e?e:n,o((e=>{i(e),t&&t(e)}),[t])]}Ca.displayName="AnchoredOverlay";const Aa=r=>e.createElement(wa,Or({role:"menuitem"},r));Aa.displayName="ActionMenuItem",Aa.displayName="ActionMenu.Item";const za=({anchorContent:r,renderAnchor:t=(r=>e.createElement(jt,r)),anchorRef:a,onAction:i,open:d,setOpen:l,overlayProps:c,items:s,...f})=>{const[b,g]=Na(d,l,!1),p=$t(a),u=o((()=>g(!0)),[g]),m=o((()=>g(!1)),[g]),h=n((()=>null===t?null:e=>t({"aria-label":"menu",children:r,...e})),[r,t]),x=n((()=>s.map((e=>({...e,role:"menuitem",onAction:(r,t)=>{var o;const a=null!==(o=e.onAction)&&void 0!==o?o:i;null==a||a(r,t),t.defaultPrevented||m()}})))),[s,i,m]);return e.createElement(Ca,{renderAnchor:h,anchorRef:p,open:b,onOpen:u,onClose:m,overlayProps:c},e.createElement($a,Or({},f,{role:"menu",items:x})))};za.displayName="ActionMenuBase",za.displayName="ActionMenu";const Ra=Object.assign(za,{Divider:la,Item:Aa});const La=s.img.attrs((e=>({height:e.size,width:e.size})))`
  display: inline-block;
  overflow: hidden; // Ensure page layout in Firefox should images fail to load
  line-height: ${Cr("lineHeights.condensedUltra")};
  vertical-align: middle;
  border-radius: ${e=>function({size:e,square:r}){return r?e&&e<=24?"4px":"6px":"50%"}(e)};
  ${Nr};
  ${et}
`;La.defaultProps={size:20,alt:"",square:!1};const Ma=s(La)`
  position: absolute;
  right: -15%;
  bottom: -9%;
  box-shadow: ${Cr("shadows.avatar.childShadow")};
`,Oa=({children:r,...t})=>{const o=e.Children.map(r,((r,t)=>e.isValidElement(r)?0===t?e.cloneElement(r,{size:40}):e.createElement(Ma,Or({bg:"bg.canvas"},r.props,{size:20})):r));return e.createElement(rt,Or({position:"relative",display:"inline-flex"},t),o)};Oa.displayName="AvatarPair",Oa.displayName="AvatarPair";var Ha,Pa={exports:{}};
/*!
  Copyright (c) 2018 Jed Watson.
  Licensed under the MIT License (MIT), see
  http://jedwatson.github.io/classnames
*/Ha=Pa,function(){var e={}.hasOwnProperty;function r(){for(var t=[],o=0;o<arguments.length;o++){var a=arguments[o];if(a){var n=typeof a;if("string"===n||"number"===n)t.push(a);else if(Array.isArray(a)){if(a.length){var i=r.apply(null,a);i&&t.push(i)}}else if("object"===n)if(a.toString===Object.prototype.toString)for(var d in a)e.call(a,d)&&a[d]&&t.push(d);else t.push(a.toString())}}return t.join(" ")}Ha.exports?(r.default=r,Ha.exports=r):window.classNames=r}();var Wa=Pa.exports;const Da=s.span`
  display: flex;
  position: relative;
  height: 20px;
  min-width: ${e=>1===e.count?"20px":2===e.count?"30px":"38px"};

  .pc-AvatarItem {
    flex-shrink: 0;
    height: 20px;
    width: 20px;
    box-shadow: 0 0 0 1px ${Cr("colors.bg.canvas")};
    margin-left: -11px;
    position: relative;
    overflow: hidden;
    transition: margin 0.2s ease-in-out, opacity 0.2s ease-in-out, visibility 0.2s ease-in-out,
      box-shadow 0.1s ease-in-out;

    &:first-child {
      margin-left: 0;
    }
    &:nth-child(n + 4) {
      display: none;
    }
  }

  &.pc-AvatarStack--two {
    min-width: 30px;
    .pc-AvatarItem {
      &:nth-child(n + 3) {
        display: none;
      }
    }
  }

  &.pc-AvatarStack--three-plus {
    min-width: 38px;
    .pc-AvatarItem {
      &:nth-child(3) {
        opacity: ${55}%;
        margin-left: -17px;
      }
      &:nth-child(4) {
        opacity: ${40}%;
        margin-left: -17px;
      }
      &:nth-child(5) {
        opacity: ${25}%;
        margin-left: -17px;
      }
      &:nth-child(n + 4) {
        display: block;
      }
      &:nth-child(n + 6) {
        opacity: 0;
        visibility: hidden;
      }
    }
  }

  &.pc-AvatarStack--right {
    justify-content: flex-end;
    .pc-AvatarItem {
      margin-left: 0 !important;
      margin-right: -11px;

      &:first-child {
        margin-right: 0;
      }
    }

    .pc-AvatarStackBody {
      flex-direction: row-reverse;

      &:hover {
        .pc-AvatarItem {
          margin-right: ${Cr("space.1")}!important;
          margin-left: 0 !important;

          &:first-child {
            margin-right: 0 !important;
          }
        }
      }
    }
  }

  &.pc-AvatarStack--three-plus.pc-AvatarStack--right {
    .pc-AvatarItem {
      &:nth-child(3) {
        margin-right: -17px;
      }
      &:nth-child(4) {
        margin-right: -17px;
      }
      &:nth-child(5) {
        margin-right: -17px;
      }
    }
  }

  .pc-AvatarStackBody:hover {
    width: auto;

    .pc-AvatarItem {
      margin-left: ${Cr("space.1")};
      opacity: 100%;
      visibility: visible;
      box-shadow: 0 0 0 4px ${Cr("colors.bg.canvas")};
      &:first-child {
        margin-left: 0;
      }
    }
  }

  ${Nr}
  ${et};
`,ja=({children:r,alignRight:t,...o})=>{const a=e.Children.count(r),n=Wa({"pc-AvatarStack--two":2===a,"pc-AvatarStack--three-plus":a>2,"pc-AvatarStack--right":t});return e.createElement(Da,Or({count:a,className:n},o),e.createElement(rt,{position:"absolute",display:"flex",width:"38px",className:"pc-AvatarStackBody"},(r=>e.Children.map(r,((r,t)=>e.isValidElement(r)?e.cloneElement(r,{className:Wa(r.props.className,"pc-AvatarItem"),sx:{zIndex:10-t,...r.props.sx}}):r)))(r)))};ja.displayName="AvatarStack";const Fa=s.a`
  display: inline-block;
  padding: 2px 6px;
  font-size: ${Cr("fontSizes.0")};
  font-family: ${Cr("fonts.mono")};
  color: ${Cr("colors.branchName.text")};
  background-color: ${Cr("colors.branchName.bg")};
  border-radius: ${Cr("radii.2")};
  ${Nr};
  ${et};
`,_a=s.li`
  display: inline-block;
  white-space: nowrap;
  list-style: none;
  &::after {
    padding-right: 0.5em;
    padding-left: 0.5em;
    color: ${Cr("colors.text.disabled")};
    font-size: ${Cr("fontSizes.1")};
    content: '/';
  }
  &:first-child {
    margin-left: 0;
  }
  &:last-child {
    &::after {
      content: none;
    }
  }
`,Ua=s.nav`
  display: flex;
  justify-content: space-between;
  ${Nr};
  ${Mr};
  ${et};
`;function Ga({className:r,children:t,theme:o,...a}){const n=Wa(r,"Breadcrumb"),i=e.Children.map(t,(r=>e.createElement(_a,{theme:o},r)));return e.createElement(Ua,Or({className:n,"aria-label":"breadcrumb",theme:o},a),e.createElement(rt,{as:"ol",my:0,pl:0},i))}Ga.displayName="Breadcrumb";const Ka=s.a.attrs((e=>({activeClassName:"string"==typeof e.to?"selected":"",className:Wa(e.selected&&"selected",e.className),"aria-current":e.selected?"page":null})))`
  color: ${Cr("colors.text.link")};
  display: inline-block;
  font-size: ${Cr("fontSizes.1")};
  text-decoration: none;
  &:hover {
    text-decoration: underline;
  }
  &.selected {
    color: ${Cr("colors.text.primary")};
    pointer-events: none;
  }
  ${Nr}
  ${et};
`;Ga.displayName="Breadcrumb",Ka.displayName="Breadcrumb.Item";var Xa=Object.assign(Ga,{Item:Ka});const Va={top:"Bottom",right:"Left",bottom:"Top",left:"Right"},Ya={top:"Left",right:"Top",bottom:"Left",left:"Top"};function qa(e,r,t){const o=Va[e].toLowerCase(),a=Ya[e].toLowerCase();return{[o]:"100%",[r||a]:r?t:"50%"}}const Ja=kr({prop:"bg",key:"colors"}),Za=kr({prop:"borderColor",key:"colors"}),Qa=kr({prop:"borderWidth",key:"borderWidths",scale:[0,1]});function en(r){var t;const o=e.useContext(p),a={...r,theme:null!==(t=r.theme)&&void 0!==t?t:o},{bg:n}=Ja(a),{borderColor:i}=Za(a),{borderWidth:d}=Qa(a),{size:l=8,location:c="bottom"}=r,[s,f]=function(e){const[r,t]=e.split("-");return[r,t]}(c),b=Ya[s],g=[-l,0],u=[0,l],m=[l,0],h=`M${g}L${u}L${m}L${g}Z`,x=`M${g}L${u}L${m}`,v={top:`translate(${[l,2*l]}) rotate(180)`,right:`translate(${[0,l]}) rotate(-90)`,bottom:`translate(${[l,0]})`,left:`translate(${[2*l,l]}) rotate(90)`}[s];return e.createElement("svg",{width:2*l,height:2*l,style:{pointerEvents:"none",position:"absolute",...qa(s,f,l),[`margin${b}`]:f?null:-l}},e.createElement("g",{transform:v},e.createElement("path",{d:h,fill:n}),e.createElement("path",{d:x,fill:"none",stroke:i,strokeWidth:d})))}en.displayName="Caret",en.locations=["top","top-left","top-right","right","right-top","right-bottom","bottom","bottom-left","bottom-right","left","left-top","left-bottom"],en.defaultProps={bg:"bg.canvas",borderColor:"border.primary",borderWidth:1};const rn={small:56,medium:96,large:128},tn=s.div`
  display: ${e=>e.inline?"inline-flex":"flex"};
  align-items: center;
  justify-content: center;
  background-color: ${Cr("colors.bg.canvas")};
  border-radius: 50%;
  box-shadow: ${Cr("shadows.shadow.medium")};
  ${Nr};
  ${({size:e,variant:r="medium"})=>{const t=(o=e,!isNaN(parseFloat(o))&&isFinite(o)?e:rn[r]);var o;return{width:t,height:t}}};
  ${et};
`,on=s(ko)`
  height: auto;
  max-width: 60%;
  max-height: 55%;
`;tn.defaultProps={inline:!1},on.displayName="CircleBadge.Icon";var an=Object.assign(tn,{Icon:on});function nn(r){const{size:t,as:o}=r,{icon:a,bg:n,...i}=r;return e.createElement(rt,{as:o,bg:n,overflow:"hidden",borderWidth:0,size:t,borderRadius:"50%",borderStyle:"solid",borderColor:"border.primary"},e.createElement(rt,Or({display:"flex"},i,{alignItems:"center",justifyContent:"center"}),e.createElement(a,{size:t})))}nn.displayName="CircleOcticon",nn.defaultProps={...rt.defaultProps,size:32};const dn=s.span`
  display: inline-block;
  padding: 2px 5px;
  font-size: ${Cr("fontSizes.0")};
  font-weight: ${Cr("fontWeights.bold")};
  line-height: ${Cr("lineHeights.condensedUltra")};
  border-radius: 20px;
  ${({scheme:e,...r})=>({color:"secondary"===e?Cr("colors.counter.text")(r):"primary"===e?Cr("colors.counter.primary.text")(r):Cr("colors.counter.text")(r)})};
  ${({scheme:e,...r})=>({backgroundColor:"secondary"===e?Cr("colors.counter.bg")(r):"primary"===e?Cr("colors.counter.primary.bg")(r):Cr("colors.counter.bg")(r)})};
  ${Nr};

  &:empty {
    display: none;
  }

  ${et};
`,ln=s.details`
  & > summary {
    list-style: none;
  }
  & > summary::-webkit-details-marker {
    display: none;
  }

  ${Nr}
  ${et};
`;ln.displayName="Details";const cn=()=>null;function sn(e){const r=e;return r.tabIndex>=0&&!r.disabled&&function(e){return!e.hidden&&(!e.type||"hidden"!==e.type)&&(e.offsetWidth>0||e.offsetHeight>0)}(r)}const fn=s.span`
  ${Ar};
  ${Nr};
  ${et};
`,bn=()=>null,gn=s.div`
  box-shadow: ${Cr("shadows.shadow.large")};
  border-radius: ${Cr("radii.2")};
  position: fixed;
  top: 0;
  left: 50%;
  transform: translateX(-50%);
  max-height: 80vh;
  z-index: 999;
  margin: 10vh auto;
  background-color: ${Cr("colors.bg.primary")};
  width: ${e=>e.narrow?"320px":e.wide?"640px":"440px"};
  outline: none;

  @media screen and (max-width: 750px) {
    width: 100vw;
    margin: 0;
    border-radius: 0;
    height: 100vh;
  }

  ${Rr};
  ${Nr};
  ${et};
`,pn=s(rt)`
  border-radius: ${Cr("radii.2")} ${Cr("radii.2")} 0px 0px;
  border-bottom: 1px solid ${Cr("colors.border.primary")};
  display: flex;

  @media screen and (max-width: 750px) {
    border-radius: 0px;
  }

  ${et};
`;function un({theme:r,children:t,backgroundColor:o="gray.1",...a}){return e.Children.toArray(t).every((e=>"string"==typeof e))&&(t=e.createElement(fn,{theme:r,color:"text.primary",fontSize:1,fontWeight:"bold",fontFamily:"sans-serif"},t)),e.createElement(pn,Or({theme:r,p:3,backgroundColor:o},a),t)}un.displayName="DialogHeader";const mn=s.span`
  &:before {
    position: fixed;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    display: block;
    cursor: default;
    content: ' ';
    background: transparent;
    z-index: 99;
    background: ${Cr("colors.fade.black50")};
  }
`,hn=i((({children:r,onDismiss:n=bn,isOpen:i,initialFocusRef:d,returnFocusRef:l,...c},s)=>{const f=t(null),b=Co(s),g=t(null),p=()=>{n(),l&&l.current&&l.current.focus()},{getDialogProps:u}=function({modalRef:e,overlayRef:r,isOpen:t,onDismiss:n=cn,initialFocusRef:i,closeButtonRef:d}){const l=o((t=>{e.current&&r.current&&!e.current.contains(t.target)&&r.current.contains(t.target)&&n()}),[n,e,r]);a((()=>{if(t)return document.addEventListener("click",l),()=>{document.removeEventListener("click",l)}}),[t,l]),a((()=>{t&&(i&&i.current?i.current.focus():d&&d.current&&d.current.focus())}),[t,i,d]);const c=o(((r,t)=>{if(e.current){const o=Array.from(e.current.querySelectorAll("*")).filter(sn);if(0===o.length)return;r.preventDefault();const a=document.activeElement;if(!a)return;const n=o.indexOf(a)+t,i=1===t?0:o.length-1;return o[n]||o[i]}}),[e]),s=o((e=>{const r=e.shiftKey?-1:1,t=c(e,r);t&&t.focus()}),[c]),f=o((e=>{switch(e.key){case"Tab":s(e);break;case"Escape":n(),e.stopPropagation()}}),[s,n]);return{getDialogProps:()=>({onKeyDown:f})}}({modalRef:b,onDismiss:p,isOpen:i,initialFocusRef:d,closeButtonRef:g,returnFocusRef:l,overlayRef:f});return i?e.createElement(e.Fragment,null,e.createElement(mn,{ref:f}),e.createElement(gn,Or({tabIndex:-1,ref:b,role:"dialog","aria-modal":"true"},c,u()),e.createElement(io,{ref:g,onClick:p,sx:{position:"absolute",top:"16px",right:"16px"}}),r)):null}));un.defaultProps={backgroundColor:"bg.tertiary"},un.propTypes={...rt.propTypes},un.displayName="Dialog.Header",hn.displayName="Dialog";var xn=Object.assign(hn,{Header:un});const vn=s(ln)`
  position: relative;
  display: inline-block;
`,yn=({children:r,className:t,...o})=>{const{getDetailsProps:a}=st({closeOnOutsideClick:!0});return e.createElement(vn,Or({className:t},a(),o),r)};yn.displayName="Dropdown";const wn=({children:r,...t})=>e.createElement(jt,Or({as:"summary","aria-haspopup":"true"},t),r,e.createElement(Bn,null));wn.displayName="DropdownButton";const Bn=s.div`
  border: 4px solid transparent;
  margin-left: 12px;
  border-top-color: currentcolor;
  border-bottom-width: 0;
  content: '';
  display: inline-block;
  height: 0;
  vertical-align: middle;
  width: 0;
  ${Nr};
  ${et};
`,$n=s.ul`
  background-clip: padding-box;
  background-color: ${Cr("colors.bg.overlay")};
  border: 1px solid ${Cr("colors.border.overlay")};
  border-radius: ${Cr("radii.2")};
  box-shadow: ${Cr("shadows.dropdown.shadow")};
  left: 0;
  list-style: none;
  margin-top: 2px;
  padding: 5px 0 5px 0 !important; //TODO: fix this override on our markdown styles
  position: absolute;
  top: 100%;
  width: 160px;
  z-index: 100;

  &::before {
    position: absolute;
    display: inline-block;
    content: '';
  }

  &::after {
    position: absolute;
    display: inline-block;
    content: '';
  }

  &::before {
    border: 8px solid transparent;
    border-bottom-color: ${Cr("colors.bg.overlay")};
  }

  &::after {
    border: 7px solid transparent;
    border-bottom-color: ${Cr("colors.bg.overlay")};
  }

  // stylelint-disable-next-line selector-max-type
  > ul {
    list-style: none;
  }
  ${e=>e.direction?((e,r)=>({w:`\n      top: 0;\n      right: 100%;\n      left: auto;\n      width: auto;\n      margin-top: 0;\n      margin-right: 10px;\n\n      &::before {\n        top: 10px;\n        right: -16px;\n        left: auto;\n        border-color: transparent;\n        border-left-color: ${Cr("colors.border.overlay")(e)};\n      }\n\n      &::after {\n        top: 11px;\n        right: -14px;\n        left: auto;\n        border-color: transparent;\n        border-left-color: ${Cr("colors.border.overlay")(e)};\n      }\n    `,e:`\n      top: 0;\n      left: 100%;\n      width: auto;\n      margin-top: 0;\n      margin-left: 10px;\n\n      &::before {\n        top: 10px;\n        left: -16px;\n        border-color: transparent;\n        border-right-color: ${Cr("colors.border.overlay")(e)};\n      }\n\n      &::after {\n        top: 11px;\n        left: -14px;\n        border-color: transparent;\n        border-right-color: ${Cr("colors.border.overlay")(e)};\n      }\n    `,ne:`\n      top: auto;\n      bottom: 100%;\n      left: 0;\n      margin-bottom: 3px;\n\n      &::before,\n      &::after {\n        top: auto;\n        right: auto;\n      }\n\n      &::before {\n        bottom: -8px;\n        left: 9px;\n        border-top: 8px solid ${Cr("colors.border.overlay")(e)};\n        border-bottom: 0;\n        border-left: 8px solid transparent;\n      }\n\n      &::after {\n        bottom: -7px;\n        left: 10px;\n        border-top: 7px solid ${Cr("colors.border.overlay")(e)};\n        border-right: 7px solid transparent;\n        border-bottom: 0;\n        border-left: 7px solid transparent;\n      }\n    `,s:"\n      right: 50%;\n      left: auto;\n      transform: translateX(50%);\n\n      &::before {\n        top: -16px;\n        right: 50%;\n        transform: translateX(50%);\n      }\n\n      &::after {\n        top: -14px;\n        right: 50%;\n        transform: translateX(50%);\n      }\n    ",sw:"\n      right: 0;\n      left: auto;\n\n      &::before {\n        top: -16px;\n        right: 9px;\n        left: auto;\n      }\n\n      &::after {\n        top: -14px;\n        right: 10px;\n        left: auto;\n      }\n    ",se:"\n      &::before {\n        top: -16px;\n        left: 9px;\n      }\n\n      &::after {\n        top: -14px;\n        left: 10px;\n      }\n    "}[r]))(e.theme,e.direction):""};
  ${Nr};
  ${et};
`,kn=s.li`
  display: block;
  padding: ${Cr("space.1")} 10px ${Cr("space.1")} 15px;
  overflow: hidden;
  color: ${Cr("colors.text.primary")};
  text-overflow: ellipsis;
  white-space: nowrap;
  a {
    color: ${Cr("colors.text.primary")};
    text-decoration: none;
    display: block;
    overflow: hidden;
    color: ${Cr("colors.text.primary")};
    text-overflow: ellipsis;
    white-space: nowrap;
  }

  &:focus,
  a:focus {
    color: ${Cr("colors.state.hover.primaryText")};
    text-decoration: none;
    background-color: ${Cr("colors.state.hover.primaryBg")};
  }

  &:hover,
  &:hover a {
    color: ${Cr("colors.state.hover.primaryText")};
    text-decoration: none;
    background-color: ${Cr("colors.state.hover.primaryBg")};
    outline: none;
  }
  ${Nr};
  ${et};
`;$n.defaultProps={direction:"sw"},$n.displayName="Dropdown.Menu",kn.displayName="Dropdown.Item",wn.defaultProps=jt.defaultProps,wn.displayName="Dropdown.Button",Bn.displayName="Dropdown.Caret",yn.defaultProps=ln.defaultProps;var Sn=Object.assign(yn,{Caret:Bn,Menu:$n,Item:kn,Button:wn});const Tn=e.forwardRef((({children:r,...t},o)=>e.createElement(jt,Or({ref:o,type:"button"},t),r,e.createElement(ko,{icon:oo,ml:1}))));function En({renderAnchor:r=(r=>e.createElement(Tn,r)),anchorRef:t,placeholder:a,selectedItem:i,onChange:d,overlayProps:l,items:c,open:s,onOpenChange:f,...b}){const[g,p]=Na(s,f,!1),u=o((()=>p(!0)),[p]),m=o((()=>p(!1)),[p]),h=$t(t),x=n((()=>null===r?null:e=>{var t;return r({...e,children:null!==(t=null==i?void 0:i.text)&&void 0!==t?t:a})}),[a,r,null==i?void 0:i.text]),v=n((()=>c.map((e=>({...e,role:"option",selected:e===i,onAction:(r,t)=>{var o;null===(o=e.onAction)||void 0===o||o.call(e,r,t),t.defaultPrevented||(m(),null==d||d(e===i?void 0:e))}})))),[c,d,m,i]);return e.createElement(Ca,{renderAnchor:x,anchorRef:h,open:g,onOpen:u,onClose:m,overlayProps:l},e.createElement($a,Or({},b,{role:"listbox",items:v})))}En.displayName="DropdownMenu",En.displayName="DropdownMenu";const In=s.div`
  ${Nr};
  display: flex;
  align-items: stretch;

  summary {
    border-radius: 0;
    border-top-left-radius: ${Cr("radii.2")};
    border-bottom-left-radius: ${Cr("radii.2")};
    border-right: 0;
  }
  .TextInput-wrapper {
    border-radius: 0;
    border-top-right-radius: ${Cr("radii.2")};
    border-bottom-right-radius: ${Cr("radii.2")};
    z-index: 1; // Allows the focus outline to show on top of the dropdown.
  }

  ${et}
`,Cn=s.ul`
  list-style-type: none;
  ${Nr};
  ${et};
`,Nn=({children:r,...t})=>{const o=e.Children.map(r,(r=>e.createElement("li",null,r)));return e.createElement(Cn,t,o)};Nn.displayName="FilterList";const An=s.a`
  position: relative;
  display: block;
  padding: ${e=>e.small?`${Cr("space.1")(e)} 10px`:`${Cr("space.2")(e)} 11px`};
  margin: ${e=>e.small?"0 0 2px":"0 0 5px 0"};
  overflow: hidden;
  font-size: ${Cr("fontSizes.1")};
  color: ${e=>e.selected?Cr("colors.state.selected.primaryText"):Cr("colors.text.secondary")};
  background-color: ${e=>e.selected?Cr("colors.state.selected.primaryBg"):""}!important;
  text-decoration: none;
  text-overflow: ellipsis;
  white-space: nowrap;
  cursor: pointer;
  border-radius: ${Cr("radii.1")};
  &:hover {
    text-decoration: none;
    background-color: ${Cr("colors.bg.tertiary")};
  }
  &:active {
    color: ${Cr("colors.state.selected.primaryText")};
    background-color: ${Cr("colors.state.selected.primaryBg")};
  }
  .count {
    float: right;
    font-weight: ${Cr("fontWeights.bold")};
  }
  ${Nr};
  ${et};
`;function zn({children:r,count:t,...o}){return e.createElement(An,o,t&&e.createElement("span",{title:"results",className:"count"},t),r)}zn.displayName="FilterListItem",Nn.defaultProps={m:0,p:0},zn.displayName="FilterList.Item";var Rn=Object.assign(Nn,{Item:zn});const Ln=be({variants:{default:{color:"alert.info.text",backgroundColor:"alert.info.bg",borderColor:"alert.info.border",svg:{color:"alert.info.icon"}},success:{color:"alert.success.text",backgroundColor:"alert.success.bg",borderColor:"alert.success.border",svg:{color:"alert.success.icon"}},danger:{color:"alert.error.text",backgroundColor:"alert.error.bg",borderColor:"alert.error.border",svg:{color:"alert.error.icon"}},warning:{color:"alert.warn.text",backgroundColor:"alert.warn.bg",borderColor:"alert.warn.border",svg:{color:"alert.warn.icon"}}}}),Mn=s.div`
  position: relative;
  color: ${Cr("colors.text.primary")};
  padding: ${Cr("space.3")};
  border-style: solid;
  border-width: ${e=>e.full?"1px 0px":"1px"};
  border-radius: ${e=>e.full?"0":Cr("radii.2")};
  margin-top: ${e=>e.full?"-1px":"0"};

  p:last-child {
    margin-bottom: 0;
  }

  svg {
    margin-right: ${Cr("space.2")};
  }

  ${Nr};
  ${Ln};
  ${et};
`;Mn.defaultProps={variant:"default"};const On=s.div`
  margin: ${Cr("space.3")} 0;
  font-weight: ${Cr("fontWeights.normal")};
  ${Nr};
  ${et};
`,Hn=s.label`
  display: block;
  margin: 0 0 ${Cr("space.2")};
  font-size: ${Cr("fontSizes.1")};
  font-weight: ${Cr("fontWeights.bold")};
  ${Ar};
  ${Nr};
  ${et};
`;Hn.displayName="FormGroup.Label";var Pn=Object.assign(On,{Label:Hn});const Wn=s.div`
  z-index: 32;
  display: flex;
  padding: ${Cr("space.3")};
  font-size: ${Cr("fontSizes.1")};
  line-height: ${Cr("lineHeights.default")};
  color: ${Cr("colors.header.text")};
  background-color: ${Cr("colors.header.bg")};
  align-items: center;
  flex-wrap: nowrap;

  ${Nr}
  ${zr}
  ${et};
`,Dn=s.div`
  display: flex;
  margin-right: ${Cr("space.3")};
  align-self: stretch;
  align-items: center;
  flex-wrap: nowrap;

  ${({full:e})=>e&&g(["flex:auto;"])};

  ${Nr};
  ${zr};
  ${et};
`;Dn.displayName="Header.Item";const jn=s.a.attrs((({to:e})=>"string"==typeof e?{"aria-current":"page"}:{}))`
  font-weight: ${Cr("fontWeights.bold")};
  color: ${Cr("colors.header.logo")};
  white-space: nowrap;
  cursor: pointer;
  text-decoration: none;
  display: flex;
  align-items: center;

  &:hover,
  &:focus {
    color: ${Cr("colors.header.text")};
  }

  ${Nr};
  ${zr};
  ${Ar};
  ${et};
`;jn.displayName="Header.Link";var Fn=Object.assign(Wn,{Link:jn,Item:Dn});const _n=s.h2`
  font-weight: ${Cr("fontWeights.bold")};
  font-size: ${Cr("fontSizes.5")};
  margin: 0;
  ${Ar};
  ${Nr};
  ${et};
`;_n.defaultProps={theme:x};const Un=s.span`
  ${Nr}
  & * {
    margin-right: ${Cr("space.1")};
  }
  & *:last-child {
    margin-right: 0;
  }
  ${et};
`,Gn=g(["margin-top:-1px;margin-bottom:-1px;color:",";border:"," solid ",";box-shadow:none;",";",";background-color:transparent;"],Cr("colors.label.secondary.text"),Cr("borderWidths.1"),Cr("colors.label.border"),cr,Nr),Kn=be({variants:{small:{fontSize:0,lineHeight:"16px",padding:"0px 8px"},medium:{fontSize:0,lineHeight:"20px",padding:"0 8px"},large:{fontSize:0,lineHeight:"24px",padding:"0 12px"},xl:{fontSize:1,lineHeight:"16px",padding:"8px 12px"}}}),Xn=s.span`
  display: inline-block;
  font-weight: ${Cr("fontWeights.semibold")};
  color: ${Cr("colors.text.inverse")};
  border-radius: ${Cr("radii.3")};

  &:hover {
    text-decoration: none;
  }

  ${Kn}
  ${Nr}
  ${e=>e.dropshadow?"box-shadow: inset 0 -1px 0 rgba(27, 31, 35, 0.12)":""}
  ${e=>e.outline?Gn:""} // must be last to override other values
  ${et}
`;Xn.defaultProps={bg:"label.primary.border",variant:"medium"};const Vn=R({hoverColor:{property:"color",scale:"colors"}}),Yn=s.a`
  color: ${e=>e.muted?Cr("colors.text.secondary")(e):Cr("colors.text.link")(e)};
  text-decoration: ${e=>e.underline?"underline":"none"};
  &:hover {
    text-decoration: ${e=>e.muted?"none":"underline"};
    ${e=>e.hoverColor?Vn:e.muted?`color: ${Cr("colors.text.link")(e)}`:""};
  }
  &:is(button) {
    display: inline-block;
    padding: 0;
    font-size: inherit;
    white-space: nowrap;
    cursor: pointer;
    user-select: none;
    background-color: transparent;
    border: 0;
    appearance: none;
  }
  ${Ar};
  ${Nr};
  ${et};
`,qn=s.div`
  position: relative;
  padding-top: ${Cr("space.4")};
  padding-bottom: ${Cr("space.4")};
  margin-bottom: ${Cr("space.4")};
  border-bottom: 1px solid ${Cr("colors.border.primary")};
  ${Nr};
  ${et};
`;const Jn=s.a`
  display: inline-block;
  min-width: 32px;
  padding: 5px 10px;
  font-style: normal;
  line-height: 20px;
  color: ${Cr("colors.text.primary")};
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  cursor: pointer;
  user-select: none;
  text-decoration: none;

  margin-right: ${Cr("space.1")};

  &:last-child {
    margin-right: 0;
  }

  border: ${Cr("borderWidths.1")} solid transparent;
  border-radius: ${Cr("radii.2")};
  transition: border-color 0.2s cubic-bezier(0.3, 0, 0.5, 1);

  &:hover,
  &:focus {
    text-decoration: none;
    border-color: ${Cr("colors.border.primary")};
    outline: 0;
    transition-duration: 0.1s;
  }

  &:active {
    border-color: ${Cr("colors.border.secondary")};
  }

  &[rel='prev'],
  &[rel='next'] {
    color: ${Cr("colors.text.link")};
  }

  &[aria-current],
  &[aria-current]:hover {
    color: ${Cr("colors.state.selected.primaryText")};
    background-color: ${Cr("colors.state.selected.primaryBg")};
    border-color: transparent;
  }

  &[aria-disabled],
  &[aria-disabled]:hover {
    color: ${Cr("colors.text.disabled")}; // check
    cursor: default;
    border-color: transparent;
  }

  @supports (clip-path: polygon(50% 0, 100% 50%, 50% 100%)) {
    &[rel='prev']::before,
    &[rel='next']::after {
      display: inline-block;
      width: 16px;
      height: 16px;
      vertical-align: text-bottom;
      content: '';
      background-color: currentColor;
    }

    // chevron-left
    &[rel='prev']::before {
      margin-right: ${Cr("space.1")};
      clip-path: polygon(
        9.8px 12.8px,
        8.7px 12.8px,
        4.5px 8.5px,
        4.5px 7.5px,
        8.7px 3.2px,
        9.8px 4.3px,
        6.1px 8px,
        9.8px 11.7px,
        9.8px 12.8px
      );
    }

    // chevron-right
    &[rel='next']::after {
      margin-left: ${Cr("space.1")};
      clip-path: polygon(
        6.2px 3.2px,
        7.3px 3.2px,
        11.5px 7.5px,
        11.5px 8.5px,
        7.3px 12.8px,
        6.2px 11.7px,
        9.9px 8px,
        6.2px 4.3px,
        6.2px 3.2px
      );
    }
  }

  ${Nr};
`;function Zn({theme:r,pageCount:t,currentPage:o,onPageChange:a,hrefBuilder:n,marginPageCount:i,showPages:d,surroundingPageCount:l}){const c=e.useCallback((e=>r=>a(r,e)),[a]),s=e.useMemo((()=>function(e,r,t,o,a){const n=[];if(t){const t=[],i=r=>{r>=1&&r<=e&&t.push(r)};let d=r-a,l=r+a;if(d<1&&l>e)d=1,l=e;else if(d<1)for(;d<1;)d++,l++;else if(l>e)for(;l>e;)d--,l--;for(let r=1;r<=o;r++){const t=r,o=e-(r-1);t>=d?l++:i(t),o<=l?d--:i(o)}for(let e=d;e<=l;e++)i(e);const c=t.slice().sort(((e,r)=>e-r)).filter(((e,r,t)=>!r||e!==t[r-1]));for(let e=0;e<c.length;e++){const t=c[e],o=t===r;0===e?(1!==t&&n.push({type:"BREAK",num:1}),n.push({type:"NUM",num:t,selected:o})):(1==t-c[e-1]||n.push({type:"BREAK",num:t-1}),n.push({type:"NUM",num:t,selected:o}))}const s=n[n.length-1];"NUM"===s.type&&s.num!==e&&n.push({type:"BREAK",num:e})}const i={type:"NEXT",num:r+1,disabled:r===e};return[{type:"PREV",num:r-1,disabled:1===r},...n,i]}(t,o,!!d,i,l)),[t,o,d,i,l]);return e.useMemo((()=>s.map((t=>{const{props:o,key:a,content:i}=function(e,r,t){const o={};let a="",n="";switch(e.type){case"PREV":n="page-prev",a="Previous",e.disabled?Object.assign(o,{as:"span","aria-disabled":"true"}):Object.assign(o,{rel:"prev",href:r(e.num),"aria-label":"Previous Page",onClick:t});break;case"NEXT":n="page-next",a="Next",e.disabled?Object.assign(o,{as:"span","aria-disabled":"true"}):Object.assign(o,{rel:"next",href:r(e.num),"aria-label":"Next Page",onClick:t});break;case"NUM":n=`page-${e.num}`,a=String(e.num),e.selected?Object.assign(o,{as:"em","aria-current":"page"}):Object.assign(o,{href:r(e.num),"aria-label":`Page ${e.num}`,onClick:t});break;case"BREAK":n=`page-${e.num}-break`,a="",Object.assign(o,{as:"span","aria-disabled":!0})}return{props:o,key:n,content:a}}(t,n,c(t.num));return e.createElement(Jn,Or({},o,{key:a,theme:r}),i)}))),[s,n,c,r])}const Qn=s.nav`
  margin-top: 20px;
  margin-bottom: 15px;
  text-align: center;
  ${et};
`;function ei({theme:r,pageCount:t,currentPage:o,onPageChange:a=ti,hrefBuilder:n=ri,marginPageCount:i=1,showPages:d=!0,surroundingPageCount:l=2,...c}){const s=Zn({theme:r,pageCount:t,currentPage:o,onPageChange:a,hrefBuilder:n,marginPageCount:i,showPages:d,surroundingPageCount:l});return e.createElement(Qn,Or({"aria-label":"Pagination"},c,{theme:r}),e.createElement(rt,{display:"inline-block",theme:r},s))}function ri(e){return`#${e}`}function ti(){}function oi(r){const{bg:t,border:o,borderColor:a,theme:n}=r,{caret:i,children:d,...l}=r,c={bg:t,borderColor:a,borderWidth:o,location:i,theme:n};return e.createElement(tt,Or({sx:{position:"relative"}},l),d,e.createElement(en,c))}ei.displayName="Pagination",ei.defaultProps={hrefBuilder:ri,marginPageCount:1,onPageChange:ti,showPages:!0,surroundingPageCount:2},oi.displayName="PointerBox";const ai=s.div.attrs((({className:e,caret:r})=>({className:Wa(e,`caret-pos--${r}`)})))`
  position: ${e=>e.relative?"relative":"absolute"};
  z-index: 100;
  display: ${e=>e.open?"block":"none"};

  ${Nr};
  ${Rr};
  ${Lr};
  ${et};
`,ni=s(rt)`
  border: 1px solid ${Cr("colors.border.primary")};
  border-radius: ${Cr("radii.2")};
  position: relative;
  width: 232px;
  margin-right: auto;
  margin-left: auto;
  padding: ${Cr("space.4")};
  background-color: ${Cr("colors.bg.overlay")};

  ${Nr};
  ${Rr};

  // Carets
  &::before,
  &::after {
    position: absolute;
    left: 50%;
    display: inline-block;
    content: '';
  }

  &::before {
    top: -${Cr("space.3")};
    margin-left: -9px;
    border: ${Cr("space.2")} solid transparent; // TODO: solid?
    border-bottom-color: ${Cr("colors.border.overlay")};
  }

  &::after {
    top: -14px;
    margin-left: -${Cr("space.2")};
    border: 7px solid transparent; // todo: solid
    border-bottom-color: ${Cr("colors.bg.overlay")};
  }

  // Bottom-oriented carets
  ${ai}.caret-pos--bottom & ,
  ${ai}.caret-pos--bottom-right & ,
  ${ai}.caret-pos--bottom-left & {
    &::before,
    &::after {
      top: auto;
      border-bottom-color: transparent;
    }

    &::before {
      bottom: -${Cr("space.3")};
      border-top-color: ${Cr("colors.border.overlay")};
    }

    &::after {
      bottom: -14px;
      // stylelint-disable-next-line primer/borders
      border-top-color: ${Cr("colors.bg.overlay")};
    }
  }

  // Top & Bottom: Right-oriented carets
  ${ai}.caret-pos--top-right & ,
  ${ai}.caret-pos--bottom-right & {
    right: -9px;
    margin-right: 0;

    &::before,
    &::after {
      left: auto;
      margin-left: 0;
    }

    &::before {
      right: 20px;
    }

    &::after {
      right: 21px;
    }
  }

  // Top & Bottom: Left-oriented carets
  ${ai}.caret-pos--top-left & ,
  ${ai}.caret-pos--bottom-left & {
    left: -9px;
    margin-left: 0;

    &::before,
    &::after {
      left: ${Cr("space.4")};
      margin-left: 0;
    }

    &::after {
      left: calc(${Cr("space.4")} + 1px);
    }
  }

  // Right- & Left-oriented carets
  ${ai}.caret-pos--right & ,
  ${ai}.caret-pos--right-top & ,
  ${ai}.caret-pos--right-bottom & ,
  ${ai}.caret-pos--left & ,
  ${ai}.caret-pos--left-top & ,
  ${ai}.caret-pos--left-bottom & {
    &::before,
    &::after {
      top: 50%;
      left: auto;
      margin-left: 0;
      border-bottom-color: transparent;
    }

    &::before {
      // stylelint-disable-next-line primer/spacing
      margin-top: calc((${Cr("space.2")} + 1px) * -1);
    }

    &::after {
      margin-top: -${Cr("space.2")};
    }
  }

  // Right-oriented carets
  ${ai}.caret-pos--right & ,
  ${ai}.caret-pos--right-top & ,
  ${ai}.caret-pos--right-bottom & {
    &::before {
      right: -${Cr("space.3")};
      border-left-color: ${Cr("colors.border.overlay")};
    }

    &::after {
      right: -14px;
      // stylelint-disable-next-line primer/borders
      border-left-color: ${Cr("colors.bg.overlay")};
    }
  }

  // Left-oriented carets
  ${ai}.caret-pos--left & ,
  ${ai}.caret-pos--left-top & ,
  ${ai}.caret-pos--left-bottom & {
    &::before {
      left: -${Cr("space.3")};
      border-right-color: ${Cr("colors.border.overlay")};
    }

    &::after {
      left: -14px;
      // stylelint-disable-next-line primer/borders
      border-right-color: ${Cr("colors.bg.overlay")};
    }
  }

  // Right & Left: Top-oriented carets
  ${ai}.caret-pos--right-top & ,
  ${ai}.caret-pos--left-top & {
    &::before,
    &::after {
      top: ${Cr("space.4")};
    }
  }

  // Right & Left: Bottom-oriented carets
  ${ai}.caret-pos--right-bottom & ,
  ${ai}.caret-pos--left-bottom & {
    &::before,
    &::after {
      top: auto;
    }

    &::before {
      bottom: ${Cr("space.3")};
    }

    &::after {
      bottom: calc(${Cr("space.3")} + 1px);
    }
  }

  ${et};
`;ai.defaultProps={caret:"top"},ni.displayName="Popover.Content";var ii=Object.assign(ai,{Content:ni});const di=s.span`
  width: ${e=>e.progress?`${e.progress}%`:0};
  ${Nr}
`,li={small:"5px",large:"10px",default:"8px"},ci=s.span`
  display: ${e=>e.inline?"inline-flex":"flex"};
  overflow: hidden;
  background-color: ${Cr("colors.border.primary")};
  border-radius: ${Cr("radii.1")};
  height: ${e=>li[e.barSize||"default"]};
  ${Nr}
  ${me}
  ${et};
`;function si({progress:r,bg:t,theme:o,...a}){return e.createElement(ci,Or({theme:o},a),e.createElement(di,{progress:r,bg:t,theme:o}))}function fi(e,r,t){const n=o((o=>{const a=()=>{t(!0)},n=r=>{const t=Array.from(e.current.querySelectorAll('[role^="menuitem"]:not([hidden]):not([disabled]):not([aria-disabled="true"])')),o=document.activeElement,a=t.indexOf(o),n=r?t[a+1]:t[a-1],i=r?t[0]:t[t.length-1];return n||i};if(!(o instanceof KeyboardEvent))return;const i=o.target instanceof Element&&"SUMMARY"===o.target.tagName;switch(o.key){case"Escape":r&&((()=>{t(!1);const r=e.current.querySelector("summary");r&&r.focus()})(),o.preventDefault(),o.stopPropagation());break;case"ArrowDown":{i&&!r&&a();const e=n(!0);e&&e.focus(),o.preventDefault()}break;case"ArrowUp":{i&&!r&&a();const e=n(!1);e&&e.focus(),o.preventDefault()}break;case" ":case"Enter":{const r=document.activeElement;r&&(e=>{const r=e.getAttribute("role");return"menuitem"===r||"menuitemcheckbox"===r||"menuitemradio"===r})(r)&&r.closest("details")===e&&(o.preventDefault(),o.stopPropagation(),r.click())}}}),[e,r,t]);a((()=>{const r=e.current;if(r)return r.addEventListener("keydown",n),()=>{r.removeEventListener("keydown",n)}}),[e,n])}si.displayName="ProgressBar",si.defaultProps={bg:"bg.successInverse",barSize:"default"};const bi=l({}),gi=g(["padding:"," ",";margin:0;font-size:",";font-weight:",";color:",";background-color:",";border-bottom:"," solid ",";"],Cr("space.1"),Cr("space.3"),Cr("fontSizes.0"),Cr("fontWeights.bold"),Cr("colors.text.tertiary"),Cr("colors.bg.tertiary"),Cr("borderWidths.1"),Cr("colors.selectMenu.borderSecondary")),pi=s.div`
  ${gi}
  ${Nr}
  ${et};
`;pi.displayName="SelectMenu.Divider";var ui=L(ee,P,H,M,W,_,G,X,j,re,ge,pe,ue),mi=new RegExp("^("+ui.propNames.join("|")+")$");const hi=be({variants:{small:{minHeight:"28px",px:2,py:"3px",fontSize:0,lineHeight:"20px"},large:{px:2,py:"10px",fontSize:3}}}),xi=s.input`
  border: 0;
  font-size: inherit;
  font-family: inherit;
  background-color: transparent;
  -webkit-appearance: none;
  color: inherit;
  width: 100%;
  &:focus {
    outline: 0;
  }
`,vi=s.span`
  display: inline-flex;
  align-items: stretch;
  min-height: 34px;
  font-size: ${Cr("fontSizes.1")};
  line-height: 20px;
  color: ${Cr("colors.text.primary")};
  vertical-align: middle;
  background-repeat: no-repeat; // Repeat and position set for form states (success, error, etc)
  background-position: right 8px center; // For form validation. This keeps images 8px from right and centered vertically.
  border: 1px solid ${Cr("colors.border.primary")};
  border-radius: ${Cr("radii.2")};
  outline: none;
  box-shadow: ${Cr("shadows.shadow.inset")};

  ${e=>e.hasIcon?g(["padding:0;"]):g(["padding:6px 12px;"])}

  .TextInput-icon {
    align-self: center;
    color: ${Cr("colors.icon.tertiary")};
    margin: 0 ${Cr("space.2")};
    flex-shrink: 0;
  }

  &:focus-within {
    border-color: ${Cr("colors.state.focus.border")};
    box-shadow: ${Cr("shadows.state.focus.shadow")};
  }

  ${e=>e.contrast&&g(["background-color:",";"],Cr("colors.input.contrastBg"))}

  ${e=>e.disabled&&g(["color:",";background-color:",";border-color:",";"],Cr("colors.text.secondary"),Cr("colors.input.disabledBg"),Cr("colors.input.disabledBorder"))}

  ${e=>e.block&&g(["display:block;width:100%;"])}

  // Ensures inputs don't zoom on mobile but are body-font size on desktop
  @media (min-width: ${Cr("breakpoints.1")}) {
    font-size: ${Cr("fontSizes.1")};
  }
  ${Nr}
  ${me}
  ${xe}
  ${ye}
  ${hi}
  ${et};
`,yi=e.forwardRef((({icon:r,contrast:t,className:o,block:a,disabled:n,theme:i,sx:d,...l},c)=>{const s=Wa(o,"TextInput-wrapper"),f=function(e){var r={};for(var t in e)mi.test(t)&&(r[t]=e[t]);return r}(l),b=function(e){var r={};for(var t in e)mi.test(t)||(r[t]=e[t]);return r}(l);return e.createElement(vi,Or({className:s,hasIcon:!!r,block:a,theme:i,disabled:n,contrast:t,sx:d},f),r&&e.createElement(r,{className:"TextInput-icon"}),e.createElement(xi,Or({ref:c,disabled:n},b)))}));yi.defaultProps={type:"text"},yi.displayName="TextInput";const wi=s.form`
  padding: ${Cr("space.3")};
  margin: 0;
  border-bottom: ${Cr("borderWidths.1")} solid ${Cr("colors.selectMenu.borderSecondary")};
  background-color: ${Cr("colors.bg.overlay")};
  ${Nr};

  @media (min-width: ${Cr("breakpoints.0")}) {
    padding: ${Cr("space.2")};
  }

  ${et};
`,Bi=i((({theme:r,value:o,sx:n,...i},d)=>{const l=t(null),s=null!=d?d:l,{open:f}=c(bi);return a((()=>{var e;f&&(null===(e=l.current)||void 0===e||e.focus())}),[f]),e.createElement(wi,{theme:r,sx:n},e.createElement(yi,Or({theme:r,ref:s,width:"100%",block:!0,value:o,contrast:!0},i)))}));Bi.displayName="SelectMenu.Filter";const $i=g(["margin-top:-1px;padding:"," ",";font-size:",";color:",";text-align:center;border-top:"," solid ",";@media (min-width:","){padding:"," ",";}"],Cr("space.2"),Cr("space.3"),Cr("fontSizes.0"),Cr("colors.text.tertiary"),Cr("borderWidths.1"),Cr("colors.selectMenu.borderSecondary"),Cr("breakpoints.0"),Cr("space.1"),Cr("space.2")),ki=s.footer`
  ${$i}
  ${Nr}
  ${et};
`;ki.displayName="SelectMenu.Footer";const Si=s.h3`
  color: ${Cr("colors.text.primary")};
  flex: auto;
  font-size: ${Cr("fontSizes.1")};
  font-weight: ${Cr("fontWeights.bold")};
  margin: 0;

  @media (min-width: ${Cr("breakpoints.0")}) {
    font-size: inherit;
  }
`,Ti=s.header`
  display: flex;
  flex: none; // fixes header from getting squeezed in Safari iOS
  padding: ${Cr("space.3")};
  border-bottom: ${Cr("borderWidths")} solid ${Cr("colors.selectMenu.borderSecondary")};
  ${Nr}
  ${Ar}

  @media (min-width: ${Cr("breakpoints.0")}) {
    padding-top: ${Cr("space.2")};
    padding-bottom: ${Cr("space.2")};
  }

  ${et};
`,Ei=({children:r,theme:t,...o})=>e.createElement(Ti,Or({theme:t},o),e.createElement(Si,{theme:t},r));Ei.displayName="SelectMenuHeader",Ei.displayName="SelectMenu.Header";const Ii=g(["display:flex;align-items:center;padding:",";overflow:hidden;text-align:left;cursor:pointer;background-color:",";border:0;border-bottom:"," solid ",";color:",";text-decoration:none;font-size:",";font-family:inherit;width:100%;&:hover{text-decoration:none;}&:focus{outline:none;}&[hidden]{display:none !important;}@media (min-width:","){padding-top:",";padding-bottom:",";}.SelectMenu-icon{width:",";margin-right:",";flex-shrink:0;}.SelectMenu-selected-icon{visibility:hidden;transition:transform 0.12s cubic-bezier(0.5,0.1,1,0.5),visibility 0s 0.12s linear;transform:scale(0);}&[aria-checked='true']{font-weight:500;color:",";.SelectMenu-selected-icon{visibility:visible;transition:transform 0.12s cubic-bezier(0,0,0.2,1),visibility 0s linear;transform:scale(1);}}@media (hover:hover){&:hover,&:active,&:focus{background-color:",";}}@media (hover:none){&:focus,&:active{background-color:",";}-webkit-tap-highlight-color:",";}"],Cr("space.3"),Cr("colors.bg.overlay"),Cr("borderWidths.1"),Cr("colors.selectMenu.borderSecondary"),Cr("colors.text.secondary"),Cr("fontSizes.0"),Cr("breakpoints.0"),Cr("space.2"),Cr("space.2"),Cr("space.3"),Cr("space.2"),Cr("colors.text.primary"),Cr("colors.state.hover.secondaryBg"),Cr("colors.bg.secondary"),Cr("colors.selectMenu.tapHighlight")),Ci=s.a.attrs((()=>({role:"menuitemcheckbox"})))`
  ${Ii}
  ${Nr}
  ${et};
`,Ni=i((({children:r,selected:o,theme:a,onClick:n,...i},d)=>{const l=c(bi),s=t(null),f=null!=d?d:s;return e.createElement(Ci,Or({ref:f},i,{theme:a,onClick:e=>{var r;(n&&n(e),e.defaultPrevented)||(null===(r=l.setOpen)||void 0===r||r.call(l,!1))},"aria-checked":o}),e.createElement(ko,{theme:a,className:"SelectMenu-icon SelectMenu-selected-icon",icon:Jt}),r)}));Ni.defaultProps={selected:!1},Ni.displayName="SelectMenu.Item";const Ai=g(["position:relative;padding:0;margin:0;flex:auto;overflow-x:hidden;overflow-y:auto;background-color:",";-webkit-overflow-scrolling:touch;@media (hover:hover){.SelectMenuTab:focus{background-color:",";}.SelectMenuTab:not([aria-checked='true']):hover{color:",";background-color:",";}.SelectMenuTab:not([aria-checked='true']):active{color:",";background-color:",";}}"],Cr("colors.bg.overlay"),Cr("colors.selectMenu.tapFocusBg"),Cr("colors.text.primary"),Cr("colors.bg.tertiary"),Cr("colors.text.primary"),Cr("colors.bg.tertiary")),zi=s.div`
  ${Ai}
  ${Nr}
  ${et};
`;zi.displayName="SelectMenu.List";const Ri={small:"16px",medium:"32px",large:"64px"};function Li({size:r="medium",...t}){const o=Ri[r];return e.createElement("svg",Or({height:o,width:o,viewBox:"0 0 16 16",fill:"none"},t),e.createElement("circle",{cx:"8",cy:"8",r:"7",stroke:"currentColor",strokeOpacity:"0.25",strokeWidth:"2",vectorEffect:"non-scaling-stroke"}),e.createElement("path",{d:"M15 8a7.002 7.002 0 00-7-7",stroke:"currentColor",strokeWidth:"2",strokeLinecap:"round",vectorEffect:"non-scaling-stroke"}))}Li.displayName="Spinner";const Mi=s(Li)`
  @keyframes rotate-keyframes {
    100% {
      transform: rotate(360deg);
    }
  }

  animation: rotate-keyframes 1s linear infinite;

  ${Nr}
  ${et}
`;Mi.displayName="Spinner";const Oi=s.div`
  padding: ${Cr("space.6")} ${Cr("space.4")};
  text-align: center;
  background-color: ${Cr("colors.bg.overlay")};
  ${Nr}
`,Hi=r=>e.createElement(Oi,r,e.createElement(Mi,null));Hi.displayName="SelectMenuLoadingAnimation";const Pi=u(["0%{opacity:0;transform:scale(0.9);}"]),Wi=g(["position:relative;z-index:99;display:flex;",";max-height:",";margin:auto 0;",";overflow:hidden;pointer-events:auto;flex-direction:column;background-color:",";border-radius:",";box-shadow:",";animation:"," 0.12s cubic-bezier(0,0.1,0.1,1) backwards;@media (min-width:","){height:auto;max-height:350px;margin:"," 0 "," 0;font-size:",";border:"," solid ",";border-radius:",";box-shadow:",";}"],(e=>e.filter?"height: 80%":""),(e=>e.filter?"none":"66%"),(e=>e.filter?"margin-top: 0":""),Cr("colors.bg.overlay"),Cr("radii.2"),Cr("shadows.shadow.small"),Pi,Cr("breakpoints.0"),Cr("space.1"),Cr("space.3"),Cr("fontSizes.0"),Cr("borderWidths.1"),Cr("colors.border.overlay"),Cr("radii.2"),Cr("shadows.shadow.small")),Di=g(["position:fixed;top:0;right:0;bottom:0;left:0;z-index:99;display:flex;padding:",";pointer-events:none;flex-direction:column;&::before{position:absolute;top:0;right:0;bottom:0;left:0;pointer-events:none;content:'';background-color:",";@media (min-width:","){display:none;}}@media (min-width:","){position:absolute;top:auto;right:",";bottom:auto;left:auto;padding:0;}"],Cr("space.3"),Cr("colors.selectMenu.backdropBg"),Cr("breakpoints.0"),Cr("breakpoints.0"),(e=>"right"===e.align?"0":"auto")),ji=s.div`
  ${Wi}
  ${me}
`,Fi=s.div`
  ${Di}
  ${Nr}
  ${et};
`,_i=e.forwardRef((({children:r,theme:t,width:o,...a},n)=>e.createElement(Fi,Or({theme:t},a,{role:"menu",ref:n}),e.createElement(ji,{theme:t,width:o},r))));_i.defaultProps={align:"left",width:"300px"},_i.displayName="SelectMenu.Modal";const Ui=g(["flex:1;padding:"," ",";font-size:",";font-weight:500;color:",";text-align:center;background-color:transparent;border:0;box-shadow:inset 0 -1px 0 ",";@media (min-width:","){flex:none;padding:"," ",";border:"," solid transparent;border-bottom-width:0;border-top-left-radius:",";border-top-right-radius:",";}&[aria-selected='true']{z-index:1;color:",";background-color:",";box-shadow:0 0 0 1px ",";@media (min-width:","){border-color:",";box-shadow:none;}}&:focus{background-color:",";}"],Cr("space.2"),Cr("space.3"),Cr("fontSizes.0"),Cr("colors.text.tertiary"),Cr("colors.selectMenu.borderSecondary"),Cr("breakpoints.0"),Cr("space.1"),Cr("space.3"),Cr("borderWidths.1"),Cr("radii.2"),Cr("radii.2"),Cr("colors.text-primary"),Cr("colors.bg.overlay"),Cr("colors.selectMenu.borderSecondary"),Cr("breakpoints.0"),Cr("colors.selectMenu.borderSecondary"),Cr("colors.state.hover.secondaryBg")),Gi=s.button`
  ${Ui}
  ${Nr}
  ${et};
`,Ki=({tabName:r="",index:t,className:o,onClick:n,...i})=>{const d=c(bi);a((()=>{var e;d.selectedTab||0!==t||(null===(e=d.setSelectedTab)||void 0===e||e.call(d,r))}),[t,d,r]);const l=d.selectedTab===r;return e.createElement(Gi,Or({role:"tab",className:Wa("SelectMenuTab",o),"aria-selected":l,onClick:e=>{var t;(n&&n(e),e.defaultPrevented)||(null===(t=d.setSelectedTab)||void 0===t||t.call(d,r))}},i),r)};Ki.displayName="SelectMenuTab",Ki.displayName="SelectMenu.Tab";const Xi=s.div`
  border-top: ${Cr("borderWidths.1")} solid ${Cr("colors.selectMenu.borderSecondary")};
  ${Nr}
  ${et};
`,Vi=({tabName:r,className:t,children:o,...a})=>{const n=c(bi);return e.createElement(Xi,Or({role:"tabpanel",className:t,hidden:n.selectedTab!==r},a),e.createElement(zi,null,o))};Vi.displayName="TabPanel",Vi.displayName="SelectMenu.TabPanel";const Yi=g(["display:flex;flex-shrink:0;margin-bottom:-1px;-webkit-overflow-scrolling:touch;overflow-x:auto;overflow-y:hidden;&::-webkit-scrollbar{display:none;}@media (min-width:","){padding:0 ",";margin-top:",";}"],Cr("breakpoints.0"),Cr("space.2"),Cr("space.3")),qi=s.div`
  ${Yi}
  ${Nr}
  ${et};
`,Ji=({children:r,...t})=>e.createElement(qi,Or({role:"tablist"},t),r);Ji.displayName="SelectMenuTabs",Ji.displayName="SelectMenu.Tabs";const Zi=s.details`
  ${"\n  // Remove marker added by the display: list-item browser default\n  > summary {\n    list-style: none;\n  }\n  // Remove marker added by details polyfill\n  > summary::before {\n    display: none;\n  }\n  // Remove marker added by Chrome\n  > summary::-webkit-details-marker {\n    display: none;\n  }\n"}
  ${Nr}
  ${et};
`,Qi=e.forwardRef((({children:n,initialTab:i="",as:d,...l},c)=>{const s=t(null),f=null!=c?c:s,[b,g]=r(i),[p,u]=r(!1),m={selectedTab:b,setSelectedTab:g,setOpen:u,open:p,initialTab:i},h=o((e=>{"current"in f&&f.current&&!f.current.contains(e.target)&&(e.defaultPrevented||u(!1))}),[f,u]);return a((()=>{if(p)return document.addEventListener("click",h),()=>{document.removeEventListener("click",h)}}),[p,h]),fi(f,p,u),e.createElement(bi.Provider,{value:m},e.createElement(Zi,Or({ref:f},l,{open:p,onToggle:function(e){u(e.target.open)}}),n))}));Qi.displayName="SelectMenu";var ed=Object.assign(Qi,{MenuContext:bi,List:zi,Divider:pi,Filter:Bi,Footer:ki,Item:Ni,Modal:_i,Tabs:Ji,Tab:Ki,TabPanel:Vi,Header:Ei,LoadingAnimation:Hi});function rd({variant:r,className:t,bordered:o,children:a,...n}){const i=Wa(t,`variant-${"lightweight"===r?"lightweight":"normal"}`);return o||(n={...n,borderWidth:0}),e.createElement(rt,Or({borderWidth:"1px",borderStyle:"solid",borderColor:"border.primary",borderRadius:2,as:"nav",className:i},n),a)}rd.displayName="SideNavBase";const td=s(rd)`
  background-color: ${Cr("colors.bg.secondary")};

  ${e=>e.bordered&&g(["& > &{border-left:0;border-right:0;border-bottom:0;}"])}

  ${Nr};
  ${et};
`,od=g(["background-color:",";outline:none;text-decoration:none;"],Cr("colors.state.hover.secondaryBg")),ad=g(["color:",";text-decoration:none;outline:none;"],Cr("colors.text.primary")),nd=s(Yn).attrs((e=>"string"==typeof e.to||e.selected?{"aria-current":"page"}:{}))`
  position: relative;
  display: block;
  ${e=>"full"===e.variant&&g(["display:flex;align-items:center;justify-content:space-between;"])}
  width: 100%;
  text-align: left;
  font-size: ${Cr("fontSizes.1")};

  & > ${td} {
    border-bottom: none;
  }

  ${td}.variant-normal > & {
    color: ${Cr("colors.text.primary")};
    padding: ${Cr("space.3")};
    border: 0;
    border-top: ${Cr("borderWidths.1")} solid ${Cr("colors.border.secondary")};

    &:first-child {
      border-top: 0;
      border-top-right-radius: ${Cr("radii.2")};
      border-top-left-radius: ${Cr("radii.2")};
    }

    &:last-child {
      border-bottom-right-radius: ${Cr("radii.2")};
      border-bottom-left-radius: ${Cr("radii.2")};
    }

    // Bar on the left
    &::before {
      position: absolute;
      top: 0;
      bottom: 0;
      left: 0;
      z-index: 1;
      width: 3px;
      pointer-events: none;
      content: '';
    }

    &:hover {
      ${od}
    }

    &:focus {
      ${od}
      box-shadow: ${Cr("shadows.state.focus.shadow")};
      z-index: 1;
    }

    &[aria-current='page'],
    &[aria-selected='true'] {
      background-color: ${Cr("colors.sidenav.selectedBg")};

      // Bar on the left
      &::before {
        background-color: ${Cr("colors.sidenav.borderActive")};
      }
    }
  }

  ${td}.variant-lightweight > & {
    padding: ${Cr("space.1")} 0;
    color: ${Cr("colors.text.link")};

    &:hover {
      ${ad}
    }

    &:focus {
      ${ad}
      box-shadow: ${Cr("shadows.state.focus.shadow")};
      z-index: 1;
    }

    &[aria-current='page'],
    &[aria-selected='true'] {
      color: ${Cr("colors.text.primary")};
      font-weight: ${Cr("fontWeights.semibold")};
    }
  }

  ${et};
`;td.defaultProps={variant:"normal"},nd.defaultProps={variant:"normal"},nd.displayName="SideNav.Link";var id=Object.assign(td,{Link:nd});const dd={issueOpened:ro,pullOpened:Qt,issueClosed:eo,pullClosed:Qt,pullMerged:Zt,draft:Qt},ld=be({prop:"status",variants:{issueClosed:{backgroundColor:"prState.closed.bg",color:"prState.closed.text",borderColor:"prState.closed.border"},pullClosed:{backgroundColor:"prState.closed.bg",color:"prState.closed.text",borderColor:"prState.closed.border"},pullMerged:{backgroundColor:"prState.merged.bg",color:"prState.merged.text",borderColor:"prState.merged.border"},issueOpened:{backgroundColor:"prState.open.bg",color:"prState.open.text",borderColor:"prState.open.border"},pullOpened:{backgroundColor:"prState.open.bg",color:"prState.open.text",borderColor:"prState.open.border"},draft:{backgroundColor:"prState.draft.bg",color:"prState.draft.text",borderColor:"prState.draft.border"}}}),cd=be({prop:"variant",variants:{small:{paddingX:2,paddingY:1,fontSize:0},normal:{paddingX:"12px",paddingY:2,fontSize:1}}}),sd=s.span`
  display: inline-flex;
  align-items: center;
  font-weight: ${Cr("fontWeights.bold")};
  line-height: 16px;
  color: ${Cr("colors.bg.primary")};
  text-align: center;
  border-radius: ${Cr("radii.3")};
  border-width: 1px;
  border-style: solid;
  ${ld};
  ${cd};
  ${Nr};
  ${et};
`;function fd({children:r,status:t,variant:o,...a}){const n="small"===o?{width:"1em"}:{};return e.createElement(sd,Or({},a,{variant:o,status:t}),t&&e.createElement(ko,Or({mr:1},n,{icon:dd[t]||to})),r)}fd.displayName="StateLabel",fd.defaultProps={variant:"normal"};const bd=s.nav`
  display: flex;
  justify-content: space-between;

  .SubNav-body {
    display: flex;
    margin-bottom: -1px;

    > * {
      margin-left: ${Cr("space.2")};
    }

    > *:first-child {
      margin-left: 0;
    }
  }

  .SubNav-actions {
    align-self: center;
  }

  ${Nr};
  ${Mr};
  ${et};
`;function gd({actions:r,className:t,children:o,label:a,...n}){const i=Wa(t,"SubNav");return e.createElement(bd,Or({className:i,"aria-label":a},n),e.createElement("div",{className:"SubNav-body"},o),r&&e.createElement("div",{className:"SubNav-actions"},r))}function pd(r){return e.createElement(rt,Or({display:"flex"},r))}gd.displayName="SubNav",pd.displayName="SubNavLinks";const ud=s.a.attrs((e=>({activeClassName:"string"==typeof e.to?"selected":"",className:Wa("SubNav-item",e.selected&&"selected",e.className)})))`
  padding-left: ${Cr("space.3")};
  padding-right: ${Cr("space.3")};
  font-weight: ${Cr("fontWeights.semibold")};
  font-size: ${Cr("fontSizes.1")};
  line-height: 20px; //custom value for SubNav
  min-height: 34px; //custom value for SubNav
  color: ${Cr("colors.text.primary")};
  text-align: center;
  text-decoration: none;
  border-top: 1px solid ${Cr("colors.border.primary")};
  border-bottom: 1px solid ${Cr("colors.border.primary")};
  border-right: 1px solid ${Cr("colors.border.primary")};
  display: flex;
  align-items: center;

  &:first-of-type {
    border-top-left-radius: ${Cr("radii.2")};
    border-bottom-left-radius: ${Cr("radii.2")};
    border-left: 1px solid ${Cr("colors.border.primary")};
  }

  &:last-of-type {
    border-top-right-radius: ${Cr("radii.2")};
    border-bottom-right-radius: ${Cr("radii.2")};
  }

  &:hover,
  &:focus {
    text-decoration: none;
    background-color: ${Cr("colors.bg.tertiary")};
    transition: 0.2s ease;

    .SubNav-octicon {
      color: ${Cr("colors.icon.secondary")};
    }
  }

  &.selected {
    color: ${Cr("colors.state.selected.primaryText")};
    background-color: ${Cr("colors.state.selected.primaryBg")};
    border-color: ${Cr("colors.state.selected.primaryBorder")};
    .SubNav-octicon {
      color: ${Cr("colors.state.selected.primaryText")};
    }
  }

  ${Nr};
  ${et};
`;ud.displayName="SubNav.Link",pd.displayName="SubNav.Links";var md=Object.assign(gd,{Link:ud,Links:pd});const hd=s.div`
  margin-top: 0;
  border-bottom: 1px solid ${Cr("colors.border.primary")};
  ${Nr}
  ${et}
`,xd=s.nav`
  display: flex;
  margin-bottom: -1px;
  overflow: auto;
`;function vd({children:r,"aria-label":t,...o}){return e.createElement(hd,o,e.createElement(xd,{"aria-label":t},r))}vd.displayName="TabNav";const yd=s.a.attrs((e=>({activeClassName:"string"==typeof e.to?"selected":"",className:Wa("TabNav-item",e.selected&&"selected",e.className)})))`
  padding: 8px 12px;
  font-size: ${Cr("fontSizes.1")};
  line-height: 20px;
  color: ${Cr("colors.text.primary")};
  text-decoration: none;
  background-color: transparent;
  border: 1px solid transparent;
  border-bottom: 0;

  &:hover,
  &:focus {
    color: ${Cr("colors.text.primary")};
    text-decoration: none;
  }

  &.selected {
    color: ${Cr("colors.text.primary")};
    border-color: ${Cr("colors.border.primary")};
    border-top-right-radius: ${Cr("radii.2")};
    border-top-left-radius: ${Cr("radii.2")};
    background-color: ${Cr("colors.bg.canvas")};
  }

  ${Nr};
  ${et};
`;yd.displayName="TabNav.Link";var wd=Object.assign(vd,{Link:yd});const Bd=s(rt).withConfig({displayName:"Timeline",componentId:"c4dq2e-0"})(["display:flex;flex-direction:column;"," ",";"],(e=>e.clipSidebar&&g([".Timeline-Item:first-child{padding-top:0;}.Timeline-Item:last-child{padding-bottom:0;}"])),et),$d=s(rt).attrs((e=>({className:Wa("Timeline-Item",e.className)}))).withConfig({displayName:"Timeline__TimelineItem",componentId:"c4dq2e-1"})(["position:relative;padding:"," 0;margin-left:",";&::before{position:absolute;top:0;bottom:0;left:0;display:block;width:2px;content:'';background-color:",";}"," ",";",";"],Cr("space.3"),Cr("space.3"),Cr("colors.border.secondary"),(e=>e.condensed&&g(["padding-top:",";padding-bottom:0;&:last-child{padding-bottom:",";}.TimelineItem-Badge{height:16px;margin-top:",";margin-bottom:",";color:",";background-color:",";border:0;}"],Cr("space.1"),Cr("space.3"),Cr("space.2"),Cr("space.2"),Cr("colors.icon.tertiary"),Cr("colors.bg.canvas"))),Nr,et);var kd=s(rt).withConfig({displayName:"Timeline___StyledBox",componentId:"c4dq2e-2"})(["border-radius:50%;border:2px solid ",";"],(e=>e._css));const Sd=r=>e.createElement(rt,{position:"relative",zIndex:1},e.createElement(kd,Or({display:"flex",className:Wa(r.className,"TimelineItem-Badge"),flexShrink:0,overflow:"hidden",color:"icon.secondary",bg:"timeline.badgeBg",width:"32px",height:"32px",mr:2,ml:"-15px",alignItems:"center",justifyContent:"center"},r,{_css:Cr("colors.bg.canvas")}),r.children));Sd.displayName="TimelineBadge";const Td=s(rt).withConfig({displayName:"Timeline__TimelineBody",componentId:"c4dq2e-3"})(["min-width:0;max-width:100%;margin-top:",";color:",";flex:auto;font-size:",";",";"],Cr("space.1"),Cr("colors.timeline.text"),Cr("fontSizes.1"),et),Ed=s(rt).withConfig({displayName:"Timeline__TimelineBreak",componentId:"c4dq2e-4"})(["position:relative z-index:1;height:24px;margin:0;margin-bottom:-",";margin-left:0;background-color:",";border:0;border-top:"," solid ",";",";"],Cr("space.3"),Cr("colors.bg.canvas"),Cr("space.1"),Cr("colors.border.primary"),et);$d.displayName="Timeline.Item",Sd.displayName="Timeline.Badge",Td.displayName="Timeline.Body",Ed.displayName="Timeline.Break";var Id=Object.assign(Bd,{Item:$d,Badge:Sd,Body:Td,Break:Ed});const Cd=s.span`
  position: relative;

  &::before {
    position: absolute;
    z-index: 1000001;
    display: none;
    width: 0px;
    height: 0px;
    color: ${Cr("colors.tooltip.bg")};
    pointer-events: none;
    content: '';
    border: 6px solid transparent;
    opacity: 0;
  }

  &::after {
    position: absolute;
    z-index: 1000000;
    display: none;
    padding: 0.5em 0.75em;
    font: normal normal 11px/1.5 ${Cr("fonts.normal")};
    -webkit-font-smoothing: subpixel-antialiased;
    color: ${Cr("colors.tooltip.text")};
    text-align: center;
    text-decoration: none;
    text-shadow: none;
    text-transform: none;
    letter-spacing: normal;
    word-wrap: break-word;
    white-space: pre;
    pointer-events: none;
    content: attr(aria-label);
    background: ${Cr("colors.tooltip.bg")};
    border-radius: ${Cr("radii.1")};
    opacity: 0;
  }

  // delay animation for tooltip
  @keyframes tooltip-appear {
    from {
      opacity: 0;
    }

    to {
      opacity: 1;
    }
  }

  &:hover,
  &:active,
  &:focus {
    &::before,
    &::after {
      display: inline-block;
      text-decoration: none;
      animation-name: tooltip-appear;
      animation-duration: 0.1s;
      animation-fill-mode: forwards;
      animation-timing-function: ease-in;
      animation-delay: 0.4s;
    }
  }

  &.tooltipped-no-delay:hover,
  &.tooltipped-no-delay:active,
  &.tooltipped-no-delay:focus {
    &::before,
    &::after {
      animation-delay: 0s;
    }
  }

  &.tooltipped-multiline:hover,
  &.tooltipped-multiline:active,
  &.tooltipped-multiline:focus {
    &::after {
      display: table-cell;
    }
  }

  // Tooltipped south
  &.tooltipped-s,
  &.tooltipped-se,
  &.tooltipped-sw {
    &::after {
      top: 100%;
      right: 50%;
      margin-top: 6px;
    }

    &::before {
      top: auto;
      right: 50%;
      bottom: -7px;
      margin-right: -6px;
      border-bottom-color: ${Cr("colors.tooltip.bg")};
    }
  }

  &.tooltipped-se {
    &::after {
      right: auto;
      left: 50%;
      margin-left: -${Cr("space.3")};
    }
  }

  &.tooltipped-sw::after {
    margin-right: -${Cr("space.3")};
  }

  // Tooltips above the object
  &.tooltipped-n,
  &.tooltipped-ne,
  &.tooltipped-nw {
    &::after {
      right: 50%;
      bottom: 100%;
      margin-bottom: 6px;
    }

    &::before {
      top: -7px;
      right: 50%;
      bottom: auto;
      margin-right: -6px;
      border-top-color: ${Cr("colors.tooltip.bg")};
    }
  }

  &.tooltipped-ne {
    &::after {
      right: auto;
      left: 50%;
      margin-left: -${Cr("space.3")};
    }
  }

  &.tooltipped-nw::after {
    margin-right: -${Cr("space.3")};
  }

  // Move the tooltip body to the center of the object.
  &.tooltipped-s::after,
  &.tooltipped-n::after {
    transform: translateX(50%);
  }

  // Tooltipped to the left
  &.tooltipped-w {
    &::after {
      right: 100%;
      bottom: 50%;
      margin-right: 6px;
      transform: translateY(50%);
    }

    &::before {
      top: 50%;
      bottom: 50%;
      left: -7px;
      margin-top: -6px;
      border-left-color: ${Cr("colors.tooltip.bg")};
    }
  }

  // tooltipped to the right
  &.tooltipped-e {
    &::after {
      bottom: 50%;
      left: 100%;
      margin-left: 6px;
      transform: translateY(50%);
    }

    &::before {
      top: 50%;
      right: -7px;
      bottom: 50%;
      margin-top: -6px;
      border-right-color: ${Cr("colors.tooltip.bg")};
    }
  }

  &.tooltipped-multiline {
    &::after {
      width: max-content;
      max-width: 250px;
      word-wrap: break-word;
      white-space: pre-line;
      border-collapse: separate;
    }

    &.tooltipped-s::after,
    &.tooltipped-n::after {
      right: auto;
      left: 50%;
      transform: translateX(-50%);
    }

    &.tooltipped-w::after,
    &.tooltipped-e::after {
      right: 100%;
    }
  }

  &.tooltipped-align-right-2::after {
    right: 0;
    margin-right: 0;
  }

  &.tooltipped-align-right-2::before {
    right: 15px;
  }

  &.tooltipped-align-left-2::after {
    left: 0;
    margin-left: 0;
  }

  &.tooltipped-align-left-2::before {
    left: 10px;
  }
  ${Nr};
  ${et};
`;function Nd({direction:r="n",children:t,className:o,text:a,noDelay:n,align:i,wrap:d,...l}){const c=Wa(o,`tooltipped-${r}`,i&&`tooltipped-align-${i}-2`,n&&"tooltipped-no-delay",d&&"tooltipped-multiline");return e.createElement(Cd,Or({role:"tooltip","aria-label":a},l,{className:c}),t)}Nd.displayName="Tooltip",Nd.alignments=["left","right"],Nd.directions=["n","ne","e","se","s","sw","w","nw"];const Ad=s.nav`
  display: flex;
  justify-content: space-between;
  border-bottom: 1px solid ${Cr("colors.border.secondary")};
  &.UnderlineNav--right {
    justify-content: flex-end;

    .UnderlineNav-item {
      margin-right: 0;
      margin-left: ${Cr("space.3")};
    }

    .UnderlineNav-actions {
      flex: 1 1 auto;
    }
  }
  &.UnderlineNav--full {
    display: block;
  }

  .UnderlineNav-body {
    display: flex;
    margin-bottom: -1px;
  }

  .UnderlineNav-actions {
    align-self: center;
  }

  ${Nr};
  ${et};
`;function zd({actions:r,className:t,align:o,children:a,full:n,label:i,theme:d,...l}){const c=Wa(t,"UnderlineNav",o&&`UnderlineNav--${o}`,n&&"UnderlineNav--full");return e.createElement(Ad,Or({className:c,"aria-label":i,theme:d},l),e.createElement("div",{className:"UnderlineNav-body"},a),r&&e.createElement("div",{className:"UnderlineNav-actions"},r))}zd.displayName="UnderlineNav";const Rd=s.a.attrs((e=>({activeClassName:"string"==typeof e.to?"selected":"",className:Wa("UnderlineNav-item",e.selected&&"selected",e.className)})))`
  padding: ${Cr("space.3")} ${Cr("space.2")};
  margin-right: ${Cr("space.3")};
  font-size: ${Cr("fontSizes.1")};
  line-height: ${Cr("lineHeights.default")};
  color: ${Cr("colors.underlinenav.text")};
  text-align: center;
  border-bottom: 2px solid transparent;
  text-decoration: none;

  &:hover,
  &:focus {
    color: ${Cr("colors.underlinenav.textHover")};
    text-decoration: none;
    border-bottom-color: ${Cr("colors.border.tertiary")};
    transition: 0.2s ease;

    .UnderlineNav-octicon {
      color: ${Cr("colors.text.tertiary")};
    }
  }

  &.selected {
    color: ${Cr("colors.underlinenav.textActive")};
    border-bottom-color: ${Cr("colors.underlinenav.borderActive")};

    .UnderlineNav-octicon {
      color: ${Cr("colors.underlinenav.iconActive")};
    }
  }

  ${Nr};
  ${et};
`;Rd.displayName="UnderlineNav.Link";var Ld=Object.assign(zd,{Link:Rd});export{it as Absolute,ka as ActionList,Ra as ActionMenu,La as Avatar,Oa as AvatarPair,ja as AvatarStack,Wr as BaseStyles,tt as BorderBox,rt as Box,Fa as BranchName,Xa as Breadcrumb,jt as Button,io as ButtonClose,Ft as ButtonDanger,_t as ButtonGroup,Kt as ButtonInvisible,Ut as ButtonOutline,Gt as ButtonPrimary,Xt as ButtonTableList,en as Caret,an as CircleBadge,nn as CircleOcticon,ea as ConfirmationDialog,dn as CounterLabel,ln as Details,xn as Dialog,Sn as Dropdown,Tn as DropdownButton,En as DropdownMenu,Rn as FilterList,In as FilteredSearch,dt as Fixed,Mn as Flash,ot as Flex,Pn as FormGroup,at as Grid,Fn as Header,_n as Heading,Xn as Label,Un as LabelGroup,Yn as Link,Ia as Overlay,qn as Pagehead,ei as Pagination,oi as PointerBox,ii as Popover,nt as Position,si as ProgressBar,lt as Relative,ed as SelectMenu,id as SideNav,Mi as Spinner,fd as StateLabel,ct as Sticky,ko as StyledOcticon,md as SubNav,wd as TabNav,fn as Text,yi as TextInput,qr as ThemeProvider,Id as Timeline,Nd as Tooltip,ia as Truncate,Ld as UnderlineNav,x as theme,Cr as themeGet,Zr as useColorSchemeVar,ra as useConfirm,st as useDetails,Bt as useOnEscapePress,ut as useOnOutsideClick,vt as useOpenAndCloseFocus,kt as useOverlay,ft as useSafeTimeout,Jr as useTheme};
