// @primer/components/lib/AvatarPair.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Avatar = _interopRequireDefault(require("./Avatar"));

var _constants = require("./constants");

var _ = require(".");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const ChildAvatar = (0, _styledComponents.default)(_Avatar.default).withConfig({
  displayName: "AvatarPair__ChildAvatar",
  componentId: "sc-1jbk273-0"
})(["position:absolute;right:-15%;bottom:-9%;box-shadow:", ";"], (0, _constants.get)('shadows.avatar.childShadow'));

const AvatarPair = ({
  children,
  ...rest
}) => {
  const avatars = _react.default.Children.map(children, (child, i) => {
    if (! /*#__PURE__*/_react.default.isValidElement(child)) return child;
    return i === 0 ? /*#__PURE__*/_react.default.cloneElement(child, {
      size: 40
    }) : /*#__PURE__*/_react.default.createElement(ChildAvatar, _extends({
      bg: "bg.canvas"
    }, child.props, {
      size: 20
    }));
  });

  return /*#__PURE__*/_react.default.createElement(_.Box, _extends({
    position: "relative",
    display: "inline-flex"
  }, rest), avatars);
};

AvatarPair.displayName = "AvatarPair";
// styled() changes this
AvatarPair.displayName = 'AvatarPair';
var _default = AvatarPair;
exports.default = _default;
