// @primer/components/lib/Breadcrumb.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classnames = _interopRequireDefault(require("classnames"));

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Box = _interopRequireDefault(require("./Box"));

var _constants = require("./constants");

var _sx = _interopRequireDefault(require("./sx"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const SELECTED_CLASS = 'selected';

const Wrapper = _styledComponents.default.li.withConfig({
  displayName: "Breadcrumb__Wrapper",
  componentId: "c1r3ff-0"
})(["display:inline-block;white-space:nowrap;list-style:none;&::after{padding-right:0.5em;padding-left:0.5em;color:", ";font-size:", ";content:'/';}&:first-child{margin-left:0;}&:last-child{&::after{content:none;}}"], (0, _constants.get)('colors.text.disabled'), (0, _constants.get)('fontSizes.1'));

const BreadcrumbBase = _styledComponents.default.nav.withConfig({
  displayName: "Breadcrumb__BreadcrumbBase",
  componentId: "c1r3ff-1"
})(["display:flex;justify-content:space-between;", ";", ";", ";"], _constants.COMMON, _constants.FLEX, _sx.default);

function Breadcrumb({
  className,
  children,
  theme,
  ...rest
}) {
  const classes = (0, _classnames.default)(className, 'Breadcrumb');

  const wrappedChildren = _react.default.Children.map(children, child => /*#__PURE__*/_react.default.createElement(Wrapper, {
    theme: theme
  }, child));

  return /*#__PURE__*/_react.default.createElement(BreadcrumbBase, _extends({
    className: classes,
    "aria-label": "breadcrumb",
    theme: theme
  }, rest), /*#__PURE__*/_react.default.createElement(_Box.default, {
    as: "ol",
    my: 0,
    pl: 0
  }, wrappedChildren));
}

Breadcrumb.displayName = "Breadcrumb";

const BreadcrumbItem = _styledComponents.default.a.attrs(props => ({
  activeClassName: typeof props.to === 'string' ? 'selected' : '',
  className: (0, _classnames.default)(props.selected && SELECTED_CLASS, props.className),
  'aria-current': props.selected ? 'page' : null
})).withConfig({
  displayName: "Breadcrumb__BreadcrumbItem",
  componentId: "c1r3ff-2"
})(["color:", ";display:inline-block;font-size:", ";text-decoration:none;&:hover{text-decoration:underline;}&.selected{color:", ";pointer-events:none;}", " ", ";"], (0, _constants.get)('colors.text.link'), (0, _constants.get)('fontSizes.1'), (0, _constants.get)('colors.text.primary'), _constants.COMMON, _sx.default);

Breadcrumb.displayName = 'Breadcrumb';
BreadcrumbItem.displayName = 'Breadcrumb.Item';

var _default = Object.assign(Breadcrumb, {
  Item: BreadcrumbItem
});

exports.default = _default;
