// @primer/components/lib/Flash.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styledSystem = require("styled-system");

var _constants = require("./constants");

var _sx = _interopRequireDefault(require("./sx"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const variants = (0, _styledSystem.variant)({
  variants: {
    default: {
      color: 'alert.info.text',
      backgroundColor: 'alert.info.bg',
      borderColor: 'alert.info.border',
      svg: {
        color: 'alert.info.icon'
      }
    },
    success: {
      color: 'alert.success.text',
      backgroundColor: 'alert.success.bg',
      borderColor: 'alert.success.border',
      svg: {
        color: 'alert.success.icon'
      }
    },
    danger: {
      color: 'alert.error.text',
      backgroundColor: 'alert.error.bg',
      borderColor: 'alert.error.border',
      svg: {
        color: 'alert.error.icon'
      }
    },
    warning: {
      color: 'alert.warn.text',
      backgroundColor: 'alert.warn.bg',
      borderColor: 'alert.warn.border',
      svg: {
        color: 'alert.warn.icon'
      }
    }
  }
});

const Flash = _styledComponents.default.div.withConfig({
  displayName: "Flash",
  componentId: "sc-4y90bn-0"
})(["position:relative;color:", ";padding:", ";border-style:solid;border-width:", ";border-radius:", ";margin-top:", ";p:last-child{margin-bottom:0;}svg{margin-right:", ";}", ";", ";", ";"], (0, _constants.get)('colors.text.primary'), (0, _constants.get)('space.3'), props => props.full ? '1px 0px' : '1px', props => props.full ? '0' : (0, _constants.get)('radii.2'), props => props.full ? '-1px' : '0', (0, _constants.get)('space.2'), _constants.COMMON, variants, _sx.default);

Flash.defaultProps = {
  variant: 'default'
};
var _default = Flash;
exports.default = _default;
