// @primer/components/lib/Position.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Sticky = exports.Relative = exports.Fixed = exports.Absolute = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Box = _interopRequireDefault(require("./Box"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

/**
 * @deprecated Use the Box component instead (i.e. <Position> → <Box>)
 */
const Position = (0, _styledComponents.default)(_Box.default).withConfig({
  displayName: "Position",
  componentId: "fla1y3-0"
})([""]);

/**
 * @deprecated Use the Box component instead (i.e. <Absolute> → <Box position="absolute">)
 */
var _default = Position; // Absolute

exports.default = _default;

/**
 * @deprecated Use the Box component instead (i.e. <Absolute> → <Box position="absolute">)
 */
const Absolute = /*#__PURE__*/_react.default.forwardRef((props, ref) => {
  return /*#__PURE__*/_react.default.createElement(Position, _extends({}, props, {
    position: "absolute",
    ref: ref
  }));
});

exports.Absolute = Absolute;
Absolute.displayName = 'Absolute'; // Fixed

/**
 * @deprecated Use the Box component instead (i.e. <Fixed> → <Box position="fixed">)
 */
const Fixed = /*#__PURE__*/_react.default.forwardRef((props, ref) => {
  return /*#__PURE__*/_react.default.createElement(Position, _extends({}, props, {
    position: "fixed",
    ref: ref
  }));
});

exports.Fixed = Fixed;
Fixed.displayName = 'Fixed'; // Relative

/**
 * @deprecated Use the Box component instead (i.e. <Relative> → <Box position="relative">)
 */
const Relative = /*#__PURE__*/_react.default.forwardRef((props, ref) => {
  return /*#__PURE__*/_react.default.createElement(Position, _extends({}, props, {
    position: "relative",
    ref: ref
  }));
});

exports.Relative = Relative;
Relative.displayName = 'Relative'; // Sticky

/**
 * @deprecated Use the Box component instead (i.e. <Sticky> → <Box position="sticky">)
 */
const Sticky = /*#__PURE__*/_react.default.forwardRef((props, ref) => {
  return /*#__PURE__*/_react.default.createElement(Position, _extends({}, props, {
    position: "sticky",
    ref: ref
  }));
});

exports.Sticky = Sticky;
Sticky.defaultProps = {
  top: 0,
  zIndex: 1
};
Sticky.displayName = 'Sticky';
