// @primer/components/lib/Position.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Sticky = exports.Relative = exports.Fixed = exports.Absolute = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Box = _interopRequireDefault(require("./Box"));

var _constants = require("./constants");

var _sx = _interopRequireDefault(require("./sx"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const Position = (0, _styledComponents.default)(_Box.default).withConfig({
  displayName: "Position",
  componentId: "fla1y3-0"
})(["", ";", ";"], _constants.POSITION, _sx.default);
var _default = Position; // Absolute

exports.default = _default;

const Absolute = _react.default.forwardRef((props, ref) => {
  return /*#__PURE__*/_react.default.createElement(Position, _extends({}, props, {
    position: "absolute",
    ref: ref
  }));
});

exports.Absolute = Absolute;
Absolute.displayName = 'Absolute'; // Fixed

const Fixed = _react.default.forwardRef((props, ref) => {
  return /*#__PURE__*/_react.default.createElement(Position, _extends({}, props, {
    position: "fixed",
    ref: ref
  }));
});

exports.Fixed = Fixed;
Fixed.displayName = 'Fixed'; // Relative

const Relative = _react.default.forwardRef((props, ref) => {
  return /*#__PURE__*/_react.default.createElement(Position, _extends({}, props, {
    position: "relative",
    ref: ref
  }));
});

exports.Relative = Relative;
Relative.displayName = 'Relative'; // Sticky

const Sticky = _react.default.forwardRef((props, ref) => {
  return /*#__PURE__*/_react.default.createElement(Position, _extends({}, props, {
    position: "sticky",
    ref: ref
  }));
});

exports.Sticky = Sticky;
Sticky.defaultProps = {
  top: 0,
  zIndex: 1
};
Sticky.displayName = 'Sticky';
