// @primer/components/lib/ProgressBar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styledSystem = require("styled-system");

var _constants = require("./constants");

var _sx = _interopRequireDefault(require("./sx"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const Bar = _styledComponents.default.span.withConfig({
  displayName: "ProgressBar__Bar",
  componentId: "sc-1tiva13-0"
})(["width:", ";", ""], props => props.progress ? `${props.progress}%` : 0, _constants.COMMON);

const sizeMap = {
  small: '5px',
  large: '10px',
  default: '8px'
};

const ProgressContainer = _styledComponents.default.span.withConfig({
  displayName: "ProgressBar__ProgressContainer",
  componentId: "sc-1tiva13-1"
})(["display:", ";overflow:hidden;background-color:", ";border-radius:", ";height:", ";", " ", " ", ";"], props => props.inline ? 'inline-flex' : 'flex', (0, _constants.get)('colors.border.primary'), (0, _constants.get)('radii.1'), props => sizeMap[props.barSize || 'default'], _constants.COMMON, _styledSystem.width, _sx.default);

function ProgressBar({
  progress,
  bg,
  theme,
  ...rest
}) {
  return /*#__PURE__*/_react.default.createElement(ProgressContainer, _extends({
    theme: theme
  }, rest), /*#__PURE__*/_react.default.createElement(Bar, {
    progress: progress,
    bg: bg,
    theme: theme
  }));
}

ProgressBar.displayName = "ProgressBar";
ProgressBar.defaultProps = {
  bg: 'bg.successInverse',
  barSize: 'default'
};
var _default = ProgressBar;
exports.default = _default;
